## Outline and motivation

While we followed a well-established workflow in our previous analysis, there
were still many decisions we had to make about which analysis tools and
parameters to use. As I discussed in **Chapter \@ref(tools)** methods for
analysing scRNA-seq data have developed at a rapid rate and there are always
new software packages to try. New methods should improve on what already exists
or approach the data in a different. Even when a dataset is well understood
applying new methods can reinforce what has already been seen or reveal
interesting new information. For the remainder of this chapter I will explore
some of the alternative analytical choices we could have made and approaches we
could have used in our analysis of the kidney organoid dataset, and what
difference they may have made to our interpretation of this data. I have
selected these methods to look at because they either do something
significantly different to what we used in our previous analysis or they have
been shown to be effective in comparison studies. This chapter also provides an
illustration of some of the thought processes involved in an analysis project.
An outline of the analysis I have performed and where it will be discussed in
this chapter is shown in Figure \@ref(fig:workflow). I begin by looking at some
of the pre-processing steps that are often overlooked, particularly methods for
selecting cell-containing droplets. I also compare the alignment-based
quantification produced by the standard pre-processing pipeline to a
pseudoalignment quantification. The next section explores some of the choices
that can be made when performing quality control to select high-quality cells.
The main aim of this analysis is to identify the cell types present in kidney
organoids which requires clustering together similar cells. Clustering methods
often require a set of genes to be chosen and here I have trialled an
alternative gene selection method and compared the resulting clusters to the
previously published analysis. In the final section of this chapter I apply two
newly developed analysis approaches to this dataset to see what additional
information they can add. Details of the analysis including the code used and
exploratory visualisations are displayed on a website at
https://lazappi.github.io/phd-thesis-analysis/ [@Zappia2019-kh] and the
analysis methods used are provided in Appendix \@ref(analysis-methods).

```{r workflow, fig.cap = "(ref:cap-workflow)", fig.scap = "(ref:scap-workflow)", out.width = "100%"}
knitr::include_graphics(here::here("figures/05-workflow.png"))
```

(ref:scap-workflow) Outline of analysis of the kidney organoid dataset.

(ref:cap-workflow) Outline of analysis of the kidney organoid dataset. Filled coloured boxes indicate stages of the analysis. Purple stages contribute to the final results shown here while blue stages were performed as a comparison. Grey stages were performed in the published analysis but are not shown here. The main analysis workflow is indicated with black arrows and comparison stages with pointed grey arrows. The published workflow is shown using smaller open grey arrows with dotted lines. Green shaded boxes show the software tools used and pink shaded boxes indicate the sections of this chapter that discuss the results of those stages.
