## Preprocessing

The first stage in any analysis of sequencing data is to convert the files
returned by the sequencing machine to a format suitable for analysis. For
scRNA-seq data produced using the 10x Genomics Chromium platform the standard
tool for accomplishing this task is the Cell Ranger software produced by the
company. This pipeline performs a number of tasks including demultiplexing the
BCL base call files returned by Illumina sequencers and converting them to more
common FASTQ sequence files, aligning reads to a reference genome, counting
reads overlapping annotated genes, separating cells by barcode and
deduplicating UMIs. Some of these stages are performed by other tools such as
Illumina's bcl2fastq software and the STAR splice-aware aligner [@Dobin2013-gx]
but with wrappers designed to handle specific features of 10x libraries. For
the analysis presented here I have performed preprocessing of the first batch
of three organoid samples using Cell Ranger (v3.0.1) with the GRCh38 reference
genome provided by 10x Genomics (v3.0.0) based on the ENSEMBL gene annotation
(v93). Each sample was processed separately to produce an expression matrix and
then these were aggregated without any normalisation to form a single dataset.

### Droplet selection

One of the final steps in preprocessing a dataset from a droplet-based cell
capture method is to select droplets that contain cells. Many droplets are
produced during the cell capture process but most of these will not contain
cells and any reads associated with them will be the result of ambient RNA
present in solution. The original Cell Ranger selection algorithm (used in our
previous publications) required an expected number of cells for each sample
($N$). The 99th percentile of the total number of UMI counts in the top $N$
cells was then taken as a robust estimate of the maximum number of counts in a
cell and any droplet with at least 10 percent of this maximum was selected as a
cell [@Zheng2017-mm] (Figure \@ref(fig:droplet-selection)A). A similar approach
to select droplets that fall above the knee point of the cumulative fraction of
reads with respect to increasing total count [@Macosko2015-rl]. While these
methods effectively select most cells they risk overlooking cells that have a
lower RNA content than most cells in the sample. The emptyDrops method
[@Lun2018-lb] in the DropletUtils package provides an alternative approach
(Figure \@ref(fig:droplet-selection)B). A threshold on total counts is selected
below which droplets are assumed to be empty and these empty droplets are used
to produce a profile of the ambient RNA in the sample. Here I have set that
threshold to `r params$pre$empty_thresh` counts. A Monte Carlo sampling
approach is then used to calculate a p-value indicating whether a particular
droplet is likely to have come from the ambient distribution. A traditional
threshold is also used to select cells with more total counts than the knee
point when droplets are ordered by total counts. Cell Ranger v3 uses a modified
version of the emptyDrops procedure where the original selection is substituted
method for the knee point threshold (Figure \@ref(fig:droplet-selection)C).

```{r droplet-selection, fig.cap = "(ref:cap-droplet-selection)", fig.scap = "(ref:scap-droplet-selection)", out.width = "100%"}
knitr::include_graphics(here::here("figures/droplet-selection.png"))
```

(ref:scap-droplet-selection) Droplet selection methods applied to the kidney organoid dataset.

(ref:cap-droplet-selection) Droplet selection methods applied to the kidney organoid dataset. Ordered plots of the total counts per cell coloured by selection method. (A) The traditional Cell Ranger approach which applies a threshold to the total counts per a droplet based on an expected number of cells in a sample. (B) The emptyDrops approach which tests whether droplets are significantly different from the ambient RNA profile. (C) Cell Ranger v3 uses a method that combines aspects of both approaches.

As we would expect the original Cell Ranger algorithm is the most conservative,
selecting only `r params$pre$n_default` cells (Figure
\@ref(fig:selection-comparison)A). The emptyDrops and Cell Ranger v3 methods,
which use a testing procedure rather than a strict threshold, identify many
more cell-containing droplets. Many of these cells are identified by both
methods but there are also a large number that are identified only by
emptyDrops. This is surprising as the two software package provide different
implementations of what is largely the same algorithm. Some of these differences
may be due to the parameters I have chosen for emptyDrops but it is difficult
to say if these are different to those that are used by Cell Ranger. This
highlights some of the trade-offs inherent in using a pre-designed pipeline for
processing any dataset. While it is convenient to have a single tools that
automates a series of tasks this often comes at the cost of less control over
each stage than would be possible if using a series of individual tools.

```{r selection-comparison, fig.cap = "(ref:cap-selection-comparison)", fig.scap = "(ref:scap-selection-comparison)", out.width = "100%"}
knitr::include_graphics(here::here("figures/selection-comparison.png"))
```

(ref:scap-selection-comparison) UpSet plot comparing droplet selection methods.

(ref:cap-selection-comparison) UpSet plot comparing droplet selection methods. Cell Ranger v3 and emptyDrops identify significantly more cells than the traditional Cell Ranger approach. My use of emptyDrops also identifies a set of cells that are overlooked by the automated Cell Ranger v3 procedure.

It is possible that my usage of emptyDrops has been too permissive and
identified cells in droplets that are actually empty but it is difficult to
tell at this stage. In the following section I will perform further quality
control of cells anyway so for now I will keep any cells that were identified
by either the emptyDrops or Cell Ranger methods. Of the
`r params$pre$n_droplets` droplets with at least one count present in the raw
dataset `r params$pre$n_cells` were selected by at least one of these methods.

### Alevin comparison

Cell Ranger is based on the traditional workflow for processing RNA-seq reads
which is to align them to a reference genome then count how many overlap
annotated genes. In recent years it has become common to use methods such as
Salmon [@Patro2017-ka] and Kallisto [@Bray2016-tm] that directly estimate
transcript expression without alignment. While these approaches do not produce
an exact location for each read they can be much faster which is appealing for
scRNA-seq data as there are many more cells to quantify. However, factors such
as cell barcodes, UMIs and different model assumptions mean that these tools
have had to be adapted for scRNA-seq data. Alevin [@Srivastava2018-fy] is the
scRNA-seq mode available in Salmon.

Alevin has it's own method for identifying which droplets contain cells. First
the knee point in cumulative distribution of barcode frequencies is identified
and any droplets above this are assumed to contain cells. The Levenshtein
distance [@Levenshtein1966-fq] to the barcodes of all the other droplets is
calculated and if they are not sufficiently close they are considered to be
noise and the reads discarded. Alevin performs a second round of cell
whitelisting following quantification. Droplets are divided into three zones,
high-quality (top half above the knee point), ambiguous (bottom half above the
knee point) and low-quality (well below the knee point). A na√Øve Bayes
classifier based on a set of features including the fraction of reads mapped,
the fraction of mitochondrial reads and the duplication rate is then used to
classify droplets as either high or low-quality. This is very different to the
emptyDrops approach, using a machine learning technique rather than a
statistical test and technical features rather than the overall expression
profile, and is more similar to approaches that have been suggested for quality
control of cells [@Ilicic2016-wy]. Figure \@ref(fig:alevin-comparison)A shows a
comparison of this approach to the two methods I used to select cells in the
previous section. Most of the cells are identified by all three methods but
just like emptyDrops, Alevin finds a large number of cells that are overlooked
by Cell Ranger v3. What is particularly striking is that there is very little
overlap in the extra cells identified by Alevin and emptyDrops suggesting that
the differences between the two approaches may be important. However I should
note that the Alevin uses it's own quantification as input for classification
while I provided the counts from Cell Ranger to emptyDrops so some of the
differences may be due to the data they were given rather than the selection
methods themselves. Alevin (and Cell Ranger v3) also processes the three
samples separately while I performed emptyDrops on the whole dataset.

```{r alevin-comparison, fig.cap = "(ref:cap-alevin-comparison)", fig.scap = "(ref:scap-alevin-comparison)", out.width = "100%"}
knitr::include_graphics(here::here("figures/alevin-comparison.png"))
```

(ref:scap-alevin-comparison) Comparison to quantification using Alevin.

(ref:cap-alevin-comparison) Comparison to quantification using Alevin. (A) UpSet plot comparing the droplets identified as cells by the Alevin, Cell Ranger v3 and emptyDrops methods. (B) Scatter plot comparing total counts per cell as estimated by Alevin and the alignment based Cell Ranger pipeline. Points are coloured according to the method that identified them as cells (Alevin only (green), alignment based only (blue), both (pink). Thin purple line shows $y = x$, thick blue line shows a linear fit for cells identified by both methods. (C) Scatter plot comparing mean counts per gene as estimated by the two pipelines, points colours and lines as in (B). (D) Relationship between the total counts per gene and the percentage of cells with a zero count for each pipeline.

Similar to Cell Ranger the quantification stage of Alevin involves several
steps including mapping reads and deduplicating UMIs. Alevin maps reads to the
transcriptome and groups them by cell barcode and UMI. To deduplicate UMIs
Alevin considers the mapping location and the UMI sequence (to account for
sequencing errors). Because reads have been mapped to the transcriptome rather
than the genome there is an additional layer of information that can be useful
for identifying if reads with the same UMI map to the same location that avoids
over collapsing UMIs. Following these steps the transcriptome level expression
estimates are aggregated to the gene level for analysis. Figure
\@ref(fig:alevin-comparison)B shows a comparison of the total counts per cell
between Alevin and the alignment based approach using Cell Ranger. For cells
that were identified by both pipelines (pink) the total counts are very
similar. Cells identified just by the alignment based approach (blue) tend to
have fewer counts than those identified only by Alevin. There are more
difference if we look at the mean counts per gene (Figure
\@ref(fig:alevin-comparison)C). Most of the variability is at low expression
levels but there are some genes that are lowly expressed by one method but have
relatively high expression in the other. One reason for this could be either
method missing (or incorrectly assigning) counts altogether. Figure
\@ref(fig:alevin-comparison)D shows the relationship between the total counts
per gene and the percentage of cells with zero counts. This relationship is
noisier for the Alevin quantification, particularly below the trend line, and
also show more genes with a very low percentage of zeros.

While there are differences between the results from the two approaches there
is nothing in this comparison to suggest that either is inaccurate. Genes that
are found to be expressed by only one of the methods could possibly lead to
differences in downstream interpretation but it is very difficult to say which
is a better representation of true expression and it is unlikely that all genes
providing similar evidence (for example markers of a cell type) would be
affected in the same way. On this evidence Alevin presents a viable alternative
to alignment based quantification for scRNA-seq data, particularly for large
datasets where computational time is a concern. However, when alignment is
required anyway it is simpler and avoids repetition to use that to produce an
expression matrix for further analysis. For the remainder of the analysis I have
used the alignment-based quantification of this dataset.
