## Clustering

Now that we have a dataset containing high-quality cells, the next step in my
analysis is to perform clustering. It is necessary to use some technique to
group and label cells. For this kind of dataset that contains an unknown set
of cell types, an unsupervised approach is a good fit. Clustering allows us to
group cells based purely on their expression profiles without making any
assumptions about what those groups should be. Here I am going to use the
graph-based clustering method available in the Seurat package. The first step in
this approach is to select a subset of genes to use.

### Gene selection

Most genes will either be expressed at a consistent level across all cells in a
dataset or have too few counts to provide enough information about differences
between cells. To reduce the noise in the dataset and the computational cost of
analysis we want to select a set of genes that will be useful for separating
cells. An intuitive approach is to filter based on the mean expression level
to remove lowly expressed genes, and the gene variability to select those that
explain differences between cells. This is the default approach used in Seurat
(Figure \@ref(fig:gene-selection)A). The thresholds I have chosen here select
`r params$clust$sel_seurat` highly variable genes.

```{r gene-selection, fig.cap = "(ref:cap-gene-selection)", fig.scap = "(ref:scap-gene-selection)", out.width = "100%"}
knitr::include_graphics(here::here("figures/05-gene-selection.png"))
```

(ref:scap-gene-selection) Selecting genes for clustering.

(ref:cap-gene-selection) Selecting genes for clustering. (A) Scatter plot of mean gene expression level against dispersion. The default Seurat selection method sets thresholds on these axes. Purple lines indicate thresholds and selected genes are coloured blue. (B) The M3Drop method fits the relationship (purple line) between average expression (x-axis) and dropout rate (y-axis). Genes that have significantly more zeros than expected are selected (pink). (C) Comparison of gene selection methods. The dropout rate calculated by M3Drop is plotted against the dispersion calculated by Seurat. Points are coloured according to whether they were selected by Seurat (blue), M3Drop (pink) or both (green). (D) Bar plot of the number of genes selected by each method and the intersection. Most genes are identified by only one approach.

The M3Drop package suggests an alternative approach that considers the
proportion of zero counts per gene rather than the variability
[@Andrews2018-xa]. Here the idea is that if a gene is important for identifying
a cell type then its expression will be concentrated in a subset of cells and
there will be more zero counts than would be expected based on the overall mean
expression. The original M3Drop method fits a Michaelis-Menten function to the
relationships between mean expression and dropout rate (proportion of zeros)
but for UMI datasets that have not been sequenced to saturation the authors
suggest their NBDrop method instead, which fits a library size adjusted
negative binomial model. A binomial test is then used to identify genes that
have significantly more zeros than would be expected from the fitted model
(Figure \@ref(fig:gene-selection)B). By using a statistical test this method
removes the need to set multiple thresholds. Here M3Drop has selected
`r params$clust$sel_m3drop` genes with excess zeros.

If we compare the sets of genes selected by these two methods we see that a
relatively small subset are identified by both of them (Figure
\@ref(fig:gene-selection)D). These differences can be explained by the
differences between variability and zero counts (Figure
\@ref(fig:gene-selection)C). For example, it is possible for a gene to vary a
lot between different cells but within an expression range that doesn't result
in more zeros. Alternatively, a gene with relatively low dispersion may still
have excess zeros if the underlying expression is low enough. For the
previously published analysis we used the Seurat selection method, so here I am
going to use the genes identified by M3Drop and see what effect that has on the
clustering results.

### Resolution selection

There are several more steps in the Seurat clustering procedure. Once a set of
genes is selected they are used to perform a PCA and a set of principal
components is selected. Distances are then calculated between cells in this PCA
space and the nearest neighbours for each cell are identified. By comparing the
sets of neighbours for two cells and computing the Jaccard index, a shared
nearest neighbour graph can be constructed where each node is a cell and the
edges are the overlap in their neighbourhoods. A graph modularity optimisation
algorithm such as Louvain modularity optimisation [@Blondel2008-ym] can then be
used to partition the graph and form clusters of cells. This graph-based
clustering approach has been shown to be effective in comparisons of scRNA-seq
clustering methods [@Duo2018-qb; @Freytag2018-ic]. One of the decisions that
needs to be made during this procedure is the value of the resolution parameter
to use, which affects the number of clusters that are produced. As I have
discussed in Chapter \@ref(simulation), clustering trees are one tool that can
be helpful in making this decision. Figure \@ref(fig:resolution-selection)A
shows a clustering tree for this dataset with resolutions between zero and one.

```{r resolution-selection, fig.cap = "(ref:cap-resolution-selection)", fig.scap = "(ref:scap-resolution-selection)", out.width = "100%"}
knitr::include_graphics(here::here("figures/05-resolution-selection.png"))
```

(ref:scap-resolution-selection) Selecting a clustering resolution.

(ref:cap-resolution-selection) Selecting a clustering resolution. (A) Clustering tree showing resolutions between zero and one. Pink box shows the selected resolution. (B) Clustering trees with nodes coloured by a selection of published marker genes. Expression signals can help to select a clustering resolution.

The structure of this tree can tell us about how clusters are forming in this
dataset. My general approach is to select a resolution where most of the
branches have formed but before there are many low in-proportion edges crossing
over between branches that can indicate that clusters are unstable. Displaying
expression levels of genes that are expected to be important in the dataset
adds extra information to the structure of the tree (Figure
\@ref(fig:resolution-selection)B). Here I look to identify at which resolution
we see clear, high-intensity signals for these genes. That is, a single or few
clusters where they are highly expressed. Based on these trees I have selected
a resolution of `r params$clust$res` which produces `r params$clust$n_clusts`
clusters.

### Cluster validation

The cluster assignments of the individual cells can be displayed in a reduced
dimensional space such as that produced by UMAP [@McInnes2018-ul;
@Becht2018-zh] (Figure \@ref(fig:cluster-validation)A). The dimensions shown in
this visualisation and the cluster labels that are used to colour it are
produced using different algorithms. By combining them we can confirm that
the two approaches produce similar results. Clusters can have different
densities in these plots so it is also useful to know how many cells they
contain (Figure \@ref(fig:cluster-validation)B). Now that we have a set of
clusters we can check to see if they are associated with technical artifacts
rather than biological factors.

```{r cluster-validation, fig.cap = "(ref:cap-cluster-validation)", fig.scap = "(ref:scap-cluster-validation)", out.width = "100%"}
knitr::include_graphics(here::here("figures/05-cluster-validation.png"))
```

(ref:scap-cluster-validation) Validation of identified clusters.

(ref:cap-cluster-validation) Validation of identified clusters. (A) Scatter plot of cells in UMAP space coloured by assigned cluster. (B) Bar chart of the number of cells in each cluster. (C) Stacked bar chart showing the proportion of cells in each cluster identified by each droplet selection method, Cell Ranger v3 only (blue), EmptyDrops only (green) or both (pink). (D) Stacked bar chart showing the proportion of cells in each cluster assigned to the G1 (pink), G2M (blue) or S (green) phases of the cell cycle. (E) SinaPlot [@Sidiropoulos2018-ll] showing the total number of counts per cell in each cluster. (F) SinaPlot showing the percentage of counts assigned to mitochondrial genes per cell in each cluster.

Figure \@ref(fig:cluster-validation)C shows the proportion of cells in each
cluster that were identified by the two droplet selection methods I used, Cell
Ranger v3 and EmptyDrops. Most of the additional cells identified by EmptyDrops
have been placed in two of the clusters. The cells in these clusters also have
fewer total counts (Figure \@ref(fig:cluster-validation)E) and a lower
percentage of mitochondrial counts (Figure \@ref(fig:cluster-validation)F).
This suggests that those clusters are at least in part a result of the lower
count cells identified by EmptyDrops and it may be appropriate to remove them
from the dataset. However, a large proportion of the cells in those clusters
were also identified by Cell Ranger v3 and an alternative hypothesis is that
these additional cells belong to a cell type that is smaller or contains less
RNA for some other reason. Looking at the marker genes associated with those
clusters can help to answer that question (Section \@ref(marker-genes)).

### Comparison to published clusters

Because we have previously analysed and interpreted this dataset I can compare
these clusters to the previously published clusters. It is important to
remember the differences between the two analyses. The published analysis
included a fourth organoid from a separate batch that I have omitted here and
was analysed using the standard Cell Ranger (v1.3.1) and Seurat workflows
without EmptyDrops droplet selection or M3Drop gene selection. I also used
different quality control thresholds and parameters at some stages. Figure
\@ref(fig:cluster-comparison)A shows the proportion of cells in each cluster
that were present in the published analysis. All of the clusters have some
additional cells which is to be expected given that the use of the newer
droplet selection methods produced a dataset with around 1000 more cells. The
two clusters that contain the cells only identified by EmptyDrops (Cluster 4 and
Cluster 6) stand out as having a very low proportion of cells from the previous
analysis.

```{r cluster-comparison, fig.cap = "(ref:cap-cluster-comparison)", fig.scap = "(ref:scap-cluster-comparison)", out.width = "100%"}
knitr::include_graphics(here::here("figures/05-cluster-comparison.png"))
```

(ref:scap-cluster-comparison) Comparison to previously published clusters.

(ref:cap-cluster-comparison) Comparison to previously published clusters. (A) Bar chart showing the proportion of cells in each cluster that were present in the previously published analysis. (B) Heatmap showing the Jaccard index between previously published clusters and the clusters from the current analysis. High values (yellow) indicate that many cells are shared between two clusters while low values (blue) indicate there is little overlap.

To compare the two sets of clusters I have calculated the Jaccard index between
each cluster from the previously published analysis and each cluster from the
analysis presented here (Figure \@ref(fig:cluster-comparison)B). Some of the
clusters (such as Cluster 5) very clearly match up with a published cluster
while others have overlaps with multiple similar clusters. For example, Cluster
2 which of cells from the two cell cycle clusters in the published analysis.
There are four stromal clusters in both analyses but for two of them (Cluster 0
and Cluster 3) the cells are divided differently. The differences in clustering
are the result of all the decisions made to this stage of the analysis
including which cells and genes have been selected. Without a known truth it is
difficult to determine which is better or more correct or if they simply
provide different views of the same data with emphasis on different features
and it is possible that both sets of clusters are biologically meaningful. The
two EmptyDrops clusters do not show obvious signals but they do have some
overlap with the published endothelial clusters which suggests that that they
may be a previously overlooked endothelial population. To confirm if this is
the case we need to check whether they express known endothelial markers.
Identifying marker genes for each cluster through differential expression
testing can show us which genes are unique to each cluster.
