## Connecting clusters

Clustering analysis is used to group cells and identify the cell types that are
present in a dataset but by itself doesn't tell us anything about how those
groups relate to each other. In the developmental context where there are
multiple differentiating cell types this is obviously something of interest.
Some aspects of these relationships can be inferred from the proximity of
clusters in reduced dimensional space or known biology of the assigned cell
types, but this is largely based on intuition rather than information in the
data. Computational methods to calculate developmental trajectories in
scRNA-seq data have been a big focus of early scRNA-seq bioinformatics
research. In our published analysis we used the Monocle package to create a
psuedotime ordering for cells in the nephron lineage. However, methods like
this do not perform as well on larger datasets with many cell types from
possibly unconnected lineages such as we have in the full organoid dataset.
Here I will demonstrate two alternative approaches for connecting clusters and
adding context to the current analysis.

### Partition-based graph abstraction

Partition-based graph abstraction (PAGA) is a method for reducing the
complexity of a dataset by estimating the connectivity between clusters
[@Wolf2017-vq]. The result is a reduced graph representation of the dataset
where each node is a cluster and the edges show how connected they are (Figure
\@ref(fig:paga-results)A). PAGA is available as part of the scanpy package
[@Wolf2018-na] and starts with the same kind of shared nearest neighbour cell
graph that is used Seurat's graph-based clustering. The number of edges between
cells in different clusters is compared to what would be expected if edges were
assigned randomly and used to calculate a connectivity between clusters that
can be interpreted as the confidence in the presence of an actual connection
between them.

The results for the organoid dataset show high connectivity between the various
stromal clusters that make up the majority of cells. Connectivity between some
of the other clusters also confirms what we would expect, for example the
podocyte cluster is strongly connected to the epithelial cluster and two
off-target neural clusters (glial and neural progenitor) are strongly
connected. The position of the two clusters containing emptyDrops cells is less
clear with connections seen between them but not as clearly with the other
clusters they would be expected to be related to, particularly between the two
endothelial clusters.

```{r paga-results, fig.cap = "(ref:cap-paga-results)", fig.scap = "(ref:scap-paga-results)", out.width = "100%"}
knitr::include_graphics(here::here("figures/paga-results.png"))
```

(ref:scap-paga-results) Results of partition-based graph abstraction (PAGA).

(ref:cap-paga-results) Results of partition-based graph abstraction (PAGA). (A) The cluster connectivity graph produced by PAGA. Each node represents a cluster and edges show the connectivity between clusters. The size of nodes indicates the number of cells in each cluster and the edge colour and transparency shows the connectivity from low (blue) to high (yellow). (B). The underlying cell shared nearest neighbour graph. Nodes are cells coloured by cluster and edges are coloured by connectivity.

In some ways the graph produced by PAGA is similar but complementary to a
clustering tree in that it considers connectivity between clusters at the same
resolution rather than overlaps between clusters from neighbouring resolutions.
In fact the authors of the method emphasise the ability to explore a dataset at
different resolutions by producing PAGA graphs from different sets of clusters.
The development of two independent methods based on building graphs of clusters
suggests to me that this is an approach that could have more general
applications. The edges produced by PAGA can show which clusters are related
but they don't provide any directionality to that relationship. One way to add
some directionality to the dataset is to perform a random walk on the graph to
calculate a diffusion pseudotime. However this approach requires manually
defining a root cell to use. An alternative approach for assigning
directionality to cells involves looking at the splicing information available
in the reads themselves.

### Cell velocity

The majority of the reads in an scRNA-seq experiment come from mature mRNA
where the introns have been removed through splicing, however some immature
mRNA molecules that still retain their intronic sequences are also sequenced.
The concept of cell velocity looks at these unspliced reads to determine which
genes are being actively transcribed. By looking at which cellular processes
are being switched on or off it is possible to infer an overall transcriptional
direction for each cell. This approach was popularised for scRNA-seq data by
the velocyto package [@La_Manno2018-tv]. Velocyto works in two steps. First it
inspects the aligned reads for each cell but instead of only counting those
that are inside an annotated exon it also generates a counts matrix of
unspliced reads for each gene. These two matrixes are then combined to estimate
a velocity for each cell by solving a set of differential equations. The
velocities can then be projected onto a reduced dimensional space by looking at
a correlation-based transition matrix on a nearest-neighbour graph in that
space (Figure \@ref(fig:cell-velocity)A). When there are more than a few
hundred cells it is difficult to look at the velocity for each individual cell
so it can be useful to summarise them in some way. Velocyto calculates a
velocity field across the whole dataset by forming a regular grid of points and
using gaussian smoothing to calculate an average of the nearby cells, providing
an indication of the overall velocity of an area (Figure
\@ref(fig:cell-velocity)B).

```{r cell-velocity, fig.cap = "(ref:cap-cell-velocity)", fig.scap = "(ref:scap-cell-velocity)", out.width = "100%"}
knitr::include_graphics(here::here("figures/cell-velocity.png"))
```

(ref:scap-cell-velocity) Results of cell velocity analysis using velocyto.

(ref:cap-cell-velocity) Results of cell velocity analysis using velocyto. (A) Cell velocity estimates for individual cells. Points show cells in UMAP space coloured according to cluster. Arrows show the cell velocity of each cell with direction indicating the transcriptional profile the cell is headed towards and length giving the rate of change. (B) Cell velocity field across the whole dataset. Arrows show the average velocity for cells in each region of a grid.

Looking at the velocity field we see some of the expected developmental
trajectories. For example cells in the epithelial cluster are headed towards
the more mature podocyte population. Despite having a limited connectivity in
the PAGA results we see the endothelial-like cluster containing emptyDrops
cells trending towards the other endothelial cluster. The other emptyDrops
cluster is again more ambiguous showing velocity towards both endothelial and
stromal clusters. This suggests that if there are immune cells present they may
only be a subset of this population. We see little evidence of any clear
directionality in the stromal clusters, likely because these cell types are
more mature and are not actively changing their transcription profiles. There
is an important caveat to look at this velocyto results which is that the
velocity arrows depend on the space they are projected into and viewing them in
different embeddings may give different results. Finding a more robust way to
visualise cell velocity would be a useful area of research that would improve
the ability to interpret these results.
