## Discussion

In this chapter I have presented an analysis of a kidney organoid scRNA-seq
dataset, demonstrating a range of computational tools that can be used and
focusing on decisions made during analysis and how these decisions affect the
results. The dataset is a subset of the data presented in two previous
publications where we investigated variability in the kidney organoid
differentiation protocol and profiled the cell types that are present in these
organoids. Our previous analysis included data from two batches of organoids as
well as integration with a published human foetal kidney dataset but for this
chapter I chose to use just the first batch of data which contains cells from
three organoids. Data integration is an important area of research that has
resulted in a range of methods produced recently, but it is still unclear which
approaches are most effective. It can also be difficult to transfer the results
of integration between analysis tools. As I wanted to demonstrate a range of
tools here I have chosen to work with a single dataset, but how best to combine
datasets is worthy of further consideration.

The most significant difference in my new analysis comes from one of the
earliest steps. During the pre-processing stage of analysing droplet-based
scRNA-seq data empty droplets must be separated from those that contain cells.
While this step can have a significant impact on the results of an analysis it
is often overlooked. I found that methods such as EmptyDrops that test for
differences from the ambient RNA profile identified many more cells than the
thresholding approach traditionally used in the Cell Ranger pipeline. Retaining
cells that were identified by EmptyDrops or the more recent Cell Ranger v3
which implements a similar approach produced a dataset of `r params$pre$n_cells`
cells. There are two possible explanations for the additional cells identified
by these approaches. Either they are more sensitive at correctly selecting
cell-containing droplets or they are more permissive and have incorrectly
selected empty droplets. Quality control of the selected cells can help to
determine which of these is true.

Typically, low-quality cells are removed from a dataset by setting thresholds
on quantities that indicate that a cell is damaged or otherwise uninformative.
I used a data-driven approach to select filtering thresholds that were a given
number of MADs for the median for the total counts per cell, total expressed
features per cell and the percentage of counts assigned to mitochondrial genes
(which is a proxy for the health of a cell). These thresholds removed
`r params$pre$n_cells - params$qc$n_cells` cells leaving `r params$qc$n_cells`,
including many of the additional cells identified by EmptyDrops. The fact that
these cells were retained suggests that they are not obviously empty droplets
and are likely to contain biologically relevant information. Using a statistical
approach to select thresholds means that they can be automatically adapted to
different datasets but it does assume that most cells are high-quality. For this
reason it is still important to visualise distributions and where the thresholds
lie. When I compared the the thresholding approach to a fully-automated approach
based on performing a PCA on a set of quality control metrics and detecting
outliers I found that thresholding identified many more low-quality cells and
was much easier to interpret.

The additional cells identified by EmptyDrops passed quality control but their
biological significance does not become clear until later on in the analysis.
Similar to the previously published analysis I used the graph-based clustering
method in the Seurat package to form groups of cells with similar expression
profiles. By using the clustering tree visualisation presented in **Chapter
\@ref(clust-trees)** I chose a resolution of `r params$clust$res` which
produced `r params$clust$n_clusts` clusters. Most of the additional cells
identified by EmptyDrops belong to two clusters (Cluster 4 and Cluster 6), but
because these cells weren't present in the published analysis there is no clear
overlap to associate them with a published cell type. To better interpret all
the clusters I performed differential gene expression testing using edgeR,
testing the cells in each cluster against the other cells in the dataset and
looking for marker genes that are more highly expressed in a single cluster.
Cluster 4 showed clear up-regulation of endothelial markers, while Cluster 6
had significant results for markers from various cell types including several
genes associated with immune cells. Although not an expected part of the kidney
organoid differentiation process, there is a plausible pathway by which immune
cells could emerge, and when this dataset was combined with a human foetal
kidney dataset we saw a small number of cells that clustered with immune cells
and seemed to share similar expression. However, a significant amount of
additional validation would be required to confirm that immune cells are
present and that this apparent population isn't just the result of technical
artifacts associated with the lower number of counts per cell. The other
clusters were consistent with the previously published analysis and I assigned
them to several kidney cell types, including several types of stroma,
epithelium, mature podocytes and endothelium. I also observed several small
off-target populations that were described previously, including neural
progenitors, glial cells and muscle progenitors.

The Seurat procedure has been shown to perform well in comparison and consists
of several steps: selecting a set of genes to use, performing PCA on those
genes and selecting a set of principal components, building a shared nearest
neighbour graph in PCA space, and using a modularity optimisation algorithm to
partition the graph into clusters. For the published analysis we selected genes
using the standard Seurat method which sets thresholds on gene expression level
and variability. I compared genes selected in this way to
those selected by the M3Drop method, which tests for genes with a greater
number of zero counts than would be expected under a negative binomial model.
Despite the significant differences between the two sets of genes the clusters
identified by Seurat using the M3Drop genes were very similar to those in the
published analysis. This suggests that graph-based clustering is fairly robust
to exactly which genes are selected, as long as they capture enough of the
variability in the dataset. The obvious difference between the two clusterings
is the clusters containing the EmptyDrops cells but it is reassuring that
including these cells did not affect the other clusters and suggests that Seurat
is also robust to the presence of some low-quality cells.

A standard analysis of an scRNA-seq dataset often ends here, but I also wanted
to demonstrate how some newly developed methods can provide additional context
to clusters. PAGA is a method that explores the connectivity between clusters,
showing how they are related and reducing the dataset to a graph of clusters.
When applied here I saw a high amount of connectivity amongst stromal clusters,
and some of the expected developmental trajectories were also highly connected.
Other clusters that we would expect to be similar, such as the two endothelial
clusters, were not so related. Overall PAGA was not particularly effective when
used on this dataset but I believe that is more likely to be a result of this
specific dataset or the parameters selected rather than the approach itself.
Another approach which gives additional context to a dataset is calculating
cell velocity, which provides a transcriptional direction for each cell.
Averaged across the dataset it can reveal where populations are headed or which
populations are transcriptionally stable. The cell velocity field also
highlighted some expected differentiation processes and showed cells in the
possible immune cell population trending in opposite directions, suggesting
that this cluster might contain sub-populations. Potentially this cluster may
contain a mixture of immune cells, other cell types and low-quality cells.

Overall this new analysis is broadly consistent with the results we previously
published, despite using different tools at several stages and making different
decisions about how to set parameters and thresholds. The main differences
between the two analyses are the result of the droplet selection step, with the
test-based methods identifying many more cells. Including these cells in the
analysis resulted in cluster containing a large number of cells only identified
by EmptyDrops and showing a potential immune-like expression profile we hadn't
observed before. A small change in parameters at this early stage could have
hidden this population or introduced more low-quality cells, either of which
would have had a large impact of the rest of the analysis. This emphasises the
importance of carefully considering decisions during analysis, visualising the
data to check and confirm them and iterating over an analysis if required. The
overall consistency in results is reassuring and suggests that Seurat is robust
to both the genes that are selected for clustering and the cells that are
present. However, there is a large amount of heterogeneity in this dataset with
many different cell types that can easily be separated. For an analysis where
the goal was to identify more subtle differences, such as an experiment
comparing patient-derived cells with cells from a corrected cell line, I would
expect these decisions to be more important and to have a bigger impact on how
the results are interpreted. Finally, new methods can extract additional
information from an scRNA-seq dataset and provide additional evidence that
helps with interpretation. I expect that methods like these will become a
standard part of analyses, with multiple methods that approach a dataset in
different ways being used to identify cell types and confirm hypotheses.
