<!DOCTYPE html>
<html >

<head>

  <meta charset="UTF-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <title>Tools and techniques for single-cell RNA sequencing data</title>
  <meta name="description" content="Tools and techniques for single-cell RNA sequencing data">
  <meta name="generator" content="bookdown 0.7 and GitBook 2.6.7">

  <meta property="og:title" content="Tools and techniques for single-cell RNA sequencing data" />
  <meta property="og:type" content="book" />
  
  
  
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Tools and techniques for single-cell RNA sequencing data" />
  
  
  

<meta name="author" content="Luke Zappia">



  <meta name="viewport" content="width=device-width, initial-scale=1">
  <meta name="apple-mobile-web-app-capable" content="yes">
  <meta name="apple-mobile-web-app-status-bar-style" content="black">
  
  
<link rel="prev" href="2-tools.html">
<link rel="next" href="4-clust-trees.html">
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />










</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./"></a></li>
<li class="divider"></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Front matter</a><ul>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#abstract"><i class="fa fa-check"></i>Abstract</a></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#declaration"><i class="fa fa-check"></i>Declaration</a></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#preface"><i class="fa fa-check"></i>Preface</a></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#acknowledgements"><i class="fa fa-check"></i>Acknowledgements</a></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#list-of-copyright"><i class="fa fa-check"></i>List of copyright</a></li>
</ul></li>
<li class="chapter" data-level="1" data-path="1-introduction.html"><a href="1-introduction.html"><i class="fa fa-check"></i><b>1</b> Introduction</a><ul>
<li class="chapter" data-level="1.1" data-path="1-introduction.html"><a href="1-introduction.html#intro-dogma"><i class="fa fa-check"></i><b>1.1</b> The central dogma</a></li>
<li class="chapter" data-level="1.2" data-path="1-introduction.html"><a href="1-introduction.html#intro-RNAseq"><i class="fa fa-check"></i><b>1.2</b> RNA sequencing</a><ul>
<li class="chapter" data-level="1.2.1" data-path="1-introduction.html"><a href="1-introduction.html#library-preparation"><i class="fa fa-check"></i><b>1.2.1</b> Library preparation</a></li>
<li class="chapter" data-level="1.2.2" data-path="1-introduction.html"><a href="1-introduction.html#high-throughput-sequencing"><i class="fa fa-check"></i><b>1.2.2</b> High-throughput sequencing</a></li>
<li class="chapter" data-level="1.2.3" data-path="1-introduction.html"><a href="1-introduction.html#analysis-of-rna-seq-data"><i class="fa fa-check"></i><b>1.2.3</b> Analysis of RNA-seq data</a></li>
</ul></li>
<li class="chapter" data-level="1.3" data-path="1-introduction.html"><a href="1-introduction.html#intro-scRNAseq"><i class="fa fa-check"></i><b>1.3</b> Single-cell RNA sequencing</a><ul>
<li class="chapter" data-level="1.3.1" data-path="1-introduction.html"><a href="1-introduction.html#early-single-cell-capture-technologies"><i class="fa fa-check"></i><b>1.3.1</b> Early single-cell capture technologies</a></li>
<li class="chapter" data-level="1.3.2" data-path="1-introduction.html"><a href="1-introduction.html#droplet-based-cell-capture"><i class="fa fa-check"></i><b>1.3.2</b> Droplet-based cell capture</a></li>
<li class="chapter" data-level="1.3.3" data-path="1-introduction.html"><a href="1-introduction.html#unique-molecular-identifiers"><i class="fa fa-check"></i><b>1.3.3</b> Unique Molecular Identifiers</a></li>
<li class="chapter" data-level="1.3.4" data-path="1-introduction.html"><a href="1-introduction.html#recent-advances-in-scrna-seq-protocols"><i class="fa fa-check"></i><b>1.3.4</b> Recent advances in scRNA-seq protocols</a></li>
</ul></li>
<li class="chapter" data-level="1.4" data-path="1-introduction.html"><a href="1-introduction.html#intro-analysis"><i class="fa fa-check"></i><b>1.4</b> Analysing scRNA-seq data</a><ul>
<li class="chapter" data-level="1.4.1" data-path="1-introduction.html"><a href="1-introduction.html#pre-processing-and-quality-control"><i class="fa fa-check"></i><b>1.4.1</b> Pre-processing and quality control</a></li>
<li class="chapter" data-level="1.4.2" data-path="1-introduction.html"><a href="1-introduction.html#normalisation-and-integration"><i class="fa fa-check"></i><b>1.4.2</b> Normalisation and integration</a></li>
<li class="chapter" data-level="1.4.3" data-path="1-introduction.html"><a href="1-introduction.html#grouping-cells"><i class="fa fa-check"></i><b>1.4.3</b> Grouping cells</a></li>
<li class="chapter" data-level="1.4.4" data-path="1-introduction.html"><a href="1-introduction.html#ordering-cells"><i class="fa fa-check"></i><b>1.4.4</b> Ordering cells</a></li>
<li class="chapter" data-level="1.4.5" data-path="1-introduction.html"><a href="1-introduction.html#gene-detection-and-interpretation"><i class="fa fa-check"></i><b>1.4.5</b> Gene detection and interpretation</a></li>
<li class="chapter" data-level="1.4.6" data-path="1-introduction.html"><a href="1-introduction.html#alternative-analyses"><i class="fa fa-check"></i><b>1.4.6</b> Alternative analyses</a></li>
<li class="chapter" data-level="1.4.7" data-path="1-introduction.html"><a href="1-introduction.html#evaluation-of-scrna-seq-analysis-methods"><i class="fa fa-check"></i><b>1.4.7</b> Evaluation of scRNA-seq analysis methods</a></li>
</ul></li>
<li class="chapter" data-level="1.5" data-path="1-introduction.html"><a href="1-introduction.html#intro-kidney"><i class="fa fa-check"></i><b>1.5</b> Kidney development</a><ul>
<li class="chapter" data-level="1.5.1" data-path="1-introduction.html"><a href="1-introduction.html#structure-and-function"><i class="fa fa-check"></i><b>1.5.1</b> Structure and function</a></li>
<li class="chapter" data-level="1.5.2" data-path="1-introduction.html"><a href="1-introduction.html#stages-of-development"><i class="fa fa-check"></i><b>1.5.2</b> Stages of development</a></li>
<li class="chapter" data-level="1.5.3" data-path="1-introduction.html"><a href="1-introduction.html#growing-kidney-organoids"><i class="fa fa-check"></i><b>1.5.3</b> Growing kidney organoids</a></li>
<li class="chapter" data-level="1.5.4" data-path="1-introduction.html"><a href="1-introduction.html#kidney-scrna-seq-studies"><i class="fa fa-check"></i><b>1.5.4</b> Kidney scRNA-seq studies</a></li>
</ul></li>
<li class="chapter" data-level="1.6" data-path="1-introduction.html"><a href="1-introduction.html#intro-overview"><i class="fa fa-check"></i><b>1.6</b> Thesis overview and aims</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="2-tools.html"><a href="2-tools.html"><i class="fa fa-check"></i><b>2</b> The scRNA-seq tools landscape</a><ul>
<li class="chapter" data-level="2.1" data-path="2-tools.html"><a href="2-tools.html#introduction-1"><i class="fa fa-check"></i><b>2.1</b> Introduction</a></li>
<li class="chapter" data-level="2.2" data-path="2-tools.html"><a href="2-tools.html#scrna-tools-publication"><i class="fa fa-check"></i><b>2.2</b> scRNA-tools publication</a></li>
<li class="chapter" data-level="2.3" data-path="2-tools.html"><a href="2-tools.html#tools-current"><i class="fa fa-check"></i><b>2.3</b> The current scRNA-tools database</a></li>
<li class="chapter" data-level="2.4" data-path="2-tools.html"><a href="2-tools.html#tools-usage"><i class="fa fa-check"></i><b>2.4</b> Usage of the scRNA-tools website</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="3-simulation.html"><a href="3-simulation.html"><i class="fa fa-check"></i><b>3</b> Simulating scRNA-seq data</a><ul>
<li class="chapter" data-level="3.1" data-path="3-simulation.html"><a href="3-simulation.html#introduction-2"><i class="fa fa-check"></i><b>3.1</b> Introduction</a></li>
<li class="chapter" data-level="3.2" data-path="3-simulation.html"><a href="3-simulation.html#splatter-publication"><i class="fa fa-check"></i><b>3.2</b> Splatter publication</a></li>
<li class="chapter" data-level="3.3" data-path="3-simulation.html"><a href="3-simulation.html#splatter-update"><i class="fa fa-check"></i><b>3.3</b> Updates to Splatter</a><ul>
<li class="chapter" data-level="3.3.1" data-path="3-simulation.html"><a href="3-simulation.html#performance-of-current-simulations"><i class="fa fa-check"></i><b>3.3.1</b> Performance of current simulations</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="4" data-path="4-clust-trees.html"><a href="4-clust-trees.html"><i class="fa fa-check"></i><b>4</b> Visualising clustering across resolutions</a><ul>
<li class="chapter" data-level="4.1" data-path="4-clust-trees.html"><a href="4-clust-trees.html#introduction-3"><i class="fa fa-check"></i><b>4.1</b> Introduction</a></li>
<li class="chapter" data-level="4.2" data-path="4-clust-trees.html"><a href="4-clust-trees.html#clustree-publication"><i class="fa fa-check"></i><b>4.2</b> Clustering trees publication</a></li>
<li class="chapter" data-level="4.3" data-path="4-clust-trees.html"><a href="4-clust-trees.html#clustree-overlay"><i class="fa fa-check"></i><b>4.3</b> Overlaying clustering trees</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="5-analysis.html"><a href="5-analysis.html"><i class="fa fa-check"></i><b>5</b> Analysis of kidney organoid scRNA-seq data</a><ul>
<li class="chapter" data-level="5.1" data-path="5-analysis.html"><a href="5-analysis.html#introduction-4"><i class="fa fa-check"></i><b>5.1</b> Introduction</a></li>
<li class="chapter" data-level="5.2" data-path="5-analysis.html"><a href="5-analysis.html#summary-of-published-work"><i class="fa fa-check"></i><b>5.2</b> Summary of published work</a></li>
<li class="chapter" data-level="5.3" data-path="5-analysis.html"><a href="5-analysis.html#outline-and-motivation"><i class="fa fa-check"></i><b>5.3</b> Outline and motivation</a></li>
<li class="chapter" data-level="5.4" data-path="5-analysis.html"><a href="5-analysis.html#pre-processing"><i class="fa fa-check"></i><b>5.4</b> Pre-processing</a><ul>
<li class="chapter" data-level="5.4.1" data-path="5-analysis.html"><a href="5-analysis.html#droplet-selection"><i class="fa fa-check"></i><b>5.4.1</b> Droplet selection</a></li>
<li class="chapter" data-level="5.4.2" data-path="5-analysis.html"><a href="5-analysis.html#alevin-comparison"><i class="fa fa-check"></i><b>5.4.2</b> Alevin comparison</a></li>
</ul></li>
<li class="chapter" data-level="5.5" data-path="5-analysis.html"><a href="5-analysis.html#quality-control"><i class="fa fa-check"></i><b>5.5</b> Quality control</a></li>
<li class="chapter" data-level="5.6" data-path="5-analysis.html"><a href="5-analysis.html#clustering"><i class="fa fa-check"></i><b>5.6</b> Clustering</a><ul>
<li class="chapter" data-level="5.6.1" data-path="5-analysis.html"><a href="5-analysis.html#gene-selection"><i class="fa fa-check"></i><b>5.6.1</b> Gene selection</a></li>
<li class="chapter" data-level="5.6.2" data-path="5-analysis.html"><a href="5-analysis.html#resolution-selection"><i class="fa fa-check"></i><b>5.6.2</b> Resolution selection</a></li>
<li class="chapter" data-level="5.6.3" data-path="5-analysis.html"><a href="5-analysis.html#cluster-validation"><i class="fa fa-check"></i><b>5.6.3</b> Cluster validation</a></li>
<li class="chapter" data-level="5.6.4" data-path="5-analysis.html"><a href="5-analysis.html#comparison-to-published-clusters"><i class="fa fa-check"></i><b>5.6.4</b> Comparison to published clusters</a></li>
</ul></li>
<li class="chapter" data-level="5.7" data-path="5-analysis.html"><a href="5-analysis.html#marker-genes"><i class="fa fa-check"></i><b>5.7</b> Marker gene detection</a></li>
<li class="chapter" data-level="5.8" data-path="5-analysis.html"><a href="5-analysis.html#connecting-clusters"><i class="fa fa-check"></i><b>5.8</b> Connecting clusters</a><ul>
<li class="chapter" data-level="5.8.1" data-path="5-analysis.html"><a href="5-analysis.html#partition-based-graph-abstraction"><i class="fa fa-check"></i><b>5.8.1</b> Partition-based graph abstraction</a></li>
<li class="chapter" data-level="5.8.2" data-path="5-analysis.html"><a href="5-analysis.html#cell-velocity"><i class="fa fa-check"></i><b>5.8.2</b> Cell velocity</a></li>
</ul></li>
<li class="chapter" data-level="5.9" data-path="5-analysis.html"><a href="5-analysis.html#discussion"><i class="fa fa-check"></i><b>5.9</b> Discussion</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="6-conclusion.html"><a href="6-conclusion.html"><i class="fa fa-check"></i><b>6</b> Conclusion</a></li>
<li class="chapter" data-level="" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i>References</a></li>
<li class="appendix"><span><b>Appendix</b></span></li>
<li class="chapter" data-level="A" data-path="A-splatter-additional.html"><a href="A-splatter-additional.html"><i class="fa fa-check"></i><b>A</b> Splatter additional files</a><ul>
<li class="chapter" data-level="A.1" data-path="A-splatter-additional.html"><a href="A-splatter-additional.html#additional-figures"><i class="fa fa-check"></i><b>A.1</b> Additional figures</a></li>
<li class="chapter" data-level="A.2" data-path="A-splatter-additional.html"><a href="A-splatter-additional.html#session-information"><i class="fa fa-check"></i><b>A.2</b> Session information</a></li>
</ul></li>
<li class="chapter" data-level="B" data-path="B-splatter-docs.html"><a href="B-splatter-docs.html"><i class="fa fa-check"></i><b>B</b> Splatter documentation</a><ul>
<li class="chapter" data-level="B.1" data-path="B-splatter-docs.html"><a href="B-splatter-docs.html#splatter-vignette"><i class="fa fa-check"></i><b>B.1</b> Splatter vignette</a></li>
<li class="chapter" data-level="B.2" data-path="B-splatter-docs.html"><a href="B-splatter-docs.html#splat-params-vignette"><i class="fa fa-check"></i><b>B.2</b> Splat parameters vignette</a></li>
<li class="chapter" data-level="B.3" data-path="B-splatter-docs.html"><a href="B-splatter-docs.html#splatter-manual"><i class="fa fa-check"></i><b>B.3</b> Splatter manual</a></li>
</ul></li>
<li class="chapter" data-level="C" data-path="C-simulation-comparison.html"><a href="C-simulation-comparison.html"><i class="fa fa-check"></i><b>C</b> Simulation comparison</a><ul>
<li class="chapter" data-level="C.1" data-path="C-simulation-comparison.html"><a href="C-simulation-comparison.html#sim-timings"><i class="fa fa-check"></i><b>C.1</b> Timings</a></li>
<li class="chapter" data-level="C.2" data-path="C-simulation-comparison.html"><a href="C-simulation-comparison.html#sim-packages"><i class="fa fa-check"></i><b>C.2</b> Package versions</a></li>
</ul></li>
<li class="chapter" data-level="D" data-path="D-clustree-docs.html"><a href="D-clustree-docs.html"><i class="fa fa-check"></i><b>D</b> clustree documentation</a><ul>
<li class="chapter" data-level="D.1" data-path="D-clustree-docs.html"><a href="D-clustree-docs.html#clustree-vignette"><i class="fa fa-check"></i><b>D.1</b> clustree vignette</a></li>
<li class="chapter" data-level="D.2" data-path="D-clustree-docs.html"><a href="D-clustree-docs.html#clustree-manual"><i class="fa fa-check"></i><b>D.2</b> clustree manual</a></li>
</ul></li>
<li class="chapter" data-level="E" data-path="E-organoid-paper.html"><a href="E-organoid-paper.html"><i class="fa fa-check"></i><b>E</b> Kidney organoid publication</a></li>
<li class="chapter" data-level="F" data-path="F-analysis-methods.html"><a href="F-analysis-methods.html"><i class="fa fa-check"></i><b>F</b> Analysis methods</a><ul>
<li class="chapter" data-level="F.1" data-path="F-analysis-methods.html"><a href="F-analysis-methods.html#pre-processing-1"><i class="fa fa-check"></i><b>F.1</b> Pre-processing</a><ul>
<li class="chapter" data-level="F.1.1" data-path="F-analysis-methods.html"><a href="F-analysis-methods.html#droplet-selection-1"><i class="fa fa-check"></i><b>F.1.1</b> Droplet selection</a></li>
<li class="chapter" data-level="F.1.2" data-path="F-analysis-methods.html"><a href="F-analysis-methods.html#alevin"><i class="fa fa-check"></i><b>F.1.2</b> Alevin</a></li>
</ul></li>
<li class="chapter" data-level="F.2" data-path="F-analysis-methods.html"><a href="F-analysis-methods.html#quality-control-1"><i class="fa fa-check"></i><b>F.2</b> Quality control</a></li>
<li class="chapter" data-level="F.3" data-path="F-analysis-methods.html"><a href="F-analysis-methods.html#clustering-1"><i class="fa fa-check"></i><b>F.3</b> Clustering</a><ul>
<li class="chapter" data-level="F.3.1" data-path="F-analysis-methods.html"><a href="F-analysis-methods.html#gene-selection-1"><i class="fa fa-check"></i><b>F.3.1</b> Gene selection</a></li>
<li class="chapter" data-level="F.3.2" data-path="F-analysis-methods.html"><a href="F-analysis-methods.html#graph-based-clustering"><i class="fa fa-check"></i><b>F.3.2</b> Graph-based clustering</a></li>
</ul></li>
<li class="chapter" data-level="F.4" data-path="F-analysis-methods.html"><a href="F-analysis-methods.html#marker-genes-1"><i class="fa fa-check"></i><b>F.4</b> Marker genes</a></li>
<li class="chapter" data-level="F.5" data-path="F-analysis-methods.html"><a href="F-analysis-methods.html#partition-based-graph-abstraction-paga"><i class="fa fa-check"></i><b>F.5</b> Partition-based graph abstraction (PAGA)</a></li>
<li class="chapter" data-level="F.6" data-path="F-analysis-methods.html"><a href="F-analysis-methods.html#cell-velocity-1"><i class="fa fa-check"></i><b>F.6</b> Cell velocity</a></li>
<li class="chapter" data-level="F.7" data-path="F-analysis-methods.html"><a href="F-analysis-methods.html#other-packages"><i class="fa fa-check"></i><b>F.7</b> Other packages</a></li>
</ul></li>
<li class="chapter" data-level="G" data-path="G-session-information-1.html"><a href="G-session-information-1.html"><i class="fa fa-check"></i><b>G</b> Session information</a><ul>
<li class="chapter" data-level="G.1" data-path="G-session-information-1.html"><a href="G-session-information-1.html#important-packages"><i class="fa fa-check"></i><b>G.1</b> Important packages</a></li>
<li class="chapter" data-level="G.2" data-path="G-session-information-1.html"><a href="G-session-information-1.html#full-session-information"><i class="fa fa-check"></i><b>G.2</b> Full session information</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Tools and techniques for single-cell RNA sequencing data</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="simulation" class="section level1">
<h1><span class="header-section-number">Chapter 3</span> Simulating scRNA-seq data</h1>
<blockquote>
<p><strong>“Essentially, all models are wrong, but some are useful.”</strong></p>
<p>— George E. P. Box and Norman R. Draper</p>
<p><em>Empirical Model-Building and Response Surfaces</em>, 1987</p>
</blockquote>
<p><img src="figures/03-simulation.png" width="100%" style="display: block; margin: auto;" /></p>
<div id="introduction-2" class="section level2">
<h2><span class="header-section-number">3.1</span> Introduction</h2>
<p>For a computational analysis method to be accepted and used by the community it
needs to demonstrate that is is effective at the task it aims to complete.
Ideally this can be achieved by evaluating the performance of the method on a
real dataset where the results are already known. Unfortunately, in many cases
such gold standard datasets are not available. This is particularly true for
genomic data where it is difficult to know what the truth is or the known truth
is limited to only small sections of the genome. It is possible to create some
genomic datasets where the truth is known, for example through carefully
performed mixing experiments <span class="citation">[175]</span>, but these often do not capture the
true biological complexity. In many cases the most effective way to evaluate an
analysis method during development is by testing it on simulated datasets.
Simulations have the additional advantage of being relatively cheap and easy to
produce, allowing exploration of a wide range of possible parameters. This is
the approach taken by many early methods for scRNA-seq analysis. Unfortunately,
these simulations were often not well explained, code for reproducing them was
not available and perhaps most importantly they didn’t show that the synthetic
datasets were similar to real scRNA-seq data.</p>
<p>This chapter presents Splatter, a software package for simulating scRNA-seq
datasets presented in a publication in <em>Genome Biology</em> <span class="citation">[201]</span>.
Splatter is designed to provide a consistent, easy-to-use interface for
multiple previously developed scRNA-seq simulation models. We do this by
providing a consistent framework for each model defined by two functions: one
which estimates parameters from a real dataset and a second that generates a
synthetic dataset using those parameters. Each simulation model has different
assumptions and reproduces different aspects of scRNA-seq data, and we explain
these differences in our publication. We also present Splat, our own simulation
model based on the Gamma-Poisson distribution. This model includes several
important aspects of scRNA-seq data including highly expressed outliers genes,
differences in library sizes between cells, a relationship between the mean and
the variance of each gene and the ability to add a dropout effect linked to
gene expression. When designing the Splat simulation our goal was to accurately
reproduce scRNA-seq count data rather than test a specific method, with the
result being that the model is highly flexible and able to generate a range of
scenarios including datasets with multiple groups of cells, batch effects and
continuous trajectories.</p>
<p>In the publication we compare how well each simulation reproduces a range of
scRNA-seq datasets, including UMI and full-length protocols, different capture
methods, and homogenous and complex tissues. We found that the Splat simulation
was a good match for some of these datasets across a range of methods, however
it was also clear that some models more faithfully reproduced different aspects
of the data, particularly for datasets from different sources. The Splatter R
package is available for download from Bioconductor
(<a href="https://bioconductor.org/packages/splatter" class="uri">https://bioconductor.org/packages/splatter</a>). Additional files for the
publication are provided in <strong>Appendix <a href="A-splatter-additional.html#splatter-additional">A</a></strong>. Following
the publication (<strong>Section <a href="3-simulation.html#splatter-publication">3.2</a></strong>) I discuss updates to
the software package and provide an updated comparison of the current
simulation models (<strong>Section <a href="3-simulation.html#splatter-update">3.3</a></strong>). Vignettes describing
the use of Splatter (<strong>Appendix <a href="B-splatter-docs.html#splatter-vignette">B.1</a></strong>) and the parameters
of the Splat model (<strong>Appendix <a href="B-splatter-docs.html#splat-params-vignette">B.2</a></strong>) are included as
appendices, as is the full Splatter manual (<strong>Appendix
<a href="B-splatter-docs.html#splatter-manual">B.3</a></strong>).</p>
</div>
<div id="splatter-publication" class="section level2">
<h2><span class="header-section-number">3.2</span> Splatter publication</h2>
<p>You can read the Splatter paper <a href="https://genomebiology.biomedcentral.com/articles/10.1186/s13059-017-1305-0">here</a>.</p>
</div>
<div id="splatter-update" class="section level2">
<h2><span class="header-section-number">3.3</span> Updates to Splatter</h2>
<p>I have continued to develop the Splatter package since publication, improving
the package infrastructure and implementing additional simulation models. This
section briefly describes changes to the software since the publication and
provides an up-to-date comparison of the current simulation models.</p>
<p>The most significant change to the infrastructure of Splatter is switching from
using the SCESet object provided by early versions of the scater package as the
output of simulation functions, to using the SingleCellExperiment object
<span class="citation">[202]</span>. This object has been available since mid-2017 and is intended to
be the core object for all Bioconductor packages working with scRNA-seq data.
It is based on Bioconductor’s SummarizedExperiment object which consists of
feature by sample matrices linked to annotation data frames for both features
and samples. The SingleCellExperiment object adds features specifically useful
for scRNA-seq data, including slots for storing the results of dimensionality
reduction techniques and normalisation size factors, as well as conveniently
named accessors for information commonly used in analyses. In practical terms,
changing objects has not changed how Splatter is used but it does ensure that
the results of Splatter simulations are immediately compatible with tools in
the Bioconductor ecosystem. Display of the Splatter parameters objects has also
been improved, including colouring of output to highlight which parameters have
been changed from the default values.</p>
<p>Minor changes have also been made to the Splat simulation model. At the request
of users we have modified the parameters controlling additional dropout to
allow them to be specified by whole experiment (the previous method), sample
batch, cell group or individual cell. This allows users investigating the
effects of dropout to simulate datasets where different cell types or batches
are differently affected by the dropout phenomenon. Many of the other
parameters could already be specified at different levels, but extending this to
the dropout parameters provides extra flexibility to the Splat model. In the
original model the total number of reads in each cell was sampled from a
log-normal distribution, however some users have come across datasets where the
normal distribution was a better fit, which caused the Splat estimation
procedure to fail. We now allow either the log-normal or normal distributions
to be used for this part of the model and the estimation procedure attempts to
identify which is more appropriate for each input dataset.</p>
<p>Another model that has seen changes is that from the BASiCS package. The
authors of BASiCS have made significant improvements to how they estimate
parameters, particularly for datasets without internal spike-ins but where
multiple batches exist. BASiCS now uses non-linear regression to capture the
trend between gene mean and over-dispersion <span class="citation">[110]</span>. This method is now
used as part of the default estimation procedure for the BASiCS model within
Splatter, resulting in simulations that are a much better match for the
original dataset.</p>
<p>I have also added four new simulation models to Splatter, bringing the number of
models to 10. These new models are:</p>
<ul>
<li>mfa – Simulation of a bifurcating pseudotime trajectory. This simulation can
optionally include genes with changes in expression along the trajectory and
added dropout <span class="citation">[203]</span>.</li>
<li>PhenoPath – Pseudotime trajectory including genes that show differential
expression, changes along pseudotime, interaction effects or combinations of
these <span class="citation">[204]</span>.</li>
<li>ZINB-WaVE – Sophisticated zero-inflated negative-binomial distribution based
simulation that includes cell and gene-level covariates <span class="citation">[63]</span>.</li>
<li>SparseDC – Simulates a set of clusters across two conditions, where
some clusters may be present in only one condition <span class="citation">[205]</span>.</li>
</ul>
<p>More details about the models used by these packages are available in the
associated publications and package documentation.</p>
<div id="performance-of-current-simulations" class="section level3">
<h3><span class="header-section-number">3.3.1</span> Performance of current simulations</h3>

<p>To demonstrate the performance of the current simulation models I have repeated
some of the analysis from the publication using the Tung dataset
<span class="citation">[206]</span>. Parameters for each of the models were estimated from a sample
of 500 cells from the Tung data and synthetic datasets were produced for each
model using these parameters. The time taken for the estimation and simulation
steps is shown in <strong>Appendix <a href="C-simulation-comparison.html#sim-timings">C.1</a></strong> and versions of the packages
used in <strong>Appendix <a href="C-simulation-comparison.html#sim-packages">C.2</a></strong>. The comparison functions within
Splatter were then used to evaluate the differences between the simulated
datasets and the real data they were based on. Figure
<a href="3-simulation.html#fig:simulation-summary">3.1</a> summarises this evaluation by showing the ranking
of the median absolute deviation (MAD) for each model across a range of
metrics.</p>
<div class="figure" style="text-align: center"><span id="fig:simulation-summary"></span>
<img src="figures/03-sim-summary.png" alt="Performance of current Splatter simulations. Synthetic datasets were produced using parameters estimated from 500 random cells from the Tung dataset. Plot shows the rank of median absolute deviation (MAD) from the real dataset on a variety of measures from least similar (blue) to most similar (yellow). The ZINB-WaVE model best reproduces this dataset, followed by the updated BASiCS model. The Splat model also performs well despite having a less sophisticated estimation procedure." width="100%" />
<p class="caption">
Figure 3.1: Performance of current Splatter simulations. Synthetic datasets were produced using parameters estimated from 500 random cells from the Tung dataset. Plot shows the rank of median absolute deviation (MAD) from the real dataset on a variety of measures from least similar (blue) to most similar (yellow). The ZINB-WaVE model best reproduces this dataset, followed by the updated BASiCS model. The Splat model also performs well despite having a less sophisticated estimation procedure.
</p>
</div>


<p>This comparison clearly shows that the ZINB-WaVE model produces the closest
match to this dataset. The similarity to the real dataset suggests that the
model used by ZINB-WaVE, and particularly the procedure used to estimate the
parameters in it, is effective for scRNA-seq data. ZINB-WaVE was originally
designed for dimensionality reduction but also has applications in
normalisation and differential expression testing <span class="citation">[64]</span>. In
contrast to the previously published analysis, the BASiCS model is also a very
good match, with the improvement in performance coming from the new
regression-based estimation method. Unfortunately, the BASiCS normalisation
method relies on knowing predefined groups of cells, which limits its
usefulness for exploratory experiments. The Splat method does not match the
performance of ZINB-WaVE and BASiCS but still does a reasonable job of
reproducing this dataset, despite having a much simpler (and quicker)
estimation procedure. While producing realistic datasets is extremely
important, the real strength of the Splat model lies in its flexibility to
simulate many kinds of experiment and its reporting of intermediate values that
can be used to evaluate methods. As observed in the publication, we again see
that the zero-inflated versions of the Splat and Lun2 models perform worse than
the regular versions on this UMI dataset. However, this is in contrast to the
extremely good performance of ZINB-WaVE which also uses a zero-inflated model.
Taken together these results suggest that both regular and zero-inflated models
can be effective for this type of data, if the parameters are estimated
correctly.</p>
<p>The other three new simulation models (mfa, PhenoPath and SparseDC) do not do a
particularly good job of reproducing this dataset, and in particular mfa
produces a dataset with a very different number of counts per cell. These
simulations have mostly been designed to demonstrate the analysis methods in
these packages and the estimation procedures associated with them are
relatively simple. This result underlines that the key to producing a realistic
simulation of any kind of data lies not so much in the simulation model itself
but in choosing good parameters for that model. As we describe in the
publication, there are caveats around the estimation procedures for some of the
other models, specifically the Lun model, which only estimates the number of
genes and cells, and the scDD model, which is designed for a high-quality
filtered dataset unlike the raw dataset used here. Another important
consideration is that the comparison shown here is based on a single dataset.
As we showed in the publication, different datasets have different biological
and technical features which may be reproduced more accurately by alternative
simulation models.</p>
<p>Comparing simulations and real datasets is a difficult task. For the
comparisons in the Splatter publication we focused on simple summary statistics
such as mean gene expression or percentage of zeros per cell and the
relationships between them. However, there are other measures that could be
considered such as correlation between features or estimated dispersions.
Statistical tests can also be used to formally evaluate the similarity of
datasets. The countsimQC package <span class="citation">[207]</span> can be used to produce a
comprehensive report comparing count datasets in a variety of ways and could be
a useful tool for any future evaluations of simulation models.</p>

</div>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="2-tools.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="4-clust-trees.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"google": false,
"linkedin": false,
"weibo": false,
"instapper": false,
"vk": false,
"all": ["facebook", "google", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"download": [["thesis.pdf", "PDF"], ["thesis.docx", "Word"]],
"toc": {
"collapse": "section"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://cdn.bootcss.com/mathjax/2.7.1/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:" && /^https?:/.test(src))
      src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
