<!DOCTYPE html>
<html >

<head>

  <meta charset="UTF-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <title>Tools and techniques for single-cell RNA sequencing data</title>
  <meta name="description" content="Tools and techniques for single-cell RNA sequencing data">
  <meta name="generator" content="bookdown 0.7 and GitBook 2.6.7">

  <meta property="og:title" content="Tools and techniques for single-cell RNA sequencing data" />
  <meta property="og:type" content="book" />
  
  
  
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Tools and techniques for single-cell RNA sequencing data" />
  
  
  

<meta name="author" content="Luke Zappia">



  <meta name="viewport" content="width=device-width, initial-scale=1">
  <meta name="apple-mobile-web-app-capable" content="yes">
  <meta name="apple-mobile-web-app-status-bar-style" content="black">
  
  

<link rel="next" href="1-introduction.html">
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />










</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./"></a></li>
<li class="divider"></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Front matter</a><ul>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#abstract"><i class="fa fa-check"></i>Abstract</a></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#declaration"><i class="fa fa-check"></i>Declaration</a></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#preface"><i class="fa fa-check"></i>Preface</a></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#acknowledgements"><i class="fa fa-check"></i>Acknowledgements</a></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#list-of-copyright"><i class="fa fa-check"></i>List of copyright</a></li>
</ul></li>
<li class="chapter" data-level="1" data-path="1-introduction.html"><a href="1-introduction.html"><i class="fa fa-check"></i><b>1</b> Introduction</a><ul>
<li class="chapter" data-level="1.1" data-path="1-introduction.html"><a href="1-introduction.html#intro-dogma"><i class="fa fa-check"></i><b>1.1</b> The central dogma</a></li>
<li class="chapter" data-level="1.2" data-path="1-introduction.html"><a href="1-introduction.html#intro-RNAseq"><i class="fa fa-check"></i><b>1.2</b> RNA sequencing</a><ul>
<li class="chapter" data-level="1.2.1" data-path="1-introduction.html"><a href="1-introduction.html#library-preparation"><i class="fa fa-check"></i><b>1.2.1</b> Library preparation</a></li>
<li class="chapter" data-level="1.2.2" data-path="1-introduction.html"><a href="1-introduction.html#high-throughput-sequencing"><i class="fa fa-check"></i><b>1.2.2</b> High-throughput sequencing</a></li>
<li class="chapter" data-level="1.2.3" data-path="1-introduction.html"><a href="1-introduction.html#analysis-of-rna-seq-data"><i class="fa fa-check"></i><b>1.2.3</b> Analysis of RNA-seq data</a></li>
</ul></li>
<li class="chapter" data-level="1.3" data-path="1-introduction.html"><a href="1-introduction.html#intro-scRNAseq"><i class="fa fa-check"></i><b>1.3</b> Single-cell RNA sequencing</a><ul>
<li class="chapter" data-level="1.3.1" data-path="1-introduction.html"><a href="1-introduction.html#early-single-cell-capture-technologies"><i class="fa fa-check"></i><b>1.3.1</b> Early single-cell capture technologies</a></li>
<li class="chapter" data-level="1.3.2" data-path="1-introduction.html"><a href="1-introduction.html#droplet-based-cell-capture"><i class="fa fa-check"></i><b>1.3.2</b> Droplet-based cell capture</a></li>
<li class="chapter" data-level="1.3.3" data-path="1-introduction.html"><a href="1-introduction.html#unique-molecular-identifiers"><i class="fa fa-check"></i><b>1.3.3</b> Unique Molecular Identifiers</a></li>
<li class="chapter" data-level="1.3.4" data-path="1-introduction.html"><a href="1-introduction.html#recent-advances-in-scrna-seq-protocols"><i class="fa fa-check"></i><b>1.3.4</b> Recent advances in scRNA-seq protocols</a></li>
</ul></li>
<li class="chapter" data-level="1.4" data-path="1-introduction.html"><a href="1-introduction.html#intro-analysis"><i class="fa fa-check"></i><b>1.4</b> Analysing scRNA-seq data</a><ul>
<li class="chapter" data-level="1.4.1" data-path="1-introduction.html"><a href="1-introduction.html#pre-processing-and-quality-control"><i class="fa fa-check"></i><b>1.4.1</b> Pre-processing and quality control</a></li>
<li class="chapter" data-level="1.4.2" data-path="1-introduction.html"><a href="1-introduction.html#normalisation-and-integration"><i class="fa fa-check"></i><b>1.4.2</b> Normalisation and integration</a></li>
<li class="chapter" data-level="1.4.3" data-path="1-introduction.html"><a href="1-introduction.html#grouping-cells"><i class="fa fa-check"></i><b>1.4.3</b> Grouping cells</a></li>
<li class="chapter" data-level="1.4.4" data-path="1-introduction.html"><a href="1-introduction.html#ordering-cells"><i class="fa fa-check"></i><b>1.4.4</b> Ordering cells</a></li>
<li class="chapter" data-level="1.4.5" data-path="1-introduction.html"><a href="1-introduction.html#gene-detection-and-interpretation"><i class="fa fa-check"></i><b>1.4.5</b> Gene detection and interpretation</a></li>
<li class="chapter" data-level="1.4.6" data-path="1-introduction.html"><a href="1-introduction.html#alternative-analyses"><i class="fa fa-check"></i><b>1.4.6</b> Alternative analyses</a></li>
<li class="chapter" data-level="1.4.7" data-path="1-introduction.html"><a href="1-introduction.html#evaluation-of-scrna-seq-analysis-methods"><i class="fa fa-check"></i><b>1.4.7</b> Evaluation of scRNA-seq analysis methods</a></li>
</ul></li>
<li class="chapter" data-level="1.5" data-path="1-introduction.html"><a href="1-introduction.html#intro-kidney"><i class="fa fa-check"></i><b>1.5</b> Kidney development</a><ul>
<li class="chapter" data-level="1.5.1" data-path="1-introduction.html"><a href="1-introduction.html#structure-and-function"><i class="fa fa-check"></i><b>1.5.1</b> Structure and function</a></li>
<li class="chapter" data-level="1.5.2" data-path="1-introduction.html"><a href="1-introduction.html#stages-of-development"><i class="fa fa-check"></i><b>1.5.2</b> Stages of development</a></li>
<li class="chapter" data-level="1.5.3" data-path="1-introduction.html"><a href="1-introduction.html#growing-kidney-organoids"><i class="fa fa-check"></i><b>1.5.3</b> Growing kidney organoids</a></li>
<li class="chapter" data-level="1.5.4" data-path="1-introduction.html"><a href="1-introduction.html#kidney-scrna-seq-studies"><i class="fa fa-check"></i><b>1.5.4</b> Kidney scRNA-seq studies</a></li>
</ul></li>
<li class="chapter" data-level="1.6" data-path="1-introduction.html"><a href="1-introduction.html#intro-overview"><i class="fa fa-check"></i><b>1.6</b> Thesis overview and aims</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="2-tools.html"><a href="2-tools.html"><i class="fa fa-check"></i><b>2</b> The scRNA-seq tools landscape</a><ul>
<li class="chapter" data-level="2.1" data-path="2-tools.html"><a href="2-tools.html#introduction-1"><i class="fa fa-check"></i><b>2.1</b> Introduction</a></li>
<li class="chapter" data-level="2.2" data-path="2-tools.html"><a href="2-tools.html#scrna-tools-publication"><i class="fa fa-check"></i><b>2.2</b> scRNA-tools publication</a></li>
<li class="chapter" data-level="2.3" data-path="2-tools.html"><a href="2-tools.html#tools-current"><i class="fa fa-check"></i><b>2.3</b> The current scRNA-tools database</a></li>
<li class="chapter" data-level="2.4" data-path="2-tools.html"><a href="2-tools.html#tools-usage"><i class="fa fa-check"></i><b>2.4</b> Usage of the scRNA-tools website</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="3-simulation.html"><a href="3-simulation.html"><i class="fa fa-check"></i><b>3</b> Simulating scRNA-seq data</a><ul>
<li class="chapter" data-level="3.1" data-path="3-simulation.html"><a href="3-simulation.html#introduction-2"><i class="fa fa-check"></i><b>3.1</b> Introduction</a></li>
<li class="chapter" data-level="3.2" data-path="3-simulation.html"><a href="3-simulation.html#splatter-publication"><i class="fa fa-check"></i><b>3.2</b> Splatter publication</a></li>
<li class="chapter" data-level="3.3" data-path="3-simulation.html"><a href="3-simulation.html#splatter-update"><i class="fa fa-check"></i><b>3.3</b> Updates to Splatter</a><ul>
<li class="chapter" data-level="3.3.1" data-path="3-simulation.html"><a href="3-simulation.html#performance-of-current-simulations"><i class="fa fa-check"></i><b>3.3.1</b> Performance of current simulations</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="4" data-path="4-clust-trees.html"><a href="4-clust-trees.html"><i class="fa fa-check"></i><b>4</b> Visualising clustering across resolutions</a><ul>
<li class="chapter" data-level="4.1" data-path="4-clust-trees.html"><a href="4-clust-trees.html#introduction-3"><i class="fa fa-check"></i><b>4.1</b> Introduction</a></li>
<li class="chapter" data-level="4.2" data-path="4-clust-trees.html"><a href="4-clust-trees.html#clustree-publication"><i class="fa fa-check"></i><b>4.2</b> Clustering trees publication</a></li>
<li class="chapter" data-level="4.3" data-path="4-clust-trees.html"><a href="4-clust-trees.html#clustree-overlay"><i class="fa fa-check"></i><b>4.3</b> Overlaying clustering trees</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="5-analysis.html"><a href="5-analysis.html"><i class="fa fa-check"></i><b>5</b> Analysis of kidney organoid scRNA-seq data</a><ul>
<li class="chapter" data-level="5.1" data-path="5-analysis.html"><a href="5-analysis.html#introduction-4"><i class="fa fa-check"></i><b>5.1</b> Introduction</a></li>
<li class="chapter" data-level="5.2" data-path="5-analysis.html"><a href="5-analysis.html#summary-of-published-work"><i class="fa fa-check"></i><b>5.2</b> Summary of published work</a></li>
<li class="chapter" data-level="5.3" data-path="5-analysis.html"><a href="5-analysis.html#outline-and-motivation"><i class="fa fa-check"></i><b>5.3</b> Outline and motivation</a></li>
<li class="chapter" data-level="5.4" data-path="5-analysis.html"><a href="5-analysis.html#pre-processing"><i class="fa fa-check"></i><b>5.4</b> Pre-processing</a><ul>
<li class="chapter" data-level="5.4.1" data-path="5-analysis.html"><a href="5-analysis.html#droplet-selection"><i class="fa fa-check"></i><b>5.4.1</b> Droplet selection</a></li>
<li class="chapter" data-level="5.4.2" data-path="5-analysis.html"><a href="5-analysis.html#alevin-comparison"><i class="fa fa-check"></i><b>5.4.2</b> Alevin comparison</a></li>
</ul></li>
<li class="chapter" data-level="5.5" data-path="5-analysis.html"><a href="5-analysis.html#quality-control"><i class="fa fa-check"></i><b>5.5</b> Quality control</a></li>
<li class="chapter" data-level="5.6" data-path="5-analysis.html"><a href="5-analysis.html#clustering"><i class="fa fa-check"></i><b>5.6</b> Clustering</a><ul>
<li class="chapter" data-level="5.6.1" data-path="5-analysis.html"><a href="5-analysis.html#gene-selection"><i class="fa fa-check"></i><b>5.6.1</b> Gene selection</a></li>
<li class="chapter" data-level="5.6.2" data-path="5-analysis.html"><a href="5-analysis.html#resolution-selection"><i class="fa fa-check"></i><b>5.6.2</b> Resolution selection</a></li>
<li class="chapter" data-level="5.6.3" data-path="5-analysis.html"><a href="5-analysis.html#cluster-validation"><i class="fa fa-check"></i><b>5.6.3</b> Cluster validation</a></li>
<li class="chapter" data-level="5.6.4" data-path="5-analysis.html"><a href="5-analysis.html#comparison-to-published-clusters"><i class="fa fa-check"></i><b>5.6.4</b> Comparison to published clusters</a></li>
</ul></li>
<li class="chapter" data-level="5.7" data-path="5-analysis.html"><a href="5-analysis.html#marker-genes"><i class="fa fa-check"></i><b>5.7</b> Marker gene detection</a></li>
<li class="chapter" data-level="5.8" data-path="5-analysis.html"><a href="5-analysis.html#connecting-clusters"><i class="fa fa-check"></i><b>5.8</b> Connecting clusters</a><ul>
<li class="chapter" data-level="5.8.1" data-path="5-analysis.html"><a href="5-analysis.html#partition-based-graph-abstraction"><i class="fa fa-check"></i><b>5.8.1</b> Partition-based graph abstraction</a></li>
<li class="chapter" data-level="5.8.2" data-path="5-analysis.html"><a href="5-analysis.html#cell-velocity"><i class="fa fa-check"></i><b>5.8.2</b> Cell velocity</a></li>
</ul></li>
<li class="chapter" data-level="5.9" data-path="5-analysis.html"><a href="5-analysis.html#discussion"><i class="fa fa-check"></i><b>5.9</b> Discussion</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="6-conclusion.html"><a href="6-conclusion.html"><i class="fa fa-check"></i><b>6</b> Conclusion</a></li>
<li class="chapter" data-level="" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i>References</a></li>
<li class="appendix"><span><b>Appendix</b></span></li>
<li class="chapter" data-level="A" data-path="A-splatter-additional.html"><a href="A-splatter-additional.html"><i class="fa fa-check"></i><b>A</b> Splatter additional files</a><ul>
<li class="chapter" data-level="A.1" data-path="A-splatter-additional.html"><a href="A-splatter-additional.html#additional-figures"><i class="fa fa-check"></i><b>A.1</b> Additional figures</a></li>
<li class="chapter" data-level="A.2" data-path="A-splatter-additional.html"><a href="A-splatter-additional.html#session-information"><i class="fa fa-check"></i><b>A.2</b> Session information</a></li>
</ul></li>
<li class="chapter" data-level="B" data-path="B-splatter-docs.html"><a href="B-splatter-docs.html"><i class="fa fa-check"></i><b>B</b> Splatter documentation</a><ul>
<li class="chapter" data-level="B.1" data-path="B-splatter-docs.html"><a href="B-splatter-docs.html#splatter-vignette"><i class="fa fa-check"></i><b>B.1</b> Splatter vignette</a></li>
<li class="chapter" data-level="B.2" data-path="B-splatter-docs.html"><a href="B-splatter-docs.html#splat-params-vignette"><i class="fa fa-check"></i><b>B.2</b> Splat parameters vignette</a></li>
<li class="chapter" data-level="B.3" data-path="B-splatter-docs.html"><a href="B-splatter-docs.html#splatter-manual"><i class="fa fa-check"></i><b>B.3</b> Splatter manual</a></li>
</ul></li>
<li class="chapter" data-level="C" data-path="C-simulation-comparison.html"><a href="C-simulation-comparison.html"><i class="fa fa-check"></i><b>C</b> Simulation comparison</a><ul>
<li class="chapter" data-level="C.1" data-path="C-simulation-comparison.html"><a href="C-simulation-comparison.html#sim-timings"><i class="fa fa-check"></i><b>C.1</b> Timings</a></li>
<li class="chapter" data-level="C.2" data-path="C-simulation-comparison.html"><a href="C-simulation-comparison.html#sim-packages"><i class="fa fa-check"></i><b>C.2</b> Package versions</a></li>
</ul></li>
<li class="chapter" data-level="D" data-path="D-clustree-docs.html"><a href="D-clustree-docs.html"><i class="fa fa-check"></i><b>D</b> clustree documentation</a><ul>
<li class="chapter" data-level="D.1" data-path="D-clustree-docs.html"><a href="D-clustree-docs.html#clustree-vignette"><i class="fa fa-check"></i><b>D.1</b> clustree vignette</a></li>
<li class="chapter" data-level="D.2" data-path="D-clustree-docs.html"><a href="D-clustree-docs.html#clustree-manual"><i class="fa fa-check"></i><b>D.2</b> clustree manual</a></li>
</ul></li>
<li class="chapter" data-level="E" data-path="E-organoid-paper.html"><a href="E-organoid-paper.html"><i class="fa fa-check"></i><b>E</b> Kidney organoid publication</a></li>
<li class="chapter" data-level="F" data-path="F-analysis-methods.html"><a href="F-analysis-methods.html"><i class="fa fa-check"></i><b>F</b> Analysis methods</a><ul>
<li class="chapter" data-level="F.1" data-path="F-analysis-methods.html"><a href="F-analysis-methods.html#pre-processing-1"><i class="fa fa-check"></i><b>F.1</b> Pre-processing</a><ul>
<li class="chapter" data-level="F.1.1" data-path="F-analysis-methods.html"><a href="F-analysis-methods.html#droplet-selection-1"><i class="fa fa-check"></i><b>F.1.1</b> Droplet selection</a></li>
<li class="chapter" data-level="F.1.2" data-path="F-analysis-methods.html"><a href="F-analysis-methods.html#alevin"><i class="fa fa-check"></i><b>F.1.2</b> Alevin</a></li>
</ul></li>
<li class="chapter" data-level="F.2" data-path="F-analysis-methods.html"><a href="F-analysis-methods.html#quality-control-1"><i class="fa fa-check"></i><b>F.2</b> Quality control</a></li>
<li class="chapter" data-level="F.3" data-path="F-analysis-methods.html"><a href="F-analysis-methods.html#clustering-1"><i class="fa fa-check"></i><b>F.3</b> Clustering</a><ul>
<li class="chapter" data-level="F.3.1" data-path="F-analysis-methods.html"><a href="F-analysis-methods.html#gene-selection-1"><i class="fa fa-check"></i><b>F.3.1</b> Gene selection</a></li>
<li class="chapter" data-level="F.3.2" data-path="F-analysis-methods.html"><a href="F-analysis-methods.html#graph-based-clustering"><i class="fa fa-check"></i><b>F.3.2</b> Graph-based clustering</a></li>
</ul></li>
<li class="chapter" data-level="F.4" data-path="F-analysis-methods.html"><a href="F-analysis-methods.html#marker-genes-1"><i class="fa fa-check"></i><b>F.4</b> Marker genes</a></li>
<li class="chapter" data-level="F.5" data-path="F-analysis-methods.html"><a href="F-analysis-methods.html#partition-based-graph-abstraction-paga"><i class="fa fa-check"></i><b>F.5</b> Partition-based graph abstraction (PAGA)</a></li>
<li class="chapter" data-level="F.6" data-path="F-analysis-methods.html"><a href="F-analysis-methods.html#cell-velocity-1"><i class="fa fa-check"></i><b>F.6</b> Cell velocity</a></li>
<li class="chapter" data-level="F.7" data-path="F-analysis-methods.html"><a href="F-analysis-methods.html#other-packages"><i class="fa fa-check"></i><b>F.7</b> Other packages</a></li>
</ul></li>
<li class="chapter" data-level="G" data-path="G-session-information-1.html"><a href="G-session-information-1.html"><i class="fa fa-check"></i><b>G</b> Session information</a><ul>
<li class="chapter" data-level="G.1" data-path="G-session-information-1.html"><a href="G-session-information-1.html#important-packages"><i class="fa fa-check"></i><b>G.1</b> Important packages</a></li>
<li class="chapter" data-level="G.2" data-path="G-session-information-1.html"><a href="G-session-information-1.html#full-session-information"><i class="fa fa-check"></i><b>G.2</b> Full session information</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Tools and techniques for single-cell RNA sequencing data</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="header">
<h1 class="title">Tools and techniques for single-cell RNA sequencing data</h1>
<h4 class="author"><em>Luke Zappia</em></h4>
</div>
<div id="front-matter" class="section level1 unnumbered">
<h1>Front matter</h1>

<!-- This document is used to add the front matter to formats other than LaTeX
(mostly word). These sections can be removed if you don't want them or the whole
file excluded from _bookdown.yml.--->
<div id="abstract" class="section level2 unnumbered">
<h2>Abstract</h2>
<p>RNA sequencing of individual cells allows us to take a snapshot of the dynamic
processes within a cell and explore differences between cell types. As this
technology has developed over the last few years it has been rapidly adopted by
researchers in areas such as developmental biology, and many single-cell RNA
sequencing datasets are now available. Coinciding with the development of
protocols for producing single-cell RNA sequencing data there has been a
simultaneous burst in the development of computational analysis methods. My
thesis explores the computational tools and techniques for analysing
single-cell RNA sequencing data. I present a database that charts the release
of analysis software, where it has been published and what it can be used for,
as well as a website that makes this information publicly available. I also
present two of my own tools and techniques including Splatter, a software
package for easily simulating single-cell datasets from multiple models, and
clustering trees, a visualisation approach for inspecting clustering at
multiple resolutions. In the final part of my thesis I perform analysis of a
dataset from kidney organoids to demonstrate and compare some current analysis
methods. Taken together, my thesis covers many aspects of the tools and
techniques for single-cell RNA sequencing by describing the approaches that are
available, presenting software that can help in developing and evaluating
methods, introducing an approach for aiding one of the most common analysis
tasks, and showing how tools can be used to extract meaning from a real
dataset.</p>
</div>
<div id="declaration" class="section level2 unnumbered">
<h2>Declaration</h2>
<p>This is to certify that:</p>
<ol style="list-style-type: lower-roman">
<li><p>this thesis comprises only my original work towards the degree of Doctor of
Philosophy except where indicated in the preface;</p></li>
<li><p>due acknowledgement has been made in the text to all other material
used; and</p></li>
<li><p>this thesis is fewer than the 100,000 words in length, exclusive of tables,
maps, bibliographies and appendices</p></li>
</ol>
<p><!-- Signature line will be placed here --></p>
</div>
<div id="preface" class="section level2 unnumbered">
<h2>Preface</h2>
<p>This preface provides a summary of the chapters in this thesis and describes my
contribution to them as well as the contributions of my collaborators and
supervisors. This is a thesis <em>with</em> publication and where publications form
part of a chapter they are listed here. The following publications are included
as part of this thesis:</p>
<p><strong>Zappia L</strong>, Phipson B, Oshlack A. “Exploring the single-cell RNA-seq
analysis landscape with the scRNA-tools database.” <em>PLoS Computational
Biology.</em> 2018. DOI:
<a href="https://doi.org/10.1371/journal.pcbi.1006245">10.1371/journal.pcbi.1006245</a></p>
<p><strong>Zappia L</strong>, Phipson B, Oshlack A. “Splatter: simulation of single-cell RNA
sequencing data.” <em>Genome Biology.</em> 2017. DOI:
<a href="https://doi.org/10.1186/s13059-017-1305-0">10.1186/s13059-017-1305-0</a></p>
<p><strong>Zappia L</strong>, Oshlack A. “Clustering trees: a visualization for evaluating
clusterings at multiple resolutions.” <em>GigaScience.</em> 2018. DOI:
<a href="https://doi.org/10.1093/gigascience/giy083">10.1093/gigascience/giy083</a></p>
<p>These publications are included as they appear online and are designed to be
read as stand-alone documents. Sections within these publications are not
included in the table of contents, and references are available at the end of
each publication rather than in the reference list for this thesis. The
contributions of authors to these papers are explained below. I am the first
author on these publications and contributed more than 50 percent of the work
towards them including drafting, editing and revising the manuscripts. My
co-authors have provided signed declarations acknowledging and supporting my
contributions which have been submitted along with this thesis. Where publicly
available datasets have been used these have been appropriately cited.</p>
<p><strong><a href="1-introduction.html#introduction">Chapter 1: Introduction</a></strong> is an original work providing a
background and overview relevant to understanding my work in this thesis
including an introduction to RNA sequencing, single-cell RNA sequencing and
kidney function and development.</p>
<p><strong><a href="2-tools.html#tools">Chapter 2: The scRNA-seq tools landscape</a></strong> is an original work
describing a database of software tools for analysing single-cell RNA
sequencing data which has been published in <em>PLoS Computational Biology</em> as
“Exploring the single-cell RNA-seq analysis landscape with the scRNA-tools
database”. In addition to this publication I developed a website displaying
the information in this database which is available at
<a href="https://www.scRNA-tools.org" class="uri">https://www.scRNA-tools.org</a>. The database and code for building the website is
available on GitHub at <a href="https://github.com/Oshlack/scRNA-tools" class="uri">https://github.com/Oshlack/scRNA-tools</a> under an MIT
license.</p>
<p>Contributions to the work in this chapter:</p>
<ul>
<li><p>I compiled and regularly updated the database of tools.</p></li>
<li><p>I designed and built the public website used to display the database. Breon
Schmidt provided assistance with implementing some of the website
functionality. Some of the code for processing the database was based on a
script written by Sean Davis.</p></li>
<li><p>I performed the analysis of the database presented in the publication.</p></li>
<li><p>I wrote the first draft of the manuscript and produced all the figures in the
publication.</p></li>
<li><p>Alicia Oshlack provided advice on planning the manuscript and edited draft
versions.</p></li>
<li><p>Belinda Phipson contributed to writing the manuscript.</p></li>
</ul>
<p><strong><a href="3-simulation.html#simulation">Chapter 3: Simulating scRNA-seq data</a></strong> is an original work
describing a software package for simulating single-cell RNA sequencing
expression data. This work was published in <em>Genome Biology</em> as “Splatter:
simulation of single-cell RNA sequencing data”. The software package described
in this publication is available through Bioconductor at
<a href="https://bioconductor.org/packages/splatter" class="uri">https://bioconductor.org/packages/splatter</a> and the code is shared on GitHub at
<a href="https://github.com/Oshlack/splatter" class="uri">https://github.com/Oshlack/splatter</a> under a GPL-3.0 license.</p>
<p>Contributions to the work in this chapter:</p>
<ul>
<li><p>I designed and implemented the Splatter R package described in this chapter.</p></li>
<li><p>Belinda Phipson contributed to the design of the Splat simulation method
described in the publication and provided statistical advice.</p></li>
<li><p>I conducted the analysis presented in the publication and produced the figures
shown.</p></li>
<li><p>Belinda Phipson performed pre-processing for some of the public datasets used.</p></li>
<li><p>Alicia Oshlack helped to design and plan the analysis presented in the
publication.</p></li>
<li><p>I wrote the first draft of the manuscript and performed revisions.</p></li>
<li><p>Alicia Oshlack assisted with planning the manuscript and edited drafts.</p></li>
<li><p>Belinda Phipson helped write sections of the manuscript and edited drafts.</p></li>
<li><p>Jovana Maksimovic and Sarah Blood proofread a draft of the manuscript and
provided comments.</p></li>
</ul>
<p><strong><a href="4-clust-trees.html#clust-trees">Chapter 4: Visualising clustering across resolutions</a></strong> is an
original work describing a visualisation for showing clustering results across
multiple resolutions and helping to select a clustering resolution to use. This
work has been published in <em>GigaScience</em> as “Clustering trees: a visualization
for evaluating clusterings at multiple resolutions” and a software package
implementing the algorithm described is available from CRAN at
<a href="https://cran.r-project.org/package=clustree" class="uri">https://cran.r-project.org/package=clustree</a>. The source code for this package
can be found on GitHub at <a href="https://github.com/lazappi/clustree" class="uri">https://github.com/lazappi/clustree</a> under a GPL-3.0
license.</p>
<p>Contributions to the work in this chapter:</p>
<ul>
<li><p>I designed the clustering trees algorithm described in this chapter.</p></li>
<li><p>I designed and built the clustree R package that implements this algorithm.</p></li>
<li><p>I performed the analysis presented in the publication and designed and
produced the figures shown.</p></li>
<li><p>Alicia Oshlack provided advice on the design and planning of the analysis to
present.</p></li>
<li><p>I planned and wrote the first draft of the manuscript.</p></li>
<li><p>Alicia Oshlack provided advice on the structure of the manuscript and edited
draft versions.</p></li>
<li><p>I performed revisions and drafted responses to reviewers.</p></li>
<li><p>Marek Cmero read and provided comments on a draft of the manuscript.</p></li>
</ul>
<p><strong><a href="5-analysis.html#analysis">Chapter 5: Analysis of kidney organoid scRNA-seq data</a></strong> is an
original work where I performed a re-analysis of a previously published
single-cell RNA sequencing experiment from kidney organoids in order to
demonstrate a range of analysis tools and decisions during analysis.</p>
<p>Contributions to the work in this chapter:</p>
<ul>
<li><p>The dataset is publicly available from the Gene Expression Omnibus under
accession GSE114802.</p></li>
<li><p>I performed pre-processing of the dataset</p></li>
<li><p>I designed and performed the analysis with input from Alicia Oshlack, Belinda
Phipson, Melissa Little and Alex Combes.</p></li>
<li><p>Alex Combes helped with interpreting gene lists describing cell types.</p></li>
<li><p>I designed and created the figures shown in this chapter.</p></li>
</ul>
<p><strong><a href="6-conclusion.html#conclusion">Chapter 6: Conclusion</a></strong> is an original work summarising the
work in this thesis, placing it in the wider context of single-cell RNA
sequencing analysis and outlining potential directions of the field.</p>
<p><strong>Other publications that I have contributed to during my candidature but are
not presented in this thesis</strong></p>
<p>Phipson B, <strong>Zappia L</strong>, Oshlack A. “Gene length and detection bias in single
cell RNA sequencing protocols.” <em>F1000 Research.</em> DOI:
<a href="https://doi.org/10.12688/f1000research.11290.1">10.12688/f1000research.11290.1</a>.</p>
<p>Phipson B, Er PX, Combes AN, Forbes TA, Howden SE, <strong>Zappia L</strong>, Yen HJ, Lawlor
KT, Hale LJ, Sun J, Wolvetang E, Takasato M, Oshlack A, Little MH. “Evaluation
of variability in human kidney organoids.” <em>Nature Methods.</em> 2019. DOI:
<a href="http://doi.org/10.1038/s41592-018-0253-2">10.1038/s41592-018-0253-2</a>.</p>
<p>Combes AN<sup>+</sup>, <strong>Zappia L<sup>+</sup></strong>, Er PX, Oshlack A, Little MH. “Single-cell
analysis reveals congruence between kidney organoids and human fetal kidney.”
<em>Genome Medicine.</em> 2019. DOI:
<a href="https://doi.org/10.1186/s13073-019-0615-0">10.1186/s13073-019-0615-0</a></p>
<p>Kumar SV, Er PX, Lawlor KT, Motazedian A, Scurr M, Ghobrial I, Combes AN,
<strong>Zappia L</strong>, Oshlack A, Stanley EG, Little MH. “Kidney micro-organoids in
suspension culture as a scalable source of human pluripotent stem cell-derived
kidney cells.” <em>Development.</em> 2019. DOI:
<a href="http://doi.org/10.1242/dev.172361">10.1242/dev.172361</a></p>
<p>Combes AN, Phipson B, Lawlor KT, Dorison A, Patrick R, <strong>Zappia L</strong>, Harvey RP,
Oshlack, A, Little MH. “Single cell analysis of the developing mouse kidney
provides deeper insight into marker gene expression and ligand-receptor
crosstalk.” <em>Development.</em> 2019. DOI:
<a href="http://doi.org/10.1242/dev.178673">10.1242/dev.178673</a></p>
<p><sup>+</sup> <em>Authors contributed equally.</em></p>
</div>
<div id="acknowledgements" class="section level2 unnumbered">
<h2>Acknowledgements</h2>
<p>Thank you to my supervisors Alicia Oshlack and Melissa Little, it has been a
pleasure and a privilege to work with you over the last few years. You both
provide supportive environments that let people get the most out of themselves
and I am fortunate to have had you to guide me through my PhD.</p>
<p>That support extends to the other members of the Oshlack lab. It’s easy to come
into work every day when you get on with your colleagues and enjoy spending
time with them. A special thanks to Belinda Phipson who as being an amazing
source of advice about statistics and analysis but also just life in general.
The Little kidney development lab has also been extremely welcoming. Thank you
to all of them for making me feel a part of their group and sharing their
knowledge, particularly Alex Combes who it has been fantastic to work closely
with.</p>
<p>I would also like to thank the member of my advisory committee Andrew Pask,
Christine Wells and Edmund Crampin for their advice and encouragement.</p>
<p>Outside my work a big part of my PhD has been my involvement with COMBINE.
Thank you to everyone who has contributed to building and continuing to develop
the organisation, particularly Leah Roberts who has guided me through
everything. The Australian bioinformatics student community is extremely
fortunate to have such an organisation and I hope students and supervisors
continue to support COMBINE and make it even better.</p>
<p>Thank you to my friends and family for the support you have provided outside my
PhD, it’s important to have other things going on in the rest of my life. The
biggest thanks to my partner Sarah for her continued love and care (and
proofreading expertise).</p>
<p>Lastly I want to thank the wider scientific and programming communities.
Everyone who makes their code, software, data and analysis available makes this
kind of research possible, easier and more fun.</p>
</div>
<div id="list-of-copyright" class="section level2 unnumbered">
<h2>List of copyright</h2>
<p>The copyright of the publication “Exploring the single-cell RNA-seq analysis
landscape with the scRNA-tools database” included in <strong><a href="2-tools.html#tools">Chapter 2</a></strong>
is held by the authors including myself under a Creative Commons Attribution
License (CC BY 4.0) (<a href="https://creativecommons.org/licenses/by/4.0/" class="uri">https://creativecommons.org/licenses/by/4.0/</a>). This
license permits unrestricted use, distribution and reproduction in any medium,
provided the original authors and source are credited.</p>
<p>The copyright of the publication “Splatter: simulation of single-cell RNA
sequencing data” included in <strong><a href="3-simulation.html#simulation">Chapter 3</a></strong> is held by the
authors including myself under a Creative Commons Attribution License (CC BY
4.0). The additional files for this publication provided in
<strong><a href="A-splatter-additional.html#splatter-additional">Appendix A</a></strong> are covered by the same license.</p>
<p>The copyright of the Splatter R software package and the documentation included
in <strong><a href="B-splatter-docs.html#splatter-docs">Appendix B</a></strong> is held by the authors including myself
under a GNU General Public License v3.0 (GPL-3) license
(<a href="https://choosealicense.com/licenses/gpl-3.0/" class="uri">https://choosealicense.com/licenses/gpl-3.0/</a>). This license allows me to
reproduce the documentation in both print and electronics versions of this
thesis.</p>
<p>The copyright of the publication “Clustering trees: a visualization for
evaluating clusterings at multiple resolutions” included in
<strong><a href="4-clust-trees.html#clust-trees">Chapter 4</a></strong> is held by the authors including myself under a
Creative Commons Attribution License (CC BY 4.0).</p>
<p>The copyright of the clustree R software package and the documentation included
in <strong><a href="D-clustree-docs.html#clustree-docs">Appendix D</a></strong> is held by the authors including myself
under a GNU General Public License v3.0 (GPL-3) license.</p>
<p>The copyright of the publication “Single-cell analysis reveals congruence
between kidney organoids and human fetal kidney” included in
<strong><a href="E-organoid-paper.html#organoid-paper">Appendix E</a></strong> is held by the authors including myself under
a Creative Commons Attribution License (CC BY 4.0).</p>

</div>
</div>
            </section>

          </div>
        </div>
      </div>

<a href="1-introduction.html" class="navigation navigation-next navigation-unique" aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"google": false,
"linkedin": false,
"weibo": false,
"instapper": false,
"vk": false,
"all": ["facebook", "google", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"download": [["thesis.pdf", "PDF"], ["thesis.docx", "Word"]],
"toc": {
"collapse": "section"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://cdn.bootcss.com/mathjax/2.7.1/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:" && /^https?:/.test(src))
      src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
