% *============================================================================*
%                  UNIVERSITY OF MELBOURNE THESIS TEMPLATE
% *============================================================================*

% +----------------------------------------------------------------------------+
%  This is a Pandoc/LaTeX template for a University of Melbourne thesis designed
%  to be used as part of a bookdown project
%  (https://bookdown.org/yihui/bookdown/).
% +----------------------------------------------------------------------------+

% +----------------------------------------------------------------------------+
%  Changes:
%
%  2018-10-10 (Luke Zappia)
%    * Create first draft template
% +----------------------------------------------------------------------------+

% *============================================================================*
%  PREAMBLE
%
%  Set up the document, load packages, set parameters etc.
% *============================================================================*

% +-----Document class---------------------------------------------------------+
%  Set the class associated with the document, options in square brackets are
%  passed to the class
% +----------------------------------------------------------------------------+

\documentclass[11pt,a4paper,titlepage,twoside,openright]{style/unimelbthesis}

% +-----Packages---------------------------------------------------------------+
%  External packages used in the document
% +----------------------------------------------------------------------------+

\usepackage{amsmath}   % American Mathematics Society standards
\usepackage{amsxtra}   % Additional math symbols
\usepackage{amssymb}   % Additional math symbols
\usepackage{amsthm}    % Additional math symbols
\usepackage{latexsym}  % Additional math symbols
\usepackage{booktabs}  % Table formatting
\usepackage{longtable} % Table formatting
\usepackage{setspace}  % Line spacing
\usepackage{chemarr}   % Improved reaction arrows for chemists
%\usepackage{palatino}  % Use the palatino font family
%\usepackage{mathpazo}  % Use the palatino font family
%\usepackage{utopia}    % Use the Utopia font
%\usepackage{charter}   % Use the Bitstream Charter font
\usepackage{libertine}  % Use the Libertine font family
\usepackage{graphicx}  % Extended graphics package
\usepackage{parskip}   % Space between paragraphs
\usepackage{microtype} % Microtypography for better spacing, kerning, etc...
\usepackage{caption}   % Caption formatting
\usepackage{xcolor}    % Coloured text
\usepackage{csquotes}  % Fancy quotes
\usepackage{flafter}

\usepackage[utf8]{inputenc}       % UTF-8 input encoding
\usepackage[T1]{fontenc}          % Use 8-bit fonts
\usepackage[hidelinks]{hyperref}  % Hyperlinks
\usepackage[explicit]{titlesec}   % Fancy section headings
\usepackage[section]{placeins}    % Restrict floats to sections

% +-----Package options--------------------------------------------------------+

\setlength{\parindent}{0pt}

% Chapter headings
\titleformat{\chapter}[display]
  {
    \normalfont
    \Large
    \filleft
    \sffamily
  }
  {
    \vskip150pt
    \fontsize{150}{0}
    \textbf{
      \textcolor{black!20}{
        ~\thechapter\vskip0pt
      }
    }
  }
  {0pt}
  {
    \fontsize{40}{0}
    \textbf{
      \textcolor{black!20}{
        #1
      }
    }
    \vspace{40pt}
  }

% Unnumbered chapter headings
\titleformat{name=\chapter,numberless}[display]
  {\normalfont\LARGE\bfseries\center}
  {}
  {0pt}
  {#1}

%\titlespacing*{\chapter}{0pt}{-50pt}{40pt}

% No numbers on chapter pages
%\assignpagestyle{\chapter}{empty}

% Figure captions
\captionsetup{labelfont=bf, font=small, margin=10pt}

% +-----verbatimfont-----------------------------------------------------------+
%  Control the font for verbatim environments (used for code)
% +----------------------------------------------------------------------------+

\makeatletter
\newcommand{\verbatimfont}[1]{\renewcommand{\verbatim@font}{\ttfamily#1}}
\makeatother

% +-----Parameters-------------------------------------------------------------+
%  Set parameters for the document. To convert from YAML to LaTeX we need to add
%  the dollar signs.
% +----------------------------------------------------------------------------+

\title{Tools and techniques for single-cell RNA sequencing data}
\author{Luke Zappia}
\letters{B.Sc, M.Sc}
\orcid{0000-0001-7744-8565}
\degree{Doctor of Philosophy}
\submissionmonth{August}
\submissionyear{2019}
\department{School of Biosciences}
\university{The University of Melbourne}
\statement{Submitted in Total Fulfillment of the Requirements of the Degree of Doctor of Philosophy}

% +-----Syntax highlighting----------------------------------------------------+
%  If code chunks are included in the document this allows Pandoc to insert the
%  code highlighting macros.
% +----------------------------------------------------------------------------+


% +-----TeXCount---------------------------------------------------------------+
%  Improve word counting using TeXCount.
% +----------------------------------------------------------------------------+

% Environment aliases to trick Pandoc
\let\Begin\begin
\let\End\end

% Environment for counting text as other
%TC:envir tcother [] other
\newenvironment*{tcother}{}{}

% Count as other
%TC:group abstract 1 3
%TC:group declaration 1 3
%TC:group preface 1 3
%TC:group acknowledgements 1 3
%TC:group copyrightlist 1 3
%TC:group longtable 1 3
%TC:group appendix 1 3
%TC:group quote 1 3

% *============================================================================*
%  FRONT MATTER
%
%  Front section of the document, everything before the main text.
% *============================================================================*

\usepackage{amsthm}
\newtheorem{theorem}{Theorem}[chapter]
\newtheorem{lemma}{Lemma}[chapter]
\theoremstyle{definition}
\newtheorem{definition}{Definition}[chapter]
\newtheorem{corollary}{Corollary}[chapter]
\newtheorem{proposition}{Proposition}[chapter]
\theoremstyle{definition}
\newtheorem{example}{Example}[chapter]
\theoremstyle{definition}
\newtheorem{exercise}{Exercise}[chapter]
\theoremstyle{remark}
\newtheorem*{remark}{Remark}
\newtheorem*{solution}{Solution}
\begin{document}

\urlstyle{same}

\verbatimfont{\scriptsize}

\begin{frontmatter}

\frontmatterheadings

% +-----Title------------------------------------------------------------------+

  \maketitle

% +-----Abstract---------------------------------------------------------------+

  \begin{abstract}
    RNA sequencing of individual cells allows us to take a snapshot of the dynamic processes within a cell and explore differences between cell types. As this technology has developed over the last few years it has been rapidly adopted by researchers in areas such as developmental biology, and many single-cell RNA sequencing datasets are now available. Coinciding with the development of protocols for producing single-cell RNA sequencing data there has been a simultaneous burst in the development of computational analysis methods. My thesis explores the computational tools and techniques for analysing single-cell RNA sequencing data. I present a database that charts the release of analysis software, where it has been published and what it can be used for, as well as a website that makes this information publicly available. I also present two of my own tools and techniques including Splatter, a software package for easily simulating single-cell datasets from multiple models, and clustering trees, a visualisation approach for inspecting clustering at multiple resolutions. In the final part of my thesis I perform analysis of a dataset from kidney organoids to demonstrate and compare some current analysis methods. Taken together, my thesis covers many aspects of the tools and techniques for single-cell RNA sequencing by describing the approaches that are available, presenting software that can help in developing and evaluating methods, introducing an approach for aiding one of the most common analysis tasks, and showing how tools can be used to extract meaning from a real dataset.
  \end{abstract}

% +-----Declaration------------------------------------------------------------+

  \begin{declaration}
    This is to certify that:
    
    \begin{enumerate}
    \def\labelenumi{\roman{enumi}.}
    \item
      this thesis comprises only my original work towards the degree of Doctor of Philosophy except where indicated in the preface;
    \item
      due acknowledgement has been made in the text to all other material used; and
    \item
      this thesis is fewer than the 100,000 words in length, exclusive of tables, maps, bibliographies and appendices
    \end{enumerate}
  \end{declaration}

% +-----Preface----------------------------------------------------------------+

  \begin{preface}
    This preface provides a summary of the chapters in this thesis and describes my contribution to them as well as the contributions of my collaborators and supervisors. This is a thesis \emph{with} publication and where publications form part of a chapter they are listed here. The following publications are included as part of this thesis:
    
    \textbf{Zappia L}, Phipson B, Oshlack A. ``Exploring the single-cell RNA-seq analysis landscape with the scRNA-tools database.'' \emph{PLoS Computational Biology.} 2018. DOI: \href{https://doi.org/10.1371/journal.pcbi.1006245}{10.1371/journal.pcbi.1006245}
    
    \textbf{Zappia L}, Phipson B, Oshlack A. ``Splatter: simulation of single-cell RNA sequencing data.'' \emph{Genome Biology.} 2017. DOI: \href{https://doi.org/10.1186/s13059-017-1305-0}{10.1186/s13059-017-1305-0}
    
    \textbf{Zappia L}, Oshlack A. ``Clustering trees: a visualization for evaluating clusterings at multiple resolutions.'' \emph{GigaScience.} 2018. DOI: \href{https://doi.org/10.1093/gigascience/giy083}{10.1093/gigascience/giy083}
    
    These publications are included as they appear online and are designed to be read as stand-alone documents. Sections within these publications are not included in the table of contents, and references are available at the end of each publication rather than in the reference list for this thesis. The contributions of authors to these papers are explained below. I am the first author on these publications and contributed more than 50 percent of the work towards them including drafting, editing and revising the manuscripts. My co-authors have provided signed declarations acknowledging and supporting my contributions which have been submitted along with this thesis. Where publicly available datasets have been used these have been appropriately cited.
    
    \textbf{\protect\hyperlink{introduction}{Chapter 1: Introduction}} is an original work providing a background and overview relevant to understanding my work in this thesis including an introduction to RNA sequencing, single-cell RNA sequencing and kidney function and development.
    
    \textbf{\protect\hyperlink{tools}{Chapter 2: The scRNA-seq tools landscape}} is an original work describing a database of software tools for analysing single-cell RNA sequencing data which has been published in \emph{PLoS Computational Biology} as ``Exploring the single-cell RNA-seq analysis landscape with the scRNA-tools database''. In addition to this publication I developed a website displaying the information in this database which is available at \url{https://www.scRNA-tools.org}. The database and code for building the website is available on GitHub at \url{https://github.com/Oshlack/scRNA-tools} under an MIT license.
    
    Contributions to the work in this chapter:
    
    \begin{itemize}
    \item
      I compiled and regularly updated the database of tools.
    \item
      I designed and built the public website used to display the database. Breon Schmidt provided assistance with implementing some of the website functionality. Some of the code for processing the database was based on a script written by Sean Davis.
    \item
      I performed the analysis of the database presented in the publication.
    \item
      I wrote the first draft of the manuscript and produced all the figures in the publication.
    \item
      Alicia Oshlack provided advice on planning the manuscript and edited draft versions.
    \item
      Belinda Phipson contributed to writing the manuscript.
    \end{itemize}
    
    \textbf{\protect\hyperlink{simulation}{Chapter 3: Simulating scRNA-seq data}} is an original work describing a software package for simulating single-cell RNA sequencing expression data. This work was published in \emph{Genome Biology} as ``Splatter: simulation of single-cell RNA sequencing data''. The software package described in this publication is available through Bioconductor at \url{https://bioconductor.org/packages/splatter} and the code is shared on GitHub at \url{https://github.com/Oshlack/splatter} under a GPL-3.0 license.
    
    Contributions to the work in this chapter:
    
    \begin{itemize}
    \item
      I designed and implemented the Splatter R package described in this chapter.
    \item
      Belinda Phipson contributed to the design of the Splat simulation method described in the publication and provided statistical advice.
    \item
      I conducted the analysis presented in the publication and produced the figures shown.
    \item
      Belinda Phipson performed pre-processing for some of the public datasets used.
    \item
      Alicia Oshlack helped to design and plan the analysis presented in the publication.
    \item
      I wrote the first draft of the manuscript and performed revisions.
    \item
      Alicia Oshlack assisted with planning the manuscript and edited drafts.
    \item
      Belinda Phipson helped write sections of the manuscript and edited drafts.
    \item
      Jovana Maksimovic and Sarah Blood proofread a draft of the manuscript and provided comments.
    \end{itemize}
    
    \textbf{\protect\hyperlink{clust-trees}{Chapter 4: Visualising clustering across resolutions}} is an original work describing a visualisation for showing clustering results across multiple resolutions and helping to select a clustering resolution to use. This work has been published in \emph{GigaScience} as ``Clustering trees: a visualization for evaluating clusterings at multiple resolutions'' and a software package implementing the algorithm described is available from CRAN at \url{https://cran.r-project.org/package=clustree}. The source code for this package can be found on GitHub at \url{https://github.com/lazappi/clustree} under a GPL-3.0 license.
    
    Contributions to the work in this chapter:
    
    \begin{itemize}
    \item
      I designed the clustering trees algorithm described in this chapter.
    \item
      I designed and built the clustree R package that implements this algorithm.
    \item
      I performed the analysis presented in the publication and designed and produced the figures shown.
    \item
      Alicia Oshlack provided advice on the design and planning of the analysis to present.
    \item
      I planned and wrote the first draft of the manuscript.
    \item
      Alicia Oshlack provided advice on the structure of the manuscript and edited draft versions.
    \item
      I performed revisions and drafted responses to reviewers.
    \item
      Marek Cmero read and provided comments on a draft of the manuscript.
    \end{itemize}
    
    \textbf{\protect\hyperlink{analysis}{Chapter 5: Analysis of kidney organoid scRNA-seq data}} is an original work where I performed a re-analysis of a previously published single-cell RNA sequencing experiment from kidney organoids in order to demonstrate a range of analysis tools and decisions during analysis.
    
    Contributions to the work in this chapter:
    
    \begin{itemize}
    \item
      The dataset is publicly available from the Gene Expression Omnibus under accession GSE114802.
    \item
      I performed pre-processing of the dataset
    \item
      I designed and performed the analysis with input from Alicia Oshlack, Belinda Phipson, Melissa Little and Alex Combes.
    \item
      Alex Combes helped with interpreting gene lists describing cell types.
    \item
      I designed and created the figures shown in this chapter.
    \end{itemize}
    
    \textbf{\protect\hyperlink{conclusion}{Chapter 6: Conclusion}} is an original work summarising the work in this thesis, placing it in the wider context of single-cell RNA sequencing analysis and outlining potential directions of the field.
    
    \textbf{Other publications that I have contributed to during my candidature but are not presented in this thesis}
    
    Phipson B, \textbf{Zappia L}, Oshlack A. ``Gene length and detection bias in single cell RNA sequencing protocols.'' \emph{F1000 Research.} DOI: \href{https://doi.org/10.12688/f1000research.11290.1}{10.12688/f1000research.11290.1}.
    
    Phipson B, Er PX, Combes AN, Forbes TA, Howden SE, \textbf{Zappia L}, Yen HJ, Lawlor KT, Hale LJ, Sun J, Wolvetang E, Takasato M, Oshlack A, Little MH. ``Evaluation of variability in human kidney organoids.'' \emph{Nature Methods.} 2019. DOI: \href{http://doi.org/10.1038/s41592-018-0253-2}{10.1038/s41592-018-0253-2}.
    
    Combes AN\textsuperscript{+}, \textbf{Zappia L\textsuperscript{+}}, Er PX, Oshlack A, Little MH. ``Single-cell analysis reveals congruence between kidney organoids and human fetal kidney.'' \emph{Genome Medicine.} 2019. DOI: \href{https://doi.org/10.1186/s13073-019-0615-0}{10.1186/s13073-019-0615-0}
    
    Kumar SV, Er PX, Lawlor KT, Motazedian A, Scurr M, Ghobrial I, Combes AN, \textbf{Zappia L}, Oshlack A, Stanley EG, Little MH. ``Kidney micro-organoids in suspension culture as a scalable source of human pluripotent stem cell-derived kidney cells.'' \emph{Development.} 2019. DOI: \href{http://doi.org/10.1242/dev.172361}{10.1242/dev.172361}
    
    Combes AN, Phipson B, Lawlor KT, Dorison A, Patrick R, \textbf{Zappia L}, Harvey RP, Oshlack, A, Little MH. ``Single cell analysis of the developing mouse kidney provides deeper insight into marker gene expression and ligand-receptor crosstalk.'' \emph{Development.} 2019. DOI: \href{http://doi.org/10.1242/dev.178673}{10.1242/dev.178673}
    
    \textsuperscript{+} \emph{Authors contributed equally.}
  \end{preface}

% +-----Acknowledgements-------------------------------------------------------+

  \begin{acknowledgements}
    Thank you to my supervisors Alicia Oshlack and Melissa Little, it has been a pleasure and a privilege to work with you over the last few years. You both provide supportive environments that let people get the most out of themselves and I am fortunate to have had you to guide me through my PhD.
    
    That support extends to the other members of the Oshlack lab. It's easy to come into work every day when you get on with your colleagues and enjoy spending time with them. A special thanks to Belinda Phipson who as being an amazing source of advice about statistics and analysis but also just life in general. The Little kidney development lab has also been extremely welcoming. Thank you to all of them for making me feel a part of their group and sharing their knowledge, particularly Alex Combes who it has been fantastic to work closely with.
    
    I would also like to thank the member of my advisory committee Andrew Pask, Christine Wells and Edmund Crampin for their advice and encouragement.
    
    Outside my work a big part of my PhD has been my involvement with COMBINE. Thank you to everyone who has contributed to building and continuing to develop the organisation, particularly Leah Roberts who has guided me through everything. The Australian bioinformatics student community is extremely fortunate to have such an organisation and I hope students and supervisors continue to support COMBINE and make it even better.
    
    Thank you to my friends and family for the support you have provided outside my PhD, it's important to have other things going on in the rest of my life. The biggest thanks to my partner Sarah for her continued love and care (and proofreading expertise).
    
    Lastly I want to thank the wider scientific and programming communities. Everyone who makes their code, software, data and analysis available makes this kind of research possible, easier and more fun.
  \end{acknowledgements}

% +-----Table of contents------------------------------------------------------+

  \hypersetup{linkcolor=black}
  \setcounter{tocdepth}{2}
  \tableofcontents

% +-----List of tables---------------------------------------------------------+


% +-----List of figures--------------------------------------------------------+

  \listoffigures

% +-----List of copyright------------------------------------------------------+

  \begin{copyrightlist}
    The copyright of the publication ``Exploring the single-cell RNA-seq analysis landscape with the scRNA-tools database'' included in \textbf{\protect\hyperlink{tools}{Chapter 2}} is held by the authors including myself under a Creative Commons Attribution License (CC BY 4.0) (\url{https://creativecommons.org/licenses/by/4.0/}). This license permits unrestricted use, distribution and reproduction in any medium, provided the original authors and source are credited.
    
    The copyright of the publication ``Splatter: simulation of single-cell RNA sequencing data'' included in \textbf{\protect\hyperlink{simulation}{Chapter 3}} is held by the authors including myself under a Creative Commons Attribution License (CC BY 4.0). The additional files for this publication provided in \textbf{\protect\hyperlink{splatter-additional}{Appendix A}} are covered by the same license.
    
    The copyright of the Splatter R software package and the documentation included in \textbf{\protect\hyperlink{splatter-docs}{Appendix B}} is held by the authors including myself under a GNU General Public License v3.0 (GPL-3) license (\url{https://choosealicense.com/licenses/gpl-3.0/}). This license allows me to reproduce the documentation in both print and electronics versions of this thesis.
    
    The copyright of the publication ``Clustering trees: a visualization for evaluating clusterings at multiple resolutions'' included in \textbf{\protect\hyperlink{clust-trees}{Chapter 4}} is held by the authors including myself under a Creative Commons Attribution License (CC BY 4.0).
    
    The copyright of the clustree R software package and the documentation included in \textbf{\protect\hyperlink{clustree-docs}{Appendix D}} is held by the authors including myself under a GNU General Public License v3.0 (GPL-3) license.
    
    The copyright of the publication ``Single-cell analysis reveals congruence between kidney organoids and human fetal kidney'' included in \textbf{\protect\hyperlink{organoid-paper}{Appendix E}} is held by the authors including myself under a Creative Commons Attribution License (CC BY 4.0).
  \end{copyrightlist}

\end{frontmatter}

% *============================================================================*
%  MAIN MATTER
%
%  Main part of the document. Includes all chapters and appendices.
% *============================================================================*


\begin{mainmatter}

\mainmatterheadings

\hypertarget{introduction}{%
\chapter{Introduction}\label{introduction}}

\begin{quote}
\textbf{See, people they don't understand}

\textbf{No, girlfriends, they can't understand}

\textbf{Your grandsons, they won't understand}

\textbf{On top of this, I ain't ever gonna understand\ldots{}}

--- The Strokes

\emph{Last Nite}, 2001
\end{quote}

\begin{center}\includegraphics[width=1\linewidth]{/home/travis/build/lazappi/phd-thesis/figures/01-introduction} \end{center}

\clearpage

This introduction chapter provides the background and overview necessary for understanding my work in this thesis, including an introduction to basic cell biology and kidney development, the technologies we use to measure those processes in the cell and the computational methods we use to analyse the data these technologies produce. \textbf{Section \ref{intro-dogma} - \protect\hyperlink{intro-dogma}{The central dogma}} describes how information flows within a cell and the molecules involved in these processes. \textbf{Section \ref{intro-RNAseq} - \protect\hyperlink{intro-RNAseq}{RNA sequencing}} covers the technologies used to measure the dynamic changes in some of these molecules and the established computational methods that are commonly used to extract meaning from this kind of data. In \textbf{Section \ref{intro-scRNAseq} - \protect\hyperlink{intro-scRNAseq}{Single-cell RNA sequencing}} I discuss new technologies that have enabled measurements on the level of individual cells while \textbf{Section \ref{intro-analysis} - \protect\hyperlink{intro-analysis}{Analysing scRNA-seq data}} outlines the types of analysis that can be performed on this data and the tools and methods that perform them. \textbf{Section \ref{intro-kidney} - \protect\hyperlink{intro-kidney}{Kidney development}} provides a brief introduction to kidney structure and development and how it can be studied using stem cell technologies. This section provides background for \textbf{Chapter \ref{analysis}} where I describe an analysis of a dataset from developing kidney organoids. My introduction ends with \textbf{Section \ref{intro-overview} - \protect\hyperlink{intro-overview}{Thesis overview and aims}} which presents an overview of my thesis aims and how I have completed them in the following chapters. An online version of this thesis is available at \url{https://lazappi.github.io/phd-thesis/}.

\hypertarget{intro-dogma}{%
\section{The central dogma}\label{intro-dogma}}

The central dogma of molecular biology describes the flow of information within a cell, from DNA to RNA to protein (Figure \ref{fig:central-dogma}). Deoxyribonucleic acid (DNA) is the long-term data storage of the cell and has a well-known double helix structure {[}1--3{]}. Each strand of the double helix consists of a series of nucleotide molecules linked by phosphate groups. These nucleotides are made of three subunits: a nitrogenous base, a deoxyribose sugar and a phosphate group. DNA nucleotides come in four species, adenine, cytosine, thymine and guanine, which have different base subunits. The sugar and phosphate groups form the backbone of each DNA strand and the two strands are bound together through hydrogen bonds between matching nucleotides known as base pairs. Guanine forms three hydrogen bonds with cytosine and adenine forms two with thymine. In computing terms DNA is similar to a hard drive in that it provides stable, consistent storage of important information.

\begin{figure}

{\centering \includegraphics[width=1\linewidth]{/home/travis/build/lazappi/phd-thesis/figures/01-central-dogma} 

}

\caption[The central dogma of molecular biology.]{The central dogma of molecular biology. Deoxyribonucleic acid (DNA) is the long-term data storage of the cell. To use this information a ribonucleic acid (RNA) molecule is produced through a process called transcription. A mature messenger RNA (mRNA) requires splicing and addition of a ploy(A) tail. Proteins are created from mRNA through translation in the ribosome.}\label{fig:central-dogma}
\end{figure}





When the cell wants to use some of this information it produces a copy of it in the form of a ribonucleic acid (RNA) molecule through a process known as transcription, similar to a computer loading information it wants to use into its random access memory. Sections of DNA that encode functional information are called genes. RNA is similar to a single strand of DNA except that the deoxyribose sugar is replaced with ribose and the thymine nucleotide is replaced with another nucleotide called uracil. Because it is single-stranded, RNA does not have a double helix structure but it can form complex shapes by binding to itself. A cell is able to create many copies of the same RNA molecule and this is referred to as the expression level. By varying RNA expression the activity of processes within a cell can be regulated. There are several different types of RNA that serve different purposes. RNA molecules that are transcribed from protein-coding genes are known as messenger RNA (mRNA). Other types of RNA include ribosomal RNA (rRNA) which forms part of the ribosome (the molecular machinery that manufactures proteins), transfer RNA (tRNA) which carries amino acids to the ribosome, micro RNA (miRNA) which have a role in regulating gene expression and long non-coding RNA (lncRNA) which are also involved in regulation among other processes. Genes are made up of regions that encode information (known as exons) that alternate with much larger non-coding regions (introns).

When an mRNA molecule is transcribed it initially contains the intronic sequences but these are removed through a process known as RNA splicing and a sequence of adenine nucleotides (a poly(A) tail) is added where transcription ends (the 3' end) to mark a mature mRNA molecule. This splicing process allows multiple forms of a protein to be produced from a single gene by selecting which exons are retained or removed. A mature mRNA transcript is converted to a protein in a complex structure called the ribosome which is made up of specialised RNA and proteins. The conversion process is known as translation because the information encoded by nucleic acids in RNA is translated to information stored as amino acids in the protein. Proteins complete most of the work required to keep a cell functioning and can be compared to the programs running on a computer. These functions require complex three dimensional structures and include tasks such as sensing things in the external environment, transporting nutrients into the cell, regulating the expression of genes, constructing new proteins, recycling molecules and facilitating metabolism. Understanding the molecules involved in the central dogma is central to our understanding of how a cell works.

\hypertarget{intro-RNAseq}{%
\section{RNA sequencing}\label{intro-RNAseq}}

By looking at DNA we can see what versions of genes are present in a cell but we cannot tell which of them are active and what processes they might be involved in. To do that we need to inspect the parts of the system that change dynamically. Ideally we might want to interrogate the proteins that are present, as they provide most of the functionality. However, while it is possible to do this using technologies such as mass spectrometry the readout produced is more difficult to interpret and the encoding is much more complex as there are 20 different amino acids compared to only four nucleotides. In contrast, RNA molecules are much easier to measure. High-throughput RNA sequencing (RNA-seq) provides a reliable method for high-quality measurement of RNA expression levels. RNA is isolated from a biological sample, converted to complementary DNA (cDNA) and provided as input to a sequencing machine. The output of an RNA-seq experiment is millions of short nucleotide sequences originating from the RNA transcripts present in the sample. Compared to older techniques for measuring RNA, such as probe-based microarrays, RNA-seq requires no prior knowledge of existing sequences in order to measure a sample and is effective over a much greater range of expression levels.

\hypertarget{library-preparation}{%
\subsection{Library preparation}\label{library-preparation}}

The first step in preparing a sample for RNA-seq is to chemically lyse the cells, disrupting the structure of the cell wall and releasing the molecules inside. RNA molecules can then be isolated, typically using a chemical process called phenol/chloroform extraction, although this can also be done by physically separating different types of molecules by passing the sample through a silica column. The majority of the RNA in a cell is ribosomal RNA, usually more than 80 percent {[}4{]}. Most of the time this type of RNA is not of interest and because the total amount of sequencing in an experiment is limited sequencing rRNA would reduce the ability to detect less abundant species. To select mature mRNA transcripts, oligonucleotide probes that bind to the poly(A) tail can be used, but a downside of this approach is that it won't capture immature mRNA or other types of RNA molecules. An alternative method is ribosomal RNA depletion where a different kind of probe specific to each species is used that binds to the rRNA allowing it to be removed. The choice of selection method has been shown to introduce different biases into the resulting data {[}5{]}.

The Illumina sequencing typically used for RNA-seq experiments can only read short sequences of nucleotides of approximately 40--400 base pairs. Most mRNA molecules are longer than this so to read the full length of a transcript it must first be fragmented into smaller parts. Most sequencing machines also only work with DNA, not directly with RNA, so the sample must first be reverse-transcribed using a retroviral enzyme to produce a single strand of cDNA. Many protocols have been designed for this step with each requiring a specific primer sequence to be joined to the RNA molecules. The complementary strand of cDNA is produced using a second enzyme that is usually involved in copying DNA for cell division. For some protocols fragmentation is performed after conversion to cDNA rather than at the RNA stage.

Once the cDNA has been produced it is usually necessary to attach adaptor sequences that are used to bind the molecules and initiate sequencing (Figure \ref{fig:adaptors}). These adaptors may also contain multiplexing barcodes that tag all the molecules in a sample and allow multiple samples to be sequenced at once. It has become standard practice to perform paired-end sequencing, where a section of nucleotides is read from one end of a fragment before it is flipped and the other end read, and this process requires an additional set of adaptors. At each of the stages of library preparation there are quality control steps to be performed to make sure a high-quality cDNA sample is loaded on to the sequencing machine.

\begin{figure}

{\centering \includegraphics[width=1\linewidth]{/home/travis/build/lazappi/phd-thesis/figures/01-adaptors} 

}

\caption[Illustration of adapter arrangements for the Illumina platform.]{Illustration of adapter arrangements for the Illumina platform ((A) minimal adapter components, (B) \enquote{in-line} barcode configuration for paired-end sequencing, (C) index configuration for paired-end sequencing). Sequencing primers SP1 (primary) and SP2 (paired-end) allow initiation of synthesis of the insert sequence. Index or barcode sequences allow multiplexing of multiple samples with the index being sequenced from a separate index primer (IP). (Image adapted from RNA-seqlopedia, \url{http://rnaseq.uoregon.edu} {[}6{]}).}\label{fig:adaptors}
\end{figure}





\hypertarget{high-throughput-sequencing}{%
\subsection{High-throughput sequencing}\label{high-throughput-sequencing}}

Most RNA-seq experiments are sequenced on an Illumina machine using their Sequence by Synthesis technology (Figure \ref{fig:sequencing}). In this process the two strands of cDNA fragments are separated and the adaptors bind to oligonucleotides probes coating a flow cell. The other end of the fragment can bind to a second oligonucleotide forming a bridge structure where an enzyme synthesises the complementary DNA strand. This process of separation of strands and synthesis of new complementary strands is repeated until clusters of DNA fragments with the same sequence are formed. Once the clusters contain enough fragments the adaptor at one end of each fragment is cleaved leaving single-stranded DNA attached to the flow cell at one end.

\begin{figure}

{\centering \includegraphics[width=0.8\linewidth]{/home/travis/build/lazappi/phd-thesis/figures/01-sequencing} 

}

\caption[The Illumina Sequencing by Synthesis process.]{The Illumina Sequencing by Synthesis process. Nucleotide fragments bind to a flow cell and clusters are formed through bridge amplification. Strands are primed and nucleotides tagged with fluorescent labels are added to the flow cell along with a DNA polymerase enzyme. The lanes of the flow cell are then scanned to produce an image. The fluorescent tags are cleaved and 3'-OH blocking groups are added, preparing the strands for another round of nucleotide incorporation. Image adapted from Mardis, \enquote{Next-Generation DNA Sequencing Methods} {[}7{]}.}\label{fig:sequencing}
\end{figure}





The sequencing process now begins. Nucleotides tagged with fluorescent markers are added and can bind to the next available position on a fragment if they are complementary. By adding all four nucleotides at once they compete for each position, reducing the chance of an incorrect match. Any unbound nucleotides are washed away before a laser excites the fluorescent tags and an image is taken. Each nucleotide is tagged with a different colour and the order of colours produced by a cluster shows the sequence of nucleotides in a fragment. For paired-end sequencing the fragments can be flipped and the sequencing process repeated at the other end. The images from the sequencing machine are processed to produce millions of short nucleotide sequences known as reads that are the starting point for computational analysis.

\hypertarget{analysis-of-rna-seq-data}{%
\subsection{Analysis of RNA-seq data}\label{analysis-of-rna-seq-data}}

Many types of analyses can be performed using RNA-seq data, such as identification of variants in the genetic sequence or detection of previously unannotated transcripts, but the most common kind of analysis is to look for differences in the expression level of genes between groups of samples (Figure \ref{fig:RNA-seq}). To do this, reads are first aligned to a reference genome and the number of reads overlapping each gene is counted. In contrast to aligners designed for DNA sequencing, RNA-seq aligners such as STAR {[}8{]}, HISAT2 {[}9{]} and subread {[}10{]} must take into account the splicing structure of mRNA transcripts which causes parts of some reads to align in different locations in the genome. The alignment step is computationally intensive and can take a significant amount of time. More recently tools such as kallisto {[}11{]} and Salmon {[}12{]} have been developed which attempt to directly quantify expression by estimating the probability that a read comes from a particular annotated transcript. These approaches are orders of magnitude faster than true genome alignment and potentially produce more accurate quantification, at the cost of having an exact genomic position for each read.

\begin{figure}

{\centering \includegraphics[width=1\linewidth]{/home/travis/build/lazappi/phd-thesis/figures/01-RNA-seq} 

}

\caption[A typical RNA-seq differential expression testing workflow.]{A typical RNA-seq differential expression testing workflow. An expression matrix is created from raw reads either by aligning them to a reference genome and counting those that overlap annotated genes or by alignment-free quantification (pink). Data analysis (blue) consists of several steps including quality control of samples and features, normalisation to remove technical differences between samples, testing for differential expression, gene set testing to identify enriched signatures, and visualisation. Results must be carefully interpreted (green) to extract meaning.}\label{fig:RNA-seq}
\end{figure}





At this stage the result is a matrix of counts, known as an expression matrix, where the rows are features (usually genes), the columns are samples and the values show the expression level of a particular feature in a sample. As these counts result from a sampling process they can be modelled using common statistical distributions. One option is the Poisson distribution which describes the probability of a given number of events happening in a fixed amount of time, however this distribution assumes that the mean and variance of each feature is equal. This can be true for replicates of the same RNA-seq sample but does not represent the extra biological variability between different samples {[}13{]}. A better fit is the negative binomial (or Gamma-Poisson) distribution which includes an over-dispersion parameter, allowing the variance in expression to be larger than the mean. While each feature is quantified for each sample, the observed expression levels are related to the total number of reads and the expression level of other features. For example, if a gene is highly expressed in a particular sample it will soak up a large proportion of the available reads, while in another sample where it is less expressed those reads are more likely to be sampled from other genes. Another complication of RNA-seq data is that the number of features (tens of thousands) is much larger than the number of samples (usually only a few per group) which violates the assumptions of many traditional statistical techniques. Technical variation is another well-known problem in high-throughput genomics studies, for example it has been estimated that only 17.8 percent of allele-specific expression is due to biological variation with the rest being technical noise {[}14{]}.

Both the edgeR {[}15,16{]} and DESeq {[}17{]} (and later the DESeq2 {[}18{]}) packages model RNA-seq data using the negative binomial distribution, but before differences in expression levels are tested the technical differences between samples should be removed through normalisation. Early approaches include simple transformations like Reads (or Fragments) Per Kilobase per Million (RPKM/FPKM) {[}19{]} or Transcripts Per Million (TPM) {[}20{]} which correct for the total number of reads per cell and gene length. The edgeR package uses the Trimmed-Mean of M values (TMM) method where a scaling factor for each sample is produced using the weighted mean of log ratios between samples for each gene, after excluding the highest expressed genes and those with the largest log ratio {[}13{]}. DESeq has a similar method that uses the median of the ratio of the expression of each gene to the geometric mean expression across all samples. The limma package {[}21{]} uses an alternative approach where a method called voom transforms the data so that it is suitable for linear modelling methods originally designed for RNA microarray technology {[}22{]}. When an experiment has been conducted in multiple batches and there are significant differences between them, alternative normalisation methods such as Remove Unwanted Variation (RUV) may be required {[}23{]}. RUV estimates unknown factors of unwanted variation using negative control genes (genes that are expected to be consistent between samples). These factors can then be included in the generalised linear modelling frameworks used for testing differential expression. Over time the methods in these packages have been refined and new tests developed allowing for the routine analysis of many RNA-seq experiments.

\hypertarget{intro-scRNAseq}{%
\section{Single-cell RNA sequencing}\label{intro-scRNAseq}}

Traditional bulk RNA-seq experiments average the transcriptome across the millions of cells in a sample. Recently it has become possible to perform single-cell RNA sequencing (scRNA-seq) and investigate the transcriptome at the resolution of individual cells. There are many situations where it is important to understand how specific cell types react to development or perturbations. This is often hindered in bulk analyses which may be affected by the unknown proportions of cell types in a sample. Studies into gene expression in specific cell types previously required selecting and isolating the cells of interest, which separates them from the other cell types they are usually associated with and makes it impossible to investigate interactions between them. With scRNA-seq technologies it is now possible to look at the transcriptome of all the cell types in a tissue simultaneously, which has led to a better understanding of what makes cell types distinct and the discovery of previously unknown cell types.

\hypertarget{early-single-cell-capture-technologies}{%
\subsection{Early single-cell capture technologies}\label{early-single-cell-capture-technologies}}

The first scRNA-seq protocol was published in 2009 {[}24{]}, just a year after the first bulk RNA-seq publication {[}19{]}. While this approach allowed measurements of the transcriptome in individual cells it required manual manipulation and was restricted to inspecting a few precious cells. Further studies quickly showed that cell types could be identified without sorting cells and approaches were developed to allow unbiased capture of the whole transcriptome. Since then, many scRNA-seq protocols have been developed (including CEL-Seq {[}25{]}, CEL-Seq2 {[}26{]}, Quartz-Seq {[}27{]}, Quartz-Seq2 {[}28{]} and Smart-seq2 {[}29{]}) and the number of cells in scRNA-seq experiments has scaled exponentially (Figure \ref{fig:technologies}) {[}30{]}. The first commercially available cell capture platform was the Fluidigm C1. This system uses microfluidics to passively separate cells into individual wells on a plate where they are lysed, reverse transcribed and the collected cDNA is PCR amplified. After this stage the product is extracted from the plate and libraries prepared for Illumina sequencing. Most C1 data has been produced using a 96 well plate but more recently an 800 well plate has become available, greatly increasing the number of cells that can be captured at a time. One of the disadvantages of microfluidic plate-based cell capture technologies is that the chips used have a fixed size window, meaning that only cells of a particular sizes can be captured in a single run. However, as cells are captured in individual wells they can be imaged before lysis, potentially identifying damaged or broken cells, empty wells or wells containing more than one cell. Capturing multiple cells is a known issue, with Macosko et al.~finding that when preparing a mixture of mouse and human cells, 30 percent of the resulting libraries contained transcripts from both species but only about a third of these doublets were visible in microscopy images {[}31{]}. The newer Polaris system from Fluidigm also uses microfluidics to capture cells but can select particular cells based on staining or fluorescent reporter expression and then hold them for up to 24 hours while introducing various stimuli. The cells can be imaged during this time before being lysed and prepared for RNA sequencing. This platform provides opportunities for a range of experiments that aren't possible using other capture technologies.

\begin{figure}

{\centering \includegraphics[width=1\linewidth]{/home/travis/build/lazappi/phd-thesis/figures/01-technologies} 

}

\caption[Exponential increase in the scale of scRNA-seq experiments.]{Exponential increase in the scale of scRNA-seq experiments. (a) Key jumps in scRNA-seq capture technologies including manual handling, plate-based microfluidic capture and droplet-based capture. (b) Cell numbers in representative studies using key technologies by publication date. Image from Svensson, Vento-Tormo and Teichmann \enquote{Exponential scaling of single-cell RNA-seq in the past decade} {[}30{]}.}\label{fig:technologies}
\end{figure}





\hypertarget{droplet-based-cell-capture}{%
\subsection{Droplet-based cell capture}\label{droplet-based-cell-capture}}

An alternative to using microfluidics to capture cells in wells is to capture them in nano-droplets. In this process a dissociated cell mixture is fed into a microfluidic device, while beads coated in primers enter at another input. The device is designed to form aqueous droplets within mineral oil and the inputs are arranged so that cells and beads can be simultaneously captured within a droplet. When this happens the reagents carried along with the bead lyse the cell and any poly(A) tagged RNA molecules present can bind to the capture probes on the bead. Reverse transcription and PCR amplification then begins and an individual cDNA library is produced for each cell, tagged with the unique barcode sequence present on the bead. The main advantage of droplet-based capture technologies is the ability to capture many more cells at one time, up to tens of thousands. These approaches are also less selective about cell size and produce fewer doublets. As a result they are much cheaper per cell, although as sequencing costs are fixed studies using droplet-based captures typically sequence individual cells at a much lower depth.

Droplet-based capture was popularised by the publication of the Drop-seq {[}31{]} and InDrop {[}32{]} platforms in 2015 and the updated InDrops in 2017 {[}33{]}. These are both DIY systems and although they differ in how the beads are produced, when the droplets are broken and in some aspects of the chemistry they can both be constructed on a lab bench from syringes, automatic plungers, a microscope and a small custom-made microfluidic chip. A similar commercially available platform is the 10x Genomics Chromium device which automates and streamlines much of the process (Figure \ref{fig:10xChromium}) {[}34{]}. This device uses droplet-based technologies for a range of applications including capture of cells for scRNA-seq. More specialised captures, such as those aimed at profiling immune cell receptors, are also possible and the company has recently announced kits for single-cell Assay for Transposase-Accessible Chromatin using sequencing (scATAC-seq) capture. A recent review has suggested that the 10x Chromium provides the most appropriate droplet-based capture for most experiments that aren't cost sensitive or require custom protocols {[}35{]}.

\begin{figure}

{\centering \includegraphics[width=1\linewidth]{/home/travis/build/lazappi/phd-thesis/figures/01-10xChromium} 

}

\caption[Diagram of the 10x Genomics Chromium cell capture process.]{Diagram of the 10x Genomics Chromium cell capture process. (A) Steps in the cell capture process. Barcoded gel beads are passed into a microfluidic device along with dissociated cells where they are captured in aqueous droplets in an oil solution. Cells are lysed within the droplets and mRNA is reverse transcribed to produced barcoded cDNA. Droplets are then broken and the cDNA collected for sequencing. (B) Structure of the gel bead capture probe including adaptor, cell barcode, UMI and poly(T) tail. Adapted using images from 10x Genomics.}\label{fig:10xChromium}
\end{figure}





\hypertarget{unique-molecular-identifiers}{%
\subsection{Unique Molecular Identifiers}\label{unique-molecular-identifiers}}

In contrast to plate-based capture methods, which usually provide reads along the length of RNA transcripts, droplet-based capture methods typically employ protocols which include short random nucleotide sequences known as Unique Molecular Identifiers (UMIs) {[}36{]}. Individual cells contain very small amounts of RNA (around 10--30 pg, less than 5 percent of which is mRNA) and to obtain enough cDNA for sequencing a PCR amplification step is necessary. Depending on their nucleotide sequence different transcripts may be amplified at different rates which can distort their relative proportions within a library. UMIs attempt to improve the quantification of gene expression by allowing the removal of PCR duplicates produced during amplification (Figure \ref{fig:UMIs}) {[}37{]}. The nucleotide probes used in droplet-based capture protocols include a poly(T) sequence which binds to mature mRNA molecules, a barcode sequence which is the same for every probe on a bead and 8--10 bases of UMI sequence which is unique to each probe. The UMI sequences are long enough that the probability of capturing two copies of a transcript on two probes with the same UMI is extremely low. After reverse transcription, amplification, sequencing and alignment, de-duplication can be performed by identifying reads with the same UMI that align to the same position and therefore should be PCR duplicates rather than truly expressed copies of a transcript.

\begin{figure}

{\centering \includegraphics[width=1\linewidth]{/home/travis/build/lazappi/phd-thesis/figures/01-UMIs} 

}

\caption[Unique Molecular Identifiers (UMIs) can improve scRNA-seq quantification.]{Unique Molecular Identifiers (UMIs) can improve scRNA-seq quantification. (A) UMIs are random 8--10 base pair sequences included as part of the mRNA capture probe along with the cell barcode (BC) and PCR handle. (B) The mRNA sequence can be aligned to a reference genome. (C) Each UMI is only counted once at each location, removing PCR duplicates and improving quantification.}\label{fig:UMIs}
\end{figure}





For this method to be effective each read must be associated with a UMI which means that only a small section at the 3' end of each transcript is sequenced. This has the side effect of reducing the amount of cDNA that needs to be sequenced and therefore increasing the number of cells that can be sequenced at a time. While the improvement in quantification of gene expression levels is useful for many downstream analyses, it comes at the cost of coverage across the length of a transcript, which is required for applications such as variant detection and de-novo assembly. However, reads along the length of genes have been observed in UMI datasets and are believed to come from unannotated transcription start sites or regions that contain enough adenine nucleotides to bind to the capture probes. Datasets with UMIs need extra processing steps which can be complicated by the possibility of sequencing errors in the UMI itself. Statistical methods designed for full-length data may also be affected by the different properties of a UMI dataset.

\hypertarget{recent-advances-in-scrna-seq-protocols}{%
\subsection{Recent advances in scRNA-seq protocols}\label{recent-advances-in-scrna-seq-protocols}}

Although droplet-based techniques are currently the most commonly used cell capture technologies, other approaches have been proposed that promise to capture even more cells at a lower cost per cell. These include approaches based around sub-nanolitre sized wells, for example the Seq-Well protocol {[}38{]}. A simple flow cell is constructed with an array of microwells made from polydimethylsiloxane (PDMS), a silicone rubber. A solution containing dissociated cells is flowed over the array and cells are captured in the wells by gravity. This is repeated to capture barcoded beads before adding reagents and sealing the wells. Once reactions are complete the beads can be retrieved and processed in a similar way to droplet-based capture {[}39{]}. Some cell types are difficult to capture due to their size, shape or other properties, and in some cases, particularly in tissues such as the brain, it has been suggested that single-nucleus rather than single-cell RNA-seq may be more effective {[}40--43{]}. In these protocols Nuclei are captured and processed in a similar way to cells, however the RNA within them is immature and unspliced so different reference annotations are required during analysis.

Extensions to the standard protocols have also been proposed that allow multiple measurements from the same cell (Figure \ref{fig:multimodal}). One such protocol is CITE-seq which enables measurement of the levels of selected proteins at the same time as the whole transcriptome (Figure \ref{fig:CITE-seq}A) {[}44{]}. Antibodies for the proteins of interest are labelled with short nucleotide sequences. These antibodies can then be applied to the dissociated cells and any that remain unbound are washed away before cell capture. The antibody labels are then captured along with mRNA transcripts and a size selection step is applied to separate them before library preparation. Similar antibodies can be used to allow multiplexing of samples through a process known as cell hashing (Figure \ref{fig:CITE-seq}B) {[}45{]}. In a typical scRNA-seq experiment each batch corresponds to a single sample. This complicates analysis as it is impossible to tell what is noise due to cells being processed in the same way and what is true biological signal. Cell hashing uses an antibody to a ubiquitously expressed protein but with a different nucleotide sequence for each sample. The samples can then be mixed, processed in batches and then the cells computationally separated based on which sequence they are associated with. An added benefit of this approach is the simple detection of doublets containing cells from different samples.

\begin{figure}

{\centering \includegraphics[width=1\linewidth]{/home/travis/build/lazappi/phd-thesis/figures/01-multimodal} 

}

\caption[The range of multimodal scRNA-seq technologies.]{The range of multimodal scRNA-seq technologies. Many protocols have been developed to enable multiple measurements from the same individual cells with a selection shown here. These other measurements (shaded colours) include genotype (rose bud), methylation (cornsilk), chromatin state (muted lime), transcription factor binding (periwinkle), protein expression (jade lime), activation or repression (cabbage), lineage tracing (aquamarine blue), spatial location (light sky blue) or electrophysiology (cotton candy). Image from \url{https://github.com/arnavm/multimodal-scRNA-seq} available under a Creative Commons Attribution-NonCommercial-ShareAlike 4.0 International (CC BY-NC-SA 4.0) license {[}46{]}.}\label{fig:multimodal}
\end{figure}





\begin{figure}

{\centering \includegraphics[width=1\linewidth]{/home/travis/build/lazappi/phd-thesis/figures/01-CITE-seq} 

}

\caption[Extensions to droplet-based capture protocols using nucleotide tagged antibodies.]{Extensions to droplet-based capture protocols using nucleotide tagged antibodies. (A) CITE-seq allows measurement of selected cell surface markers along with the transcriptome. Nucleotide tagged antibodies are applied to cells before droplet-based capture and processing. A size selection step is used to separate antibody nucleotides before library preparation. (B) Cell hashing allows multiplexing of samples. Ubiquitous antibodies are used to tag cells in each sample with a hashtag oligonucleotide. The cells are then mixed for cell capture and processing and the hashtags used to computationally identify the origin of each cell. Images from \url{https://cite-seq.com} {[}47{]}.}\label{fig:CITE-seq}
\end{figure}





CRISPR-Cas9 gene editing has also been developed as an extension to scRNA-seq protocols. One possibility is to introduce a mutation at a known location that can then be used to demultiplex samples processed together {[}48{]}. It is possible to do this with samples from different individuals or cell lines but the advantage of a gene editing based approach is that the genetic background remains similar between samples. It is also possible to investigate the effects of introducing a mutation. Protocols like Perturb-Seq {[}49{]} introduce a range of guide RNA molecules to a cell culture, subject the cells to some stimulus then perform single-cell RNA sequencing. The introduced mutation can then be linked to the response of the cells to the stimulus and the associated broader changes in gene expression. More recently it has been proposed that CITE-seq can be combined with CRISPR-based approaches {[}50{]}. Another approach is CellTagging which introduces a barcoded lentiviral construct into cells {[}51{]}. A unique combination of barcodes is then passed on to daughter cells and can be used to trace the lineage of a cell population.

Other approaches that allow multiple measurements from the same individual cells include G\&T-seq {[}52{]} and SIDR {[}53{]} which measure RNA and genomic DNA, similar approaches which measure the exome (or protein-coding part of the genome) along with the transcriptome {[}54{]}, scMT-seq which measures RNA and DNA methylation {[}55{]}, Patch-seq which combines patch-clamp recording to measure the electrophysiology of neurons with scRNA-seq {[}56{]} and scTrio-seq which is able to measure the genome, transcriptome and methylome simultaneously {[}57{]}.

\hypertarget{intro-analysis}{%
\section{Analysing scRNA-seq data}\label{intro-analysis}}

Cell capture technologies and scRNA-seq protocols have developed rapidly but the data they produce still presents a number of challenges. Existing approaches are inefficient, capturing around 10 percent of transcripts in a cell {[}58{]}. When combined with the low sequencing depth per cell this results in a limited sensitivity and an inability to detect lowly expressed transcripts. The small amount of starting material also contributes to high levels of technical noise, complicating downstream analysis and making it difficult to detect biological differences {[}59{]}. In order to capture cells they must first be dissociated into single-cell suspensions but this step can be non-trivial. Some tissues or cell types may be more difficult to separate than others and the treatments required to break them apart may affect the health of the cells and their transcriptional profiles. It has been suggested that damage caused by dissociation can be minimised by using a cold-active protease to separate cells {[}60{]}. Other cell types may be too big or have other characteristics that prevent them being captured. Multiple cells may be captured together or empty wells or droplets sequenced making quality control of datasets an important consideration.

As well as increasing technical noise the small amounts of starting material and low sequencing depth mean there are many occasions where zero counts are recorded, indicating no measured expression for a particular gene in a particular cell. These zero counts often represent the true biological state we are interested in as we that different genes will be expressed by different cell types. However, zeros can also be the result of confounding biological factors such as stage in the cell cycle, transcriptional bursting and environmental interactions which cause genuine changes in expression but that might not be of interest to a particular study. On top of this there are effects that are purely technical factors. In particular, sampling effects, which can result in \enquote{dropout} events where a transcript is truly expressed in a sample but is not observed in the sequencing data {[}61{]}. In bulk experiments these effects are limited by averaging across the cells in a sample and by the greater sequencing depth, but for single-cell experiments they can present a significant challenge for analysis as methods must account for the missing information and the large number of zeros may cause the assumptions of existing methods to be violated.

One approach to tackling the problem of too many zeros is to use zero-inflated versions of common distributions {[}62--65{]}. However, it is still debatable whether scRNA-seq datasets, particularly those from droplet-based capture protocols {[}66{]}, are truly zero inflated or if the additional zeros are better modelled with standard distributions with lower means. Another approach to analysis is to impute some of the zeros, replacing them with estimates of how expressed those genes truly are based on their expression in similar cells using methods such as MAGIC {[}67{]}, SAVER {[}68,69{]} or scImpute {[}70{]}. However, imputation comes with the risk of introducing false structure that is not actually present in the samples {[}71{]}.

Bulk RNA-seq experiments usually involve predefined groups of samples, for example cancer cells and normal tissue, different tissue types or treatment and control groups. It is possible to design scRNA-seq experiments in the same way by sorting cells into known groups based on surface markers, sampling them at a series of time points or comparing treatment groups, but often single-cell experiments are more exploratory. Many of the single-cell studies to date have sampled developing or mature tissues and attempted to profile the cell types that are present {[}32,72--77{]}. This approach is best exemplified by the Human Cell Atlas project which is attempting to produce a reference of the transcriptional profiles of all the cell types in the human body {[}78{]}. Similar projects exist for other species and specific tissues {[}79--88{]}.

As scRNA-seq datasets have become more widely available, a standard analysis workflow has developed which can be applied to many experiments {[}89,90{]}. This workflow can be divided into four phases shown in Figure \ref{fig:phases}: 1) Data acquisition, pre-processing of samples to produce a cell by gene expression matrix, 2) Data cleaning, quality control to refine the dataset used for analysis, 3) Cell assignment, grouping or ordering of cells based on their transcriptional profile, and 4) Gene identification to find genes that represent particular groups and can be used to interpret them. Within each phase a range of processes may be used and there are now many tools available for completing each of them, with over 450 tools currently available. An introduction to the phases of scRNA-seq analysis is provided here but the analysis tools landscape is more fully explored in \textbf{Chapter \ref{tools}}.

\begin{figure}

{\centering \includegraphics[width=1\linewidth]{/home/travis/build/lazappi/phd-thesis/figures/01-phases} 

}

\caption[Phases of a standard scRNA-seq analysis workflow.]{Phases of a standard scRNA-seq analysis workflow. In Phase 1 (Data acquisition) samples are pre-processed to produce an expression matrix that undergoes quality control and filtering in Phase 2 (Data cleaning). Phase 3 (Cell assignment) groups or orders cells according to their transcriptional profile and in Phase 4 (Gene identification) genes describing those groups are discovered and used to interpret them.}\label{fig:phases}
\end{figure}





\hypertarget{pre-processing-and-quality-control}{%
\subsection{Pre-processing and quality control}\label{pre-processing-and-quality-control}}

The result of a sequencing experiment is typically a set of image files from the sequencer or a FASTQ file containing nucleotide reads but for most analyses we use an expression matrix where each column is a cell, each row is a feature and the values indicate expression levels. To produce this matrix there is a series of pre-processing steps, typically beginning will some quality control of the raw reads. Reads are then aligned to a reference genome and the number of reads overlapping annotated features (genes or transcripts) is counted. Probabilistic quantification methods can be applied to full-length scRNA-seq datasets but have required adaptations such as the alevin method for UMI-based datasets {[}91{]} in the Salmon package. When using conventional alignment, UMI samples need extra processing with tools like UMI-tools {[}92{]}, umis {[}93{]} or zUMIs {[}94{]} in order to assign cell barcodes and deduplicate UMIs. For datasets produced using the 10x Chromium platform the company provides the Cell Ranger software which is a complete pre-processing pipeline that also includes an automated downstream analysis. Other packages such as scPipe {[}95{]} also aim to streamline this process with some such as Falco {[}96{]} designed to work on scalable cloud-based infrastructure which may be required as bigger datasets continue to be produced.

Quality control of individual cells is important as experiments will contain low-quality cells that can be uninformative or lead to misleading results. Particular types of cells that are commonly removed include damaged cells, doublets where multiple cells have been captured together {[}97,98{]} and empty droplets or wells that have been sequenced but do not contain a cell {[}99{]}. Quality control can be performed on various levels including the quality scores of the reads themselves and how or where they align to features of the expression matrix. The Cellity package attempts to automate this process by inspecting a series of biological and technical features and using machine learning methods to distinguish between high and low-quality cells {[}100{]}. However, the authors found that many of the features were cell type specific and more work needs to be done to make this approach more generally applicable. The scater package {[}101{]} emphasises a more exploratory approach to quality control at the expression matrix level by providing a series of functions for visualising various features of a dataset. These plots can then be used for selecting thresholds for removing cells. Plate-based capture platforms can produce additional biases based on the location of individual wells, a problem which is addressed by the OEFinder package which attempts to identify and visualise these \enquote{ordering effects}{[}102{]}.

Filtering and selection of features also deserves attention. Genes or transcripts that are lowly expressed are typically removed from datasets in order to reduce computational time and the effect of multiple-testing correction but it is unclear how many counts indicate that a gene is truly \enquote{expressed}. Many downstream analyses operate on a selected subset of genes and how they are selected can have a dramatic effect on their results {[}103{]}. These features are often selected based on how variable they are across the dataset but this may be a result of noise rather than biological importance. Alternative selection methods have been proposed such as M3Drop which identifies genes that have more zeros than would be expected based on their mean expression {[}104{]}.

\hypertarget{normalisation-and-integration}{%
\subsection{Normalisation and integration}\label{normalisation-and-integration}}

Effective normalisation is just as crucial for single-cell experiments as it is for bulk RNA-seq datasets. FPKM or TPM transformations can be used, but for UMI data the gene length correction is not required as reads only come from the ends of transcripts {[}105{]}. Normalisation methods designed for detecting differential expression between bulk samples such as TMM or the DESeq method can be applied, but it is unclear how suitable they in the single-cell context. Many of the early normalisation methods developed specifically for scRNA-seq data made use of spike-ins. These are synthetic RNA sequences added to cells in known quantities such as the External RNA Controls Consortium (ERCC) control mixes, a set of 92 transcripts from 250 to 2000 base pairs long based on bacterial plasmids. Brennecke et al. {[}106{]}, Ding et al. {[}107{]} and Grn, Kester and van Oudenaarden {[}58{]} all propose methods for estimating technical variance using spike-ins, as does Bayesian Analysis of Single-Cell Sequencing data (BASiCS) {[}108{]}. Using spike-ins for normalisation assumes that they properly capture the dynamics of the underlying dataset and even if this is the case it is restricted to protocols where they can be added which does not include droplet-based capture techniques. The scran package implements a method that doesn't rely on spike-ins, instead using a pooling approach to compensate for the large number of zero counts where expression levels are summed across similar cells before calculating size factors that are deconvolved back to the original cells {[}109{]}. The BASiCS method has also been adapted to experiments without spike-ins by integrating data replicated across batches {[}110{]}, but only for designed experiments where groups are known in advance.

Early scRNA-seq studies often made use of only a single sample but as technologies have become cheaper and more widely available it is common to see studies with multiple batches or making use of publicly available data produced by other groups. While this expands the potential insights to be gained it presents a problem as to how to integrate these datasets {[}111{]}. A range of computational approaches for performing integration have been developed including bbknn {[}112{]}, ClusterMap {[}113{]}, kBET {[}114{]}, LIGER {[}115{]}, matchSCore {[}116{]}, Scanorama {[}117{]} and scMerge {[}118{]}. The alignment approach in the Seurat package {[}119{]} uses Canonical Correlation Analysis (CCA) {[}120{]} to identify a multi-dimensional subspace that is consistent between datasets. Dynamic Time Warping (DTW) {[}122{]} is then used to stretch and align these dimensions so that the datasets are similarly spread along them. Some tasks can be performed using these aligned dimensions, but as the original expression matrix is unchanged the integration is not used for other tasks such as differential expression testing. The authors of scran use a Mutual Nearest Neighbours (MNN) approach that calculates a cosine distance between cells in different datasets then identifies those that share a neighbourhood {[}123{]}. Batch correction vectors can then be calculated and subtracted from one dataset to overlay them. A recent update to the Seurat method combines these approaches by applying CCA before using the MNN approach to identify \enquote{anchor} cells that have common features in the different datasets {[}124{]}.

\hypertarget{grouping-cells}{%
\subsection{Grouping cells}\label{grouping-cells}}

Grouping similar cells is a key step in analysing scRNA-seq datasets that is not usually required for bulk experiments and as such it has been a key focus of methods development. Over one hundred tools have been released for clustering cells which attempt to address a range of technical and biological challenges {[}125{]}. Some of these methods include SINgle CEll RNA-seq profiling Analysis (SINCERA) {[}126{]}, Single-Cell Consensus Clustering (SC3) {[}127{]}, single-cell latent variable model (scLVM) {[}76{]} and Spanning-tree Progression Analysis of Density-normalised Events (SPADE) {[}128{]}, as well as BackSPIN which was used to identify nine cell types and 47 distinct subclasses in the mouse cortex and hippocampus in one of the earliest studies to demonstrate the possibilities of scRNA-seq {[}72{]}. All of these tools attempt to cluster similar cells together based on their expression profiles, forming groups of cells of the same type. The clustering method in the Seurat package {[}129{]} has become particularly popular and has been shown to perform well, particularly on UMI datasets {[}130,131{]}. This method begins by selecting a set of highly variable genes then performing PCA on them. A set of dimensions is then selected that contains most of the variation in the dataset. Alternatively, if Seurat's alignment method has been used to integrate datasets the aligned CCA dimensions are used instead. Next, a Shared Nearest Neighbours (SNN) graph is constructed by considering the distance between cells in this multidimensional space and the overlap between shared neighbourhoods. Seurat uses a Euclidean distance but it has been suggested that correlations can provide better results {[}132{]}. In order to separate cells into clusters, a community detection algorithm such as Louvain optimisation {[}133{]} is run on the graph with a resolution parameter that controls the number of clusters that are produced. Selecting this parameter and similar parameters in other methods is difficult but important as the number of clusters selected can affect the interpretation of results. I address this problem with a visualisation-based approach in \textbf{Chapter \ref{clust-trees}}.

For tissue types that are well understood or where comprehensive references are available an alternative to unsupervised clustering is to directly classify cells. This can be done using a gating approach based on the expression of known marker genes similar to that commonly used for flow cytometry experiments. Alternatively, machine learning algorithms can be used to perform classification based on the overall expression profile. Methods such as scmap {[}134{]}, scPred {[}135{]}, CaSTLe {[}136{]} and Moana {[}137{]} take this approach. Classification has the advantage of making use of existing knowledge and avoids manual annotation and interpretation of clusters which can often be difficult and time consuming. However, it is biased by what is present in the reference datasets used and typically cannot reveal previously unknown cell types or states. As projects like the Human Cell Atlas produce well-annotated references based on scRNA-seq data the viability of classification and other reference-based methods will improve.

\hypertarget{ordering-cells}{%
\subsection{Ordering cells}\label{ordering-cells}}

In some studies, for example in development where stem cells are differentiating into mature cell types, it may make sense to order cells along a continuous trajectory from one cell type to another instead of assigning them to distinct groups. Trajectory analysis was pioneered by the Monocle package which used dimensionality reduction and computation of a minimum spanning tree to explore a model of skeletal muscle differentiation {[}77{]}. Since then the Monocle algorithm has been updated {[}138{]} and a range of others developed including TSCAN {[}139{]}, SLICER {[}140{]}, CellTree {[}141{]}, Sincell {[}142{]}, Mpath {[}143{]} and Slingshot {[}144{]}. In their review of trajectory inference methods, Cannoodt, Saelens and Saeys break the process into two steps (Figure \ref{fig:trajectory-inference}) {[}145{]}. In the first step, dimensionality reduction techniques such as principal component analysis (PCA) {[}146{]}, t-distributed stochastic neighbourhood embedding (t-SNE) {[}147{]} or uniform manifold approximation and projection (UMAP) {[}148{]} are used to project cells into lower dimensions where the cells are clustered or a graph constructed between them. The trajectory is then created by finding a path through the cells and ordering the cells along it. The same group of authors have conducted a comprehensive comparison of ordering methods evaluating their performance on a range of real and synthetic datasets {[}149{]}. To do so the authors had to develop metrics for comparing trajectories from different methods and built a comprehensive infrastructure for running and evaluating software tools. They found that the performance of methods depended on the underlying topology of the data and that multiple complementary trajectory inference approaches should be used to better what is occurring.

\begin{figure}

{\centering \includegraphics[width=0.9\linewidth]{/home/travis/build/lazappi/phd-thesis/figures/01-trajectory-inference} 

}

\caption[Trajectory inference framework as described by Cannoodt, Saelens and Saeys.]{Trajectory inference framework as described by Cannoodt, Saelens and Saeys. In the first stage, dimensionality reduction is used to convert the data to a simpler representation. In the second stage a trajectory is identified and the cells ordered. Image from Cannoodt, Saelens and Saeys \enquote{Computational methods for trajectory inference from single-cell transcriptomics} {[}145{]}.}\label{fig:trajectory-inference}
\end{figure}





An alternative continuous approach is the cell velocity technique {[}150{]} introduced for scRNA-seq data in the velocyto package {[}151{]}. RNA-seq studies typically focus on the expression of mature mRNA molecules but a sample will also contain immature mRNA that are yet to be spliced. Examining the reads assigned to introns can indicate newly transcribed mRNA molecules and therefore which genes are currently active. Instead of assigning cells to discrete groups or along a continuous path velocyto uses reads from unspliced regions to place them in a space and create a vector indicating the direction in which the transcriptional profile is heading. This vector can show that a cell is differentiating in a particular way or that a specific transcriptional program has been activated.

Deciding on which cell assignment approach to use depends on the source of the data, the goals of the study and the questions that are being asked. Both grouping and ordering can be informative and it is often useful to attempt both on a dataset and see how they compare.

\hypertarget{gene-detection-and-interpretation}{%
\subsection{Gene detection and interpretation}\label{gene-detection-and-interpretation}}

Once cells are assigned by clustering or ordering the problem is to interpret what these groups represent. For clustered datasets this is usually done by identifying genes that are differentially expressed across the groups or marker genes that are expressed in a single cluster. Many methods have been suggested for testing differential expression, some of which take in to account the unique features of scRNA-seq data, for example D3E {[}152{]}, DEsingle {[}65{]}, MAST {[}153{]}, scDD {[}154{]} and SCDE {[}155{]}. The large number of cells in scRNA-seq datasets means that some of the problems that made standard statistical tests unsuitable for bulk RNA-seq experiments do not apply. Simple methods like the unpaired Wilcoxon rank-sum test (or Mann-Whitney U test), Student's t-test or logistic regression may give reasonable results in this setting. Methods originally developed for bulk experiments have also been applied to scRNA-seq datasets but the assumptions they make may not be appropriate for single-cell data. Methods such as ZINB-WaVe {[}64{]} may be required to transform the data so that bulk testing methods are appropriate. A comprehensive evaluation of 36 differential expression testing approaches found that methods developed for bulk RNA-seq did not perform worse than scRNA-seq specific methods, however, the performance of bulk methods depended on how lowly-expressed genes were filtered {[}156{]}. The authors also observed that some methods were biased in the types of features they tended to detect as differentially expressed. Often the goal is not to find all the genes that are differentially expressed between groups but to identify genes which uniquely mark particular clusters. This goal is open to alternative approaches, such as the Gini coefficient which measures unequal distribution across a population. Another approach is to construct machine learning classifiers for each gene to distinguish between one group and all other cells. Genes that give good classification performance should be good indicators of what is specific to that cluster. It is also possible to identify genes that might have the same mean between groups but differ in variance or other characteristics of their expression distribution.

When cells have been ordered along a continuous trajectory the task is slightly different. Instead of testing for a difference in means between two groups, the goal is to find genes that have a relationship between expression and pseudotime. This can be accomplished by fitting splines to the relationship between pseudotime and expression and testing the fitting coefficients. For more complex trajectories it can also be useful to find genes that are differently expressed along each side of a branch point. Monocle's BEAM (Branch Expression Analysis Modelling) method does this using a likelihood ratio test between splines where the branch assignments are known or unknown {[}157{]}. Genes that are associated with a trajectory are important in their own right as they describe the biology along a path but they can also be used to identify cell types at end points.

Interpreting the meaning of detected marker genes is a difficult task and is likely to remain so. Some methods have been developed for this task such as celaref which suggests cluster labels based on similarity of marker genes to an already characterised reference dataset. Gene set testing to identify related categories such as Gene Ontology terms can also help but often it is necessary to rely on the results of previous functional studies. Ultimately this can only be reliably done by working closely with experts who have significant domain knowledge in the cell types being studied. An additional concern for unsupervised scRNA-seq studies is that the same genes are used for clustering or ordering and determining what those clusters or trajectories mean. This is a problem addressed by Zhang, Kamath and Tse who suggest a differential expression test using a long-tailed distribution for testing genes following clustering {[}158{]}.

\hypertarget{alternative-analyses}{%
\subsection{Alternative analyses}\label{alternative-analyses}}

Some uses of scRNA-seq data fall outside the most common workflow and methods have been developed for a range of other purposes. For example, methods have been designed for assigning haplotypes to cells {[}159{]}, detecting allele-specific expression {[}160--162{]}, identifying alternative splicing {[}163--165{]} or calling single nucleotide or complex genomic variants {[}166--168{]}. Other methods have been designed for specific cell types or tissues such as reconstructing immune cell receptors on B-cells (BraCeR {[}169{]}, BRAPeS {[}170{]}, VDJPuzzle {[}171{]}) or T-cells (TraCeR {[}172{]}, TRAPeS {[}173{]}) or interrogating the development of cancer samples (HoneyBADGER {[}174{]}, SSrGE {[}166{]}). Most future studies can be expected to continue to follow common practice but it is also expected that researchers will continue to push the boundaries of what it is possible to study using scRNA-seq technologies.

\hypertarget{evaluation-of-scrna-seq-analysis-methods}{%
\subsection{Evaluation of scRNA-seq analysis methods}\label{evaluation-of-scrna-seq-analysis-methods}}

Although the analysis for many scRNA-seq studies follow a standard workflow there is wide variation in the tools that are used. We are now at the stage where there are multiple software packages for completing every stage of analysis. Deciding which tools to use can be difficult and depends on a number of factors including effectiveness, robustness, scalability, availability, ease of use and quality of documentation. In terms of effectiveness, publications describing analysis methods should demonstrate two things: 1) they can perform the task they are designed for (at least as well as existing methods) and 2) performing that task leads to biological insights. Answering the first question is difficult using real datasets as often the underlying truth is not known. For this reason simulation techniques are commonly used to produce synthetic datasets in order to evaluate methods. I present a software package for simulating scRNA-seq data in \textbf{Chapter \ref{simulation}}. Simulation is an efficient and flexible approach for producing gold standard datasets but synthetic data can never fully reproduce a real dataset. An alternative approach is to carefully construct a gold standard biological dataset by combining samples where the true differences are known {[}175{]}. These datasets can be extremely useful but are difficult, time-consuming and expensive to produce and can only reproduce a limited set of scenarios. Comprehensive evaluations of methods have already been conducted for some aspects of scRNA-seq analysis including clustering {[}130,131{]}, trajectory inference {[}149{]} and differential expression testing {[}156{]} but these will need to continue to be performed and updated as the field matures and new methods are developed.

\hypertarget{intro-kidney}{%
\section{Kidney development}\label{intro-kidney}}

One area that has particularly benefitted from the possibilities created by scRNA-seq technology is developmental biology. Although the genes involved in the development of many organs are now well understood, arriving at this knowledge has required many painstaking experiments to investigate a single gene at a time. During development cells are participating in a continuous dynamic process involving the maturation from one cell type to another and the creation of new cell types. Single-cell RNA-seq captures a snapshot of the expression of all the genes involved in this process, allowing the transcriptome of intermediate and mature cells to be studied. This has revealed that some of the genes thought to be markers of specific cell types are more widely expressed or involved in other processes.

\hypertarget{structure-and-function}{%
\subsection{Structure and function}\label{structure-and-function}}

The kidney is the organ responsible for filtering the blood in order to remove waste products. In humans, kidneys grow as a pair of functional organs with each being around the size of an adult fist and weighing about 150 grams. Blood flows into the kidney via the renal artery and the blood vessels form a tree-like branching structure with ever smaller capillaries (Figure \ref{fig:kidney}A). At the end of these branches are nephrons, the functional filtration unit of the kidney (Figure \ref{fig:kidney}B). Humans can have around 1 million nephrons {[}176{]} that are formed during development and just after birth, however they cannot be regenerated during adulthood. A capillary loop is formed inside a structure at the end of the nephron called a glomerulus and surrounded by Bowman's capsule. Here specialised cells called podocytes create a structure called the slit diaphragm that allows water, metal ions and small molecules to be filtered while keeping blood cells and larger species such as proteins trapped within the bloodstream. The rest of the nephron is divided into segments that are responsible for different processes involved in balancing the concentration of these species in the filtrate (Figure \ref{fig:kidney}C). The tubular segments of the nephron are surrounded by capillaries, allowing molecules to be transferred between the filtrate and blood as required. The first segment of the nephron is the proximal tubule. Here common biomolecules such as glucose, amino acids and bicarbonate are reabsorbed into the bloodstream, as is most of the water present. Other molecules including urea and ammonium ions are secreted from the blood into the filtrate at his stage. The proximal tubule is followed by the loop of Henle and the distal tubule where ions are reabsorbed including potassium, chlorine, magnesium and calcium. The final segment is the collecting duct where salt concentrations are balanced by exchanging sodium in the filtrate for potassium in the bloodstream using a process controlled by the hormone aldosterone. The remaining filtrate is then passed to the ureter where it is carried to the bladder and collected as urine while the blood leaves via the renal vein. In order to perform this complex series of reabsorption and secretion steps, each segment of the nephron is made up specialised cell types with their own sets of signalling and transporter proteins. The filtration process is repeated about 12 times every hour with around 200 litres of blood being filtered during a day. Aside from removing waste and maintaining the balance of molecular species in the bloodstream, the kidneys also play a role in the activation of vitamin D. Other important functions include synthesising the hormones erythropoietin, which stimulates red blood cell production, and renin, which is part of the pathway that controls fluid volume and the constriction of arteries to regulate blood pressure.

\begin{figure}

{\centering \includegraphics[width=0.9\linewidth]{/home/travis/build/lazappi/phd-thesis/figures/01-kidney} 

}

\caption[Structure of the kidney and nephron.]{Structure of the kidney and nephron. (A) Blood flows into the kidney via the renal artery which branches into ever smaller capillaries. (B) At the end of these capillaries is the nephron, the functional filtration unit of the kidney, which is surrounded by a capillary network. (C) Different segments of the kidney are responsible for transferring specific molecular species between the bloodstream and the filtrate. Figure adapted using images from OpenStax College via Wikimedia Commons under a CC BY 3.0 license.}\label{fig:kidney}
\end{figure}





Chronic kidney disease is a major health problem in Australia with 1 percent of the population (237\thinspace800 people) diagnosed with it in 2017 {[}177{]} and it being considered a contributory factor for 13 percent of deaths {[}178{]}. Early stages of the disease can be managed but once it becomes severe the only treatment options are dialysis, which is expensive, time consuming and unpleasant, or a kidney transplant. There are also a range of genetic developmental kidney disorders that have limited treatment options and can profoundly affect quality of life. Understanding how the kidney grows and develops is key to developing new treatments that may improve kidney function or repair damage.

\hypertarget{stages-of-development}{%
\subsection{Stages of development}\label{stages-of-development}}

The kidney develops from a region of the early embryo called the intermediate mesoderm and occurs in three phases with a specific spatial and temporal order {[}179{]}. The first phase results in the pronephros which consists of 6--10 pairs of tubules that forms the mature kidney in most primitive vertebrates such as hagfish. By about the fourth week of human embryonic development this structure dies off and is replaced by the mesonephros, which is the form of kidney present in most fish and amphibians. The mesonephros is functional during weeks 4--8 of human embryonic development before degenerating, although parts of its duct system go on to form part of the male reproductive system. The final phase of human kidney development results in the metanephros which begins developing at around five weeks and continues to around week 36 to become the permanent and functional kidney {[}180{]}. Individual nephrons grow in a similar series of stages. Cells from the duct that will become the ureter begin to invade the surrounding metanephric mesenchyme forming a ureteric bud. Interactions between these cell types, including Wnt signalling, cause mesenchymal cells to condense around the ureteric bud forming a stem cell population known as the cap mesenchyme, which expresses genes such as Six2 and Cited1. Cells from the cap mesenchyme first form a renal vesicle, a primitive structure with a lumen, which extends to form an S-shaped body (Figure \ref{fig:nephrogenesis}). By this stage the lumen has joined with the ureteric bud to form a continuous tubule. The s-shaped body continues to elongate with podocytes beginning to develop and form a glomerulus at one end and other specialised cells arising along the length of the tubule to form the various nephron segments. Several signalling pathways and cell--cell interactions are involved in this process, including Notch signalling. While all nephrons form before birth they continue to elongate and mature postnatally.

\begin{figure}

{\centering \includegraphics[width=0.7\linewidth]{/home/travis/build/lazappi/phd-thesis/figures/01-nephrogenesis} 

}

\caption[Diagram of the stages of nephron maturation.]{Diagram of the stages of nephron maturation. The nephron begins as a pre-tubular aggregate (PA) which forms a renal vesicle (RV), comma-shaped body (CSB), S-shaped body (SSB), capillary loop nephron and mature nephron. The connection between the forming nephron and the lumen of the adjacent ureteric epithelium forms at late renal vesicle stage. Adapted from Little \enquote{Improving our resolution of kidney morphogenesis across time and space} {[}180{]}.}\label{fig:nephrogenesis}
\end{figure}





Most of our understanding of kidney development comes from studies using mouse models and other model species. While these have greatly added to our knowledge they do not completely replicate human kidney development and there are known to be significant differences in the developmental timeline, signalling pathways and gene expression between species {[}181{]}. To better understand human kidney development we need models that reproduce the human version of this process.

\hypertarget{growing-kidney-organoids}{%
\subsection{Growing kidney organoids}\label{growing-kidney-organoids}}

One alternative model of human kidney development is to grow miniature organs in a lab. Known as organoids, these tissues are grown from pluripotent stem cells provided with the right sequence of conditions and growth factors {[}182{]}. Naturally occurring embryonic stem cells can be used but a more feasible approach is to reprogram mature cell types (typically fibroblasts from skin samples) using a method discovered by Takahashi and Yamanaka {[}183,184{]}. Under this protocol, cells are supplied with the transcription factors Oct3/4, Sox2 and Klf4 using retroviral transduction. The resulting cells have the ability to differentiate into any cell type and are known as induced pluripotent stem cells (iPSCs). By culturing iPSCs under the right conditions the course of differentiation can be directed, and protocols for growing a range of tissues have been developed including stomach, intestine, liver, lung and brain {[}185{]}. The first protocol for growing kidney organoids was published in 2015 by Takasato et al. (Figure \ref{fig:organoids}) {[}186{]}.

\begin{figure}

{\centering \includegraphics[width=1\linewidth]{/home/travis/build/lazappi/phd-thesis/figures/01-organoids} 

}

\caption[Diagram of the Takasato et al.~kidney organoid protocol.]{Diagram of the Takasato et al.~kidney organoid protocol. iPSCs are cultured on a plate in the presence of CHIR followed by FGF9. After about a week cells are formed into three-dimensional pellets and a short CHIR pulse is applied. The pellets continue to be cultured in FGF9 for a further 10 days before all growth factors are removed. The organoids now contain tubular structures replicating human nephrons. Adapted from Takasato et al. \enquote{Kidney organoids from human iPS cells contain multiple lineages and model human nephrogenesis} {[}186{]}.}\label{fig:organoids}
\end{figure}





Using this protocol, iPSCs are first grown on a plate where Wnt signalling is induced by the presence of CHIR, an inhibitor of glycogen kinase synthase 3. After several days of growth the growth factor FGF9 is added which is required to form the intermediate mesoderm. Following several more days of growth the cells are removed from the plate and formed into three dimensional pellets. A short pulse of CHIR is added to again induce Wnt signalling and the pellets continue to be cultured in the presence of FGF9. Growth factors are removed after about five days of 3D culture and the organoids continue to grow for a further two weeks, at which point tubular structures have formed. These kidney organoids have been extensively characterised using both immunofluorescence imaging and transcriptional profiling by RNA-seq. Imaging showed that the tubules are segmented and express markers of podocytes, proximal tubule, distal tubule and collecting duct, however individual tubules are not connected in the same way they would be in a real kidney. By comparing RNA-seq profiles with those from a range of developing tissues, the organoids from this protocol were found to be most similar to trimester one and two foetal kidney {[}186{]}. While the bulk transcriptional profiles may be similar, this analysis does not confirm that individual cells types are the same between lab-grown kidney organoids and the true developing kidney. Further studies using this protocol have shown that it is reproducible with organoids grown at the same time having very similar transcriptional profiles, but organoids from different batches can be significantly different, potentially due to differences in the rate at which they develop {[}187{]}.

While they are not a perfect model of a developing human kidney, organoids have several advantages over other models {[}188{]}. In particular, they have great potential for uses in the modelling of developmental kidney diseases {[}189{]}. Cells from a patient with a particular mutation can be reprogrammed and used to grow organoids that can then be used for functional studies or drug screening. Alternatively, gene editing techniques can be used to insert the mutation into an existing cell line or correct the mutation in the patient line, allowing comparisons on the same genetic background. Modified versions of the protocol that can produce much larger numbers of organoids, for example by growing them in swirler cultures {[}190{]}, could potentially be used to produce cells in sufficient numbers for cellular therapies. Extensive work has been done to improve the protocol in other ways as well, such as improving the maturation of the organoids or directing them more towards particular segments. Overall, kidney organoids open up many possibilities for studies to better understand kidney development and potentially help develop new treatments for kidney disease.

\hypertarget{kidney-scrna-seq-studies}{%
\subsection{Kidney scRNA-seq studies}\label{kidney-scrna-seq-studies}}

Several analyses of scRNA-seq from kidney tissues have already been performed, including of developing and adult organs from human and mouse as well as kidney organoids and specific kidney cell types {[}191,192{]}. Park et al.~analysed almost 58\thinspace000 cells from mouse kidneys and found that disease-causing genetic mutations with similar phenotypes arise in the same cell types {[}193{]}. They also identified a new transitional cell type in the collecting duct. Young et al.~compared human foetal, paediatric and adult kidney cells to cells from renal cancers and identified links to specific cells types, including that Wilms tumours are similar to specific foetal cell types and that adult renal cell carcinoma is related to a little-known proximal tubule cell type {[}194{]}. Lindstrm et al.~focused on the developing mouse and human kidney {[}195{]} while Wu et al.~examine kidney organoids {[}196{]}. They found that organoid protocols produced a range of cell types but that those cells were immature and 10--20 percent of cells came from non-kidney lineages. Many of these were neuronal and their formation could be limited by inhibiting the BDNF/NTRK2 pathway. In \textbf{Chapter \ref{analysis}} I analyse another organoid dataset generated by collaborators using the Takasato protocol for the purposes of profiling the cell types present. I demonstrate a range of analysis tools, focusing on the decisions that need to be made during the analysis and the effects they might have on the results.

\hypertarget{intro-overview}{%
\section{Thesis overview and aims}\label{intro-overview}}

The tools and techniques for analysing scRNA-seq data have developed rapidly along with the technology. My thesis aims to chart the progress of these methods over the last few years, contribute to the development in the field and apply those methods to a scRNA-seq dataset to better understand the cellular composition of kidney organoids. I have completed these aims in the following ways:

\begin{enumerate}
\def\labelenumi{\arabic{enumi}.}
\item
  Construction and maintenance of a database recording the development of scRNA-seq analysis methods and analysis of its contents (\textbf{Chapter \ref{tools}}).
\item
  Development of a software package and method for easily simulating realistic scRNA-seq datasets that can be used for the evaluation and development of analysis methods (\textbf{Chapter \ref{simulation}}).
\item
  Development of an algorithm for visualising clustering results at multiple resolutions that can be used to select clustering parameters and a software package that implements it (\textbf{Chapter \ref{clust-trees}}).
\item
  Application of scRNA-seq analysis methods to a kidney organoid dataset in order to uncover and better understand the cells types that are present and demonstrate differences between approaches (\textbf{Chapter \ref{analysis}}).
\end{enumerate}

\hypertarget{tools}{%
\chapter{The scRNA-seq tools landscape}\label{tools}}

\begin{quote}
\textbf{Have you ever seen a sound}

\textbf{Have you listened to an image}

\textbf{Have you ever touched a thought}

\textbf{Have you ever tasted nothing}

\textbf{Have you ever told a lie}

\textbf{That was true more than truth}

\textbf{Because truth it had lied}

\textbf{All its life when it spoke to you?}

--- The Cat Empire

\emph{Miserere}, 2005
\end{quote}

\begin{center}\includegraphics[width=1\linewidth]{/home/travis/build/lazappi/phd-thesis/figures/02-tools-sketch} \end{center}

\clearpage

\hypertarget{introduction-1}{%
\section{Introduction}\label{introduction-1}}

When I began my PhD in early 2016, single-cell RNA sequencing technologies were just beginning to become widely available. Since then there has been a rapid uptake of the technologies and there are now many studies using this approach. Along with the growth in the adoption of scRNA-seq technologies, there has been an explosion in the number of software tools for analysing these datasets. This chapter charts the growth in the scRNA-seq analysis landscape over time.

In 2016 there were relatively few analysis methods available and to answer questions like how many tools perform a particular task, which areas are developers focusing on or is there a tool for doing a particular task, I began to record details about them. Inspired by similar projects such as Sean Davis's Awesome Single Cell page {[}197{]} I decided to make this collection public. This turned out to be useful to other researchers and over time a simple spreadsheet became the scRNA-tools database and website (\url{https://scRNA-tools.org}). Our paper published in \emph{PLoS Computational Biology} describing this resource forms the main part of this chapter {[}198{]}.

By having access to details about existing analysis tools we were able to explore how the field has developed. We found that computational researchers had focused their efforts on analysis tasks specific to scRNA-seq data, such as clustering and ordering of cells or handling the larger numbers of zero counts. We also saw that many of the tools performed tasks common to several stages of analysis, including various approaches to dimensionality reduction and visualisation of data and results. Details recorded in the database show that developers of scRNA-seq analysis tools tend to embrace an open-source and open-science approach, for example most tools are developed on GitHub where others can ask questions and submit improvements. The majority of tools are also available under open-source licenses allowing their code to be reused for other purposes, although there is also a significant proportion that do not have any associated license. Tools are commonly made public by releasing a preprint publication, making them quickly available to the community and giving early adopters a chance to contribute to their development.

Sections at the end of this chapter present a version of some the analysis presented in the paper based on the most recent version of the database (\textbf{Section \ref{tools-current}}), as well as details about the usage of the scRNA-tools website over time (\textbf{Section \ref{tools-usage}}).

\hypertarget{scrna-tools-publication}{%
\section{scRNA-tools publication}\label{scrna-tools-publication}}

\frame{\includegraphics[page=1, width=\textwidth]{figures/02-scRNAtools-paper-cropped}}\clearpage
\frame{\includegraphics[page=2, width=\textwidth]{figures/02-scRNAtools-paper-cropped}}\clearpage
\frame{\includegraphics[page=3, width=\textwidth]{figures/02-scRNAtools-paper-cropped}}\clearpage
\frame{\includegraphics[page=4, width=\textwidth]{figures/02-scRNAtools-paper-cropped}}\clearpage
\frame{\includegraphics[page=5, width=\textwidth]{figures/02-scRNAtools-paper-cropped}}\clearpage
\frame{\includegraphics[page=6, width=\textwidth]{figures/02-scRNAtools-paper-cropped}}\clearpage
\frame{\includegraphics[page=7, width=\textwidth]{figures/02-scRNAtools-paper-cropped}}\clearpage
\frame{\includegraphics[page=8, width=\textwidth]{figures/02-scRNAtools-paper-cropped}}\clearpage
\frame{\includegraphics[page=9, width=\textwidth]{figures/02-scRNAtools-paper-cropped}}\clearpage
\frame{\includegraphics[page=10, width=\textwidth]{figures/02-scRNAtools-paper-cropped}}\clearpage
\frame{\includegraphics[page=11, width=\textwidth]{figures/02-scRNAtools-paper-cropped}}\clearpage
\frame{\includegraphics[page=12, width=\textwidth]{figures/02-scRNAtools-paper-cropped}}\clearpage
\frame{\includegraphics[page=13, width=\textwidth]{figures/02-scRNAtools-paper-cropped}}\clearpage
\frame{\includegraphics[page=14, width=\textwidth]{figures/02-scRNAtools-paper-cropped}}\clearpage

\hypertarget{tools-current}{%
\section{The current scRNA-tools database}\label{tools-current}}

Since the paper was published in June 2018, 227 tools have been added to the scRNA-tools database, taking the total to 459. Figure \ref{fig:tools-plot} shows an overview of the current database as of 1 August 2019. Overall the analysis landscape has not changed significantly over the last few months.

\begin{figure}

{\centering \includegraphics[width=1\linewidth]{/home/travis/build/lazappi/phd-thesis/figures/02-tools} 

}

\caption[Current state of the scRNA-tools database.]{Current state of the scRNA-tools database. (A) Size of the database over time. (B) Publication status of tools in the database. Most tools are published as either peer-reviewed publications or preprints. (C) Associated software licenses. Most tools are covered by open-source licenses but around a quarter of tools still have no license. (D) Software platforms used by tools, some tools use more than one language. R is the most common programming language followed by Python. (E) Number of tools that complete each analysis task. Tasks associated with multiple analysis phases are the most common, along with those that assign cells.}\label{fig:tools-plot}
\end{figure}





Over half of tools are still associated with a peer-reviewed publication with another third described in preprints (Figure \ref{fig:tools-plot}B). Versions of the GNU Public License (GPL) continue to be the most common license for scRNA-seq software tools, followed by the more permissive MIT license (Figure \ref{fig:tools-plot}C). The use of preprints to quickly share methods and observations and open-source licenses which allow code to be reused and adapted reflects the general willingness to share that I have observed in the scRNA-seq bioinformatics community during my PhD. Unfortunately, around a quarter of tools do not have associated licenses, which can limit their usefulness for other developers. I believe that in most cases this is not an attempt by authors to restrict how their work is used, but a lack of awareness about the importance of clear licensing. By displaying licenses on the scRNA-tools website I hope to encourage developers to consider how they license their work in the future.

R continues to be the most popular programming platform and is used in some way by around two thirds of scRNA-seq analysis tools (Figure \ref{fig:tools-plot}D). Around a third of tools are built using Python which is the next most common programming language, followed by C++ and MATLAB which are each used by just under 10 percent of tools. The dominance of R reflects the statistical nature of many of the tasks involved in analysing scRNA-seq data as well as the historical usage of the platform for bulk RNA-seq and other genomic data types. The Bioconductor project {[}199{]} has been integral to promoting the use of R for analysing genomic data and already includes a range of tools for scRNA-seq analysis {[}200{]}. However, as the size of scRNA-seq datasets continues to increase and approaches to analysis change to include more machine learning methods I expect the platforms used to change to some degree. In particular we will see more use of compiled languages like C++, Swift, Julia and Go used for their improved performance. While MATLAB is commonly used within a small section of the academic community, it is proprietary software that requires a paid license and tools developed in MATLAB cannot be used by the majority of scRNA-seq analysts.

The analysis tasks that tools perform reflect the advantages and uses of scRNA-seq data (Figure \ref{fig:tools-plot}E). The most common categories include those that are associated with multiple phases of analysis, particularly visualisation, for displaying results and checking distributions, and dimensionality reduction. The dimensionality reduction category includes tools that make use of a lower dimensional space as part of their analysis approach, as well as several techniques that have been specifically developed for projecting scRNA-seq data. Clustering and ordering of cells, which are tasks specific to scRNA-seq analysis, are also common categories. I expect these tasks to continue to be included in new software tools as they are important steps in scRNA-seq analysis, however I hope that the focus turns from developing new clustering and ordering methods to identifying those that are consistently effective and developing robust, easy to use and well-documented software tools that implement them. The availability of reliable and complete reference datasets should lead to the development of more that make use of references such as the classification of cells, and developers will continue to push the boundaries of what is possible using scRNA-seq data.

\hypertarget{tools-usage}{%
\section{Usage of the scRNA-tools website}\label{tools-usage}}

Since it was launched in July 2017 the scRNA-tools website has been used by people from around the world (Figure \ref{fig:tools-analytics-plot}). Usage has increased since release of the publication and the website now receives an average of over 1000 visitors per month (Figure \ref{fig:tools-analytics-plot}A). These users come from over 90 countries around the world (Figure \ref{fig:tools-analytics-plot}B) with the USA being by far the most common location, followed by China, Japan, Australia, the United Kingdom and countries in Western Europe (Figure \ref{fig:tools-analytics-plot}C). Aggregating by continent (Figure \ref{fig:tools-analytics-plot}D) better reflects population size and the size of academic communities, with the Americas responsible for the majority of traffic followed by Europe and Asia.

\begin{figure}

{\centering \includegraphics[width=1\linewidth]{/home/travis/build/lazappi/phd-thesis/figures/02-analytics} 

}

\caption[Usage of the scRNA-tools website.]{Usage of the scRNA-tools website. (A) Usage over time showing the number of users per day (pink), week (blue) or 30 days (green). Labelled vertical lines indicate events that may have influenced traffic. (B) World map coloured according to number of users from low (blue) to high (yellow). (C) Number of users from the top 10 most common countries. (D) Number of users by continent.}\label{fig:tools-analytics-plot}
\end{figure}





Although these figures may not seem large when compared to an average website, they represent a significant part of the scRNA-seq community. Seeing usage of the website from so many countries sustained over this period of time suggests that the database is a useful resource. I intend to continue maintain and grow the resource by adding new tools and updating those in the database. I hope to encourage contributions from the community who can easily submit changes via a form on the website.

\hypertarget{simulation}{%
\chapter{Simulating scRNA-seq data}\label{simulation}}

\begin{quote}
\textbf{\enquote{Essentially, all models are wrong, but some are useful.}}

--- George E. P. Box and Norman R. Draper

\emph{Empirical Model-Building and Response Surfaces}, 1987
\end{quote}

\begin{center}\includegraphics[width=1\linewidth]{/home/travis/build/lazappi/phd-thesis/figures/03-simulation} \end{center}

\clearpage

\hypertarget{introduction-2}{%
\section{Introduction}\label{introduction-2}}

For a computational analysis method to be accepted and used by the community it needs to demonstrate that is is effective at the task it aims to complete. Ideally this can be achieved by evaluating the performance of the method on a real dataset where the results are already known. Unfortunately, in many cases such gold standard datasets are not available. This is particularly true for genomic data where it is difficult to know what the truth is or the known truth is limited to only small sections of the genome. It is possible to create some genomic datasets where the truth is known, for example through carefully performed mixing experiments {[}175{]}, but these often do not capture the true biological complexity. In many cases the most effective way to evaluate an analysis method during development is by testing it on simulated datasets. Simulations have the additional advantage of being relatively cheap and easy to produce, allowing exploration of a wide range of possible parameters. This is the approach taken by many early methods for scRNA-seq analysis. Unfortunately, these simulations were often not well explained, code for reproducing them was not available and perhaps most importantly they didn't show that the synthetic datasets were similar to real scRNA-seq data.

This chapter presents Splatter, a software package for simulating scRNA-seq datasets presented in a publication in \emph{Genome Biology} {[}201{]}. Splatter is designed to provide a consistent, easy-to-use interface for multiple previously developed scRNA-seq simulation models. We do this by providing a consistent framework for each model defined by two functions: one which estimates parameters from a real dataset and a second that generates a synthetic dataset using those parameters. Each simulation model has different assumptions and reproduces different aspects of scRNA-seq data, and we explain these differences in our publication. We also present Splat, our own simulation model based on the Gamma-Poisson distribution. This model includes several important aspects of scRNA-seq data including highly expressed outliers genes, differences in library sizes between cells, a relationship between the mean and the variance of each gene and the ability to add a dropout effect linked to gene expression. When designing the Splat simulation our goal was to accurately reproduce scRNA-seq count data rather than test a specific method, with the result being that the model is highly flexible and able to generate a range of scenarios including datasets with multiple groups of cells, batch effects and continuous trajectories.

In the publication we compare how well each simulation reproduces a range of scRNA-seq datasets, including UMI and full-length protocols, different capture methods, and homogenous and complex tissues. We found that the Splat simulation was a good match for some of these datasets across a range of methods, however it was also clear that some models more faithfully reproduced different aspects of the data, particularly for datasets from different sources. The Splatter R package is available for download from Bioconductor (\url{https://bioconductor.org/packages/splatter}). Additional files for the publication are provided in \textbf{Appendix \ref{splatter-additional}}. Following the publication (\textbf{Section \ref{splatter-publication}}) I discuss updates to the software package and provide an updated comparison of the current simulation models (\textbf{Section \ref{splatter-update}}). Vignettes describing the use of Splatter (\textbf{Appendix \ref{splatter-vignette}}) and the parameters of the Splat model (\textbf{Appendix \ref{splat-params-vignette}}) are included as appendices, as is the full Splatter manual (\textbf{Appendix \ref{splatter-manual}}).

\hypertarget{splatter-publication}{%
\section{Splatter publication}\label{splatter-publication}}

\frame{\includegraphics[page=1, width=\textwidth]{figures/03-splatter-paper-cropped}}\clearpage
\frame{\includegraphics[page=2, width=\textwidth]{figures/03-splatter-paper-cropped}}\clearpage
\frame{\includegraphics[page=3, width=\textwidth]{figures/03-splatter-paper-cropped}}\clearpage
\frame{\includegraphics[page=4, width=\textwidth]{figures/03-splatter-paper-cropped}}\clearpage
\frame{\includegraphics[page=5, width=\textwidth]{figures/03-splatter-paper-cropped}}\clearpage
\frame{\includegraphics[page=6, width=\textwidth]{figures/03-splatter-paper-cropped}}\clearpage
\frame{\includegraphics[page=7, width=\textwidth]{figures/03-splatter-paper-cropped}}\clearpage
\frame{\includegraphics[page=8, width=\textwidth]{figures/03-splatter-paper-cropped}}\clearpage
\frame{\includegraphics[page=9, width=\textwidth]{figures/03-splatter-paper-cropped}}\clearpage
\frame{\includegraphics[page=10, width=\textwidth]{figures/03-splatter-paper-cropped}}\clearpage
\frame{\includegraphics[page=11, width=\textwidth]{figures/03-splatter-paper-cropped}}\clearpage
\frame{\includegraphics[page=12, width=\textwidth]{figures/03-splatter-paper-cropped}}\clearpage
\frame{\includegraphics[page=13, width=\textwidth]{figures/03-splatter-paper-cropped}}\clearpage
\frame{\includegraphics[page=14, width=\textwidth]{figures/03-splatter-paper-cropped}}\clearpage
\frame{\includegraphics[page=15, width=\textwidth]{figures/03-splatter-paper-cropped}}\clearpage

\hypertarget{splatter-update}{%
\section{Updates to Splatter}\label{splatter-update}}

I have continued to develop the Splatter package since publication, improving the package infrastructure and implementing additional simulation models. This section briefly describes changes to the software since the publication and provides an up-to-date comparison of the current simulation models.

The most significant change to the infrastructure of Splatter is switching from using the SCESet object provided by early versions of the scater package as the output of simulation functions, to using the SingleCellExperiment object {[}202{]}. This object has been available since mid-2017 and is intended to be the core object for all Bioconductor packages working with scRNA-seq data. It is based on Bioconductor's SummarizedExperiment object which consists of feature by sample matrices linked to annotation data frames for both features and samples. The SingleCellExperiment object adds features specifically useful for scRNA-seq data, including slots for storing the results of dimensionality reduction techniques and normalisation size factors, as well as conveniently named accessors for information commonly used in analyses. In practical terms, changing objects has not changed how Splatter is used but it does ensure that the results of Splatter simulations are immediately compatible with tools in the Bioconductor ecosystem. Display of the Splatter parameters objects has also been improved, including colouring of output to highlight which parameters have been changed from the default values.

Minor changes have also been made to the Splat simulation model. At the request of users we have modified the parameters controlling additional dropout to allow them to be specified by whole experiment (the previous method), sample batch, cell group or individual cell. This allows users investigating the effects of dropout to simulate datasets where different cell types or batches are differently affected by the dropout phenomenon. Many of the other parameters could already be specified at different levels, but extending this to the dropout parameters provides extra flexibility to the Splat model. In the original model the total number of reads in each cell was sampled from a log-normal distribution, however some users have come across datasets where the normal distribution was a better fit, which caused the Splat estimation procedure to fail. We now allow either the log-normal or normal distributions to be used for this part of the model and the estimation procedure attempts to identify which is more appropriate for each input dataset.

Another model that has seen changes is that from the BASiCS package. The authors of BASiCS have made significant improvements to how they estimate parameters, particularly for datasets without internal spike-ins but where multiple batches exist. BASiCS now uses non-linear regression to capture the trend between gene mean and over-dispersion {[}110{]}. This method is now used as part of the default estimation procedure for the BASiCS model within Splatter, resulting in simulations that are a much better match for the original dataset.

I have also added four new simulation models to Splatter, bringing the number of models to 10. These new models are:

\begin{itemize}
\tightlist
\item
  mfa -- Simulation of a bifurcating pseudotime trajectory. This simulation can optionally include genes with changes in expression along the trajectory and added dropout {[}203{]}.
\item
  PhenoPath -- Pseudotime trajectory including genes that show differential expression, changes along pseudotime, interaction effects or combinations of these {[}204{]}.
\item
  ZINB-WaVE -- Sophisticated zero-inflated negative-binomial distribution based simulation that includes cell and gene-level covariates {[}63{]}.
\item
  SparseDC -- Simulates a set of clusters across two conditions, where some clusters may be present in only one condition {[}205{]}.
\end{itemize}

More details about the models used by these packages are available in the associated publications and package documentation.

\hypertarget{performance-of-current-simulations}{%
\subsection{Performance of current simulations}\label{performance-of-current-simulations}}

\FloatBarrier

To demonstrate the performance of the current simulation models I have repeated some of the analysis from the publication using the Tung dataset {[}206{]}. Parameters for each of the models were estimated from a sample of 500 cells from the Tung data and synthetic datasets were produced for each model using these parameters. The time taken for the estimation and simulation steps is shown in \textbf{Appendix \ref{sim-timings}} and versions of the packages used in \textbf{Appendix \ref{sim-packages}}. The comparison functions within Splatter were then used to evaluate the differences between the simulated datasets and the real data they were based on. Figure \ref{fig:simulation-summary} summarises this evaluation by showing the ranking of the median absolute deviation (MAD) for each model across a range of metrics.

\begin{figure}

{\centering \includegraphics[width=1\linewidth]{/home/travis/build/lazappi/phd-thesis/figures/03-sim-summary} 

}

\caption[Performance of current Splatter simulations.]{Performance of current Splatter simulations. Synthetic datasets were produced using parameters estimated from 500 random cells from the Tung dataset. Plot shows the rank of median absolute deviation (MAD) from the real dataset on a variety of measures from least similar (blue) to most similar (yellow). The ZINB-WaVE model best reproduces this dataset, followed by the updated BASiCS model. The Splat model also performs well despite having a less sophisticated estimation procedure.}\label{fig:simulation-summary}
\end{figure}





This comparison clearly shows that the ZINB-WaVE model produces the closest match to this dataset. The similarity to the real dataset suggests that the model used by ZINB-WaVE, and particularly the procedure used to estimate the parameters in it, is effective for scRNA-seq data. ZINB-WaVE was originally designed for dimensionality reduction but also has applications in normalisation and differential expression testing {[}64{]}. In contrast to the previously published analysis, the BASiCS model is also a very good match, with the improvement in performance coming from the new regression-based estimation method. Unfortunately, the BASiCS normalisation method relies on knowing predefined groups of cells, which limits its usefulness for exploratory experiments. The Splat method does not match the performance of ZINB-WaVE and BASiCS but still does a reasonable job of reproducing this dataset, despite having a much simpler (and quicker) estimation procedure. While producing realistic datasets is extremely important, the real strength of the Splat model lies in its flexibility to simulate many kinds of experiment and its reporting of intermediate values that can be used to evaluate methods. As observed in the publication, we again see that the zero-inflated versions of the Splat and Lun2 models perform worse than the regular versions on this UMI dataset. However, this is in contrast to the extremely good performance of ZINB-WaVE which also uses a zero-inflated model. Taken together these results suggest that both regular and zero-inflated models can be effective for this type of data, if the parameters are estimated correctly.

The other three new simulation models (mfa, PhenoPath and SparseDC) do not do a particularly good job of reproducing this dataset, and in particular mfa produces a dataset with a very different number of counts per cell. These simulations have mostly been designed to demonstrate the analysis methods in these packages and the estimation procedures associated with them are relatively simple. This result underlines that the key to producing a realistic simulation of any kind of data lies not so much in the simulation model itself but in choosing good parameters for that model. As we describe in the publication, there are caveats around the estimation procedures for some of the other models, specifically the Lun model, which only estimates the number of genes and cells, and the scDD model, which is designed for a high-quality filtered dataset unlike the raw dataset used here. Another important consideration is that the comparison shown here is based on a single dataset. As we showed in the publication, different datasets have different biological and technical features which may be reproduced more accurately by alternative simulation models.

Comparing simulations and real datasets is a difficult task. For the comparisons in the Splatter publication we focused on simple summary statistics such as mean gene expression or percentage of zeros per cell and the relationships between them. However, there are other measures that could be considered such as correlation between features or estimated dispersions. Statistical tests can also be used to formally evaluate the similarity of datasets. The countsimQC package {[}207{]} can be used to produce a comprehensive report comparing count datasets in a variety of ways and could be a useful tool for any future evaluations of simulation models.

\hypertarget{clust-trees}{%
\chapter{Visualising clustering across resolutions}\label{clust-trees}}

\begin{quote}
\textbf{\enquote{Don't do what you can't undo, until you've considered what you can't do once you've done it.}}

--- Robin Hobb

\emph{Assassin's Apprentice}, 1995
\end{quote}

\begin{center}\includegraphics[width=1\linewidth]{/home/travis/build/lazappi/phd-thesis/figures/04-clustering-trees} \end{center}

\clearpage

\hypertarget{introduction-3}{%
\section{Introduction}\label{introduction-3}}

Clustering of cells to form groups is a common task when analysing scRNA-seq data. This is different from analysis of bulk RNA-seq datasets where the groups are usually known in advance and clustering isn't required. Because clustering is a new step in analysis of scRNA-seq dat it has received a lot of attention from scRNA-seq methods developers. The need to group samples is not unique to genomic data and clustering techniques are used in many other fields for a wide variety of purposes. Whatever kind of data is of interest, and whatever clustering method is being used, a question that commonly comes up is how many clusters do we want to have? Depending on the clustering method, the number of clusters can be controlled by setting an exact value or by changing a parameter that controls the clustering resolution. The number of clusters can also be indirectly affected by the values of related parameters. The number of clusters that are used in an analysis can often have a profound affect on how the results are interpreted. Existing measures of clustering typically only consider a single clustering resolution at a time or require multiple rounds of permutations and clustering which can be infeasible for large datasets. In this chapter I propose an alternative, visualisation-based aid for deciding which clustering resolution to use.

Clusterings of the same dataset at different resolutions are often related and it is common for new clusters formed at higher resolutions to form by splitting existing clusters. However, when comparing clusterings it is not always clear what those relationships are and how significant they might be. The method I describe here was published in \emph{GigaScience} and proposes clustering datasets at multiple resolutions and considering the overlap in samples between clusters at neighbouring resolutions {[}208{]}. By doing this we can build a graph structure we call a \enquote{clustering tree}. Visualising this tree allows us to see where new clusters form, how they are related and the stability of particular clustering resolutions. In the publication we demonstrate this approach using simulated datasets, a simple dataset commonly used as an example for machine learning techniques and a complex scRNA-seq dataset from blood.

While the structure of clustering trees can help choose a clustering resolution to use for an analysis, clustering trees are more generally a compact, information-dense visualisation that can show information across clustering resolutions. This is something that is not possible with traditional visualisations used for clustering results such as t-SNE projections, and is achieved by trading individual information about each sample for summarised information about clusters and adding a resolution dimension. Overlaying important domain knowledge (such as the expression of known marker genes) onto these visualisations can be particularly informative and we also demonstrate this in our paper (\textbf{Section \ref{clustree-publication}}). Section \ref{clustree-overlay} demonstrates how clustering trees can be combined with other visualisations of scRNA-seq data.

Clustering trees can be produced using the clustree R package which is built on the tidygraph {[}209{]}, igraph {[}210{]} and ggraph {[}211{]} packages and is available from CRAN (\url{https://cran.r-project.org/package=clustree}). The clustree vignette is included as \textbf{Appendix \ref{clustree-manual}} and the manual as \textbf{Appendix \ref{clustree-manual}}.

\hypertarget{clustree-publication}{%
\section{Clustering trees publication}\label{clustree-publication}}

\frame{\includegraphics[page=1, width=\textwidth]{figures/04-clustering-trees-paper-cropped}}\clearpage
\frame{\includegraphics[page=2, width=\textwidth]{figures/04-clustering-trees-paper-cropped}}\clearpage
\frame{\includegraphics[page=3, width=\textwidth]{figures/04-clustering-trees-paper-cropped}}\clearpage
\frame{\includegraphics[page=4, width=\textwidth]{figures/04-clustering-trees-paper-cropped}}\clearpage
\frame{\includegraphics[page=5, width=\textwidth]{figures/04-clustering-trees-paper-cropped}}\clearpage
\frame{\includegraphics[page=6, width=\textwidth]{figures/04-clustering-trees-paper-cropped}}\clearpage
\frame{\includegraphics[page=7, width=\textwidth]{figures/04-clustering-trees-paper-cropped}}\clearpage
\frame{\includegraphics[page=8, width=\textwidth]{figures/04-clustering-trees-paper-cropped}}\clearpage
\frame{\includegraphics[page=9, width=\textwidth]{figures/04-clustering-trees-paper-cropped}}\clearpage

\hypertarget{clustree-overlay}{%
\section{Overlaying clustering trees}\label{clustree-overlay}}

In the clustering trees publication we focused on the use of the visualisation as a tool for selecting a clustering resolution, but more generally the clustering tree structure can be used to display information across the resolution dimension. The visualisation that is most commonly used to display scRNA-seq data is the dimensionality reduction plot produced using methods such as PCA, t-SNE {[}147{]} or UMAP {[}148{]}. In this visualisation each point is a cell placed in a two-dimensional space and coloured using factors such as the assigned cluster or the expression level of a gene. The two visualisations display complementary information. The dimensionality reduction plot shows the location of each individual sample in a low-dimensional space, while the clustering tree trades that sample level resolution for a summarised view of each cluster and swaps the spatial information for a new resolution dimension. It is possible to combine these two approaches by overlaying a clustering tree onto a dimensionality reduction plot. Figure \ref{fig:clustree-overlay} shows an example of this using the iris dataset {[}212,213{]}.

\begin{figure}

{\centering \includegraphics[width=1\linewidth]{/home/travis/build/lazappi/phd-thesis/figures/04-clustree-overlay} 

}

\caption[Clustering tree of the iris dataset overlaid on a PCA plot.]{Clustering tree of the iris dataset overlaid on a PCA plot. (A) A PCA plot of the samples in the iris dataset coloured by cluster at \(k = 5\) overlaid with a clustering tree showing clustering resolutions from \(k = 1\) to \(k = 5\). (B) The same plot shown from the direction of the x-axis (PC1), the y-axis now shows the clustering resolution and samples are jittered along the x-axis. (C) Same as (B) but from the direction of the y-axis (PC2).}\label{fig:clustree-overlay}
\end{figure}





Here instead of using a tree layout algorithm to decide where the cluster nodes are placed, we place them in the low-dimensional space at the mean position of the samples in each cluster. The easiest way to view this overlaid plot (Figure \ref{fig:clustree-overlay}A) is to imagine looking down on a landscape from above. On the ground we have the individual sample points (here coloured by their highest resolution cluster) and rising up from that we have the clustering tree with the highest resolutions placed closest to the ground and the lowest resolution at the top. As we step down the clustering tree we can see where in this space new clusters are formed. Figures \ref{fig:clustree-overlay} B and C show this same view but from a different perspective. Here imagine standing on the ground at the same level as the samples and looking towards the clustering tree. Now the x-axis shows one of the reduced dimensions (PC1 or PC2) and the y-axis show the resolution dimension. Looking at a clustering tree in this way provides an alternative view where the focus is less on the structure of the tree itself and more on how it fits into a space that researchers are more familiar with.

\hypertarget{analysis}{%
\chapter{Analysis of kidney organoid scRNA-seq data}\label{analysis}}

\begin{quote}
\textbf{\enquote{And the first lesson of all was the basic trust that he could learn. It's shocking to find how many people do not believe they can learn, and how many more believe learning to be difficult.}}

--- Frank Herbert

\emph{Dune}, 1965
\end{quote}

\begin{center}\includegraphics[width=1\linewidth]{/home/travis/build/lazappi/phd-thesis/figures/05-analysis} \end{center}

\clearpage

\hypertarget{introduction-4}{%
\section{Introduction}\label{introduction-4}}

The kidney has a branching structure where incoming blood arteries are split into smaller and smaller vessels. At the end of the smallest capillaries are nephrons, the functional unit of the kidney. Blood is filtered in the glomerulus and the filtrate passes through the proximal tubule, loop of Henle, and distal tubule, then into the collecting duct. Each section of the nephron is made up of specialised cell types responsible for balancing concentrations of particular molecular species between the filtrate and blood. To do this they produce a range of specific transporter molecules that secrete or extract molecules including metal ions, small organic molecules and water. Disruptions to these important cell types can have a profound effect on the body as the delicate balance between concentrations of molecular species is lost.

Kidney disease can develop over a lifetime as the results of a range of factors. When the condition becomes chronic the only reliable treatments currently available are dialysis, which is time consuming and unpleasant, or a kidney transplant. Some kidney conditions are purely genetic and are caused by defects in kidney development or structure rather than damage from external factors. Understanding kidney development and structure is key to understanding these diseases and developing treatments for them.

One way to investigate kidney development is to grow kidney tissue from stem cells {[}182,188{]}. By providing the correct growth factors and conditions, stem cells which could become any cell type can be directed to differentiate into kidney tissue. Part of this process involves transferring the cells from growing in a flat, two-dimensional layer on a plate to a three-dimensional environment, where different types of cells can interact with each other and form structures that replicate a nephron. These tissues are known as kidney organoids and can be used to model various aspects of kidney development, but the extent to which organoids are useful depends on how well they replicate real kidney cell types.

Organoids have typically been characterised by immunofluorescence imaging using antibodies for known markers of nephron segments and cell types. This approach makes it easy to see what structures are present in an organoid and if their morphology resembles real tissues. However, using immunofluorescence relies on prior knowledge about which proteins are important, as well as antibodies for those proteins being available. Typically only four or five colour channels are available for immunofluorescence imaging, limiting the number of proteins that can be shown at once. Bulk RNA sequencing has also been used to characterise kidney organoids. While this approach is less biased and measures expression across the transcriptome, it only provides a global average of expression across an organoid and cannot show exactly which cell types are present or which genes are expressed in the same cells. Single-cell RNA-seq addresses some of these limitations by measuring expression in individual cells, allowing cells with similar expression profiles to be identified and investigated.

As in many other areas, scRNA-seq has been rapidly embraced as a useful tool in studying kidney development and there have already been several studies investigating the cell types present in adult kidney, developing kidney and kidney organoids. During my PhD I have contributed to this aspect of several studies {[}187,190,214,215{]}. The work I was most closely involved in profiled a set of kidney organoid scRNA-seq samples to identify the cell types that were present. The analysis we presented in this publication followed a standard scRNA-seq workflow including pre-processing, quality control, clustering and ordering of cells, and detection of marker genes. But there were still many decisions we had to make during the analysis. Which tools should be used for each step and which parameters should we use for these tools? Should we have used any alternative analysis approaches? In this chapter I revisit some of those decisions, demonstrate some alternative methods and discuss what difference they might have made to the results.

\hypertarget{summary-of-published-work}{%
\section{Summary of published work}\label{summary-of-published-work}}

The dataset I present here consists of scRNA-seq data from three kidney organoids. Cells from the CRL1502-C32 human iPSC cell line were cultured according to the Takasato kidney organoid protocol {[}186{]} to day 25. Three organoid samples were taken and dissociated before capturing cells in parallel on 10x Genomics Single Cell Chips and preparing sequencing libraries using the 10x Genomics Single Cell Library Kit V2. Sequencing was performed on an Illumina HiSeq using 100 base pair paired-end reads at the Australian Genome Research Facility. The organoid differentiation and cell isolation was performed by Pei Xuan Er and Alex Combes.

This dataset was first introduced in \enquote{Evaluation of variability in human kidney organoids} {[}187{]}. This study investigated the robustness and repeatability of the organoid differentiation protocol by comparing RNA-seq data from multiple batches, cell lines and time points. We found that organoids from the same batch had highly correlated expression profiles but that there could be significant differences between batches. The genes that were differentially expressed between batches were compared with a time course of organoids taken at different stages in the protocol. We saw that many of the differences between batches were in genes involved in organoid maturation, suggesting that some batches grow faster than others. This is an important consideration for experiments that want to make comparisons between organoids, for example between disease and gene-corrected cell lines. To confirm this result at the single-cell level we compared the organoid dataset described above with a second batch containing a single organoid. Integration and clustering with Seurat {[}119,129{]} produced 13 clusters that showed expression of markers for important kidney cell types as well as some off-target populations. These clusters contained cells from all four organoids but the proportions were different, particularly between the batches, with the fourth organoid being depleted for stromal populations and showing lower expression of mature podocyte genes. While this result reinforces the findings from the bulk RNA-seq data, it is difficult to say what might be the effect of different patterning, dissociation and cell capture rather than organoid maturation. The computational and statistical analysis for this study was performed by Belinda Phipson. I performed pre-processing of the single-cell data using Cell Ranger and some of the early analysis.

While this study made use of kidney organoid scRNA-seq data it didn't fully profile the cell types present. To better understand what cell populations are present in these organoids and detectable using scRNA-seq, we performed a re-analysis of this data that was published as \enquote{Single-cell analysis reveals congruence between kidney organoids and human fetal kidney} {[}214{]}. This study was lead by Alex Combes who performed much of the lab work and interpretation of results while I performed all the computational analysis. Details of this analysis are presented online at \url{http://oshlacklab.com/combes-organoid-paper/} {[}216{]}. The full text of this publication is included in \textbf{Appendix \ref{organoid-paper}} but I will summarise the results here. Using a similar workflow based around the Seurat package we identified 13 clusters including clusters that were labelled as covering kidney stroma, podocytes, epithelium and endothelium, based on the expression of known kidney marker genes. Other clusters were associated with active phases of the cell cycle as well as off-target muscle progenitor, neural progenitor and glial cells types. Re-clustering of the podocyte and epithelial clusters revealed more detail of the nephron lineage with five clusters covering nephron progenitors, podocyte precursors, mature podocytes, proximal tubule precursors and distal tubule precursors. We used Monocle {[}77,138{]} to order the cells along a pseudotime trajectory which showed two branches, with nephron progenitors heading towards either a podocyte or tubular fate. Characterising the samples in this way allowed us to identify many of the cell types we would expect to be present in a developing human kidney, but this process cannot show exactly how similar they are to real tissue. For example we found podocytes in these organoids, but are organoid podocytes the same as normal kidney podocytes?

To answer this question we used the Seurat integration method based on canonical correlation analysis {[}120,121{]} to combine our kidney organoid datasets with an scRNA-seq dataset from human foetal kidney previously published by Lindstrm et al. {[}195{]}. Clustering of the combined dataset produced 16 clusters, most of which contained cells from both datasets and covered similar cell types to what we saw in the organoid-only analysis. Some clusters were only present in one dataset, including the off-target muscle and neural cell types in the organoids and blood cells in the foetal kidney data. One of the new clusters represented a population of immune cells. Most of these came from the foetal kidney dataset but a few originated from the organoids. While there were insufficient cells to truly say that immune cells are present in these organoids it is possible they could arise somewhere during the differentiation process.

As with the organoid data we looked more closely at the nephron lineage in the combined dataset. We saw evidence of similar developmental trajectories with more mature podocytes in the organoid samples and fewer distal tubule cells. At this resolution we also found two clusters unique to the human foetal kidney data: a stromal population that may be more stem-like in nature and originally clustered with the nephron progenitors, and a separate podocyte cluster. Some of the differences between the datasets can be explained by the differences between the samples, for example the human foetal kidney dataset comes from a section of the outer cortex rather than a whole organ. For this reason it is important to look for differences between the datasets within cells that have been labelled as the same cell type. Doing this revealed an overall stress or heatshock signature in the foetal kidney data that is most likely the result of the tissue being exposed to suboptimal conditions and made it difficult to see smaller differences. However, we were able to identify some differences, including in signalling pathways, that may suggest further avenues for directing kidney organoid differentiation. Overall our analyses of this dataset support the presence of important kidney cell types in kidney organoids and builds confidence in their usefulness for disease modelling and drug screening studies.

\hypertarget{outline-and-motivation}{%
\section{Outline and motivation}\label{outline-and-motivation}}

While we followed a well-established workflow in our previous analysis, there were still many decisions we had to make about which analysis tools and parameters to use. As I discussed in \textbf{Chapter \ref{tools}} methods for analysing scRNA-seq data have developed at a rapid rate and there are always new software packages to try. New methods should improve on what already exists or approach the data in a different. Even when a dataset is well understood applying new methods can reinforce what has already been seen or reveal interesting new information. For the remainder of this chapter I will explore some of the alternative analytical choices we could have made and approaches we could have used in our analysis of the kidney organoid dataset, and what difference they may have made to our interpretation of this data. I have selected these methods to look at because they either do something significantly different to what we used in our previous analysis or they have been shown to be effective in comparison studies. This chapter also provides an illustration of some of the thought processes involved in an analysis project. An outline of the analysis I have performed and where it will be discussed in this chapter is shown in Figure \ref{fig:workflow}. I begin by looking at some of the pre-processing steps that are often overlooked, particularly methods for selecting cell-containing droplets. I also compare the alignment-based quantification produced by the standard pre-processing pipeline to a pseudoalignment quantification. The next section explores some of the choices that can be made when performing quality control to select high-quality cells. The main aim of this analysis is to identify the cell types present in kidney organoids which requires clustering together similar cells. Clustering methods often require a set of genes to be chosen and here I have trialled an alternative gene selection method and compared the resulting clusters to the previously published analysis. In the final section of this chapter I apply two newly developed analysis approaches to this dataset to see what additional information they can add. Details of the analysis including the code used and exploratory visualisations are displayed on a website at \url{https://lazappi.github.io/phd-thesis-analysis/} {[}217{]} and the analysis methods used are provided in \textbf{Appendix \ref{analysis-methods}}.

\begin{figure}

{\centering \includegraphics[width=1\linewidth]{/home/travis/build/lazappi/phd-thesis/figures/05-workflow} 

}

\caption[Outline of analysis of the kidney organoid dataset.]{Outline of analysis of the kidney organoid dataset. Filled coloured boxes indicate stages of the analysis. Purple stages contribute to the final results shown here while blue stages were performed as a comparison. Grey stages were performed in the published analysis but are not shown here. The main analysis workflow is indicated with black arrows and comparison stages with pointed grey arrows. The published workflow is shown using smaller open grey arrows with dotted lines. Green shaded boxes show the software tools used and pink shaded boxes indicate the sections of this chapter that discuss the results of those stages.}\label{fig:workflow}
\end{figure}





\hypertarget{pre-processing}{%
\section{Pre-processing}\label{pre-processing}}

The first stage in any analysis of sequencing data is to convert the files returned by the sequencing machine to a format suitable for analysis. For scRNA-seq data produced using the 10x Genomics Chromium platform the standard tool for accomplishing this task is 10x's Cell Ranger software. This pipeline performs a number of tasks including demultiplexing the BCL base call files returned by Illumina sequencers and converting them to more common FASTQ sequence files, aligning reads to a reference genome, counting reads overlapping annotated genes, separating cells by barcode, and deduplicating UMIs. Some of these stages are performed by other tools such as Illumina's bcl2fastq software and the STAR splice-aware aligner {[}8{]} but with wrappers designed to handle specific features of 10x libraries. For the analysis presented here I have performed pre-processing of the first batch of three organoid samples using Cell Ranger (v3.0.1) with the GRCh38 reference genome provided by 10x Genomics (v3.0.0) based on the ENSEMBL gene annotation (v93). Each sample was processed separately to produce an expression matrix and then these were aggregated without any normalisation to form a single dataset.

\hypertarget{droplet-selection}{%
\subsection{Droplet selection}\label{droplet-selection}}

One of the final steps in pre-processing a dataset from a droplet-based cell capture method is to select droplets that contain cells. Many droplets are produced during the cell capture process but most of these will not contain cells and any reads associated with them will be the result of ambient RNA present in solution. The original Cell Ranger selection algorithm (used in our previous publications) required an expected number of cells for each sample (\(N\)). The 99th percentile of the total number of UMI counts in the top \(N\) cells was then taken as a robust estimate of the maximum number of counts in a cell and any droplet with at least 10 percent of this maximum was selected as a cell {[}34{]}. A similar approach is to select droplets that fall above the knee point (where curvature is minimised) of the cumulative fraction of counts with respect to increasing total count {[}31{]}. While these methods effectively select most real cells, they risk overlooking cells that have a lower RNA content than the majority of cells in the sample. The EmptyDrops method {[}99{]} in the DropletUtils package provides an alternative approach (Figure \ref{fig:droplet-selection}). A threshold on total counts is selected, below which droplets are assumed to be empty, and these empty droplets are used to produce a profile of the ambient RNA in the sample. A Monte Carlo sampling approach is then used to calculate a p-value indicating whether a particular droplet is likely to have come from the ambient distribution. A traditional threshold is also used to select cells with more total counts than the knee point when droplets are ordered by total counts. Cell Ranger v3 uses a modified version of the EmptyDrops procedure where the original selection method is substituted for the knee point threshold.

\begin{figure}

{\centering \includegraphics[width=1\linewidth]{/home/travis/build/lazappi/phd-thesis/figures/05-droplet-selection} 

}

\caption[Droplet selection using EmptyDrops.]{Droplet selection using EmptyDrops. Scatter plot of ordered total counts per droplet. Points are coloured by whether they were identified as cells (pink) or empty (black). Horizontal dotted lines show the knee point (blue), inflection point (green) and the threshold for droplets assumed to be empty (purple).}\label{fig:droplet-selection}
\end{figure}





I compared the newer EmptyDrops and Cell Ranger v3 droplet selection methods with the older Cell Ranger thresholding approach. As we would expect, the original Cell Ranger algorithm is the most conservative, selecting only 7006 cells (Figure \ref{fig:selection-comparison}A). The EmptyDrops and Cell Ranger v3 methods, which use a testing procedure rather than a strict threshold, select many more cell-containing droplets. Many of these cells are identified by both methods but there are also a large number that are identified only by EmptyDrops. This is surprising as the two software packages provide different implementations of what is largely the same algorithm. Some of these differences may be due to the parameters I have chosen for EmptyDrops but it is difficult to say if these are different to those that are used by Cell Ranger. This highlights some of the trade-offs inherent in using a pre-designed pipeline for processing any dataset. While it is convenient to have a single tool that automates a series of tasks, this often comes at the cost of less control over each stage than would be possible if using a series of individual tools.

\begin{figure}

{\centering \includegraphics[width=1\linewidth]{/home/travis/build/lazappi/phd-thesis/figures/05-selection-comparison} 

}

\caption[UpSet plot comparing droplet selection methods.]{UpSet plot {[}218{]} comparing droplet selection methods. Cell Ranger v3 and EmptyDrops identify significantly more cells than the traditional Cell Ranger approach. My use of EmptyDrops also identifies a set of cells that are overlooked by the automated Cell Ranger v3 procedure.}\label{fig:selection-comparison}
\end{figure}





It is difficult to evaluate if the extra cells identified by EmptyDrops are real cells, or if the algorithm has been too permissive and selected droplets that are actually empty. In the following section I will perform further quality control to remove low-quality cells so at this stage I chose to keep cells identified by either the EmptyDrops or Cell Ranger methods. Of the 945038 droplets with at least one count present in the raw dataset, 10002 were selected as cells by at least one of these methods.

\hypertarget{alevin-comparison}{%
\subsection{Alevin comparison}\label{alevin-comparison}}

Cell Ranger is based on the traditional workflow for processing RNA-seq reads, which is to align them to a reference genome then count how many reads overlap annotated genes. In recent years it has become common to use methods such as Salmon {[}12{]} and Kallisto {[}11{]} that directly estimate transcript expression without alignment. While these approaches do not produce an exact genomic location for each read they can be much faster, which is appealing for scRNA-seq data as there are many more cells to quantify. However, factors such as cell barcodes, UMIs and different model assumptions mean that these tools have had to be adapted for scRNA-seq data. Alevin {[}91{]} is the scRNA-seq mode available in Salmon.

Alevin has its own method for identifying which droplets contain cells. First, the knee point in the cumulative distribution of barcode frequencies is identified and any droplets above this are assumed to contain cells. The Levenshtein distance {[}219{]} between the barcodes of all droplets is calculated and if they are not sufficiently close to other barcodes they are considered to be noise and the associated reads discarded. Alevin performs a second round of cell whitelisting following quantification. Droplets are divided into three zones: high-quality (top half above the knee point), ambiguous (bottom half above the knee point) and low-quality (well below the knee point). A nave Bayes classifier based on a set of features, including the fraction of reads mapped, the fraction of mitochondrial reads and the duplication rate, is then used to classify droplets as either high or low-quality. This is very different to the EmptyDrops approach, using a machine learning technique rather than a statistical test and technical features rather than the overall expression profile, and is more similar to approaches that have been suggested for quality control of cells {[}100{]}.

Figure \ref{fig:alevin-comparison}A shows a comparison of the alevin approach to the two methods I used to select cells in the previous section. Most of the cells are identified by all three methods but just like EmptyDrops, alevin finds a large number of cells that are overlooked by Cell Ranger v3. What is particularly striking is that there is very little overlap in the extra cells identified by alevin and EmptyDrops suggesting that the differences between the two approaches may be important. However, I should note that alevin uses its own quantification as input for classification while I provided the counts from Cell Ranger to EmptyDrops, so some of the differences may be due to the data they were given rather than the selection methods themselves. Alevin (and Cell Ranger v3) also processes the three samples separately while I performed EmptyDrops on the whole dataset.

\begin{figure}

{\centering \includegraphics[width=1\linewidth]{/home/travis/build/lazappi/phd-thesis/figures/05-alevin-comparison} 

}

\caption[Comparison to quantification using alevin.]{Comparison to quantification using alevin. (A) UpSet plot comparing the droplets identified as cells by the alevin, Cell Ranger v3 and EmptyDrops methods. (B) Scatter plot comparing total counts per cell as estimated by alevin and the alignment-based Cell Ranger pipeline. Points are coloured according to the method that identified them as cells (alevin only (green), alignment-based only (blue), both (pink)). Thin purple line shows \(y = x\), thick blue line shows a linear fit for cells identified by both methods. (C) Scatter plot comparing mean counts per gene as estimated by the two pipelines, point colours and lines as in (B). (D) Relationship between the total counts per gene and the percentage of cells with a zero count for each pipeline.}\label{fig:alevin-comparison}
\end{figure}





Similar to Cell Ranger, the quantification stage of Alevin involves several steps including mapping reads and deduplicating UMIs. Alevin maps reads to the transcriptome and groups them by cell barcode and UMI. To deduplicate UMIs, alevin considers the mapping location and the UMI sequence (to account for sequencing errors). Because reads have been mapped to the transcriptome rather than the genome, there is an additional layer of information that can be useful for identifying if reads with the same UMI map to the same location. This should avoid over collapsing UMIs. Following these steps, the transcriptome level expression estimates are aggregated to the gene level for analysis. Figure \ref{fig:alevin-comparison}B shows a comparison of the total counts per cell between alevin and the alignment-based approach using Cell Ranger. For cells that were identified by both pipelines (pink) the total counts are very similar. Cells identified just by the alignment-based approach (blue) tend to have fewer counts than those identified only by alevin (green). There are more differences if we look at the mean counts per gene (Figure \ref{fig:alevin-comparison}C). Most of the variability is at low expression levels, but there are some genes that are lowly expressed by one method but have relatively high expression in the other. One reason for this could be either method missing (or incorrectly assigning) counts altogether. Figure \ref{fig:alevin-comparison}D shows the relationship between the total counts per gene and the percentage of cells with zero counts. This relationship is noisier for the alevin quantification, particularly below the trend line, and also shows more genes with a very low percentage of zeros.

While there are differences between the results from the two approaches there is nothing in this comparison to suggest that either is inaccurate. Genes that are found to be expressed by only one of the methods could possibly lead to differences in downstream interpretation but it is very difficult to say which is a better representation of true expression and it is unlikely that all genes providing similar evidence (for example markers of a cell type) would be affected in the same way. On this evidence alevin presents a viable alternative to alignment-based quantification for scRNA-seq data, particularly for large datasets where computational time is a concern. However, when alignment is required anyway, it is simpler and avoids repetition to use those alignments to produce an expression matrix for further analysis. For the remainder of the analysis I have used the alignment-based quantification of this dataset.

\hypertarget{quality-control}{%
\section{Quality control}\label{quality-control}}

In the previous section I discussed methods for selecting droplets that contain cells but there are further steps that are required in order to select high-quality cells. There are a variety of reasons that we may want to remove cells before further analysis. Cells can become damaged during processing, may not contain sufficient counts for analysis or may be the result of doublets where two cells are captured in the same droplet. Because these factors are not related just to the number of reads per cell they may not have been detected by the methods used previously and require different approaches.

Two tools that are useful for quality control of scRNA-seq data are the scater {[}101{]} and scran {[}109{]} packages. The scater package provides functions for calculating commonly used quality control metrics, adding annotations and visualisation, while scran can be used for various calculations such as assigning cell cycle phase and doublet scoring. After selecting cell-containing droplets I used these two packages to add extra information to the dataset. Additional feature annotation was retrieved from BioMart {[}220{]} using the biomaRt package {[}221{]} via functions in scater. Before calculating scater's quality control metrics I added mitochondrial genes as a feature control set, as the proportion of counts assigned to these genes can be an indicator of the health of a cell. I also used the cyclone function {[}222{]} in scran to score cell cycle activity and assign cell cycle phases to each cell. This method uses the relative expression of pairs of genes from previous studies of the cell cycle in order to score cell cycle activity. Thresholds can then be applied to these scores to assign a phase to each cell. Pre-computing a range of metrics is a valuable feature of scater as they are immediately available for exploration and filtering.

The simplest approach to quality control is to select thresholds for various metrics and to remove cells that fall outside these values. This is the approach I have taken here and in the previously published analysis. While previously these thresholds were selected by visually inspecting distributions, here I have used a more data-driven approach by selecting thresholds that are a specified number of median absolute deviations (MADs) from the median value. This is made easy by the isOutlier function in scater. The first filter I have applied is to the number of counts per cell, with thresholds on both the total number of counts per cell and the total number of features that are expressed (Figure \ref{fig:qc-thresholds}A). These filters are important as cells that fall below these thresholds do not contain enough information to be useful for analysis. A similar threshold is applied to the percentage of counts assigned to mitochondrial genes (Figure \ref{fig:qc-thresholds}B). A high proportion of mitochondrial expression is a sign of distress and can indicate that cells have become damaged during processing. Once most of the low-quality cells have been removed it becomes possible to check for doublets. Here I have used a simulation-based approach implemented in scran. This method is similar to the one suggested by DoubletFinder {[}98{]} and creates synthetic doublets by combining counts from randomly selected cells. The density of simulated doublets in the neighbourhood of each cell in PCA space is then used to calculate a doublet score to which a cutoff can be applied (Figure \ref{fig:qc-thresholds}C). For this threshold I have chosen a value that gives a number of doublets roughly equal to the expected number of the cell capture technology (around one percent for the 10x Chromium).

\begin{figure}

{\centering \includegraphics[width=1\linewidth]{/home/travis/build/lazappi/phd-thesis/figures/05-qc-thresholds} 

}

\caption[Thresholds used for quality control.]{Thresholds used for quality control. (A) Scatter plot of total counts per cell against number of features expressed. Thresholds for selecting high-quality cells are show by purple (features) and green (counts) dotted lines. Points are coloured according to whether cells were kept (blue) or removed (pink). (B) Histogram of percentage of counts assigned to mitochondrial genes with filtering thresholds shown in pink. (C) Histogram of doublet scores with threshold shown in pink. (D) PCA plot based on technical factors. Cells are coloured according to whether they were removed by both the PCA outlier method and manual filtering (orange) or just manual filtering (pink).}\label{fig:qc-thresholds}
\end{figure}





An alternative approach is to rely purely on the quality control metrics themselves rather than manually selecting thresholds. PCA can be performed using the technical metrics such as total counts per cell rather than gene expression levels, and then outliers can be detected in PCA space (Figure \ref{fig:qc-thresholds}D). This approach relies on the assumption that most cells will be high-quality and that the values that have been calculated will be sufficient to separate them from the low-quality cells. On this dataset the PCA-based approach identified a subset of the cells that were removed by my manual thresholds as low-quality. While I believe the idea of an automated quality control method is valuable, it will require more work to identify which features are most useful and it is likely that they will differ between technologies and tissue types. Additionally, any automated approach will always be more difficult to interpret than clear manual thresholds. For these reasons I have used the dataset produced by filtering cells using the manual thresholds for the rest of this analysis. I have also performed a minimal filtering of genes, selecting those that have at least one count in at least two high-quality cells. After quality control the dataset had 8059 cells and 22739 genes with a median of 7838 counts per cell and a median of 2506 genes expressed.

After removing low-quality cells it is important to inspect the data again to check that the quality control has been successful. If the selected thresholds are too severe it is possible to remove interesting populations of cells that differ from the majority of the dataset, such as a rare cell type that is smaller and therefore has less RNA. One way to do this is to look at differences in expression between the cells that have been removed and those that have been kept. Here I have calculated the predicted log fold change using the predFC function in the edgeR package and plotted it against the average expression for each gene (Figure \ref{fig:qc-validation}A). Seeing a set of interesting genes (in this case known kidney marker genes) with positive fold changes in the removed cells would suggest that the selected thresholds are too stringent. That is not the case here as can be seen from the 15 genes with greatest fold change shown in Figure \ref{fig:qc-validation}B.

\begin{figure}

{\centering \includegraphics[width=1\linewidth]{/home/travis/build/lazappi/phd-thesis/figures/05-qc-validation} 

}

\caption[Validation of quality control.]{Validation of quality control. (A) Scatter plot of genes showing average expression against predicted fold change for removed cells compared to kept cells. Points are coloured according to the difference in proportion of cells that express each gene between removed and kept cells. (B) Top 15 genes when sorted by positive fold change in removed cells. (C) t-SNE plots of cells coloured by various technical factors including sample, droplet selection method, assigned cell cycle stage, total counts, total features and percentage mitochondrial counts.}\label{fig:qc-validation}
\end{figure}





On the other hand if the selected thresholds are not strict enough we will have retained a set of low-quality cells. One way to identify these is to look at the dataset in a reduced dimensional space using techniques such as t-SNE or UMAP, highlighting technical factors that may be associated with low-quality cells. If a low-quality population is present we would expect to see it separate from the other cells. This approach is taken in Figure \ref{fig:qc-validation}C where I have coloured a t-SNE plot by organoid sample, the droplet selection method, assigned cell cycle stage, total counts, total features and percentage of counts assigned to mitochondrial genes. These plots show a small population of cells that have fewer counts and have mostly been selected using the EmptyDrops method. However, this population is not clearly separated and may represent some intermediate cell type. At this stage I will retain those cells but following clustering it may be appropriate to revisit that decision if they do not appear to represent a logical biological population.

\hypertarget{clustering}{%
\section{Clustering}\label{clustering}}

Now that we have a dataset containing high-quality cells, the next step in my analysis is to perform clustering. It is necessary to use some technique to group and label cells. For this kind of dataset that contains an unknown set of cell types, an unsupervised approach is a good fit. Clustering allows us to group cells based purely on their expression profiles without making any assumptions about what those groups should be. Here I am going to use the graph-based clustering method available in the Seurat package. The first step in this approach is to select a subset of genes to use.

\hypertarget{gene-selection}{%
\subsection{Gene selection}\label{gene-selection}}

Most genes will either be expressed at a consistent level across all cells in a dataset or have too few counts to provide enough information about differences between cells. To reduce the noise in the dataset and the computational cost of analysis we want to select a set of genes that will be useful for separating cells. An intuitive approach is to filter based on the mean expression level to remove lowly expressed genes, and the gene variability to select those that explain differences between cells. This is the default approach used in Seurat (Figure \ref{fig:gene-selection}A). The thresholds I have chosen here select 2457 highly variable genes.

\begin{figure}

{\centering \includegraphics[width=1\linewidth]{/home/travis/build/lazappi/phd-thesis/figures/05-gene-selection} 

}

\caption[Selecting genes for clustering.]{Selecting genes for clustering. (A) Scatter plot of mean gene expression level against dispersion. The default Seurat selection method sets thresholds on these axes. Purple lines indicate thresholds and selected genes are coloured blue. (B) The M3Drop method fits the relationship (purple line) between average expression (x-axis) and dropout rate (y-axis). Genes that have significantly more zeros than expected are selected (pink). (C) Comparison of gene selection methods. The dropout rate calculated by M3Drop is plotted against the dispersion calculated by Seurat. Points are coloured according to whether they were selected by Seurat (blue), M3Drop (pink) or both (green). (D) Bar plot of the number of genes selected by each method and the intersection. Most genes are identified by only one approach.}\label{fig:gene-selection}
\end{figure}





The M3Drop package suggests an alternative approach that considers the proportion of zero counts per gene rather than the variability {[}104{]}. Here the idea is that if a gene is important for identifying a cell type then its expression will be concentrated in a subset of cells and there will be more zero counts than would be expected based on the overall mean expression. The original M3Drop method fits a Michaelis-Menten function to the relationships between mean expression and dropout rate (proportion of zeros) but for UMI datasets that have not been sequenced to saturation the authors suggest their NBDrop method instead, which fits a library size adjusted negative binomial model. A binomial test is then used to identify genes that have significantly more zeros than would be expected from the fitted model (Figure \ref{fig:gene-selection}B). By using a statistical test this method removes the need to set multiple thresholds. Here M3Drop has selected 1952 genes with excess zeros.

If we compare the sets of genes selected by these two methods we see that a relatively small subset are identified by both of them (Figure \ref{fig:gene-selection}D). These differences can be explained by the differences between variability and zero counts (Figure \ref{fig:gene-selection}C). For example, it is possible for a gene to vary a lot between different cells but within an expression range that doesn't result in more zeros. Alternatively, a gene with relatively low dispersion may still have excess zeros if the underlying expression is low enough. For the previously published analysis we used the Seurat selection method, so here I am going to use the genes identified by M3Drop and see what effect that has on the clustering results.

\hypertarget{resolution-selection}{%
\subsection{Resolution selection}\label{resolution-selection}}

There are several more steps in the Seurat clustering procedure. Once a set of genes is selected they are used to perform a PCA and a set of principal components is selected. Distances are then calculated between cells in this PCA space and the nearest neighbours for each cell are identified. By comparing the sets of neighbours for two cells and computing the Jaccard index, a shared nearest neighbour graph can be constructed where each node is a cell and the edges are the overlap in their neighbourhoods. A graph modularity optimisation algorithm such as Louvain modularity optimisation {[}133{]} can then be used to partition the graph and form clusters of cells. This graph-based clustering approach has been shown to be effective in comparisons of scRNA-seq clustering methods {[}130,131{]}. One of the decisions that needs to be made during this procedure is the value of the resolution parameter to use, which affects the number of clusters that are produced. As I have discussed in \textbf{Chapter \ref{simulation}}, clustering trees are one tool that can be helpful in making this decision. Figure \ref{fig:resolution-selection}A shows a clustering tree for this dataset with resolutions between zero and one.

\begin{figure}

{\centering \includegraphics[width=1\linewidth]{/home/travis/build/lazappi/phd-thesis/figures/05-resolution-selection} 

}

\caption[Selecting a clustering resolution.]{Selecting a clustering resolution. (A) Clustering tree showing resolutions between zero and one. Pink box shows the selected resolution. (B) Clustering trees with nodes coloured by a selection of published marker genes. Expression signals can help to select a clustering resolution.}\label{fig:resolution-selection}
\end{figure}





The structure of this tree can tell us about how clusters are forming in this dataset. My general approach is to select a resolution where most of the branches have formed but before there are many low in-proportion edges crossing over between branches that can indicate that clusters are unstable. Displaying expression levels of genes that are expected to be important in the dataset adds extra information to the structure of the tree (Figure \ref{fig:resolution-selection}B). Here I look to identify at which resolution we see clear, high-intensity signals for these genes. That is, a single or few clusters where they are highly expressed. Based on these trees I have selected a resolution of 0.4 which produces 13 clusters.

\hypertarget{cluster-validation}{%
\subsection{Cluster validation}\label{cluster-validation}}

The cluster assignments of the individual cells can be displayed in a reduced dimensional space such as that produced by UMAP {[}148,223{]} (Figure \ref{fig:cluster-validation}A). The dimensions shown in this visualisation and the cluster labels that are used to colour it are produced using different algorithms. By combining them we can confirm that the two approaches produce similar results. Clusters can have different densities in these plots so it is also useful to know how many cells they contain (Figure \ref{fig:cluster-validation}B). Now that we have a set of clusters we can check to see if they are associated with technical artifacts rather than biological factors.

\begin{figure}

{\centering \includegraphics[width=1\linewidth]{/home/travis/build/lazappi/phd-thesis/figures/05-cluster-validation} 

}

\caption[Validation of identified clusters.]{Validation of identified clusters. (A) Scatter plot of cells in UMAP space coloured by assigned cluster. (B) Bar chart of the number of cells in each cluster. (C) Stacked bar chart showing the proportion of cells in each cluster identified by each droplet selection method, Cell Ranger v3 only (blue), EmptyDrops only (green) or both (pink). (D) Stacked bar chart showing the proportion of cells in each cluster assigned to the G1 (pink), G2M (blue) or S (green) phases of the cell cycle. (E) SinaPlot {[}224{]} showing the total number of counts per cell in each cluster. (F) SinaPlot showing the percentage of counts assigned to mitochondrial genes per cell in each cluster.}\label{fig:cluster-validation}
\end{figure}





Figure \ref{fig:cluster-validation}C shows the proportion of cells in each cluster that were identified by the two droplet selection methods I used, Cell Ranger v3 and EmptyDrops. Most of the additional cells identified by EmptyDrops have been placed in two of the clusters. The cells in these clusters also have fewer total counts (Figure \ref{fig:cluster-validation}E) and a lower percentage of mitochondrial counts (Figure \ref{fig:cluster-validation}F). This suggests that those clusters are at least in part a result of the lower count cells identified by EmptyDrops and it may be appropriate to remove them from the dataset. However, a large proportion of the cells in those clusters were also identified by Cell Ranger v3 and an alternative hypothesis is that these additional cells belong to a cell type that is smaller or contains less RNA for some other reason. Looking at the marker genes associated with those clusters can help to answer that question (\textbf{Section \ref{marker-genes}}).

\hypertarget{comparison-to-published-clusters}{%
\subsection{Comparison to published clusters}\label{comparison-to-published-clusters}}

Because we have previously analysed and interpreted this dataset I can compare these clusters to the previously published clusters. It is important to remember the differences between the two analyses. The published analysis included a fourth organoid from a separate batch that I have omitted here and was analysed using the standard Cell Ranger (v1.3.1) and Seurat workflows without EmptyDrops droplet selection or M3Drop gene selection. I also used different quality control thresholds and parameters at some stages. Figure \ref{fig:cluster-comparison}A shows the proportion of cells in each cluster that were present in the published analysis. All of the clusters have some additional cells which is to be expected given that the use of the newer droplet selection methods produced a dataset with around 1000 more cells. The two clusters that contain the cells only identified by EmptyDrops (Cluster 4 and Cluster 6) stand out as having a very low proportion of cells from the previous analysis.

\begin{figure}

{\centering \includegraphics[width=1\linewidth]{/home/travis/build/lazappi/phd-thesis/figures/05-cluster-comparison} 

}

\caption[Comparison to previously published clusters.]{Comparison to previously published clusters. (A) Bar chart showing the proportion of cells in each cluster that were present in the previously published analysis. (B) Heatmap showing the Jaccard index between previously published clusters and the clusters from the current analysis. High values (yellow) indicate that many cells are shared between two clusters while low values (blue) indicate there is little overlap.}\label{fig:cluster-comparison}
\end{figure}





To compare the two sets of clusters I have calculated the Jaccard index between each cluster from the previously published analysis and each cluster from the analysis presented here (Figure \ref{fig:cluster-comparison}B). Some of the clusters (such as Cluster 5) very clearly match up with a published cluster while others have overlaps with multiple similar clusters. For example, Cluster 2 which consists of cells from the two cell cycle clusters in the published analysis. There are four stromal clusters in both analyses but for two of them (Cluster 0 and Cluster 3) the cells are divided differently. The differences in clustering are the result of all the decisions made to this stage of the analysis including which cells and genes have been selected. Without a known truth it is difficult to determine which is better or more correct or if they simply provide different views of the same data with emphasis on different features and it is possible that both sets of clusters are biologically meaningful. The two EmptyDrops clusters do not show obvious signals but they do have some overlap with the published endothelial clusters which suggests that that they may be a previously overlooked endothelial population. To confirm if this is the case we need to check whether they express known endothelial markers. Identifying marker genes for each cluster through differential expression testing can show us which genes are unique to each cluster.

\hypertarget{marker-genes}{%
\section{Marker gene detection}\label{marker-genes}}

To try and identify which cell types a cluster represents we need to look more closely at what genes are expressed. Ideally we want to identify genes that are expressed in a single cluster at a relatively high level and whose function is well-known enough to interpret. These are often the same genes that are useful for techniques like immunofluorescence imaging. The most common way of identifying marker genes is to test for differential expression between the cells in a cluster and all other cells in the dataset, then inspect the significantly highly up-regulated genes. Many approaches have been suggested for this task, including a range of methods designed specifically for scRNA-seq data. Comparisons of these methods have found that the scRNA-seq specific methods do not show significantly better results than those designed for bulk RNA-seq or standard statistical tests {[}156,225{]}. For this analysis I have chosen to use the edgeR package as it performs well in these comparisons and allows for more complex designs such as including the proportion of genes that are expressed per cell (detection rate).

The edgeR method is based on the negative binomial distribution. After calculating normalisation factors to capture differences between cells and estimating common and gene-wise dispersions, a negative binomial is fitted to the dataset. We can evaluate this fit using some of the information calculated by the edgeR model such as the gene means (Figure \ref{fig:de-results}A) and proportion of zeros per gene (Figure \ref{fig:de-results}B) and comparing it to what we observe in the dataset. When looking at the comparison of means we see that at high expression levels the edgeR estimates become larger than what is observed. We also observed more zeros than expected when the proportion of zeros is low. Checking these fits can help us to decide if the dataset fits an ordinary negative binomial model or whether a zero-inflated negative model would be more appropriate, as has often been suggested for scRNA-seq data. For example, a method such as ZINB-WaVE can be used to calculate weights that can be added to the edgeR model to correct for zero inflation {[}64{]}. In this case I decided the edgeR fit was sufficiently good and these kinds of corrections were unnecessary.

\begin{figure}

{\centering \includegraphics[width=1\linewidth]{/home/travis/build/lazappi/phd-thesis/figures/05-de-results} 

}

\caption[Results of edgeR differential expression analysis.]{Results of edgeR differential expression analysis. (A) Scatter plot showing the edgeR fit of gene means. The x-axis shows the average of the observed gene mean and the expected gene mean from the edgeR fit and the y-axis shows the difference between the observed and expected means. The pink line lies at zero and the blue line is a LOESS fit. (B) Scatter plot showing the edgeR fit of gene proportion of zeros. Similar to (A) but showing the observed and expected proportion of zeros. (C) Plot of the biological coefficient as calculated by edgeR against the average log counts per million for each gene. (D) Bar chart of the number of significantly differentially expressed genes for each cluster with positive fold changes. Points show individual genes coloured by their log fold change. Position of each point on the y-axis also shows the log fold change but scaled to the height of each bar. (E) Differential expression results for some genes identified as important markers in the previously published analysis. Size of points indicates the positive log fold change and transparency indicates the negative false discovery rate.}\label{fig:de-results}
\end{figure}





Another important quantity that is calculated by the edgeR model is the biological coefficient of variation (BCV). BCV is the component of variability that captures the true underlying biological differences between replicates of the same RNA sample. Estimating BCV correctly can be particularly important for highly expressed genes. The BCV plot (Figure \ref{fig:de-results}C) shows the common dispersion across all genes (pink line), the dispersion for each individual gene (points) and a trend across the dataset (blue line). The empirical Bayes approach of edgeR borrows information between genes in order to shrink the dispersion towards the trended value to achieve more robust estimates. The shape of the trend line can be used to check how well the BCV has been fit and we expect lowly expressed genes to be more variable than those that are highly expressed. Once we have a fitted negative binomial model we can test for differences between the groups we are interested in.

There are multiple testing procedures available in edgeR but the currently recommended approach is to fit a generalised linear model and test genes using a quasi-likelihood F-test. Figure \ref{fig:de-results}D shows the number of significantly up-regulated genes for each cluster identified using this approach and the log fold change for individual genes. For most clusters we see a large number of positive marker genes but for the two clusters that contain most of the EmptyDrops cells many fewer markers have been identified. This could be a result of the lower counts in these clusters or a clue that these are not pure populations with clear markers.

It is important to look at the gene lists for each cluster in detail in order to interpret and assign cell types, but as we have previously analysed this dataset, and can be relatively confident of what it contains, I have plotted the edgeR results for a few of the genes that were previously identified to be important markers (Figure \ref{fig:de-results}E). For most of the clusters we see that only markers for one particular cell type are significant with large fold changes and these are consistent with the overlaps between the published clusters presented in the previous section. For the clusters that had little overlap with the published analysis these results help to assign them. Cluster 4 clearly shows up-regulation of endothelial markers confirming this lineage as a possible identity for these cells. The other new cluster (Cluster 6) is more ambiguous with significant results for markers of several cell types but does show up-regulation of immune-related genes. Immune cells are not part of the lineage that these kidney organoids are differentiated towards, but we did see some organoid cells that clustered with immune cells from human foetal kidney in our previous analysis and there is a plausible pathway by which they could develop. Further lab-based experiments would be required to confirm whether these cells exist in kidney organoids. Based on the detected markers I have assigned cell types to clusters as shown in Table \ref{tab:assignments}.

\begin{table}

\caption{\label{tab:assignments}Cluster assignments based on detected marker genes}
\centering
\begin{tabular}[t]{ll}
\toprule
Cluster & Assignment\\
\midrule
0 & Stroma\\
1 & Stroma\\
2 & Cell cycle\\
3 & Stroma\\
4 & Endothelium\\
5 & Podocyte\\
6 & Possible immune\\
7 & Stroma\\
8 & Endothelium\\
9 & Glial\\
10 & Epithelium\\
11 & Muscle progenitor\\
12 & Neural progenitor\\
\bottomrule
\end{tabular}
\end{table}

\hypertarget{connecting-clusters}{%
\section{Connecting clusters}\label{connecting-clusters}}

Clustering analysis is used to group cells and identify the cell types that are present in a dataset, but by itself clustering doesn't tell us anything about how those groups relate to each other. In the developmental context where there are multiple differentiating cell types these relationships are of interest. Some aspects of these relationships can be inferred from the proximity of clusters in reduced dimensional space or from the known biology of the assigned cell types, but this is largely based on intuition rather than information in the data. Computational methods to calculate developmental trajectories in scRNA-seq data have been a big focus of early scRNA-seq bioinformatics research. In our published analysis we used the Monocle package to create a pseudotime ordering for cells in the nephron lineage. However, methods like this do not perform as well on larger datasets with many cell types from possibly unconnected lineages, such as we have in the full organoid dataset. Here I will demonstrate two recently developed alternative approaches for connecting clusters and adding context to the current analysis.

\hypertarget{partition-based-graph-abstraction}{%
\subsection{Partition-based graph abstraction}\label{partition-based-graph-abstraction}}

Partition-based graph abstraction (PAGA) is a method for reducing the complexity of a dataset by estimating the connectivity between clusters {[}226{]}. The result is a reduced graph representation of the dataset where each node is a cluster and the edges show how connected they are (Figure \ref{fig:paga-results}A). PAGA is available as part of the scanpy package {[}227{]} and starts with the same kind of shared nearest neighbour cell graph that is used by Seurat's graph-based clustering. Connectivity between clusters is calculated by comparing the number of edges between cells in different clusters to the number that would be expected if edges were assigned randomly. This connectivity can be interpreted as the confidence in the presence of an actual connection between clusters.

\begin{figure}

{\centering \includegraphics[width=1\linewidth]{/home/travis/build/lazappi/phd-thesis/figures/05-paga-results} 

}

\caption[Results of partition-based graph abstraction (PAGA).]{Results of partition-based graph abstraction (PAGA). (A) The cluster connectivity graph produced by PAGA. Each node represents a cluster and edges show the connectivity between clusters. The size of nodes indicates the number of cells in each cluster and the edge colour and transparency show the connectivity from low (blue) to high (yellow). (B). The underlying cell shared nearest neighbour graph. Nodes are cells coloured by cluster and edge weight indicates connectivity.}\label{fig:paga-results}
\end{figure}





The results for the organoid dataset show high connectivity between the various stromal clusters that make up the majority of cells. Connectivity between some of the other clusters also confirms what we would expect. For example the podocyte cluster is strongly connected to the epithelial cluster and the two off-target neural clusters (glial and neural progenitor) are strongly connected to each other. The position of the two clusters containing the additional EmptyDrops cells is less clear with connections seen between them, but not as strongly with the other clusters they would be expected to be related to, particularly between the two endothelial clusters. A possible explanation for this are technical features shared by these two clusters but not the clusters they are biologically similar to. The PAGA method does not give any estimate of the stability or variability of the connectivity between clusters but this would be a meaningful area for future research.

In some ways the graph produced by PAGA is similar but complementary to a clustering tree. The PAGA graph considers connectivity between clusters at the same resolution while clustering trees use the overlap between clusters from neighbouring resolutions to build a graph. Interestingly, the authors of PAGA emphasise the ability to explore a dataset at different resolutions by creating PAGA graphs from different sets of clusters produced using different clustering parameters. The development of two independent methods based on building graphs of clusters suggests that this is an approach that could have more general applications. The edges produced by PAGA can show which clusters are related but they don't provide any directionality to that relationship. One way to add some directionality to the cluster graph suggested by the PAGA authors is to perform a random walk on the cell graph to calculate a diffusion pseudotime across the dataset. However, this approach requires manually defining a root cell to use. An alternative approach for assigning directionality to cells involves looking at the splicing information available in the reads themselves.

\FloatBarrier

\hypertarget{cell-velocity}{%
\subsection{Cell velocity}\label{cell-velocity}}

The majority of the reads in an scRNA-seq experiment come from mature mRNA where the introns have been removed through splicing, however some immature mRNA molecules that still retain their intronic sequences are also sequenced. The concept of cell velocity looks at these unspliced reads to determine which genes are being actively transcribed. By looking at which cellular processes are being switched on or off it is possible to infer an overall transcriptional direction for each cell. This approach was popularised for scRNA-seq data by the velocyto package {[}151{]}. Two steps are required for cell velocity analysis. First, velocyto inspects the aligned reads for each cell, but instead of only counting those that are inside an annotated exon it also generates a counts matrix of unspliced reads for each gene. These two matrices are then combined to estimate a velocity for each cell by solving a set of differential equations. The velocities can be projected onto a reduced dimensional space by looking at a correlation-based transition matrix on a nearest neighbour graph in that space. When there are more than a few hundred cells it is difficult to look at the velocity for each individual cell, so it can be useful to summarise them in some way. By forming a regular grid of points and using Gaussian smoothing to calculate an average of the nearby cells velocyto calculates a velocity field across the whole dataset, indicating the overall velocity of an area (Figure \ref{fig:cell-velocity}).

\begin{figure}

{\centering \includegraphics[width=1\linewidth]{/home/travis/build/lazappi/phd-thesis/figures/05-cell-velocity} 

}

\caption[Cell velocity field calculated by velocyto.]{Cell velocity field calculated by velocyto. Points show cells in UMAP space coloured according to cluster. Arrows show the average velocity for cells in each region of a grid, with direction indicating the transcriptional profile cells are headed towards and length giving the rate of change.}\label{fig:cell-velocity}
\end{figure}





Looking at the velocity field for this dataset we see some of the expected developmental trajectories. For example, cells in the epithelial cluster are headed towards the more mature podocyte population. Despite having a limited connectivity in the PAGA results we see the endothelial-like cluster containing EmptyDrops cells (Cluster 4) trending towards the other endothelial cluster. Cluster 6 is again more ambiguous, showing velocity towards both endothelial and stromal clusters. This suggests that if there are immune cells present they may only be a subset of this population. We see little evidence of any clear directionality in the stromal clusters, likely because these cell types are more mature and are not actively changing their transcription profiles. There is an important caveat when looking at these velocyto results, which is that the velocity arrows depend on the space they are projected into and viewing them in different embeddings may give different views. Finding a more robust way to visualise cell velocity would be a useful area of further research that would improve the ability to interpret these results.

\hypertarget{discussion}{%
\section{Discussion}\label{discussion}}

In this chapter I have presented an analysis of a kidney organoid scRNA-seq dataset, demonstrating a range of computational tools that can be used and focusing on decisions made during analysis and how these decisions affect the results. The dataset is a subset of the data presented in two previous publications where we investigated variability in the kidney organoid differentiation protocol and profiled the cell types that are present in these organoids. Our previous analysis included data from two batches of organoids as well as integration with a published human foetal kidney dataset but for this chapter I chose to use just the first batch of data which contains cells from three organoids. Data integration is an important area of research that has resulted in a range of methods produced recently, but it is still unclear which approaches are most effective. It can also be difficult to transfer the results of integration between analysis tools. As I wanted to demonstrate a range of tools here I have chosen to work with a single dataset, but how best to combine datasets is worthy of further consideration.

The most significant difference in my new analysis comes from one of the earliest steps. During the pre-processing stage of analysing droplet-based scRNA-seq data empty droplets must be separated from those that contain cells. While this step can have a significant impact on the results of an analysis it is often overlooked. I found that methods such as EmptyDrops that test for differences from the ambient RNA profile identified many more cells than the thresholding approach traditionally used in the Cell Ranger pipeline. Retaining cells that were identified by EmptyDrops or the more recent Cell Ranger v3 which implements a similar approach produced a dataset of 10002 cells. There are two possible explanations for the additional cells identified by these approaches. Either they are more sensitive at correctly selecting cell-containing droplets or they are more permissive and have incorrectly selected empty droplets. Quality control of the selected cells can help to determine which of these is true.

Typically, low-quality cells are removed from a dataset by setting thresholds on quantities that indicate that a cell is damaged or otherwise uninformative. I used a data-driven approach to select filtering thresholds that were a given number of MADs for the median for the total counts per cell, total expressed features per cell and the percentage of counts assigned to mitochondrial genes (which is a proxy for the health of a cell). These thresholds removed 1943 cells leaving 8059, including many of the additional cells identified by EmptyDrops. The fact that these cells were retained suggests that they are not obviously empty droplets and are likely to contain biologically relevant information. Using a statistical approach to select thresholds means that they can be automatically adapted to different datasets but it does assume that most cells are high-quality. For this reason it is still important to visualise distributions and where the thresholds lie. When I compared the the thresholding approach to a fully-automated approach based on performing a PCA on a set of quality control metrics and detecting outliers I found that thresholding identified many more low-quality cells and was much easier to interpret.

The additional cells identified by EmptyDrops passed quality control but their biological significance does not become clear until later on in the analysis. Similar to the previously published analysis I used the graph-based clustering method in the Seurat package to form groups of cells with similar expression profiles. By using the clustering tree visualisation presented in \textbf{Chapter \ref{clust-trees}} I chose a resolution of 0.4 which produced 13 clusters. Most of the additional cells identified by EmptyDrops belong to two clusters (Cluster 4 and Cluster 6), but because these cells weren't present in the published analysis there is no clear overlap to associate them with a published cell type. To better interpret all the clusters I performed differential gene expression testing using edgeR, testing the cells in each cluster against the other cells in the dataset and looking for marker genes that are more highly expressed in a single cluster. Cluster 4 showed clear up-regulation of endothelial markers, while Cluster 6 had significant results for markers from various cell types including several genes associated with immune cells. Although not an expected part of the kidney organoid differentiation process, there is a plausible pathway by which immune cells could emerge, and when this dataset was combined with a human foetal kidney dataset we saw a small number of cells that clustered with immune cells and seemed to share similar expression. However, a significant amount of additional validation would be required to confirm that immune cells are present and that this apparent population isn't just the result of technical artifacts associated with the lower number of counts per cell. The other clusters were consistent with the previously published analysis and I assigned them to several kidney cell types, including several types of stroma, epithelium, mature podocytes and endothelium. I also observed several small off-target populations that were described previously, including neural progenitors, glial cells and muscle progenitors.

The Seurat procedure has been shown to perform well in comparison and consists of several steps: selecting a set of genes to use, performing PCA on those genes and selecting a set of principal components, building a shared nearest neighbour graph in PCA space, and using a modularity optimisation algorithm to partition the graph into clusters. For the published analysis we selected genes using the standard Seurat method which sets thresholds on gene expression level and variability. I compared genes selected in this way to those selected by the M3Drop method, which tests for genes with a greater number of zero counts than would be expected under a negative binomial model. Despite the significant differences between the two sets of genes the clusters identified by Seurat using the M3Drop genes were very similar to those in the published analysis. This suggests that graph-based clustering is fairly robust to exactly which genes are selected, as long as they capture enough of the variability in the dataset. The obvious difference between the two clusterings is the clusters containing the EmptyDrops cells but it is reassuring that including these cells did not affect the other clusters and suggests that Seurat is also robust to the presence of some low-quality cells.

A standard analysis of an scRNA-seq dataset often ends here, but I also wanted to demonstrate how some newly developed methods can provide additional context to clusters. PAGA is a method that explores the connectivity between clusters, showing how they are related and reducing the dataset to a graph of clusters. When applied here I saw a high amount of connectivity amongst stromal clusters, and some of the expected developmental trajectories were also highly connected. Other clusters that we would expect to be similar, such as the two endothelial clusters, were not so related. Overall PAGA was not particularly effective when used on this dataset but I believe that is more likely to be a result of this specific dataset or the parameters selected rather than the approach itself. Another approach which gives additional context to a dataset is calculating cell velocity, which provides a transcriptional direction for each cell. Averaged across the dataset it can reveal where populations are headed or which populations are transcriptionally stable. The cell velocity field also highlighted some expected differentiation processes and showed cells in the possible immune cell population trending in opposite directions, suggesting that this cluster might contain sub-populations. Potentially this cluster may contain a mixture of immune cells, other cell types and low-quality cells.

Overall this new analysis is broadly consistent with the results we previously published, despite using different tools at several stages and making different decisions about how to set parameters and thresholds. The main differences between the two analyses are the result of the droplet selection step, with the test-based methods identifying many more cells. Including these cells in the analysis resulted in cluster containing a large number of cells only identified by EmptyDrops and showing a potential immune-like expression profile we hadn't observed before. A small change in parameters at this early stage could have hidden this population or introduced more low-quality cells, either of which would have had a large impact of the rest of the analysis. This emphasises the importance of carefully considering decisions during analysis, visualising the data to check and confirm them and iterating over an analysis if required. The overall consistency in results is reassuring and suggests that Seurat is robust to both the genes that are selected for clustering and the cells that are present. However, there is a large amount of heterogeneity in this dataset with many different cell types that can easily be separated. For an analysis where the goal was to identify more subtle differences, such as an experiment comparing patient-derived cells with cells from a corrected cell line, I would expect these decisions to be more important and to have a bigger impact on how the results are interpreted. Finally, new methods can extract additional information from an scRNA-seq dataset and provide additional evidence that helps with interpretation. I expect that methods like these will become a standard part of analyses, with multiple methods that approach a dataset in different ways being used to identify cell types and confirm hypotheses.

\hypertarget{conclusion}{%
\chapter{Conclusion}\label{conclusion}}

\begin{quote}
\textbf{\enquote{Once you've decided that something's absolutely true, you've closed your mind on it, and a closed mind doesn't go anywhere. Question everything. That's what education's all about.}}

--- David and Leigh Eddings

\emph{Belgarath the Sorcerer}, 1995
\end{quote}

\begin{center}\includegraphics[width=1\linewidth]{/home/travis/build/lazappi/phd-thesis/figures/06-conclusion} \end{center}

\clearpage

My thesis explores the computational tools and techniques used to analyse single-cell RNA sequencing data. This type of data is the result of new technologies that have been developed over the last few years. It is now possible to capture individual cells, extract and isolate RNA from them and quantify the expression levels of genes in each cell. These new technologies allow us to inspect cells in a new way and open up a range of possibilities for biological studies, but usefulness of these technologies relies on our ability to interpret the data they produce. My PhD has coincided with a surge in the development of methods that attempt to make sense of this data and my thesis aims to chart this development and contribute to it. My contributions revolve around how we make decisions during an scRNA-seq analysis, which tools do we choose, how do we know that they work and what parameter values should we use?

My thesis contains four discrete but related projects that explore scRNA-seq analysis methods and decisions made during the analysis process.

In \textbf{Chapter \ref{tools} - \protect\hyperlink{tools}{The scRNA-seq tools landscape}} I present my work developing and maintaining the scRNA-tools database, a database of software tools for analysing scRNA-seq data. I have recorded the details of over 450 tools including where they have been published, where the code is available, associated software licenses and the types of analysis they can perform. The scRNA-tools database allows researchers to identify focus areas of methods development, captures features of how tools are developed and shared, and displays trends in tool development over time. I have also designed a website interface to the database that makes this resource easily available to the research community. This website has been visited by hundreds of researchers from around the world. Over time new kinds of analysis have emerged and these could be integrated into the scRNA-tools database as categories to better organise the tools that are present. In the future the performance of the website could also be improved, particularly to better handle the large number of tools that it now displays.

\textbf{Chapter \ref{simulation} - \protect\hyperlink{simulation}{Simulating scRNA-seq data}} describes Splatter, a software package for generating synthetic scRNA-seq datasets. Developers of new analysis tools should demonstrate their tools' effectiveness and simulations are an important way of evaluating computational methods. Splatter provides a consistent, easy-to-use interface to several published simulations models, allowing researchers to select the most appropriate model for their situation. This interface includes functions for estimating parameters from real datasets, objects for storing those parameters and functions for using them to produce synthetic datasets with associated intermediate values, in a standard format. One of these models is the Splat simulation we introduced in the Splatter publication. This model is based on a negative binomial distribution and includes many of the features considered important in scRNA-seq, data including variation in total counts per cell, relationships between the mean and variance of each gene, and the ability to add additional dropout based on mean expression. The Splat model has been designed to be flexible and can be used to generate datasets with different properties, including different groups of cells, batch effects and continuous trajectories. Splatter also contains functions for assessing the similarity of simulations to real scRNA-seq datasets and we used these to compare the performance of several simulation models. The results showed good performance for the Splat model overall, however we did see that performance varied with different capture technologies and tissue types. When comparing the models currently available in Splatter I saw that those with sophisticated estimation procedures, including BASiCS and ZINB-WaVE, performed extremely well. The performance of these newer models suggests that the Splat estimation procedure could be improved and doing so would improve its usefulness for developers. The simulation model could also be extended in various ways, for example by including an underlying gene network or by making it possible to use the same set of intermediate values for multiple simulations.

One of the analysis steps that has received a large amount of attention from scRNA-seq methods developers is using unsupervised clustering to form groups of cells with similar transcriptional protocols. These approaches could be assessed using simulations and each method will have some parameters that affect how many clusters are produced. The number of clusters used in an analysis can affect how the results are interpreted but choosing these parameters is difficult. In \textbf{Chapter \ref{clust-trees} - \protect\hyperlink{clust-trees}{Visualising clustering across resolutions}} I introduce clustering trees, an alternative, visualisation-based approach for selecting parameters that affect cluster number. For this approach a dataset is clustered multiple times at different resolutions and the overlap of samples between clusters at neighbouring resolutions is used to create a graph structure that is displayed as a tree. The structure of this tree can provide clues as to which resolution is most appropriate for an analysis, and more generally this visualisation can be used to display information across resolutions. How useful the clustering tree visualisation is depends somewhat on the clustering method used and it can become cluttered when there are many clusters present. Further work could look at improving these aspects of the algorithm and adapting it for fuzzy clustering methods. I also believe that the general idea of forming a graph of clusters may have other uses in a variety of applications. The clustering trees algorithm is implemented in a software package that is publicly available from the CRAN repository.

The final project of my PhD was to analyse a kidney organoid scRNA-seq dataset. This dataset was introduced in a publication investigating variability in the organoid differentiation protocol, where it was used to confirm that much of the variability is due to differences between batches rather than between organoids from the same batch. In a second publication, for which I performed the computational analysis, this dataset was profiled more thoroughly and the cell types present were identified. These included many of the cell types expected in the developing kidney as well as some off-target populations. To confirm these cell types we integrated the organoid data with a human foetal kidney dataset and found that many of the cell types were consistent. In \textbf{Chapter \ref{analysis} - \protect\hyperlink{analysis}{Analysis of kidney organoid scRNA-seq data}} I re-analysed this dataset to demonstrate some alternative tools that we could have used in the analysis, focusing on how the decisions made at different stages affect the results. My main finding here was that using alternative droplet selection methods resulted in many more cells being included in the analysis. Continuing the analysis with these additional cells showed and that many of them were from a population which showed some evidence of an immune-like signature. While immune cells are not expected in kidney organoids, and their presence would need to be validated experimentally, there is a plausible pathway by which they could emerge and we did see hints of their presence in the published analysis. My re-analysis also showed robustness of the clustering methods used to how genes were selected and the presence of the additional cells. For both the published analysis and the analysis in this thesis chapter I have followed a reproducible analysis workflow, including producing websites to display the analysis code and results and making the version controlled code publicly available.

The rise of single-cell technologies has been staggering and over the course of my PhD the field has moved from exploratory studies with a few hundred cells to detailed investigations of complex tissues and processes with tens or hundreds of thousands of cells. Although scRNA-seq technologies have only been available for a short time, they continue to improve and over the next few years we will continue to see significant changes. New platforms and protocols will be developed and used to generate even larger and more detailed datasets. Experiments that combine scRNA-seq with other single-cell measurements such as chromatin accessibility will become more common, as will spatial transcriptomics technologies that can retain a cell's position in a tissue while measuring its transcriptional profile.

The continued improvement in single-cell technologies will require further development of computational methods for analysing the data they produce, building on the significant work that has already been done. As bigger datasets become more common, tools will need to be carefully engineered to handle the computational workload. Bigger studies will also require methods to integrate multiple samples and different types of data. This has already become a significant area of research but it is a difficult problem and I expect it will take some time for the robustness and effectiveness of these methods to be established.

As I have discussed throughout my thesis, deciding which methods to use for various tasks is a crucial part of scRNA-seq analysis. Currently these decisions are largely made based on the analyst's experience and intuition, what is common practice and how easy tools are to install and use. Benchmarking and comparison studies are vital for helping to understand which tools are effective, but they can be difficult and time consuming to perform. For some analysis tasks, such as differential expression testing, there are established evaluation metrics that can be used, but for others, such as ordering of cells or integration of datasets, it is less clear how to measure performance. Developing new metrics is vital for informative comparisons. As more benchmarking studies are performed, I hope the field crystallises around an established set of methods for common tasks and energy can be redirected towards alternative approaches, performing complex and novel analyses and improving the quality, usability and efficiency of existing tools.

Even when scRNA-seq analysis methods are reliable and effective, interpreting their results can be difficult and requires the input of a domain expert. Often there is a somewhat circular process where established knowledge is used to evaluate results from what is supposed to be an unsupervised process. To some degree this is the result of the novelty of the field and a reluctance to trust unfamiliar methods, but it is also true that we are looking at biology at a new resolution and this is challenging our current understanding. Large scale profiling projects such as the Human Cell Atlas will make interpretation easier by providing a reference to test against. When it is easier to assign identities to cells, it will become more practical to perform designed experiments such as studies that investigate the effect of a drug in a target cell type in a particular tissue. This has already begun to happen as exploratory analyses have been completed on many tissues. However, building these reference atlases is an extremely difficult task that will take many years and a well-coordinated global effort. Developing methods that best make use of these references will also require significant research but that is an effort that is already underway.

Single-cell RNA sequencing is an exciting technology that has opened up a world of possibilities, but we are still at the early stages of working out how best to use it. My contribution to this effort is in building a database that tracks the analysis tools that are available, and developing simulation software that can be used to evaluate methods. As well as deciding which tool to use, analysts need to choose parameters for those they select and I have developed a visualisation that can help choose parameters that change clustering resolution. I have also contributed to the biological literature by performing the computational analysis in a study that profiled the cell types present in kidney organoids and exploring how the decisions made during this analysis could affect the results and how they are interpreted. Overall I believe I have made a significant contribution to the emerging field of single-cell RNA sequencing data analysis during my PhD.

\hypertarget{references}{%
\chapter*{References}\label{references}}
\addcontentsline{toc}{chapter}{References}

\Begin{tcother}

\hypertarget{refs}{}
\leavevmode\hypertarget{ref-Watson1953-fx}{}%
1. Watson JD, Crick FH. ``Molecular structure of nucleic acids; a structure for deoxyribose nucleic acid.'' \emph{Nature}. 1953;171:737--8. DOI: \href{https://doi.org/10.1038/171737a0}{10.1038/171737a0}

\leavevmode\hypertarget{ref-Wilkins1953-ik}{}%
2. Wilkins MHF, Stokes AR, Wilson HR. ``Molecular structure of deoxypentose nucleic acids.'' \emph{Nature}. 1953;171:738--40. DOI: \href{https://doi.org/10.1038/171738a0}{10.1038/171738a0}

\leavevmode\hypertarget{ref-Franklin1953-pe}{}%
3. Franklin RE, Gosling RG. ``Molecular configuration in sodium thymonucleate.'' \emph{Nature}. 1953;171:740--1. DOI: \href{https://doi.org/10.1038/171740a0}{10.1038/171740a0}

\leavevmode\hypertarget{ref-Raz2011-kn}{}%
4. Raz T, Kapranov P, Lipson D, Letovsky S, Milos PM, Thompson JF. ``Protocol dependence of sequencing-based gene expression measurements.'' \emph{PloS One}. 2011;6:e19287. DOI: \href{https://doi.org/10.1371/journal.pone.0019287}{10.1371/journal.pone.0019287}

\leavevmode\hypertarget{ref-Sultan2014-xh}{}%
5. Sultan M, Amstislavskiy V, Risch T, Schuette M, Dkel S, Ralser M, Balzereit D, Lehrach H, Yaspo M-L. ``Influence of RNA extraction methods and library selection schemes on RNA-seq data.'' \emph{BMC Genomics}. 2014;15:675. DOI: \href{https://doi.org/10.1186/1471-2164-15-675}{10.1186/1471-2164-15-675}

\leavevmode\hypertarget{ref-Voelker_undated-ze}{}%
6. Voelker R, Small C, Bassham S, Catchen J, Sydes J, Cresko B. ``RNA-seqlopedia.'' Available from: \url{https://web.archive.org/web/20181218043831/https://rnaseq.uoregon.edu/} Accessed: 2018-12-18

\leavevmode\hypertarget{ref-Mardis2008-ju}{}%
7. Mardis ER. ``Next-generation DNA sequencing methods.'' \emph{Annual Review of Genomics and Human Genetics}. 2008;9:387--402. DOI: \href{https://doi.org/10.1146/annurev.genom.9.081307.164359}{10.1146/annurev.genom.9.081307.164359}

\leavevmode\hypertarget{ref-Dobin2013-gx}{}%
8. Dobin A, Davis CA, Schlesinger F, Drenkow J, Zaleski C, Jha S, Batut P, Chaisson M, Gingeras TR. ``STAR: ultrafast universal RNA-seq aligner.'' \emph{Bioinformatics}. 2013;29:15--21. DOI: \href{https://doi.org/10.1093/bioinformatics/bts635}{10.1093/bioinformatics/bts635}

\leavevmode\hypertarget{ref-Kim2015-ys}{}%
9. Kim D, Langmead B, Salzberg SL. ``HISAT: a fast spliced aligner with low memory requirements.'' \emph{Nature Methods}. 2015;12:357--60. DOI: \href{https://doi.org/10.1038/nmeth.3317}{10.1038/nmeth.3317}

\leavevmode\hypertarget{ref-Liao2013-id}{}%
10. Liao Y, Smyth GK, Shi W. ``The Subread aligner: fast, accurate and scalable read mapping by seed-and-vote.'' \emph{Nucleic Acids Research}. 2013;41:e108. DOI: \href{https://doi.org/10.1093/nar/gkt214}{10.1093/nar/gkt214}

\leavevmode\hypertarget{ref-Bray2016-tm}{}%
11. Bray NL, Pimentel H, Melsted P, Pachter L. ``Near-optimal probabilistic RNA-seq quantification.'' \emph{Nature Biotechnology}. 2016;34:525--7. DOI: \href{https://doi.org/10.1038/nbt.3519}{10.1038/nbt.3519}

\leavevmode\hypertarget{ref-Patro2017-ka}{}%
12. Patro R, Duggal G, Love MI, Irizarry RA, Kingsford C. ``Salmon provides fast and bias-aware quantification of transcript expression.'' \emph{Nature Methods}. 2017;14:417--9. DOI: \href{https://doi.org/10.1038/nmeth.4197}{10.1038/nmeth.4197}

\leavevmode\hypertarget{ref-Robinson2010-ll}{}%
13. Robinson MD, Oshlack A. ``A scaling normalization method for differential expression analysis of RNA-seq data.'' \emph{Genome Biology}. 2010;11:R25. DOI: \href{https://doi.org/10.1186/gb-2010-11-3-r25}{10.1186/gb-2010-11-3-r25}

\leavevmode\hypertarget{ref-Kim2015-mo}{}%
14. Kim JK, Kolodziejczyk AA, Illicic T, Teichmann SA, Marioni JC. ``Characterizing noise structure in single-cell RNA-seq distinguishes genuine from technical stochastic allelic expression.'' \emph{Nature Communications}. 2015;6:8687. DOI: \href{https://doi.org/10.1038/ncomms9687}{10.1038/ncomms9687}

\leavevmode\hypertarget{ref-Robinson2010-pt}{}%
15. Robinson MD, McCarthy DJ, Smyth GK. ``edgeR: a Bioconductor package for differential expression analysis of digital gene expression data.'' \emph{Bioinformatics}. 2010;26:139--40. DOI: \href{https://doi.org/10.1093/bioinformatics/btp616}{10.1093/bioinformatics/btp616}

\leavevmode\hypertarget{ref-McCarthy2012-gc}{}%
16. McCarthy DJ, Chen Y, Smyth GK. ``Differential expression analysis of multifactor RNA-Seq experiments with respect to biological variation.'' \emph{Nucleic Acids Research}. 2012;40:4288--97. DOI: \href{https://doi.org/10.1093/nar/gks042}{10.1093/nar/gks042}

\leavevmode\hypertarget{ref-Anders2010-pq}{}%
17. Anders S, Huber W. ``Differential expression analysis for sequence count data.'' \emph{Genome Biology}. 2010;11:R106. DOI: \href{https://doi.org/10.1186/gb-2010-11-10-r106}{10.1186/gb-2010-11-10-r106}

\leavevmode\hypertarget{ref-Love2014-tw}{}%
18. Love MI, Huber W, Anders S. ``Moderated estimation of fold change and dispersion for RNA-seq data with DESeq2.'' \emph{Genome Biology}. 2014;15:550. DOI: \href{https://doi.org/10.1186/s13059-014-0550-8}{10.1186/s13059-014-0550-8}

\leavevmode\hypertarget{ref-Mortazavi2008-vu}{}%
19. Mortazavi A, Williams BA, McCue K, Schaeffer L, Wold B. ``Mapping and quantifying mammalian transcriptomes by RNA-Seq.'' \emph{Nature Methods}. 2008;5:621--8. DOI: \href{https://doi.org/10.1038/nmeth.1226}{10.1038/nmeth.1226}

\leavevmode\hypertarget{ref-Wagner2012-qf}{}%
20. Wagner GP, Kin K, Lynch VJ. ``Measurement of mRNA abundance using RNA-seq data: RPKM measure is inconsistent among samples.'' \emph{Theory in Biosciences}. 2012;131:281--5. DOI: \href{https://doi.org/10.1007/s12064-012-0162-3}{10.1007/s12064-012-0162-3}

\leavevmode\hypertarget{ref-Ritchie2015-te}{}%
21. Ritchie ME, Phipson B, Wu D, Hu Y, Law CW, Shi W, Smyth GK. ``limma powers differential expression analyses for RNA-sequencing and microarray studies.'' \emph{Nucleic Acids Research}. 2015;43:e47. DOI: \href{https://doi.org/10.1093/nar/gkv007}{10.1093/nar/gkv007}

\leavevmode\hypertarget{ref-Law2014-nq}{}%
22. Law CW, Chen Y, Shi W, Smyth GK. ``voom: Precision weights unlock linear model analysis tools for RNA-seq read counts.'' \emph{Genome Biology}. 2014;15:R29. DOI: \href{https://doi.org/10.1186/gb-2014-15-2-r29}{10.1186/gb-2014-15-2-r29}

\leavevmode\hypertarget{ref-Risso2014-zw}{}%
23. Risso D, Ngai J, Speed TP, Dudoit S. ``Normalization of RNA-seq data using factor analysis of control genes or samples.'' \emph{Nature Biotechnology}. 2014;32:896--902. DOI: \href{https://doi.org/10.1038/nbt.2931}{10.1038/nbt.2931}

\leavevmode\hypertarget{ref-Tang2009-jd}{}%
24. Tang F, Barbacioru C, Wang Y, Nordman E, Lee C, Xu N, Wang X, Bodeau J, Tuch BB, Siddiqui A, Lao K, Surani MA. ``mRNA-Seq whole-transcriptome analysis of a single cell.'' \emph{Nature Methods}. 2009;6:377--82. DOI: \href{https://doi.org/10.1038/nmeth.1315}{10.1038/nmeth.1315}

\leavevmode\hypertarget{ref-Hashimshony2012-dh}{}%
25. Hashimshony T, Wagner F, Sher N, Yanai I. ``CEL-Seq: single-cell RNA-Seq by multiplexed linear amplification.'' \emph{Cell Reports}. 2012;2:666--73. DOI: \href{https://doi.org/10.1016/j.celrep.2012.08.003}{10.1016/j.celrep.2012.08.003}

\leavevmode\hypertarget{ref-Hashimshony2016-yi}{}%
26. Hashimshony T, Senderovich N, Avital G, Klochendler A, Leeuw Y de, Anavy L, Gennert D, Li S, Livak KJ, Rozenblatt-Rosen O, Dor Y, Regev A, Yanai I. ``CEL-Seq2: sensitive highly-multiplexed single-cell RNA-Seq.'' \emph{Genome Biology}. 2016;17:77. DOI: \href{https://doi.org/10.1186/s13059-016-0938-8}{10.1186/s13059-016-0938-8}

\leavevmode\hypertarget{ref-Sasagawa2013-uc}{}%
27. Sasagawa Y, Nikaido I, Hayashi T, Danno H, Uno KD, Imai T, Ueda HR. ``Quartz-Seq: a highly reproducible and sensitive single-cell RNA sequencing method, reveals non-genetic gene-expression heterogeneity.'' \emph{Genome Biology}. 2013;14:R31. DOI: \href{https://doi.org/10.1186/gb-2013-14-4-r31}{10.1186/gb-2013-14-4-r31}

\leavevmode\hypertarget{ref-Sasagawa2018-ox}{}%
28. Sasagawa Y, Danno H, Takada H, Ebisawa M, Tanaka K, Hayashi T, Kurisaki A, Nikaido I. ``Quartz-Seq2: a high-throughput single-cell RNA-sequencing method that effectively uses limited sequence reads.'' \emph{Genome Biology}. 2018;19:29. DOI: \href{https://doi.org/10.1186/s13059-018-1407-3}{10.1186/s13059-018-1407-3}

\leavevmode\hypertarget{ref-Picelli2013-or}{}%
29. Picelli S, Bjrklund K, Faridani OR, Sagasser S, Winberg G, Sandberg R. ``Smart-seq2 for sensitive full-length transcriptome profiling in single cells.'' \emph{Nature Methods}. 2013;10:1096--8. DOI: \href{https://doi.org/10.1038/nmeth.2639}{10.1038/nmeth.2639}

\leavevmode\hypertarget{ref-Svensson2018-dx}{}%
30. Svensson V, Vento-Tormo R, Teichmann SA. ``Exponential scaling of single-cell RNA-seq in the past decade.'' \emph{Nature Protocols}. 2018;13:599. DOI: \href{https://doi.org/10.1038/nprot.2017.149}{10.1038/nprot.2017.149}

\leavevmode\hypertarget{ref-Macosko2015-rl}{}%
31. Macosko EZ, Basu A, Satija R, Nemesh J, Shekhar K, Goldman M, Tirosh I, Bialas AR, Kamitaki N, Martersteck EM, Trombetta JJ, Weitz DA, Sanes JR, Shalek AK, Regev A, McCarroll SA. ``Highly parallel genome-wide expression profiling of individual cells using nanoliter droplets.'' \emph{Cell}. 2015;161:1202--14. DOI: \href{https://doi.org/10.1016/j.cell.2015.05.002}{10.1016/j.cell.2015.05.002}

\leavevmode\hypertarget{ref-Klein2015-iw}{}%
32. Klein AM, Mazutis L, Akartuna I, Tallapragada N, Veres A, Li V, Peshkin L, Weitz DA, Kirschner MW. ``Droplet barcoding for single-cell transcriptomics applied to embryonic stem cells.'' \emph{Cell}. 2015;161:1187--201. DOI: \href{https://doi.org/10.1016/j.cell.2015.04.044}{10.1016/j.cell.2015.04.044}

\leavevmode\hypertarget{ref-Zilionis2017-gu}{}%
33. Zilionis R, Nainys J, Veres A, Savova V, Zemmour D, Klein AM, Mazutis L. ``Single-cell barcoding and sequencing using droplet microfluidics.'' \emph{Nature Protocols}. 2017;12:44--73. DOI: \href{https://doi.org/10.1038/nprot.2016.154}{10.1038/nprot.2016.154}

\leavevmode\hypertarget{ref-Zheng2017-mm}{}%
34. Zheng GXY, Terry JM, Belgrader P, Ryvkin P, Bent ZW, Wilson R, Ziraldo SB, Wheeler TD, McDermott GP, Zhu J, Gregory MT, Shuga J, Montesclaros L, Underwood JG, Masquelier DA, Nishimura SY, Schnall-Levin M, Wyatt PW, et al. ``Massively parallel digital transcriptional profiling of single cells.'' \emph{Nature Communications}. 2017;8:14049. DOI: \href{https://doi.org/10.1038/ncomms14049}{10.1038/ncomms14049}

\leavevmode\hypertarget{ref-Zhang2019-vw}{}%
35. Zhang X, Li T, Liu F, Chen Y, Yao J, Li Z, Huang Y, Wang J. ``Comparative Analysis of Droplet-Based Ultra-High-Throughput Single-Cell RNA-Seq Systems.'' \emph{Molecular Cell}. 2019;73:130--142.e5. DOI: \href{https://doi.org/10.1016/j.molcel.2018.10.020}{10.1016/j.molcel.2018.10.020}

\leavevmode\hypertarget{ref-Kivioja2012-bq}{}%
36. Kivioja T, Vhrautio A, Karlsson K, Bonke M, Enge M, Linnarsson S, Taipale J. ``Counting absolute numbers of molecules using unique molecular identifiers.'' \emph{Nature Methods}. 2012;9:72--4. DOI: \href{https://doi.org/10.1038/nmeth.1778}{10.1038/nmeth.1778}

\leavevmode\hypertarget{ref-Islam2014-gq}{}%
37. Islam S, Zeisel A, Joost S, La Manno G, Zajac P, Kasper M, Lnnerberg P, Linnarsson S. ``Quantitative single-cell RNA-seq with unique molecular identifiers.'' \emph{Nature Methods}. 2014;11:163--6. DOI: \href{https://doi.org/10.1038/nmeth.2772}{10.1038/nmeth.2772}

\leavevmode\hypertarget{ref-Gierahn2017-kr}{}%
38. Gierahn TM, Wadsworth MH 2nd, Hughes TK, Bryson BD, Butler A, Satija R, Fortune S, Love JC, Shalek AK. ``Seq-Well: portable, low-cost RNA sequencing of single cells at high throughput.'' \emph{Nature Methods}. 2017;14:395--8. DOI: \href{https://doi.org/10.1038/nmeth.4179}{10.1038/nmeth.4179}

\leavevmode\hypertarget{ref-Bose2015-vn}{}%
39. Bose S, Wan Z, Carr A, Rizvi AH, Vieira G, Pe'er D, Sims PA. ``Scalable microfluidics for single-cell RNA printing and sequencing.'' \emph{Genome Biology}. 2015;16:120. DOI: \href{https://doi.org/10.1186/s13059-015-0684-3}{10.1186/s13059-015-0684-3}

\leavevmode\hypertarget{ref-Hu2017-jm}{}%
40. Hu P, Fabyanic E, Kwon DY, Tang S, Zhou Z, Wu H. ``Dissecting Cell-Type Composition and Activity-Dependent Transcriptional State in Mammalian Brains by Massively Parallel Single-Nucleus RNA-Seq.'' \emph{Molecular Cell}. 2017;68:1006--1015.e7. DOI: \href{https://doi.org/10.1016/j.molcel.2017.11.017}{10.1016/j.molcel.2017.11.017}

\leavevmode\hypertarget{ref-Gao2017-tq}{}%
41. Gao R, Kim C, Sei E, Foukakis T, Crosetto N, Chan L-K, Srinivasan M, Zhang H, Meric-Bernstam F, Navin N. ``Nanogrid single-nucleus RNA sequencing reveals phenotypic diversity in breast cancer.'' \emph{Nature Communications}. 2017;8:228. DOI: \href{https://doi.org/10.1038/s41467-017-00244-w}{10.1038/s41467-017-00244-w}

\leavevmode\hypertarget{ref-Habib2017-uh}{}%
42. Habib N, Avraham-Davidi I, Basu A, Burks T, Shekhar K, Hofree M, Choudhury SR, Aguet F, Gelfand E, Ardlie K, Weitz DA, Rozenblatt-Rosen O, Zhang F, Regev A. ``Massively parallel single-nucleus RNA-seq with DroNc-seq.'' \emph{Nature Methods}. 2017;14:955--8. DOI: \href{https://doi.org/10.1038/nmeth.4407}{10.1038/nmeth.4407}

\leavevmode\hypertarget{ref-Wu2018-xf}{}%
43. Wu H, Kirita Y, Donnelly EL, Humphreys BD. ``Advantages of Single-Nucleus over Single-Cell RNA Sequencing of Adult Kidney: Rare Cell Types and Novel Cell States Revealed in Fibrosis.'' \emph{Journal of the American Society of Nephrology: JASN}. 2018;30:23--32. DOI: \href{https://doi.org/10.1681/ASN.2018090912}{10.1681/ASN.2018090912}

\leavevmode\hypertarget{ref-Stoeckius2017-wm}{}%
44. Stoeckius M, Hafemeister C, Stephenson W, Houck-Loomis B, Chattopadhyay PK, Swerdlow H, Satija R, Smibert P. ``Simultaneous epitope and transcriptome measurement in single cells.'' \emph{Nature Methods}. 2017;14:865--8. DOI: \href{https://doi.org/10.1038/nmeth.4380}{10.1038/nmeth.4380}

\leavevmode\hypertarget{ref-Stoeckius2018-ij}{}%
45. Stoeckius M, Zheng S, Houck-Loomis B, Hao S, Yeung BZ, Mauck WM 3rd, Smibert P, Satija R. ``Cell Hashing with barcoded antibodies enables multiplexing and doublet detection for single cell genomics.'' \emph{Genome Biology}. 2018;19:224. DOI: \href{https://doi.org/10.1186/s13059-018-1603-1}{10.1186/s13059-018-1603-1}

\leavevmode\hypertarget{ref-Moudgil2019-bl}{}%
46. Moudgil A. ``Multimodal scRNA-seq.'' 2019. DOI: \href{https://doi.org/10.5281/zenodo.2628012}{10.5281/zenodo.2628012}

\leavevmode\hypertarget{ref-CITEseq_web}{}%
47. Technology Innovation Lab, New York Genome Center. ``cite-seq.com.'' Available from: \url{https://web.archive.org/web/20181218054100/https://cite-seq.com/} Accessed: 2018-12-18

\leavevmode\hypertarget{ref-Spanjaard2018-gn}{}%
48. Spanjaard B, Hu B, Mitic N, Olivares-Chauvet P, Janjuha S, Ninov N, Junker JP. ``Simultaneous lineage tracing and cell-type identification using CRISPR-Cas9-induced genetic scars.'' \emph{Nature Biotechnology}. 2018;36:469--73. DOI: \href{https://doi.org/10.1038/nbt.4124}{10.1038/nbt.4124}

\leavevmode\hypertarget{ref-Dixit2016-wp}{}%
49. Dixit A, Parnas O, Li B, Chen J, Fulco CP, Jerby-Arnon L, Marjanovic ND, Dionne D, Burks T, Raychowdhury R, Adamson B, Norman TM, Lander ES, Weissman JS, Friedman N, Regev A. ``Perturb-Seq: Dissecting Molecular Circuits with Scalable Single-Cell RNA Profiling of Pooled Genetic Screens.'' \emph{Cell}. 2016;167:1853--1866.e17. DOI: \href{https://doi.org/10.1016/j.cell.2016.11.038}{10.1016/j.cell.2016.11.038}

\leavevmode\hypertarget{ref-Mimitou2019-ju}{}%
50. Mimitou EP, Cheng A, Montalbano A, Hao S, Stoeckius M, Legut M, Roush T, Herrera A, Papalexi E, Ouyang Z, Satija R, Sanjana NE, Koralov SB, Smibert P. ``Multiplexed detection of proteins, transcriptomes, clonotypes and CRISPR perturbations in single cells.'' \emph{Nature Methods}. 2019;16:409--12. DOI: \href{https://doi.org/10.1038/s41592-019-0392-0}{10.1038/s41592-019-0392-0}

\leavevmode\hypertarget{ref-Biddy2018-uz}{}%
51. Biddy BA, Kong W, Kamimoto K, Guo C, Waye SE, Sun T, Morris SA. ``Single-cell mapping of lineage and identity in direct reprogramming.'' \emph{Nature}. 2018;564:219--24. DOI: \href{https://doi.org/10.1038/s41586-018-0744-4}{10.1038/s41586-018-0744-4}

\leavevmode\hypertarget{ref-Macaulay2015-dp}{}%
52. Macaulay IC, Haerty W, Kumar P, Li YI, Hu TX, Teng MJ, Goolam M, Saurat N, Coupland P, Shirley LM, Smith M, Van der Aa N, Banerjee R, Ellis PD, Quail MA, Swerdlow HP, Zernicka-Goetz M, Livesey FJ, et al. ``G\&T-seq: parallel sequencing of single-cell genomes and transcriptomes.'' \emph{Nature Methods}. 2015;12:519--22. DOI: \href{https://doi.org/10.1038/nmeth.3370}{10.1038/nmeth.3370}

\leavevmode\hypertarget{ref-Han2017-sc}{}%
53. Han KY, Kim K-T, Joung J-G, Son D-S, Kim YJ, Jo A, Jeon H-J, Moon H-S, Yoo CE, Chung W, Eum HH, Kim S, Kim HK, Lee JE, Ahn M-J, Lee H-O, Park D, Park W-Y. ``SIDR: simultaneous isolation and parallel sequencing of genomic DNA and total RNA from single cells.'' \emph{Genome Research}. 2017; DOI: \href{https://doi.org/10.1101/gr.223263.117}{10.1101/gr.223263.117}

\leavevmode\hypertarget{ref-Wang2018-xp}{}%
54. Wang LY, Guo J, Cao W, Zhang M, He J, Li Z. ``Integrated sequencing of exome and mRNA of large-sized single cells.'' \emph{Scientific Reports}. 2018;8:384. DOI: \href{https://doi.org/10.1038/s41598-017-18730-y}{10.1038/s41598-017-18730-y}

\leavevmode\hypertarget{ref-Hu2016-il}{}%
55. Hu Y, Huang K, An Q, Du G, Hu G, Xue J, Zhu X, Wang C-Y, Xue Z, Fan G. ``Simultaneous profiling of transcriptome and DNA methylome from a single cell.'' \emph{Genome Biology}. 2016;17:88. DOI: \href{https://doi.org/10.1186/s13059-016-0950-z}{10.1186/s13059-016-0950-z}

\leavevmode\hypertarget{ref-Cadwell2017-wu}{}%
56. Cadwell CR, Scala F, Li S, Livrizzi G, Shen S, Sandberg R, Jiang X, Tolias AS. ``Multimodal profiling of single-cell morphology, electrophysiology, and gene expression using Patch-seq.'' \emph{Nature Protocols}. 2017;12:2531. DOI: \href{https://doi.org/10.1038/nprot.2017.120}{10.1038/nprot.2017.120}

\leavevmode\hypertarget{ref-Bian2018-rt}{}%
57. Bian S, Hou Y, Zhou X, Li X, Yong J, Wang Y, Wang W, Yan J, Hu B, Guo H, Wang J, Gao S, Mao Y, Dong J, Zhu P, Xiu D, Yan L, Wen L, et al. ``Single-cell multiomics sequencing and analyses of human colorectal cancer.'' \emph{Science}. 2018;362:1060--3. DOI: \href{https://doi.org/10.1126/science.aao3791}{10.1126/science.aao3791}

\leavevmode\hypertarget{ref-Grun2014-zn}{}%
58. Grn D, Kester L, Oudenaarden A van. ``Validation of noise models for single-cell transcriptomics.'' \emph{Nature Methods}. 2014;11:637--40. DOI: \href{https://doi.org/10.1038/nmeth.2930}{10.1038/nmeth.2930}

\leavevmode\hypertarget{ref-Liu2016-wq}{}%
59. Liu S, Trapnell C. ``Single-cell transcriptome sequencing: recent advances and remaining challenges.'' \emph{F1000Research}. 2016;5. DOI: \href{https://doi.org/10.12688/f1000research.7223.1}{10.12688/f1000research.7223.1}

\leavevmode\hypertarget{ref-Adam2017-zl}{}%
60. Adam M, Potter AS, Potter SS. ``Psychrophilic proteases dramatically reduce single cell RNA-seq artifacts: A molecular atlas of kidney development.'' \emph{Development}. 2017;144:3625--32. DOI: \href{https://doi.org/10.1242/dev.151142}{10.1242/dev.151142}

\leavevmode\hypertarget{ref-Hicks2017-ut}{}%
61. Hicks SC, Townes FW, Teng M, Irizarry RA. ``Missing data and technical variability in single-cell RNA-sequencing experiments.'' \emph{Biostatistics}. 2017; DOI: \href{https://doi.org/10.1093/biostatistics/kxx053}{10.1093/biostatistics/kxx053}

\leavevmode\hypertarget{ref-Pierson2015-qp}{}%
62. Pierson E, Yau C. ``ZIFA: Dimensionality reduction for zero-inflated single-cell gene expression analysis.'' \emph{Genome Biology}. 2015;16:241. DOI: \href{https://doi.org/10.1186/s13059-015-0805-z}{10.1186/s13059-015-0805-z}

\leavevmode\hypertarget{ref-Risso2018-dg}{}%
63. Risso D, Perraudeau F, Gribkova S, Dudoit S, Vert J-P. ``A general and flexible method for signal extraction from single-cell RNA-seq data.'' \emph{Nature Communications}. 2018;9:284. DOI: \href{https://doi.org/10.1038/s41467-017-02554-5}{10.1038/s41467-017-02554-5}

\leavevmode\hypertarget{ref-Van_den_Berge2018-ip}{}%
64. Van den Berge K, Perraudeau F, Soneson C, Love MI, Risso D, Vert J-P, Robinson MD, Dudoit S, Clement L. ``Observation weights unlock bulk RNA-seq tools for zero inflation and single-cell applications.'' \emph{Genome Biology}. 2018;19:24. DOI: \href{https://doi.org/10.1186/s13059-018-1406-4}{10.1186/s13059-018-1406-4}

\leavevmode\hypertarget{ref-Miao2018-ft}{}%
65. Miao Z, Deng K, Wang X, Zhang X. ``DEsingle for detecting three types of differential expression in single-cell RNA-seq data.'' \emph{Bioinformatics}. 2018; DOI: \href{https://doi.org/10.1093/bioinformatics/bty332}{10.1093/bioinformatics/bty332}

\leavevmode\hypertarget{ref-Svensson2019-el}{}%
66. Svensson V. ``Droplet scRNA-seq is not zero-inflated.'' \emph{bioRxiv}. 2019. p. 582064. DOI: \href{https://doi.org/10.1101/582064}{10.1101/582064}

\leavevmode\hypertarget{ref-Van_Dijk2018-dk}{}%
67. Dijk D van, Sharma R, Nainys J, Yim K, Kathail P, Carr AJ, Burdziak C, Moon KR, Chaffer CL, Pattabiraman D, Bierie B, Mazutis L, Wolf G, Krishnaswamy S, Pe'er D. ``Recovering Gene Interactions from Single-Cell Data Using Data Diffusion.'' \emph{Cell}. 2018;174:P716--729.E27. DOI: \href{https://doi.org/10.1016/j.cell.2018.05.061}{10.1016/j.cell.2018.05.061}

\leavevmode\hypertarget{ref-Huang2018-ef}{}%
68. Huang M, Wang J, Torre E, Dueck H, Shaffer S, Bonasio R, Murray JI, Raj A, Li M, Zhang NR. ``SAVER: gene expression recovery for single-cell RNA sequencing.'' \emph{Nature Methods}. 2018;15:539--42. DOI: \href{https://doi.org/10.1038/s41592-018-0033-z}{10.1038/s41592-018-0033-z}

\leavevmode\hypertarget{ref-Wang2018-av}{}%
69. Wang J, Agarwal D, Huang M, Hu G, Zhou Z, Conley VB, MacMullan H, Zhang NR. ``Transfer learning in single-cell transcriptomics improves data denoising and pattern discovery.'' \emph{bioRxiv}. 2018. p. 457879. DOI: \href{https://doi.org/10.1101/457879}{10.1101/457879}

\leavevmode\hypertarget{ref-Li2018-gx}{}%
70. Li WV, Li JJ. ``An accurate and robust imputation method scImpute for single-cell RNA-seq data.'' \emph{Nature Communications}. 2018;9:997. DOI: \href{https://doi.org/10.1038/s41467-018-03405-7}{10.1038/s41467-018-03405-7}

\leavevmode\hypertarget{ref-Andrews2018-li}{}%
71. Andrews TS, Hemberg M. ``False signals induced by single-cell imputation.'' \emph{F1000Research}. 2018;7. DOI: \href{https://doi.org/10.12688/f1000research.16613.1}{10.12688/f1000research.16613.1}

\leavevmode\hypertarget{ref-Zeisel2015-rd}{}%
72. Zeisel A, Muoz-Manchado AB, Codeluppi S, Lnnerberg P, La Manno G, Jurus A, Marques S, Munguba H, He L, Betsholtz C, Rolny C, Castelo-Branco G, Hjerling-Leffler J, Linnarsson S. ``Brain structure. Cell types in the mouse cortex and hippocampus revealed by single-cell RNA-seq.'' \emph{Science}. 2015;347:1138--42. DOI: \href{https://doi.org/10.1126/science.aaa1934}{10.1126/science.aaa1934}

\leavevmode\hypertarget{ref-Patel2014-bl}{}%
73. Patel AP, Tirosh I, Trombetta JJ, Shalek AK, Gillespie SM, Wakimoto H, Cahill DP, Nahed BV, Curry WT, Martuza RL, Louis DN, Rozenblatt-Rosen O, Suv ML, Regev A, Bernstein BE. ``Single-cell RNA-seq highlights intratumoral heterogeneity in primary glioblastoma.'' \emph{Science}. 2014;344:1396--401. DOI: \href{https://doi.org/10.1126/science.1254257}{10.1126/science.1254257}

\leavevmode\hypertarget{ref-Treutlein2014-wd}{}%
74. Treutlein B, Brownfield DG, Wu AR, Neff NF, Mantalas GL, Espinoza FH, Desai TJ, Krasnow MA, Quake SR. ``Reconstructing lineage hierarchies of the distal lung epithelium using single-cell RNA-seq.'' \emph{Nature}. 2014;509:371--5. DOI: \href{https://doi.org/10.1038/nature13173}{10.1038/nature13173}

\leavevmode\hypertarget{ref-Usoskin2015-fz}{}%
75. Usoskin D, Furlan A, Islam S, Abdo H, Lnnerberg P, Lou D, Hjerling-Leffler J, Haeggstrm J, Kharchenko O, Kharchenko PV, Linnarsson S, Ernfors P. ``Unbiased classification of sensory neuron types by large-scale single-cell RNA sequencing.'' \emph{Nature Neuroscience}. 2015;18:145--53. DOI: \href{https://doi.org/10.1038/nn.3881}{10.1038/nn.3881}

\leavevmode\hypertarget{ref-Buettner2015-rq}{}%
76. Buettner F, Natarajan KN, Casale FP, Proserpio V, Scialdone A, Theis FJ, Teichmann SA, Marioni JC, Stegle O. ``Computational analysis of cell-to-cell heterogeneity in single-cell RNA-sequencing data reveals hidden subpopulations of cells.'' \emph{Nature Biotechnology}. 2015;33:155--60. DOI: \href{https://doi.org/10.1038/nbt.3102}{10.1038/nbt.3102}

\leavevmode\hypertarget{ref-Trapnell2014-he}{}%
77. Trapnell C, Cacchiarelli D, Grimsby J, Pokharel P, Li S, Morse M, Lennon NJ, Livak KJ, Mikkelsen TS, Rinn JL. ``The dynamics and regulators of cell fate decisions are revealed by pseudotemporal ordering of single cells.'' \emph{Nature Biotechnology}. 2014;32:381--6. DOI: \href{https://doi.org/10.1038/nbt.2859}{10.1038/nbt.2859}

\leavevmode\hypertarget{ref-Regev2017-tz}{}%
78. Regev A, Teichmann SA, Lander ES, Amit I, Benoist C, Birney E, Bodenmiller B, Campbell P, Carninci P, Clatworthy M, Clevers H, Deplancke B, Dunham I, Eberwine J, Eils R, Enard W, Farmer A, Fugger L, et al. ``The Human Cell Atlas.'' \emph{eLife}. 2017;6. DOI: \href{https://doi.org/10.7554/eLife.27041}{10.7554/eLife.27041}

\leavevmode\hypertarget{ref-Cao2017-ig}{}%
79. Cao J, Packer JS, Ramani V, Cusanovich DA, Huynh C, Daza R, Qiu X, Lee C, Furlan SN, Steemers FJ, Adey A, Waterston RH, Trapnell C, Shendure J. ``Comprehensive single-cell transcriptional profiling of a multicellular organism.'' \emph{Science}. 2017;357:661--7. DOI: \href{https://doi.org/10.1126/science.aam8940}{10.1126/science.aam8940}

\leavevmode\hypertarget{ref-Tabula_Muris_Consortium2018-pb}{}%
80. Tabula Muris Consortium, Overall coordination, Logistical coordination, Organ collection and processing, Library preparation and sequencing, Computational data analysis, Cell type annotation, Writing group, Supplemental text writing group, Principal investigators. ``Single-cell transcriptomics of 20 mouse organs creates a Tabula Muris.'' \emph{Nature}. 2018;562:367--72. DOI: \href{https://doi.org/10.1038/s41586-018-0590-4}{10.1038/s41586-018-0590-4}

\leavevmode\hypertarget{ref-Wu2018-xd}{}%
81. Wu B, Li Y, Liu Y, Jin K, Zhao K, An C, Li Q, Gong L, Zhao W, Hu J, Qian J, Ouyang H, Zou X. ``Cell atlas of human uterus.'' \emph{bioRxiv}. 2018. p. 267849. DOI: \href{https://doi.org/10.1101/267849}{10.1101/267849}

\leavevmode\hypertarget{ref-Han2018-kp}{}%
82. Han X, Wang R, Zhou Y, Fei L, Sun H, Lai S, Saadatpour A, Zhou Z, Chen H, Ye F, Huang D, Xu Y, Huang W, Jiang M, Jiang X, Mao J, Chen Y, Lu C, et al. ``Mapping the Mouse Cell Atlas by Microwell-Seq.'' \emph{Cell}. 2018;172:1091--1107.e17. DOI: \href{https://doi.org/10.1016/j.cell.2018.02.001}{10.1016/j.cell.2018.02.001}

\leavevmode\hypertarget{ref-Saunders2018-ut}{}%
83. Saunders A, Macosko E, Wysoker A, Goldman M, Krienen F, Rivera H de, Bien E, Baum M, Wang S, Goeva A, Nemesh J, Kamitaki N, Brumbaugh S, Kulp D, McCarroll SA. ``A Single-Cell Atlas of Cell Types, States, and Other Transcriptional Patterns from Nine Regions of the Adult Mouse Brain.'' \emph{bioRxiv}. 2018. p. 299081. DOI: \href{https://doi.org/10.1101/299081}{10.1101/299081}

\leavevmode\hypertarget{ref-Plass2018-sa}{}%
84. Plass M, Solana J, Wolf FA, Ayoub S, Misios A, Glaar P, Obermayer B, Theis FJ, Kocks C, Rajewsky N. ``Cell type atlas and lineage tree of a whole complex animal by single-cell transcriptomics.'' \emph{Science}. 2018;360. DOI: \href{https://doi.org/10.1126/science.aaq1723}{10.1126/science.aaq1723}

\leavevmode\hypertarget{ref-Bhaduri2018-nt}{}%
85. Bhaduri A, Nowakowski TJ, Pollen AA, Kriegstein AR. ``Identification of cell types in a mouse brain single-cell atlas using low sampling coverage.'' \emph{BMC Biology}. 2018;16:113. DOI: \href{https://doi.org/10.1186/s12915-018-0580-x}{10.1186/s12915-018-0580-x}

\leavevmode\hypertarget{ref-Yuan2018-qu}{}%
86. Yuan H, Yan M, Zhang G, Liu W, Deng C, Liao G, Xu L, Luo T, Yan H, Long Z, Shi A, Zhao T, Xiao Y, Li X. ``CancerSEA: a cancer single-cell state atlas.'' \emph{Nucleic Acids Research}. 2018;47:D900--8. DOI: \href{https://doi.org/10.1093/nar/gky939}{10.1093/nar/gky939}

\leavevmode\hypertarget{ref-Davie2018-eo}{}%
87. Davie K, Janssens J, Koldere D, De Waegeneer M, Pech U, Kreft , Aibar S, Makhzami S, Christiaens V, Bravo Gonzlez-Blas C, Poovathingal S, Hulselmans G, Spanier KI, Moerman T, Vanspauwen B, Geurs S, Voet T, Lammertyn J, et al. ``A Single-Cell Transcriptome Atlas of the Aging Drosophila Brain.'' \emph{Cell}. 2018;174:982--998.e20. DOI: \href{https://doi.org/10.1016/j.cell.2018.05.057}{10.1016/j.cell.2018.05.057}

\leavevmode\hypertarget{ref-Taylor2019-hy}{}%
88. Taylor DM, Aronow BJ, Tan K, Bernt K, Salomonis N, Greene CS, Frolova A, Henrickson SE, Wells A, Pei L, Jaiswal JK, Whitsett J, Hamilton KE, MacParland SA, Kelsen J, Heuckeroth RO, Potter SS, Vella LA, et al. ``The Pediatric Cell Atlas: Defining the Growth Phase of Human Development at Single-Cell Resolution.'' \emph{Developmental Cell}. 2019; DOI: \href{https://doi.org/10.1016/j.devcel.2019.03.001}{10.1016/j.devcel.2019.03.001}

\leavevmode\hypertarget{ref-Lun2016-lp}{}%
89. Lun ATL, McCarthy DJ, Marioni JC. ``A step-by-step workflow for low-level analysis of single-cell RNA-seq data with Bioconductor.'' \emph{F1000Research}. 2016;5:2122. DOI: \href{https://doi.org/10.12688/f1000research.9501.2}{10.12688/f1000research.9501.2}

\leavevmode\hypertarget{ref-Perraudeau2017-ii}{}%
90. Perraudeau F, Risso D, Street K, Purdom E, Dudoit S. ``Bioconductor workflow for single-cell RNA sequencing: Normalization, dimensionality reduction, clustering, and lineage inference.'' \emph{F1000Research}. 2017;6. DOI: \href{https://doi.org/10.12688/f1000research.12122.1}{10.12688/f1000research.12122.1}

\leavevmode\hypertarget{ref-Srivastava2019-ce}{}%
91. Srivastava A, Malik L, Smith T, Sudbery I, Patro R. ``Alevin efficiently estimates accurate gene abundances from dscRNA-seq data.'' \emph{Genome Biology}. 2019;20:65. DOI: \href{https://doi.org/10.1186/s13059-019-1670-y}{10.1186/s13059-019-1670-y}

\leavevmode\hypertarget{ref-Smith2017-nz}{}%
92. Smith T, Heger A, Sudbery I. ``UMI-tools: modeling sequencing errors in Unique Molecular Identifiers to improve quantification accuracy.'' \emph{Genome Research}. 2017;27:491--9. DOI: \href{https://doi.org/10.1101/gr.209601.116}{10.1101/gr.209601.116}

\leavevmode\hypertarget{ref-Svensson2017-zo}{}%
93. Svensson V, Natarajan KN, Ly L-H, Miragaia RJ, Labalette C, Macaulay IC, Cvejic A, Teichmann SA. ``Power analysis of single-cell RNA-sequencing experiments.'' \emph{Nature Methods}. 2017;14:381--7. DOI: \href{https://doi.org/10.1038/nmeth.4220}{10.1038/nmeth.4220}

\leavevmode\hypertarget{ref-Parekh2018-ug}{}%
94. Parekh S, Ziegenhain C, Vieth B, Enard W, Hellmann I. ``zUMIs - A fast and flexible pipeline to process RNA sequencing data with UMIs.'' \emph{GigaScience}. 2018;7. DOI: \href{https://doi.org/10.1093/gigascience/giy059}{10.1093/gigascience/giy059}

\leavevmode\hypertarget{ref-Tian2018-ub}{}%
95. Tian L, Su S, Dong X, Amann-Zalcenstein D, Biben C, Seidi A, Hilton DJ, Naik SH, Ritchie ME. ``scPipe: A flexible R/Bioconductor preprocessing pipeline for single-cell RNA-sequencing data.'' \emph{PLoS Computational Biology}. 2018;14:e1006361. DOI: \href{https://doi.org/10.1371/journal.pcbi.1006361}{10.1371/journal.pcbi.1006361}

\leavevmode\hypertarget{ref-Yang2017-yz}{}%
96. Yang A, Troup M, Lin P, Ho JWK. ``Falco: a quick and flexible single-cell RNA-seq processing framework on the cloud.'' \emph{Bioinformatics}. 2017;33:767--9. DOI: \href{https://doi.org/10.1093/bioinformatics/btw732}{10.1093/bioinformatics/btw732}

\leavevmode\hypertarget{ref-DePasquale2018-ct}{}%
97. DePasquale EAK, Schnell DJ, Valiente I, Blaxall BC, Grimes HL, Singh H, Salomonis N. ``DoubletDecon: Cell-State Aware Removal of Single-Cell RNA-Seq Doublets.'' \emph{bioRxiv}. 2018. p. 364810. DOI: \href{https://doi.org/10.1101/364810}{10.1101/364810}

\leavevmode\hypertarget{ref-McGinnis2019-at}{}%
98. McGinnis CS, Murrow LM, Gartner ZJ. ``DoubletFinder: Doublet Detection in Single-Cell RNA Sequencing Data Using Artificial Nearest Neighbors.'' \emph{Cell Systems}. 2019;8:329--337.e4. DOI: \href{https://doi.org/10.1016/j.cels.2019.03.003}{10.1016/j.cels.2019.03.003}

\leavevmode\hypertarget{ref-Lun2019-wp}{}%
99. Lun ATL, Riesenfeld S, Andrews T, Dao TP, Gomes T, participants in the 1st Human Cell Atlas Jamboree, Marioni JC. ``EmptyDrops: distinguishing cells from empty droplets in droplet-based single-cell RNA sequencing data.'' \emph{Genome Biology}. 2019;20:63. DOI: \href{https://doi.org/10.1186/s13059-019-1662-y}{10.1186/s13059-019-1662-y}

\leavevmode\hypertarget{ref-Ilicic2016-wy}{}%
100. Ilicic T, Kim JK, Kolodziejczyk AA, Bagger FO, McCarthy DJ, Marioni JC, Teichmann SA. ``Classification of low quality cells from single-cell RNA-seq data.'' \emph{Genome Biology}. 2016;17:29. DOI: \href{https://doi.org/10.1186/s13059-016-0888-1}{10.1186/s13059-016-0888-1}

\leavevmode\hypertarget{ref-McCarthy2017-ql}{}%
101. McCarthy DJ, Campbell KR, Lun ATL, Wills QF. ``Scater: pre-processing, quality control, normalization and visualization of single-cell RNA-seq data in R.'' \emph{Bioinformatics}. 2017;33:1179--86. DOI: \href{https://doi.org/10.1093/bioinformatics/btw777}{10.1093/bioinformatics/btw777}

\leavevmode\hypertarget{ref-Leng2016-it}{}%
102. Leng N, Choi J, Chu L-F, Thomson JA, Kendziorski C, Stewart R. ``OEFinder: a user interface to identify and visualize ordering effects in single-cell RNA-seq data.'' \emph{Bioinformatics}. 2016; DOI: \href{https://doi.org/10.1093/bioinformatics/btw004}{10.1093/bioinformatics/btw004}

\leavevmode\hypertarget{ref-Chen2018-ug}{}%
103. Chen B, Herring CA, Lau KS. ``pyNVR: Investigating factors affecting feature selection from scRNA-seq data for lineage reconstruction.'' \emph{Bioinformatics}. 2018; DOI: \href{https://doi.org/10.1093/bioinformatics/bty950}{10.1093/bioinformatics/bty950}

\leavevmode\hypertarget{ref-Andrews2018-xa}{}%
104. Andrews TS, Hemberg M. ``M3Drop: Dropout-based feature selection for scRNASeq.'' \emph{Bioinformatics}. 2018; DOI: \href{https://doi.org/10.1093/bioinformatics/bty1044}{10.1093/bioinformatics/bty1044}

\leavevmode\hypertarget{ref-Phipson2017-qt}{}%
105. Phipson B, Zappia L, Oshlack A. ``Gene length and detection bias in single cell RNA sequencing protocols.'' \emph{F1000Research}. 2017;6. DOI: \href{https://doi.org/10.12688/f1000research.11290.1}{10.12688/f1000research.11290.1}

\leavevmode\hypertarget{ref-Brennecke2013-pt}{}%
106. Brennecke P, Anders S, Kim JK, Koodziejczyk AA, Zhang X, Proserpio V, Baying B, Benes V, Teichmann SA, Marioni JC, Heisler MG. ``Accounting for technical noise in single-cell RNA-seq experiments.'' \emph{Nature Methods}. 2013;10:1093--5. DOI: \href{https://doi.org/10.1038/nmeth.2645}{10.1038/nmeth.2645}

\leavevmode\hypertarget{ref-Ding2015-ht}{}%
107. Ding B, Zheng L, Zhu Y, Li N, Jia H, Ai R, Wildberg A, Wang W. ``Normalization and noise reduction for single cell RNA-seq experiments.'' \emph{Bioinformatics}. 2015;31:2225--7. DOI: \href{https://doi.org/10.1093/bioinformatics/btv122}{10.1093/bioinformatics/btv122}

\leavevmode\hypertarget{ref-Vallejos2015-ef}{}%
108. Vallejos CA, Marioni JC, Richardson S. ``BASiCS: Bayesian Analysis of Single-Cell Sequencing data.'' \emph{PLoS Computational Biology}. 2015;11:e1004333. DOI: \href{https://doi.org/10.1371/journal.pcbi.1004333}{10.1371/journal.pcbi.1004333}

\leavevmode\hypertarget{ref-Lun2016-mq}{}%
109. Lun ATL, Bach K, Marioni JC. ``Pooling across cells to normalize single-cell RNA sequencing data with many zero counts.'' \emph{Genome Biology}. 2016;17:1--14. DOI: \href{https://doi.org/10.1186/s13059-016-0947-7}{10.1186/s13059-016-0947-7}

\leavevmode\hypertarget{ref-Eling2018-lp}{}%
110. Eling N, Richard AC, Richardson S, Marioni JC, Vallejos CA. ``Correcting the Mean-Variance Dependency for Differential Variability Testing Using Single-Cell RNA Sequencing Data.'' \emph{Cell Systems}. 2018;0. DOI: \href{https://doi.org/10.1016/j.cels.2018.06.011}{10.1016/j.cels.2018.06.011}

\leavevmode\hypertarget{ref-Stuart2019-ua}{}%
111. Stuart T, Satija R. ``Integrative single-cell analysis.'' \emph{Nature Reviews. Genetics}. 2019; DOI: \href{https://doi.org/10.1038/s41576-019-0093-7}{10.1038/s41576-019-0093-7}

\leavevmode\hypertarget{ref-Park2018-ri}{}%
112. Park J-E, Polanski K, Meyer K, Teichmann SA. ``Fast Batch Alignment of Single Cell Transcriptomes Unifies Multiple Mouse Cell Atlases into an Integrated Landscape.'' \emph{bioRxiv}. 2018. p. 397042. DOI: \href{https://doi.org/10.1101/397042}{10.1101/397042}

\leavevmode\hypertarget{ref-Gao2019-ir}{}%
113. Gao X, Hu D, Gogol M, Li H. ``ClusterMap: Compare multiple Single Cell RNA-Seq datasets across different experimental conditions.'' \emph{Bioinformatics}. 2019; DOI: \href{https://doi.org/10.1093/bioinformatics/btz024}{10.1093/bioinformatics/btz024}

\leavevmode\hypertarget{ref-Buttner2019-yl}{}%
114. Bttner M, Miao Z, Wolf FA, Teichmann SA, Theis FJ. ``A test metric for assessing single-cell RNA-seq batch correction.'' \emph{Nature Methods}. 2019;16:43--9. DOI: \href{https://doi.org/10.1038/s41592-018-0254-1}{10.1038/s41592-018-0254-1}

\leavevmode\hypertarget{ref-Welch2018-gb}{}%
115. Welch J, Kozareva V, Ferreira A, Vanderburg C, Martin C, Macosko E. ``Integrative inference of brain cell similarities and differences from single-cell genomics.'' \emph{bioRxiv}. 2018. p. 459891. DOI: \href{https://doi.org/10.1101/459891}{10.1101/459891}

\leavevmode\hypertarget{ref-Mereu2018-bx}{}%
116. Mereu E, Iacono G, Guillaumet-Adkins A, Moutinho C, Lunazzi G, Santos C, Miguel-Escalada I, Ferrer J, Real FX, Gut I, Heyn H. ``matchSCore: Matching Single-Cell Phenotypes Across Tools and Experiments.'' \emph{bioRxiv}. 2018. p. 314831. DOI: \href{https://doi.org/10.1101/314831}{10.1101/314831}

\leavevmode\hypertarget{ref-Hie2019-er}{}%
117. Hie B, Bryson B, Berger B. ``Efficient integration of heterogeneous single-cell transcriptomes using Scanorama.'' \emph{Nature Biotechnology}. 2019; DOI: \href{https://doi.org/10.1038/s41587-019-0113-3}{10.1038/s41587-019-0113-3}

\leavevmode\hypertarget{ref-Lin2019-lk}{}%
118. Lin Y, Ghazanfar S, Wang KYX, Gagnon-Bartsch JA, Lo KK, Su X, Han Z-G, Ormerod JT, Speed TP, Yang P, Yang JYH. ``scMerge leverages factor analysis, stable expression, and pseudoreplication to merge multiple single-cell RNA-seq datasets.'' \emph{Proceedings of the National Academy of Sciences of the United States of America}. 2019; DOI: \href{https://doi.org/10.1073/pnas.1820006116}{10.1073/pnas.1820006116}

\leavevmode\hypertarget{ref-Butler2018-js}{}%
119. Butler A, Hoffman P, Smibert P, Papalexi E, Satija R. ``Integrating single-cell transcriptomic data across different conditions, technologies, and species.'' \emph{Nature Biotechnology}. 2018; DOI: \href{https://doi.org/10.1038/nbt.4096}{10.1038/nbt.4096}

\leavevmode\hypertarget{ref-Hotelling1936-ni}{}%
120. Hotelling H. ``RELATIONS BETWEEN TWO SETS OF VARIATES.'' \emph{Biometrika}. 1936;28:321--77. DOI: \href{https://doi.org/10.1093/biomet/28.3-4.321}{10.1093/biomet/28.3-4.321}

\leavevmode\hypertarget{ref-Hardoon2004-ev}{}%
121. Hardoon DR, Szedmak S, Shawe-Taylor J. ``Canonical correlation analysis: an overview with application to learning methods.'' \emph{Neural Computation}. 2004;16:2639--64. DOI: \href{https://doi.org/10.1162/0899766042321814}{10.1162/0899766042321814}

\leavevmode\hypertarget{ref-Berndt1994-lh}{}%
122. Berndt DJ, Clifford J. ``Using Dynamic Time Warping to Find Patterns in Time Series.'' \emph{Proceedings of the 3rd International Conference on Knowledge Discovery and Data Mining}. Seattle, WA; 1994. pp. 359--70. Available from: \url{http://dl.acm.org/citation.cfm?id=3000850.3000887}

\leavevmode\hypertarget{ref-Haghverdi2018-bd}{}%
123. Haghverdi L, Lun ATL, Morgan MD, Marioni JC. ``Batch effects in single-cell RNA-sequencing data are corrected by matching mutual nearest neighbors.'' \emph{Nature Biotechnology}. 2018; DOI: \href{https://doi.org/10.1038/nbt.4091}{10.1038/nbt.4091}

\leavevmode\hypertarget{ref-Stuart2019-lq}{}%
124. Stuart T, Butler A, Hoffman P, Hafemeister C, Papalexi E, Mauck WM 3rd, Hao Y, Stoeckius M, Smibert P, Satija R. ``Comprehensive Integration of Single-Cell Data.'' \emph{Cell}. 2019;177:1888--1902.e21. DOI: \href{https://doi.org/10.1016/j.cell.2019.05.031}{10.1016/j.cell.2019.05.031}

\leavevmode\hypertarget{ref-Kiselev2019-mg}{}%
125. Kiselev VY, Andrews TS, Hemberg M. ``Challenges in unsupervised clustering of single-cell RNA-seq data.'' \emph{Nature Reviews. Genetics}. 2019; DOI: \href{https://doi.org/10.1038/s41576-018-0088-9}{10.1038/s41576-018-0088-9}

\leavevmode\hypertarget{ref-Guo2015-mf}{}%
126. Guo M, Wang H, Potter SS, Whitsett JA, Xu Y. ``SINCERA: a pipeline for single-cell RNA-seq profiling analysis.'' \emph{PLoS Computational Biology}. 2015;11:e1004575. DOI: \href{https://doi.org/10.1371/journal.pcbi.1004575}{10.1371/journal.pcbi.1004575}

\leavevmode\hypertarget{ref-Kiselev2017-an}{}%
127. Kiselev VY, Kirschner K, Schaub MT, Andrews T, Yiu A, Chandra T, Natarajan KN, Reik W, Barahona M, Green AR, Hemberg M. ``SC3: consensus clustering of single-cell RNA-seq data.'' \emph{Nature Methods}. 2017;14:483--6. DOI: \href{https://doi.org/10.1038/nmeth.4236}{10.1038/nmeth.4236}

\leavevmode\hypertarget{ref-Anchang2016-vo}{}%
128. Anchang B, Hart TDP, Bendall SC, Qiu P, Bjornson Z, Linderman M, Nolan GP, Plevritis SK. ``Visualization and cellular hierarchy inference of single-cell data using SPADE.'' \emph{Nature Protocols}. 2016;11:1264--79. DOI: \href{https://doi.org/10.1038/nprot.2016.066}{10.1038/nprot.2016.066}

\leavevmode\hypertarget{ref-Satija2015-or}{}%
129. Satija R, Farrell JA, Gennert D, Schier AF, Regev A. ``Spatial reconstruction of single-cell gene expression data.'' \emph{Nature Biotechnology}. 2015;33:495--502. DOI: \href{https://doi.org/10.1038/nbt.3192}{10.1038/nbt.3192}

\leavevmode\hypertarget{ref-Duo2018-qb}{}%
130. Du A, Robinson MD, Soneson C. ``A systematic performance evaluation of clustering methods for single-cell RNA-seq data.'' \emph{F1000Research}. 2018;7. DOI: \href{https://doi.org/10.12688/f1000research.15666.1}{10.12688/f1000research.15666.1}

\leavevmode\hypertarget{ref-Freytag2018-ic}{}%
131. Freytag S, Tian L, Lnnstedt I, Ng M, Bahlo M. ``Comparison of clustering tools in R for medium-sized 10x Genomics single-cell RNA-sequencing data.'' \emph{F1000Research}. 2018;7. DOI: \href{https://doi.org/10.12688/f1000research.15809.1}{10.12688/f1000research.15809.1}

\leavevmode\hypertarget{ref-Kim2018-xn}{}%
132. Kim T, Chen IR, Lin Y, Wang AY-Y, Yang JYH, Yang P. ``Impact of similarity metrics on single-cell RNA-seq data clustering.'' \emph{Briefings in Bioinformatics}. 2018; DOI: \href{https://doi.org/10.1093/bib/bby076}{10.1093/bib/bby076}

\leavevmode\hypertarget{ref-Blondel2008-ym}{}%
133. Blondel VD, Guillaume J-L, Lambiotte R, Lefebvre E. ``Fast unfolding of communities in large networks.'' \emph{Journal of Statistical Mechanics}. 2008;2008:P10008. DOI: \href{https://doi.org/10.1088/1742-5468/2008/10/P10008}{10.1088/1742-5468/2008/10/P10008}

\leavevmode\hypertarget{ref-Kiselev2018-nu}{}%
134. Kiselev VY, Yiu A, Hemberg M. ``scmap: projection of single-cell RNA-seq data across data sets.'' \emph{Nature Methods}. 2018; DOI: \href{https://doi.org/10.1038/nmeth.4644}{10.1038/nmeth.4644}

\leavevmode\hypertarget{ref-Alquicira-Hernandez2018-qx}{}%
135. Alquicira-Hernndez J, Sathe A, Ji HP, Nguyen Q, Powell JE. ``scPred: Cell type prediction at single-cell resolution.'' \emph{bioRxiv}. 2018. p. 369538. DOI: \href{https://doi.org/10.1101/369538}{10.1101/369538}

\leavevmode\hypertarget{ref-Lieberman2018-zg}{}%
136. Lieberman Y, Rokach L, Shay T. ``CaSTLe - Classification of single cells by transfer learning: Harnessing the power of publicly available single cell RNA sequencing experiments to annotate new experiments.'' \emph{PloS One}. 2018;13:e0205499. DOI: \href{https://doi.org/10.1371/journal.pone.0205499}{10.1371/journal.pone.0205499}

\leavevmode\hypertarget{ref-Wagner2018-xl}{}%
137. Wagner F, Yanai I. ``Moana: A robust and scalable cell type classification framework for single-cell RNA-Seq data.'' \emph{bioRxiv}. 2018. p. 456129. DOI: \href{https://doi.org/10.1101/456129}{10.1101/456129}

\leavevmode\hypertarget{ref-Qiu2017-mq}{}%
138. Qiu X, Mao Q, Tang Y, Wang L, Chawla R, Pliner HA, Trapnell C. ``Reversed graph embedding resolves complex single-cell trajectories.'' \emph{Nature Methods}. 2017;14:979--82. DOI: \href{https://doi.org/10.1038/nmeth.4402}{10.1038/nmeth.4402}

\leavevmode\hypertarget{ref-Ji2016-ws}{}%
139. Ji Z, Ji H. ``TSCAN: Pseudo-time reconstruction and evaluation in single-cell RNA-seq analysis.'' \emph{Nucleic Acids Research}. 2016; DOI: \href{https://doi.org/10.1093/nar/gkw430}{10.1093/nar/gkw430}

\leavevmode\hypertarget{ref-Welch2016-cw}{}%
140. Welch JD, Hartemink AJ, Prins JF. ``SLICER: inferring branched, nonlinear cellular trajectories from single cell RNA-seq data.'' \emph{Genome Biology}. 2016;17:106. DOI: \href{https://doi.org/10.1186/s13059-016-0975-3}{10.1186/s13059-016-0975-3}

\leavevmode\hypertarget{ref-DuVerle2016-ni}{}%
141. duVerle DA, Yotsukura S, Nomura S, Aburatani H, Tsuda K. ``CellTree: an R/bioconductor package to infer the hierarchical structure of cell populations from single-cell RNA-seq data.'' \emph{BMC Bioinformatics}. 2016;17:363. DOI: \href{https://doi.org/10.1186/s12859-016-1175-6}{10.1186/s12859-016-1175-6}

\leavevmode\hypertarget{ref-Julia2015-fz}{}%
142. Juli M, Telenti A, Rausell A. ``Sincell: an R/Bioconductor package for statistical assessment of cell-state hierarchies from single-cell RNA-seq.'' \emph{Bioinformatics}. 2015;31:3380--2. DOI: \href{https://doi.org/10.1093/bioinformatics/btv368}{10.1093/bioinformatics/btv368}

\leavevmode\hypertarget{ref-Chen2016-kx}{}%
143. Chen J, Schlitzer A, Chakarov S, Ginhoux F, Poidinger M. ``Mpath maps multi-branching single-cell trajectories revealing progenitor cell progression during development.'' \emph{Nature Communications}. 2016;7:11988. DOI: \href{https://doi.org/10.1038/ncomms11988}{10.1038/ncomms11988}

\leavevmode\hypertarget{ref-Street2018-sc}{}%
144. Street K, Risso D, Fletcher RB, Das D, Ngai J, Yosef N, Purdom E, Dudoit S. ``Slingshot: cell lineage and pseudotime inference for single-cell transcriptomics.'' \emph{BMC Genomics}. 2018;19:477. DOI: \href{https://doi.org/10.1186/s12864-018-4772-0}{10.1186/s12864-018-4772-0}

\leavevmode\hypertarget{ref-Cannoodt2016-iv}{}%
145. Cannoodt R, Saelens W, Yvan S. ``Computational methods for trajectory inference from single-cell transcriptomics.'' \emph{European Journal of Immunology}. 2016;46:2496--506. DOI: \href{https://doi.org/10.1002/eji.201646347}{10.1002/eji.201646347}

\leavevmode\hypertarget{ref-Pearson1901-gm}{}%
146. Pearson K. ``On lines and planes of closest fit to systems of points in space.'' \emph{The London, Edinburgh, and Dublin Philosophical Magazine and Journal of Science}. 1901;2:559--72. DOI: \href{https://doi.org/10.1080/14786440109462720}{10.1080/14786440109462720}

\leavevmode\hypertarget{ref-Maaten2008-ne}{}%
147. Maaten L van der, Hinton G. ``Visualizing Data using t-SNE.'' \emph{Journal of Machine Learning Research}. 2008;9:2579--605. Available from: \url{http://www.jmlr.org/papers/v9/vandermaaten08a.html}

\leavevmode\hypertarget{ref-McInnes2018-ul}{}%
148. McInnes L, Healy J. ``UMAP: Uniform Manifold Approximation and Projection for Dimension Reduction.'' 2018; Available from: \url{http://arxiv.org/abs/1802.03426}

\leavevmode\hypertarget{ref-Saelens2019-rs}{}%
149. Saelens W, Cannoodt R, Todorov H, Saeys Y. ``A comparison of single-cell trajectory inference methods.'' \emph{Nature Biotechnology}. 2019; DOI: \href{https://doi.org/10.1038/s41587-019-0071-9}{10.1038/s41587-019-0071-9}

\leavevmode\hypertarget{ref-Svensson2018-lu}{}%
150. Svensson V, Pachter L. ``RNA Velocity: Molecular Kinetics from Single-Cell RNA-Seq.'' \emph{Molecular Cell}. 2018;72:7--9. DOI: \href{https://doi.org/10.1016/j.molcel.2018.09.026}{10.1016/j.molcel.2018.09.026}

\leavevmode\hypertarget{ref-La_Manno2018-tv}{}%
151. La Manno G, Soldatov R, Zeisel A, Braun E, Hochgerner H, Petukhov V, Lidschreiber K, Kastriti ME, Lnnerberg P, Furlan A, Fan J, Borm LE, Liu Z, Bruggen D van, Guo J, He X, Barker R, Sundstrm E, et al. ``RNA velocity of single cells.'' \emph{Nature}. 2018;560:494--8. DOI: \href{https://doi.org/10.1038/s41586-018-0414-6}{10.1038/s41586-018-0414-6}

\leavevmode\hypertarget{ref-Delmans2016-bp}{}%
152. Delmans M, Hemberg M. ``Discrete distributional differential expression (D3E)--a tool for gene expression analysis of single-cell RNA-seq data.'' \emph{BMC Bioinformatics}. 2016;17:110. DOI: \href{https://doi.org/10.1186/s12859-016-0944-6}{10.1186/s12859-016-0944-6}

\leavevmode\hypertarget{ref-Finak2015-ow}{}%
153. Finak G, McDavid A, Yajima M, Deng J, Gersuk V, Shalek AK, Slichter CK, Miller HW, McElrath MJ, Prlic M, Linsley PS, Gottardo R. ``MAST: a flexible statistical framework for assessing transcriptional changes and characterizing heterogeneity in single-cell RNA sequencing data.'' \emph{Genome Biology}. 2015;16:278. DOI: \href{https://doi.org/10.1186/s13059-015-0844-5}{10.1186/s13059-015-0844-5}

\leavevmode\hypertarget{ref-Korthauer2016-yq}{}%
154. Korthauer KD, Chu L-F, Newton MA, Li Y, Thomson J, Stewart R, Kendziorski C. ``A statistical approach for identifying differential distributions in single-cell RNA-seq experiments.'' \emph{Genome Biology}. 2016;17:222. DOI: \href{https://doi.org/10.1186/s13059-016-1077-y}{10.1186/s13059-016-1077-y}

\leavevmode\hypertarget{ref-Kharchenko2014-tb}{}%
155. Kharchenko PV, Silberstein L, Scadden DT. ``Bayesian approach to single-cell differential expression analysis.'' \emph{Nature Methods}. 2014;11:740--2. DOI: \href{https://doi.org/10.1038/nmeth.2967}{10.1038/nmeth.2967}

\leavevmode\hypertarget{ref-Soneson2018-fb}{}%
156. Soneson C, Robinson MD. ``Bias, robustness and scalability in single-cell differential expression analysis.'' \emph{Nature Methods}. 2018;15:255--61. DOI: \href{https://doi.org/10.1038/nmeth.4612}{10.1038/nmeth.4612}

\leavevmode\hypertarget{ref-Qiu2017-bi}{}%
157. Qiu X, Hill A, Packer J, Lin D, Ma Y-A, Trapnell C. ``Single-cell mRNA quantification and differential analysis with Census.'' \emph{Nature Methods}. 2017;14:309--15. DOI: \href{https://doi.org/10.1038/nmeth.4150}{10.1038/nmeth.4150}

\leavevmode\hypertarget{ref-Zhang2019-rv}{}%
158. Zhang JM, Kamath GM, Tse DN. ``Valid post-clustering differential analysis for single-cell RNA-Seq.'' \emph{bioRxiv}. 2019. p. 463265. DOI: \href{https://doi.org/10.1101/463265}{10.1101/463265}

\leavevmode\hypertarget{ref-Edsgard2016-bn}{}%
159. Edsgrd D, Reinius B, Sandberg R. ``scphaser: haplotype inference using single-cell RNA-seq data.'' \emph{Bioinformatics}. 2016; DOI: \href{https://doi.org/10.1093/bioinformatics/btw484}{10.1093/bioinformatics/btw484}

\leavevmode\hypertarget{ref-Reinius2016-hu}{}%
160. Reinius B, Mold JE, Ramskld D, Deng Q, Johnsson P, Michalsson J, Frisn J, Sandberg R. ``Analysis of allelic expression patterns in clonal somatic cells by single-cell RNA-seq.'' \emph{Nature Genetics}. 2016;48:1430--5. DOI: \href{https://doi.org/10.1038/ng.3678}{10.1038/ng.3678}

\leavevmode\hypertarget{ref-Jiang2017-sb}{}%
161. Jiang Y, Zhang NR, Li M. ``SCALE: modeling allele-specific gene expression by single-cell RNA sequencing.'' \emph{Genome Biology}. 2017;18:74. DOI: \href{https://doi.org/10.1186/s13059-017-1200-8}{10.1186/s13059-017-1200-8}

\leavevmode\hypertarget{ref-Choi2019-gt}{}%
162. Choi K, Raghupathy N, Churchill GA. ``scBASE: A Bayesian mixture model for the analysis of allelic expression in single cells.'' \emph{bioRxiv}. 2019. p. 383224. DOI: \href{https://doi.org/10.1101/383224}{10.1101/383224}

\leavevmode\hypertarget{ref-Welch2016-dv}{}%
163. Welch JD, Hu Y, Prins JF. ``Robust detection of alternative splicing in a population of single cells.'' \emph{Nucleic Acids Research}. 2016;44:e73. DOI: \href{https://doi.org/10.1093/nar/gkv1525}{10.1093/nar/gkv1525}

\leavevmode\hypertarget{ref-Huang2017-lp}{}%
164. Huang Y, Sanguinetti G. ``BRIE: transcriptome-wide splicing quantification in single cells.'' \emph{Genome Biology}. 2017;18:123. DOI: \href{https://doi.org/10.1186/s13059-017-1248-5}{10.1186/s13059-017-1248-5}

\leavevmode\hypertarget{ref-Song2017-ym}{}%
165. Song Y, Botvinnik OB, Lovci MT, Kakaradov B, Liu P, Xu JL, Yeo GW. ``Single-cell alternative splicing analysis with Expedition reveals splicing dynamics during neuron differentiation.'' \emph{Molecular Cell}. 2017;67:148--161.e5. DOI: \href{https://doi.org/10.1016/j.molcel.2017.06.003}{10.1016/j.molcel.2017.06.003}

\leavevmode\hypertarget{ref-Poirion2018-mc}{}%
166. Poirion O, Zhu X, Ching T, Garmire LX. ``Using single nucleotide variations in single-cell RNA-seq to identify subpopulations and genotype-phenotype linkage.'' \emph{Nature Communications}. 2018;9:4892. DOI: \href{https://doi.org/10.1038/s41467-018-07170-5}{10.1038/s41467-018-07170-5}

\leavevmode\hypertarget{ref-Ding2019-kx}{}%
167. Ding J, Lin C, Bar-Joseph Z. ``Cell lineage inference from SNP and scRNA-Seq data.'' \emph{Nucleic Acids Research}. 2019; DOI: \href{https://doi.org/10.1093/nar/gkz146}{10.1093/nar/gkz146}

\leavevmode\hypertarget{ref-Petti2018-bj}{}%
168. Petti AA, Williams SR, Miller CA, Fiddes IT, Srivatsan SN, Chen DY, Fronick CC, Fulton RS, Church DM, Ley TJ. ``Mutation detection in thousands of acute myeloid leukemia cells using single cell RNA-sequencing.'' \emph{bioRxiv}. 2018. p. 434746. DOI: \href{https://doi.org/10.1101/434746}{10.1101/434746}

\leavevmode\hypertarget{ref-Lindeman2018-bk}{}%
169. Lindeman I, Emerton G, Mamanova L, Snir O, Polanski K, Qiao S-W, Sollid LM, Teichmann SA, Stubbington MJT. ``BraCeR: B-cell-receptor reconstruction and clonality inference from single-cell RNA-seq.'' \emph{Nature Methods}. 2018;15:563--5. DOI: \href{https://doi.org/10.1038/s41592-018-0082-3}{10.1038/s41592-018-0082-3}

\leavevmode\hypertarget{ref-Afik2018-zl}{}%
170. Afik S, Raulet G, Yosef N. ``Reconstructing B cell receptor sequences from short-read single cell RNA-sequencir with BRAPeS.'' \emph{Bioinformatics}. bioRxiv; 2018. p. e148.

\leavevmode\hypertarget{ref-Rizzetto2018-ev}{}%
171. Rizzetto S, Koppstein DNP, Samir J, Singh M, Reed JH, Cai CH, Lloyd AR, Eltahla AA, Goodnow CC, Luciani F. ``B-cell receptor reconstruction from single-cell RNA-seq with VDJPuzzle.'' \emph{Bioinformatics}. 2018;34:2846--7. DOI: \href{https://doi.org/10.1093/bioinformatics/bty203}{10.1093/bioinformatics/bty203}

\leavevmode\hypertarget{ref-Stubbington2016-xw}{}%
172. Stubbington MJT, Lnnberg T, Proserpio V, Clare S, Speak AO, Dougan G, Teichmann SA. ``T cell fate and clonality inference from single-cell transcriptomes.'' \emph{Nature Methods}. 2016;13:329--32. DOI: \href{https://doi.org/10.1038/nmeth.3800}{10.1038/nmeth.3800}

\leavevmode\hypertarget{ref-Afik2017-qw}{}%
173. Afik S, Yates KB, Bi K, Darko S, Godec J, Gerdemann U, Swadling L, Douek DC, Klenerman P, Barnes EJ, Sharpe AH, Haining WN, Yosef N. ``Targeted reconstruction of T cell receptor sequence from single cell RNA-seq links CDR3 length to T cell differentiation state.'' \emph{Nucleic Acids Research}. 2017;45:e148. DOI: \href{https://doi.org/10.1093/nar/gkx615}{10.1093/nar/gkx615}

\leavevmode\hypertarget{ref-Fan2018-ki}{}%
174. Fan J, Lee H-O, Lee S, Ryu D-E, Lee S, Xue C, Kim SJ, Kim K, Barkas N, Park PJ, Park W-Y, Kharchenko PV. ``Linking transcriptional and genetic tumor heterogeneity through allele analysis of single-cell RNA-seq data.'' \emph{Genome Research}. 2018;28:1217--27. DOI: \href{https://doi.org/10.1101/gr.228080.117}{10.1101/gr.228080.117}

\leavevmode\hypertarget{ref-Tian2019-wl}{}%
175. Tian L, Dong X, Freytag S, L Cao K-A, Su S, JalalAbadi A, Amann-Zalcenstein D, Weber TS, Seidi A, Jabbari JS, Naik SH, Ritchie ME. ``Benchmarking single cell RNA-sequencing analysis pipelines using mixture control experiments.'' \emph{Nature Methods}. 2019; DOI: \href{https://doi.org/10.1038/s41592-019-0425-8}{10.1038/s41592-019-0425-8}

\leavevmode\hypertarget{ref-Bertram2011-kk}{}%
176. Bertram JF, Douglas-Denton RN, Diouf B, Hughson MD, Hoy WE. ``Human nephron number: implications for health and disease.'' \emph{Pediatric Nephrology}. 2011;26:1529--33. DOI: \href{https://doi.org/10.1007/s00467-011-1843-8}{10.1007/s00467-011-1843-8}

\leavevmode\hypertarget{ref-Australian_Bureau_of_Statistics2018-uz}{}%
177. Australian Bureau of Statistics. ``National Health Survey: First Results, 2017-18.'' Commonwealth of Australia; Commonwealth of Australia; 2018 Dec. Available from: \url{http://www.abs.gov.au/ausstats/abs@.nsf/Lookup/by\%20Subject/4364.0.55.001~2017-18~Main\%20Features~Kidney\%20disease~65}

\leavevmode\hypertarget{ref-Australian_Bureau_of_Statistics2018-ms}{}%
178. Australian Bureau of Statistics. ``Causes of Death, Australia, 2017.'' Commonwealth of Australia; Commonwealth of Australia; 2018 Sep. Available from: \url{http://www.abs.gov.au/AUSSTATS/abs@.nsf/Lookup/3303.0Main+Features12017?OpenDocument}

\leavevmode\hypertarget{ref-Patel2013-vh}{}%
179. Patel SR, Dressler GR. ``The genetics and epigenetics of kidney development.'' \emph{Seminars in Nephrology}. 2013;33:314--26. DOI: \href{https://doi.org/10.1016/j.semnephrol.2013.05.004}{10.1016/j.semnephrol.2013.05.004}

\leavevmode\hypertarget{ref-Little2015-wm}{}%
180. Little MH. ``Improving our resolution of kidney morphogenesis across time and space.'' \emph{Current Opinion in Genetics \& Development}. 2015;32:135--43. DOI: \href{https://doi.org/10.1016/j.gde.2015.03.001}{10.1016/j.gde.2015.03.001}

\leavevmode\hypertarget{ref-Lindstrom2018-tj}{}%
181. Lindstrm NO, McMahon JA, Guo J, Tran T, Guo Q, Rutledge E, Parvez RK, Saribekyan G, Schuler RE, Liao C, Kim AD, Abdelhalim A, Ruffins SW, Thornton ME, Baskin L, Grubbs B, Kesselman C, McMahon AP. ``Conserved and Divergent Features of Human and Mouse Kidney Organogenesis.'' \emph{Journal of the American Society of Nephrology: JASN}. 2018;29:785--805. DOI: \href{https://doi.org/10.1681/ASN.2017080887}{10.1681/ASN.2017080887}

\leavevmode\hypertarget{ref-Tian2019-rv}{}%
182. Tian P, Lennon R. ``The myriad possibility of kidney organoids.'' \emph{Current Opinion in Nephrology and Hypertension}. 2019;28:211--8. DOI: \href{https://doi.org/10.1097/MNH.0000000000000498}{10.1097/MNH.0000000000000498}

\leavevmode\hypertarget{ref-Takahashi2006-ho}{}%
183. Takahashi K, Yamanaka S. ``Induction of pluripotent stem cells from mouse embryonic and adult fibroblast cultures by defined factors.'' \emph{Cell}. 2006;126:663--76. DOI: \href{https://doi.org/10.1016/j.cell.2006.07.024}{10.1016/j.cell.2006.07.024}

\leavevmode\hypertarget{ref-Nakagawa2008-bx}{}%
184. Nakagawa M, Koyanagi M, Tanabe K, Takahashi K, Ichisaka T, Aoi T, Okita K, Mochiduki Y, Takizawa N, Yamanaka S. ``Generation of induced pluripotent stem cells without Myc from mouse and human fibroblasts.'' \emph{Nature Biotechnology}. 2008;26:101--6. DOI: \href{https://doi.org/10.1038/nbt1374}{10.1038/nbt1374}

\leavevmode\hypertarget{ref-Fatehullah2016-qw}{}%
185. Fatehullah A, Tan SH, Barker N. ``Organoids as an in vitro model of human development and disease.'' \emph{Nature Cell Biology}. 2016;18:246--54. DOI: \href{https://doi.org/10.1038/ncb3312}{10.1038/ncb3312}

\leavevmode\hypertarget{ref-Takasato2015-zn}{}%
186. Takasato M, Er PX, Chiu HS, Maier B, Baillie GJ, Ferguson C, Parton RG, Wolvetang EJ, Roost MS, Chuva de Sousa Lopes SM, Little MH. ``Kidney organoids from human iPS cells contain multiple lineages and model human nephrogenesis.'' \emph{Nature}. 2015;526:564--8. DOI: \href{https://doi.org/10.1038/nature15695}{10.1038/nature15695}

\leavevmode\hypertarget{ref-Phipson2019-er}{}%
187. Phipson B, Er PX, Combes AN, Forbes TA, Howden SE, Zappia L, Yen H-J, Lawlor KT, Hale LJ, Sun J, Wolvetang E, Takasato M, Oshlack A, Little MH. ``Evaluation of variability in human kidney organoids.'' \emph{Nature Methods}. 2019;16:79--87. DOI: \href{https://doi.org/10.1038/s41592-018-0253-2}{10.1038/s41592-018-0253-2}

\leavevmode\hypertarget{ref-Little2019-hf}{}%
188. Little MH, Hale LJ, Howden SE, Kumar SV. ``Generating Kidney from Stem Cells.'' \emph{Annual Review of Physiology}. 2019;81:335--57. DOI: \href{https://doi.org/10.1146/annurev-physiol-020518-114331}{10.1146/annurev-physiol-020518-114331}

\leavevmode\hypertarget{ref-Bartfeld2017-uo}{}%
189. Bartfeld S, Clevers H. ``Stem cell-derived organoids and their application for medical research and patient treatment.'' \emph{Journal of Molecular Medicine}. 2017;95:729--38. DOI: \href{https://doi.org/10.1007/s00109-017-1531-7}{10.1007/s00109-017-1531-7}

\leavevmode\hypertarget{ref-Kumar2019-tz}{}%
190. Kumar SV, Er PX, Lawlor KT, Motazedian A, Scurr M, Ghobrial I, Combes AN, Zappia L, Oshlack A, Stanley EG, Little MH. ``Kidney micro-organoids in suspension culture as a scalable source of human pluripotent stem cell-derived kidney cells.'' \emph{Development}. 2019;146. DOI: \href{https://doi.org/10.1242/dev.172361}{10.1242/dev.172361}

\leavevmode\hypertarget{ref-Wilson2019-mw}{}%
191. Wilson PC, Humphreys BD. ``Kidney and organoid single-cell transcriptomics: the end of the beginning.'' \emph{Pediatric Nephrology}. 2019;1--7. DOI: \href{https://doi.org/10.1007/s00467-018-4177-y}{10.1007/s00467-018-4177-y}

\leavevmode\hypertarget{ref-Brazovskaja2018-fj}{}%
192. Brazovskaja A, Treutlein B, Camp JG. ``High-throughput single-cell transcriptomics on organoids.'' \emph{Current Opinion in Biotechnology}. 2018;55:167--71. DOI: \href{https://doi.org/10.1016/j.copbio.2018.11.002}{10.1016/j.copbio.2018.11.002}

\leavevmode\hypertarget{ref-Park2018-ub}{}%
193. Park J, Shrestha R, Qiu C, Kondo A, Huang S, Werth M, Li M, Barasch J, Susztk K. ``Single-cell transcriptomics of the mouse kidney reveals potential cellular targets of kidney disease.'' \emph{Science}. 2018;360:758--63. DOI: \href{https://doi.org/10.1126/science.aar2131}{10.1126/science.aar2131}

\leavevmode\hypertarget{ref-Young2018-qd}{}%
194. Young MD, Mitchell TJ, Vieira Braga FA, Tran MGB, Stewart BJ, Ferdinand JR, Collord G, Botting RA, Popescu D-M, Loudon KW, Vento-Tormo R, Stephenson E, Cagan A, Farndon SJ, Del Castillo Velasco-Herrera M, Guzzo C, Richoz N, Mamanova L, et al. ``Single-cell transcriptomes from human kidneys reveal the cellular identity of renal tumors.'' \emph{Science}. 2018;361:594--9. DOI: \href{https://doi.org/10.1126/science.aat1699}{10.1126/science.aat1699}

\leavevmode\hypertarget{ref-Lindstrom2018-mo}{}%
195. Lindstrm NO, Guo J, Kim AD, Tran T, Guo Q, De Sena Brandine G, Ransick A, Parvez RK, Thornton ME, Basking L, Grubbs B, McMahon JA, Smith AD, McMahon AP. ``Conserved and Divergent Features of Mesenchymal Progenitor Cell Types within the Cortical Nephrogenic Niche of the Human and Mouse Kidney.'' \emph{Journal of the American Society of Nephrology: JASN}. 2018;29:806--24. DOI: \href{https://doi.org/10.1681/ASN.2017080890}{10.1681/ASN.2017080890}

\leavevmode\hypertarget{ref-Wu2018-qi}{}%
196. Wu H, Uchimura K, Donnelly EL, Kirita Y, Morris SA, Humphreys BD. ``Comparative Analysis and Refinement of Human PSC-Derived Kidney Organoid Differentiation with Single-Cell Transcriptomics.'' \emph{Cell Stem Cell}. 2018;23:869--881.e8. DOI: \href{https://doi.org/10.1016/j.stem.2018.10.010}{10.1016/j.stem.2018.10.010}

\leavevmode\hypertarget{ref-Davis_undated-gy}{}%
197. Davis S, Kutum R, Zappia L, Sorenson J, Kiselev V, Olivier P, Botvinnik O, Korthauer K, Gitter A, Campbell KR, Hickey P, Tuncel MA, MikeDMorgan, markrobinsonuzh, Vallejos C, Wang Z, Salomonis N, dyl4nm4rsh4ll, et al. ``Awesome Single Cell.'' DOI: \href{https://doi.org/10.5281/zenodo.1294021}{10.5281/zenodo.1294021}

\leavevmode\hypertarget{ref-Zappia2018-ph}{}%
198. Zappia L, Phipson B, Oshlack A. ``Exploring the single-cell RNA-seq analysis landscape with the scRNA-tools database.'' \emph{PLoS Computational Biology}. 2018;14:e1006245. DOI: \href{https://doi.org/10.1371/journal.pcbi.1006245}{10.1371/journal.pcbi.1006245}

\leavevmode\hypertarget{ref-Huber2015-lz}{}%
199. Huber W, Carey VJ, Gentleman R, Anders S, Carlson M, Carvalho BS, Bravo HC, Davis S, Gatto L, Girke T, Gottardo R, Hahne F, Hansen KD, Irizarry RA, Lawrence M, Love MI, MacDonald J, Obenchain V, et al. ``Orchestrating high-throughput genomic analysis with Bioconductor.'' \emph{Nature Methods}. 2015;12:115--21. DOI: \href{https://doi.org/10.1038/nmeth.3252}{10.1038/nmeth.3252}

\leavevmode\hypertarget{ref-Amezquita2019-id}{}%
200. Amezquita RA, Carey VJ, Carpp LN, Geistlinger L, Lun ATL, Marini F, Rue-Albrecht K, Risso D, Soneson C, Waldron L, Pags H, Smith M, Huber W, Morgan M, Gottardo R, Hicks SC. ``Orchestrating Single-Cell Analysis with Bioconductor.'' \emph{bioRxiv}. 2019. p. 590562. DOI: \href{https://doi.org/10.1101/590562}{10.1101/590562}

\leavevmode\hypertarget{ref-Zappia2017-sg}{}%
201. Zappia L, Phipson B, Oshlack A. ``Splatter: simulation of single-cell RNA sequencing data.'' \emph{Genome Biology}. 2017;18:174. DOI: \href{https://doi.org/10.1186/s13059-017-1305-0}{10.1186/s13059-017-1305-0}

\leavevmode\hypertarget{ref-Lun2017-jj}{}%
202. Lun A, Risso D. ``SingleCellExperiment: S4 Classes for Single Cell Data.'' 2017. Available from: \url{http://bioconductor.org/packages/SingleCellExperiment/}

\leavevmode\hypertarget{ref-Campbell2017-yh}{}%
203. Campbell KR, Yau C. ``Probabilistic modeling of bifurcations in single-cell gene expression data using a Bayesian mixture of factor analyzers.'' \emph{Wellcome Open Research}. 2017;2:19. DOI: \href{https://doi.org/10.12688/wellcomeopenres.11087.1}{10.12688/wellcomeopenres.11087.1}

\leavevmode\hypertarget{ref-Campbell2018-dz}{}%
204. Campbell KR, Yau C. ``Uncovering pseudotemporal trajectories with covariates from single cell and bulk expression data.'' \emph{Nature Communications}. 2018;9:2442. DOI: \href{https://doi.org/10.1038/s41467-018-04696-6}{10.1038/s41467-018-04696-6}

\leavevmode\hypertarget{ref-Barron2017-nd}{}%
205. Barron M, Zhang S, Li J. ``A sparse differential clustering algorithm for tracing cell type changes via single-cell RNA-sequencing data.'' \emph{Nucleic Acids Research}. 2017;46:e14. DOI: \href{https://doi.org/10.1093/nar/gkx1113}{10.1093/nar/gkx1113}

\leavevmode\hypertarget{ref-Tung2017-ry}{}%
206. Tung P-Y, Blischak JD, Hsiao CJ, Knowles DA, Burnett JE, Pritchard JK, Gilad Y. ``Batch effects and the effective design of single-cell gene expression studies.'' \emph{Scientific Reports}. 2017;7:39921. DOI: \href{https://doi.org/10.1038/srep39921}{10.1038/srep39921}

\leavevmode\hypertarget{ref-Soneson2017-yt}{}%
207. Soneson C, Robinson MD. ``Towards unified quality verification of synthetic count data with countsimQC.'' \emph{Bioinformatics}. 2017; DOI: \href{https://doi.org/10.1093/bioinformatics/btx631}{10.1093/bioinformatics/btx631}

\leavevmode\hypertarget{ref-Zappia2018-lz}{}%
208. Zappia L, Oshlack A. ``Clustering trees: a visualization for evaluating clusterings at multiple resolutions.'' \emph{GigaScience}. 2018;7. DOI: \href{https://doi.org/10.1093/gigascience/giy083}{10.1093/gigascience/giy083}

\leavevmode\hypertarget{ref-Pedersen2018-ly}{}%
209. Pedersen TL. ``tidygraph: A Tidy API for Graph Manipulation.'' 2018. Available from: \url{https://CRAN.R-project.org/package=tidygraph}

\leavevmode\hypertarget{ref-Csardi2006-ce}{}%
210. Csardi G, Nepusz T. ``The igraph software package for complex network research.'' \emph{InterJournal, Complex Systems}. 2006;1695:1--9.

\leavevmode\hypertarget{ref-Pedersen2018-ie}{}%
211. Pedersen TL. ``ggraph: An Implementation of Grammar of Graphics for Graphs and Networks.'' 2018. Available from: \url{https://CRAN.R-project.org/package=ggraph}

\leavevmode\hypertarget{ref-Anderson1935-mt}{}%
212. Anderson E. ``The Irises of the Gaspe Peninsula.'' \emph{Bulletin of the American Iris Society}. 1935;59:2--5.

\leavevmode\hypertarget{ref-Fisher1936-wt}{}%
213. Fisher RA. ``The use of multiple measurements in taxonomic problems.'' \emph{Annals of Eugenics}. 1936;7:179--88. DOI: \href{https://doi.org/10.1111/j.1469-1809.1936.tb02137.x}{10.1111/j.1469-1809.1936.tb02137.x}

\leavevmode\hypertarget{ref-Combes2019-uv}{}%
214. Combes AN, Zappia L, Er PX, Oshlack A, Little MH. ``Single-cell analysis reveals congruence between kidney organoids and human fetal kidney.'' \emph{Genome Medicine}. 2019;11:3. DOI: \href{https://doi.org/10.1186/s13073-019-0615-0}{10.1186/s13073-019-0615-0}

\leavevmode\hypertarget{ref-Lawlor2019-iw}{}%
215. Lawlor KT, Zappia L, Lefevre J, Park J-S, Hamilton NA, Oshlack A, Little MH, Combes AN. ``Nephron progenitor commitment is a stochastic process influenced by cell migration.'' \emph{eLife}. 2019;8:e41156. DOI: \href{https://doi.org/10.7554/eLife.41156}{10.7554/eLife.41156}

\leavevmode\hypertarget{ref-Zappia2019-dy}{}%
216. Zappia L, Combes AN, Er PX, Oshlack A, Little MH. ``Combes organoid paper analysis code.'' 2019. DOI: \href{https://doi.org/10.5281/zenodo.2548990}{10.5281/zenodo.2548990}

\leavevmode\hypertarget{ref-Zappia2019-kh}{}%
217. Zappia L. ``PhD thesis analysis.'' 2019. DOI: \href{https://doi.org/10.5281/zenodo.2622384}{10.5281/zenodo.2622384}

\leavevmode\hypertarget{ref-Lex2014-hy}{}%
218. Lex A, Gehlenborg N, Strobelt H, Vuillemot R, Pfister H. ``UpSet: Visualization of Intersecting Sets.'' \emph{IEEE Transactions on Visualization and Computer Graphics}. 2014;20:1983--92. DOI: \href{https://doi.org/10.1109/TVCG.2014.2346248}{10.1109/TVCG.2014.2346248}

\leavevmode\hypertarget{ref-Levenshtein1966-fq}{}%
219. Levenshtein VI. ``Binary codes capable of correcting deletions, insertions, and reversals.'' \emph{Soviet physics doklady}. 1966. pp. 707--10.

\leavevmode\hypertarget{ref-Smedley2015-an}{}%
220. Smedley D, Haider S, Durinck S, Pandini L, Provero P, Allen J, Arnaiz O, Awedh MH, Baldock R, Barbiera G, Bardou P, Beck T, Blake A, Bonierbale M, Brookes AJ, Bucci G, Buetti I, Burge S, et al. ``The BioMart community portal: an innovative alternative to large, centralized data repositories.'' \emph{Nucleic Acids Research}. 2015;43:W589--98. DOI: \href{https://doi.org/10.1093/nar/gkv350}{10.1093/nar/gkv350}

\leavevmode\hypertarget{ref-Durinck2005-zb}{}%
221. Durinck S, Moreau Y, Kasprzyk A, Davis S, De Moor B, Brazma A, Huber W. ``BioMart and Bioconductor: a powerful link between biological databases and microarray data analysis.'' \emph{Bioinformatics}. 2005;21:3439--40. DOI: \href{https://doi.org/10.1093/bioinformatics/bti525}{10.1093/bioinformatics/bti525}

\leavevmode\hypertarget{ref-Scialdone2015-rp}{}%
222. Scialdone A, Natarajan KN, Saraiva LR, Proserpio V, Teichmann SA, Stegle O, Marioni JC, Buettner F. ``Computational assignment of cell-cycle stage from single-cell transcriptome data.'' \emph{Methods}. 2015;85:54--61. DOI: \href{https://doi.org/10.1016/j.ymeth.2015.06.021}{10.1016/j.ymeth.2015.06.021}

\leavevmode\hypertarget{ref-Becht2018-zh}{}%
223. Becht E, McInnes L, Healy J, Dutertre C-A, Kwok IWH, Ng LG, Ginhoux F, Newell EW. ``Dimensionality reduction for visualizing single-cell data using UMAP.'' \emph{Nature Biotechnology}. 2018; DOI: \href{https://doi.org/10.1038/nbt.4314}{10.1038/nbt.4314}

\leavevmode\hypertarget{ref-Sidiropoulos2018-ll}{}%
224. Sidiropoulos N, Sohi SH, Pedersen TL, Porse BT, Winther O, Rapin N, Bagger FO. ``SinaPlot: An Enhanced Chart for Simple and Truthful Representation of Single Observations Over Multiple Classes.'' \emph{Journal of Computational and Graphical Statistics: A Joint Publication of American Statistical Association, Institute of Mathematical Statistics, Interface Foundation of North America}. 2018;27:673--6. DOI: \href{https://doi.org/10.1080/10618600.2017.1366914}{10.1080/10618600.2017.1366914}

\leavevmode\hypertarget{ref-Jaakkola2016-ko}{}%
225. Jaakkola MK, Seyednasrollah F, Mehmood A, Elo LL. ``Comparison of methods to detect differentially expressed genes between single-cell populations.'' \emph{Briefings in Bioinformatics}. 2016; DOI: \href{https://doi.org/10.1093/bib/bbw057}{10.1093/bib/bbw057}

\leavevmode\hypertarget{ref-Wolf2019-fo}{}%
226. Wolf FA, Hamey FK, Plass M, Solana J, Dahlin JS, Gttgens B, Rajewsky N, Simon L, Theis FJ. ``PAGA: graph abstraction reconciles clustering with trajectory inference through a topology preserving map of single cells.'' \emph{Genome Biology}. 2019;20:59. DOI: \href{https://doi.org/10.1186/s13059-019-1663-x}{10.1186/s13059-019-1663-x}

\leavevmode\hypertarget{ref-Wolf2018-na}{}%
227. Wolf FA, Angerer P, Theis FJ. ``SCANPY: large-scale single-cell gene expression data analysis.'' \emph{Genome Biology}. 2018;19:15. DOI: \href{https://doi.org/10.1186/s13059-017-1382-0}{10.1186/s13059-017-1382-0}

\leavevmode\hypertarget{ref-R-base}{}%
228. R Core Team. ``R: A language and environment for statistical computing.'' Vienna, Austria: R Foundation for Statistical Computing; 2018.

\leavevmode\hypertarget{ref-R-SingleCellExperiment}{}%
229. Lun A, Risso D. ``SingleCellExperiment: S4 classes for single cell data.'' 2019. R package version 1.4.1

\leavevmode\hypertarget{ref-Jacomy2014-um}{}%
230. Jacomy M, Venturini T, Heymann S, Bastian M. ``ForceAtlas2, a continuous graph layout algorithm for handy network visualization designed for the Gephi software.'' \emph{PloS One}. 2014;9:e98679. DOI: \href{https://doi.org/10.1371/journal.pone.0098679}{10.1371/journal.pone.0098679}

\leavevmode\hypertarget{ref-Wickham2010-zq}{}%
231. Wickham H. ``ggplot2: Elegant Graphics for Data Analysis.'' Springer New York; 2010. ISBN: \href{https://worldcat.org/isbn/9780387981406}{9780387981406}

\leavevmode\hypertarget{ref-R-cowplot}{}%
232. Wilke CO. ``cowplot: Streamlined plot theme and plot annotations for 'ggplot2'.'' 2018. R package version 0.9.3

\leavevmode\hypertarget{ref-R-UpSetR}{}%
233. Gehlenborg N. ``UpSetR: A more scalable alternative to Venn and Euler diagrams for visualizing intersecting sets.'' 2017. R package version 1.3.3

\leavevmode\hypertarget{ref-R-gridExtra}{}%
234. Auguie B. ``GridExtra: Miscellaneous functions for "grid" graphics.'' 2017. R package version 2.3

\leavevmode\hypertarget{ref-R-ggforce}{}%
235. Pedersen TL. ``ggforce: Accelerating 'ggplot2'.'' 2018. R package version 0.1.3

\leavevmode\hypertarget{ref-R-tidyverse}{}%
236. Wickham H. ``tidyverse: Easily install and load the 'tidyverse'.'' 2017. R package version 1.2.1

\leavevmode\hypertarget{ref-R-dplyr}{}%
237. Wickham H, Franois R, Henry L, Mller K. ``dplyr: A grammar of data manipulation.'' 2018. R package version 0.7.8

\leavevmode\hypertarget{ref-R-tidyr}{}%
238. Wickham H, Henry L. ``tidyr: Easily tidy data with 'spread()' and 'gather()' functions.'' 2018. R package version 0.8.2

\leavevmode\hypertarget{ref-R-purrr}{}%
239. Henry L, Wickham H. ``purrr: Functional programming tools.'' 2018. R package version 0.2.5

\leavevmode\hypertarget{ref-R-LoomExperiment}{}%
240. Morgan M, Van Twisk D. ``LoomExperiment: LoomExperiment container.'' 2019. R package version 1.0.2

\leavevmode\hypertarget{ref-R-workflowr}{}%
241. Blischak J, Carbonetto P, Stephens M. ``workflowr: A framework for reproducible and collaborative data science.'' 2018. R package version 1.1.1

\leavevmode\hypertarget{ref-Xie2014-ha}{}%
242. Xie Y. ``knitr: A Comprehensive Tool for Reproducible Research in R.'' In: Stodden V, Leisch F, Peng RD, editors. \emph{Implementing Reproducible Research}. CRC Press; 2014. ISBN: \href{https://worldcat.org/isbn/9781466561595}{9781466561595}

\leavevmode\hypertarget{ref-Xie2016-ct}{}%
243. Xie Y. ``Dynamic Documents with R and knitr.'' CRC Press; 2016. DOI: \href{https://doi.org/10.1201/b15166}{10.1201/b15166} ISBN: \href{https://worldcat.org/isbn/9781482203547}{9781482203547}

\leavevmode\hypertarget{ref-R-knitr}{}%
244. Xie Y. ``knitr: A general-purpose package for dynamic report generation in R.'' 2018. R package version 1.20

\leavevmode\hypertarget{ref-Xie2018-tw}{}%
245. Xie Y, Allaire JJ, Grolemund G. ``R Markdown: The Definitive Guide.'' CRC Press LLC; 2018. ISBN: \href{https://worldcat.org/isbn/9781138359338}{9781138359338}

\leavevmode\hypertarget{ref-R-rmarkdown}{}%
246. Allaire J, Xie Y, McPherson J, Luraschi J, Ushey K, Atkins A, Wickham H, Cheng J, Chang W. ``rmarkdown: Dynamic documents for R.'' 2018. R package version 1.10

\leavevmode\hypertarget{ref-R-bookdown}{}%
247. Xie Y. ``bookdown: Authoring books and technical documents with R Markdown.'' 2018. R package version 0.7

\leavevmode\hypertarget{ref-Xie2016-do}{}%
248. Xie Y. ``Bookdown: Authoring Books and Technical Documents with R Markdown.'' CRC Press; 2016. ISBN: \href{https://worldcat.org/isbn/9781138700109}{9781138700109}

\leavevmode\hypertarget{ref-R-clustree}{}%
249. Zappia L, Oshlack A. ``clustree: Visualise clusterings at different resolutions.'' 2019. R package version 0.3.0

\leavevmode\hypertarget{ref-R-forcats}{}%
250. Wickham H. ``forcats: Tools for working with categorical variables (factors).'' 2018. R package version 0.3.0

\leavevmode\hypertarget{ref-R-ggplot2}{}%
251. Wickham H, Chang W, Henry L, Pedersen TL, Takahashi K, Wilke C, Woo K. ``ggplot2: Create elegant data visualisations using the grammar of graphics.'' 2018. R package version 3.1.0

\leavevmode\hypertarget{ref-R-glue}{}%
252. Hester J. ``glue: Interpreted string literals.'' 2018. R package version 1.3.0

\leavevmode\hypertarget{ref-R-googleAnalyticsR}{}%
253. Edmondson M. ``googleAnalyticsR: Google Analytics API into R.'' 2018. R package version 0.6.0

\leavevmode\hypertarget{ref-R-googleAuthR}{}%
254. Edmondson M. ``googleAuthR: Authenticate and create Google APIs.'' 2018. R package version 0.7.0

\leavevmode\hypertarget{ref-R-here}{}%
255. Mller K. ``here: A simpler way to find your files.'' 2017. R package version 0.1

\leavevmode\hypertarget{ref-R-jsonlite}{}%
256. Ooms J, Temple Lang D, Hilaiel L. ``jsonlite: A robust, high performance JSON parser and generator for R.'' 2018. R package version 1.6

\leavevmode\hypertarget{ref-Ooms2014-ay}{}%
257. Ooms J. ``The jsonlite Package: A Practical and Consistent Mapping Between JSON Data and R Objects.'' 2014; Available from: \url{http://arxiv.org/abs/1403.2805}

\leavevmode\hypertarget{ref-R-lubridate}{}%
258. Spinu V, Grolemund G, Wickham H. ``lubridate: Make dealing with dates a little easier.'' 2018. R package version 1.7.4

\leavevmode\hypertarget{ref-Grolemund2011-qn}{}%
259. Grolemund G, Wickham H. ``Dates and Times Made Easy with lubridate.'' \emph{Journal of Statistical Software, Articles}. 2011;40:1--25. DOI: \href{https://doi.org/10.18637/jss.v040.i03}{10.18637/jss.v040.i03}

\leavevmode\hypertarget{ref-R-magrittr}{}%
260. Bache SM, Wickham H. ``magrittr: A forward-pipe operator for R.'' 2014. R package version 1.5

\leavevmode\hypertarget{ref-R-packrat}{}%
261. Ushey K, McPherson J, Cheng J, Atkins A, Allaire J. ``packrat: A dependency management system for projects and their R package dependencies.'' 2018. R package version 0.4.9-3

\leavevmode\hypertarget{ref-R-RColorBrewer}{}%
262. Neuwirth E. ``RColorBrewer: ColorBrewer palettes.'' 2014. R package version 1.1-2

\leavevmode\hypertarget{ref-R-readr}{}%
263. Wickham H, Hester J, Francois R. ``readr: Read rectangular text data.'' 2018. R package version 1.3.1

\leavevmode\hypertarget{ref-R-remotes}{}%
264. Csrdi G, Wickham H, Chang W, Hester J, Morgan M, Tenenbaum D. ``remotes: R package installation from remote repositories, including 'GitHub'.'' 2018. R package version 2.0.0

\leavevmode\hypertarget{ref-R-stringr}{}%
265. Wickham H. ``stringr: Simple, consistent wrappers for common string operations.'' 2018. R package version 1.3.1

\leavevmode\hypertarget{ref-R-viridis}{}%
266. Garnier S. ``viridis: Default color maps from 'matplotlib'.'' 2018. R package version 0.5.1

\appendix

\hypertarget{splatter-additional}{%
\chapter{Splatter additional files}\label{splatter-additional}}

\hypertarget{additional-figures}{%
\section{Additional figures}\label{additional-figures}}

\frame{\includegraphics[page=1, width=\textwidth]{figures/A1-splatter-figures-cropped}}\clearpage
\frame{\includegraphics[page=2, width=\textwidth]{figures/A1-splatter-figures-cropped}}\clearpage
\frame{\includegraphics[page=3, width=\textwidth]{figures/A1-splatter-figures-cropped}}\clearpage
\frame{\includegraphics[page=4, width=\textwidth]{figures/A1-splatter-figures-cropped}}\clearpage
\frame{\includegraphics[page=5, width=\textwidth]{figures/A1-splatter-figures-cropped}}\clearpage
\frame{\includegraphics[page=6, width=\textwidth]{figures/A1-splatter-figures-cropped}}\clearpage
\frame{\includegraphics[page=7, width=\textwidth]{figures/A1-splatter-figures-cropped}}\clearpage
\frame{\includegraphics[page=8, width=\textwidth]{figures/A1-splatter-figures-cropped}}\clearpage
\frame{\includegraphics[page=9, width=\textwidth]{figures/A1-splatter-figures-cropped}}\clearpage
\frame{\includegraphics[page=10, width=\textwidth]{figures/A1-splatter-figures-cropped}}\clearpage
\frame{\includegraphics[page=11, width=\textwidth]{figures/A1-splatter-figures-cropped}}\clearpage
\frame{\includegraphics[page=12, width=\textwidth]{figures/A1-splatter-figures-cropped}}\clearpage
\frame{\includegraphics[page=13, width=\textwidth]{figures/A1-splatter-figures-cropped}}\clearpage
\frame{\includegraphics[page=14, width=\textwidth]{figures/A1-splatter-figures-cropped}}\clearpage
\frame{\includegraphics[page=15, width=\textwidth]{figures/A1-splatter-figures-cropped}}\clearpage
\frame{\includegraphics[page=16, width=\textwidth]{figures/A1-splatter-figures-cropped}}\clearpage
\frame{\includegraphics[page=17, width=\textwidth]{figures/A1-splatter-figures-cropped}}\clearpage

\hypertarget{session-information}{%
\section{Session information}\label{session-information}}

\frame{\includegraphics[page=1, width=\textwidth]{figures/A1-splatter-session-cropped}}\clearpage
\frame{\includegraphics[page=2, width=\textwidth]{figures/A1-splatter-session-cropped}}\clearpage
\frame{\includegraphics[page=3, width=\textwidth]{figures/A1-splatter-session-cropped}}\clearpage
\frame{\includegraphics[page=4, width=\textwidth]{figures/A1-splatter-session-cropped}}\clearpage
\frame{\includegraphics[page=5, width=\textwidth]{figures/A1-splatter-session-cropped}}\clearpage
\frame{\includegraphics[page=6, width=\textwidth]{figures/A1-splatter-session-cropped}}\clearpage

\hypertarget{splatter-docs}{%
\chapter{Splatter documentation}\label{splatter-docs}}

\hypertarget{splatter-vignette}{%
\section{Splatter vignette}\label{splatter-vignette}}

\frame{\includegraphics[page=1, width=\textwidth]{figures/A2-splatter-vignette-cropped}}\clearpage
\frame{\includegraphics[page=2, width=\textwidth]{figures/A2-splatter-vignette-cropped}}\clearpage
\frame{\includegraphics[page=3, width=\textwidth]{figures/A2-splatter-vignette-cropped}}\clearpage
\frame{\includegraphics[page=4, width=\textwidth]{figures/A2-splatter-vignette-cropped}}\clearpage
\frame{\includegraphics[page=5, width=\textwidth]{figures/A2-splatter-vignette-cropped}}\clearpage
\frame{\includegraphics[page=6, width=\textwidth]{figures/A2-splatter-vignette-cropped}}\clearpage
\frame{\includegraphics[page=7, width=\textwidth]{figures/A2-splatter-vignette-cropped}}\clearpage
\frame{\includegraphics[page=8, width=\textwidth]{figures/A2-splatter-vignette-cropped}}\clearpage
\frame{\includegraphics[page=9, width=\textwidth]{figures/A2-splatter-vignette-cropped}}\clearpage
\frame{\includegraphics[page=10, width=\textwidth]{figures/A2-splatter-vignette-cropped}}\clearpage
\frame{\includegraphics[page=11, width=\textwidth]{figures/A2-splatter-vignette-cropped}}\clearpage
\frame{\includegraphics[page=12, width=\textwidth]{figures/A2-splatter-vignette-cropped}}\clearpage
\frame{\includegraphics[page=13, width=\textwidth]{figures/A2-splatter-vignette-cropped}}\clearpage
\frame{\includegraphics[page=14, width=\textwidth]{figures/A2-splatter-vignette-cropped}}\clearpage
\frame{\includegraphics[page=15, width=\textwidth]{figures/A2-splatter-vignette-cropped}}\clearpage
\frame{\includegraphics[page=16, width=\textwidth]{figures/A2-splatter-vignette-cropped}}\clearpage
\frame{\includegraphics[page=17, width=\textwidth]{figures/A2-splatter-vignette-cropped}}\clearpage
\frame{\includegraphics[page=18, width=\textwidth]{figures/A2-splatter-vignette-cropped}}\clearpage
\frame{\includegraphics[page=19, width=\textwidth]{figures/A2-splatter-vignette-cropped}}\clearpage
\frame{\includegraphics[page=20, width=\textwidth]{figures/A2-splatter-vignette-cropped}}\clearpage
\frame{\includegraphics[page=21, width=\textwidth]{figures/A2-splatter-vignette-cropped}}\clearpage
\frame{\includegraphics[page=22, width=\textwidth]{figures/A2-splatter-vignette-cropped}}\clearpage
\frame{\includegraphics[page=23, width=\textwidth]{figures/A2-splatter-vignette-cropped}}\clearpage

\hypertarget{splat-params-vignette}{%
\section{Splat parameters vignette}\label{splat-params-vignette}}

\frame{\includegraphics[page=1, width=\textwidth]{figures/A2-splatter-parameters-cropped}}\clearpage
\frame{\includegraphics[page=2, width=\textwidth]{figures/A2-splatter-parameters-cropped}}\clearpage
\frame{\includegraphics[page=3, width=\textwidth]{figures/A2-splatter-parameters-cropped}}\clearpage
\frame{\includegraphics[page=4, width=\textwidth]{figures/A2-splatter-parameters-cropped}}\clearpage
\frame{\includegraphics[page=5, width=\textwidth]{figures/A2-splatter-parameters-cropped}}\clearpage
\frame{\includegraphics[page=6, width=\textwidth]{figures/A2-splatter-parameters-cropped}}\clearpage
\frame{\includegraphics[page=7, width=\textwidth]{figures/A2-splatter-parameters-cropped}}\clearpage
\frame{\includegraphics[page=8, width=\textwidth]{figures/A2-splatter-parameters-cropped}}\clearpage
\frame{\includegraphics[page=9, width=\textwidth]{figures/A2-splatter-parameters-cropped}}\clearpage
\frame{\includegraphics[page=10, width=\textwidth]{figures/A2-splatter-parameters-cropped}}\clearpage
\frame{\includegraphics[page=11, width=\textwidth]{figures/A2-splatter-parameters-cropped}}\clearpage
\frame{\includegraphics[page=12, width=\textwidth]{figures/A2-splatter-parameters-cropped}}\clearpage
\frame{\includegraphics[page=13, width=\textwidth]{figures/A2-splatter-parameters-cropped}}\clearpage
\frame{\includegraphics[page=14, width=\textwidth]{figures/A2-splatter-parameters-cropped}}\clearpage
\frame{\includegraphics[page=15, width=\textwidth]{figures/A2-splatter-parameters-cropped}}\clearpage
\frame{\includegraphics[page=16, width=\textwidth]{figures/A2-splatter-parameters-cropped}}\clearpage
\frame{\includegraphics[page=17, width=\textwidth]{figures/A2-splatter-parameters-cropped}}\clearpage
\frame{\includegraphics[page=18, width=\textwidth]{figures/A2-splatter-parameters-cropped}}\clearpage
\frame{\includegraphics[page=19, width=\textwidth]{figures/A2-splatter-parameters-cropped}}\clearpage
\frame{\includegraphics[page=20, width=\textwidth]{figures/A2-splatter-parameters-cropped}}\clearpage

\hypertarget{splatter-manual}{%
\section{Splatter manual}\label{splatter-manual}}

\frame{\includegraphics[page=1, width=\textwidth]{figures/A2-splatter-manual-cropped}}\clearpage
\frame{\includegraphics[page=2, width=\textwidth]{figures/A2-splatter-manual-cropped}}\clearpage
\frame{\includegraphics[page=3, width=\textwidth]{figures/A2-splatter-manual-cropped}}\clearpage
\frame{\includegraphics[page=4, width=\textwidth]{figures/A2-splatter-manual-cropped}}\clearpage
\frame{\includegraphics[page=5, width=\textwidth]{figures/A2-splatter-manual-cropped}}\clearpage
\frame{\includegraphics[page=6, width=\textwidth]{figures/A2-splatter-manual-cropped}}\clearpage
\frame{\includegraphics[page=7, width=\textwidth]{figures/A2-splatter-manual-cropped}}\clearpage
\frame{\includegraphics[page=8, width=\textwidth]{figures/A2-splatter-manual-cropped}}\clearpage
\frame{\includegraphics[page=9, width=\textwidth]{figures/A2-splatter-manual-cropped}}\clearpage
\frame{\includegraphics[page=10, width=\textwidth]{figures/A2-splatter-manual-cropped}}\clearpage
\frame{\includegraphics[page=11, width=\textwidth]{figures/A2-splatter-manual-cropped}}\clearpage
\frame{\includegraphics[page=12, width=\textwidth]{figures/A2-splatter-manual-cropped}}\clearpage
\frame{\includegraphics[page=13, width=\textwidth]{figures/A2-splatter-manual-cropped}}\clearpage
\frame{\includegraphics[page=14, width=\textwidth]{figures/A2-splatter-manual-cropped}}\clearpage
\frame{\includegraphics[page=15, width=\textwidth]{figures/A2-splatter-manual-cropped}}\clearpage
\frame{\includegraphics[page=16, width=\textwidth]{figures/A2-splatter-manual-cropped}}\clearpage
\frame{\includegraphics[page=17, width=\textwidth]{figures/A2-splatter-manual-cropped}}\clearpage
\frame{\includegraphics[page=18, width=\textwidth]{figures/A2-splatter-manual-cropped}}\clearpage
\frame{\includegraphics[page=19, width=\textwidth]{figures/A2-splatter-manual-cropped}}\clearpage
\frame{\includegraphics[page=20, width=\textwidth]{figures/A2-splatter-manual-cropped}}\clearpage
\frame{\includegraphics[page=21, width=\textwidth]{figures/A2-splatter-manual-cropped}}\clearpage
\frame{\includegraphics[page=22, width=\textwidth]{figures/A2-splatter-manual-cropped}}\clearpage
\frame{\includegraphics[page=23, width=\textwidth]{figures/A2-splatter-manual-cropped}}\clearpage
\frame{\includegraphics[page=24, width=\textwidth]{figures/A2-splatter-manual-cropped}}\clearpage
\frame{\includegraphics[page=25, width=\textwidth]{figures/A2-splatter-manual-cropped}}\clearpage
\frame{\includegraphics[page=26, width=\textwidth]{figures/A2-splatter-manual-cropped}}\clearpage
\frame{\includegraphics[page=27, width=\textwidth]{figures/A2-splatter-manual-cropped}}\clearpage
\frame{\includegraphics[page=28, width=\textwidth]{figures/A2-splatter-manual-cropped}}\clearpage
\frame{\includegraphics[page=29, width=\textwidth]{figures/A2-splatter-manual-cropped}}\clearpage
\frame{\includegraphics[page=30, width=\textwidth]{figures/A2-splatter-manual-cropped}}\clearpage
\frame{\includegraphics[page=31, width=\textwidth]{figures/A2-splatter-manual-cropped}}\clearpage
\frame{\includegraphics[page=32, width=\textwidth]{figures/A2-splatter-manual-cropped}}\clearpage
\frame{\includegraphics[page=33, width=\textwidth]{figures/A2-splatter-manual-cropped}}\clearpage
\frame{\includegraphics[page=34, width=\textwidth]{figures/A2-splatter-manual-cropped}}\clearpage
\frame{\includegraphics[page=35, width=\textwidth]{figures/A2-splatter-manual-cropped}}\clearpage
\frame{\includegraphics[page=36, width=\textwidth]{figures/A2-splatter-manual-cropped}}\clearpage
\frame{\includegraphics[page=37, width=\textwidth]{figures/A2-splatter-manual-cropped}}\clearpage
\frame{\includegraphics[page=38, width=\textwidth]{figures/A2-splatter-manual-cropped}}\clearpage
\frame{\includegraphics[page=39, width=\textwidth]{figures/A2-splatter-manual-cropped}}\clearpage
\frame{\includegraphics[page=40, width=\textwidth]{figures/A2-splatter-manual-cropped}}\clearpage
\frame{\includegraphics[page=41, width=\textwidth]{figures/A2-splatter-manual-cropped}}\clearpage
\frame{\includegraphics[page=42, width=\textwidth]{figures/A2-splatter-manual-cropped}}\clearpage
\frame{\includegraphics[page=43, width=\textwidth]{figures/A2-splatter-manual-cropped}}\clearpage
\frame{\includegraphics[page=44, width=\textwidth]{figures/A2-splatter-manual-cropped}}\clearpage
\frame{\includegraphics[page=45, width=\textwidth]{figures/A2-splatter-manual-cropped}}\clearpage
\frame{\includegraphics[page=46, width=\textwidth]{figures/A2-splatter-manual-cropped}}\clearpage
\frame{\includegraphics[page=47, width=\textwidth]{figures/A2-splatter-manual-cropped}}\clearpage
\frame{\includegraphics[page=48, width=\textwidth]{figures/A2-splatter-manual-cropped}}\clearpage
\frame{\includegraphics[page=49, width=\textwidth]{figures/A2-splatter-manual-cropped}}\clearpage
\frame{\includegraphics[page=50, width=\textwidth]{figures/A2-splatter-manual-cropped}}\clearpage
\frame{\includegraphics[page=51, width=\textwidth]{figures/A2-splatter-manual-cropped}}\clearpage
\frame{\includegraphics[page=52, width=\textwidth]{figures/A2-splatter-manual-cropped}}\clearpage
\frame{\includegraphics[page=53, width=\textwidth]{figures/A2-splatter-manual-cropped}}\clearpage
\frame{\includegraphics[page=54, width=\textwidth]{figures/A2-splatter-manual-cropped}}\clearpage
\frame{\includegraphics[page=55, width=\textwidth]{figures/A2-splatter-manual-cropped}}\clearpage
\frame{\includegraphics[page=56, width=\textwidth]{figures/A2-splatter-manual-cropped}}\clearpage
\frame{\includegraphics[page=57, width=\textwidth]{figures/A2-splatter-manual-cropped}}\clearpage
\frame{\includegraphics[page=58, width=\textwidth]{figures/A2-splatter-manual-cropped}}\clearpage

\hypertarget{simulation-comparison}{%
\chapter{Simulation comparison}\label{simulation-comparison}}

\hypertarget{sim-timings}{%
\section{Timings}\label{sim-timings}}

\begin{center}\includegraphics[width=1\linewidth]{/home/travis/build/lazappi/phd-thesis/figures/A3-sim-timings} \end{center}

\hypertarget{sim-packages}{%
\section{Package versions}\label{sim-packages}}

Versions of simulation packages used for the comparison.

\begin{longtable}[]{@{}lr@{}}
\toprule
Package & Version\tabularnewline
\midrule
\endhead
BASiCS & 1.4.1\tabularnewline
DESeq2 & 1.22.2\tabularnewline
edgeR & 3.22.3\tabularnewline
ggplot2 & 3.0.0\tabularnewline
limma & 3.36.2\tabularnewline
mfa & 1.4.1\tabularnewline
phenopath & 1.6.0\tabularnewline
scater & 1.10.1\tabularnewline
scDD & 1.6.1\tabularnewline
scran & 1.10.2\tabularnewline
SingleCellExperiment & 1.4.1\tabularnewline
SparseDC & 0.1.17\tabularnewline
splatter & 1.6.1\tabularnewline
zinbwave & 1.4.1\tabularnewline
\bottomrule
\end{longtable}

\hypertarget{clustree-docs}{%
\chapter{clustree documentation}\label{clustree-docs}}

\hypertarget{clustree-vignette}{%
\section{clustree vignette}\label{clustree-vignette}}

\frame{\includegraphics[page=1, width=\textwidth]{figures/A4-clustree-vignette-cropped}}\clearpage
\frame{\includegraphics[page=2, width=\textwidth]{figures/A4-clustree-vignette-cropped}}\clearpage
\frame{\includegraphics[page=3, width=\textwidth]{figures/A4-clustree-vignette-cropped}}\clearpage
\frame{\includegraphics[page=4, width=\textwidth]{figures/A4-clustree-vignette-cropped}}\clearpage
\frame{\includegraphics[page=5, width=\textwidth]{figures/A4-clustree-vignette-cropped}}\clearpage
\frame{\includegraphics[page=6, width=\textwidth]{figures/A4-clustree-vignette-cropped}}\clearpage
\frame{\includegraphics[page=7, width=\textwidth]{figures/A4-clustree-vignette-cropped}}\clearpage
\frame{\includegraphics[page=8, width=\textwidth]{figures/A4-clustree-vignette-cropped}}\clearpage
\frame{\includegraphics[page=9, width=\textwidth]{figures/A4-clustree-vignette-cropped}}\clearpage
\frame{\includegraphics[page=10, width=\textwidth]{figures/A4-clustree-vignette-cropped}}\clearpage
\frame{\includegraphics[page=11, width=\textwidth]{figures/A4-clustree-vignette-cropped}}\clearpage
\frame{\includegraphics[page=12, width=\textwidth]{figures/A4-clustree-vignette-cropped}}\clearpage
\frame{\includegraphics[page=13, width=\textwidth]{figures/A4-clustree-vignette-cropped}}\clearpage
\frame{\includegraphics[page=14, width=\textwidth]{figures/A4-clustree-vignette-cropped}}\clearpage
\frame{\includegraphics[page=15, width=\textwidth]{figures/A4-clustree-vignette-cropped}}\clearpage
\frame{\includegraphics[page=16, width=\textwidth]{figures/A4-clustree-vignette-cropped}}\clearpage
\frame{\includegraphics[page=17, width=\textwidth]{figures/A4-clustree-vignette-cropped}}\clearpage
\frame{\includegraphics[page=18, width=\textwidth]{figures/A4-clustree-vignette-cropped}}\clearpage
\frame{\includegraphics[page=19, width=\textwidth]{figures/A4-clustree-vignette-cropped}}\clearpage
\frame{\includegraphics[page=20, width=\textwidth]{figures/A4-clustree-vignette-cropped}}\clearpage
\frame{\includegraphics[page=21, width=\textwidth]{figures/A4-clustree-vignette-cropped}}\clearpage
\frame{\includegraphics[page=22, width=\textwidth]{figures/A4-clustree-vignette-cropped}}\clearpage
\frame{\includegraphics[page=23, width=\textwidth]{figures/A4-clustree-vignette-cropped}}\clearpage

\hypertarget{clustree-manual}{%
\section{clustree manual}\label{clustree-manual}}

\frame{\includegraphics[page=1, width=\textwidth]{figures/A4-clustree-manual-cropped}}\clearpage
\frame{\includegraphics[page=2, width=\textwidth]{figures/A4-clustree-manual-cropped}}\clearpage
\frame{\includegraphics[page=3, width=\textwidth]{figures/A4-clustree-manual-cropped}}\clearpage
\frame{\includegraphics[page=4, width=\textwidth]{figures/A4-clustree-manual-cropped}}\clearpage
\frame{\includegraphics[page=5, width=\textwidth]{figures/A4-clustree-manual-cropped}}\clearpage
\frame{\includegraphics[page=6, width=\textwidth]{figures/A4-clustree-manual-cropped}}\clearpage
\frame{\includegraphics[page=7, width=\textwidth]{figures/A4-clustree-manual-cropped}}\clearpage
\frame{\includegraphics[page=8, width=\textwidth]{figures/A4-clustree-manual-cropped}}\clearpage
\frame{\includegraphics[page=9, width=\textwidth]{figures/A4-clustree-manual-cropped}}\clearpage
\frame{\includegraphics[page=10, width=\textwidth]{figures/A4-clustree-manual-cropped}}\clearpage
\frame{\includegraphics[page=11, width=\textwidth]{figures/A4-clustree-manual-cropped}}\clearpage
\frame{\includegraphics[page=12, width=\textwidth]{figures/A4-clustree-manual-cropped}}\clearpage
\frame{\includegraphics[page=13, width=\textwidth]{figures/A4-clustree-manual-cropped}}\clearpage
\frame{\includegraphics[page=14, width=\textwidth]{figures/A4-clustree-manual-cropped}}\clearpage
\frame{\includegraphics[page=15, width=\textwidth]{figures/A4-clustree-manual-cropped}}\clearpage
\frame{\includegraphics[page=16, width=\textwidth]{figures/A4-clustree-manual-cropped}}\clearpage
\frame{\includegraphics[page=17, width=\textwidth]{figures/A4-clustree-manual-cropped}}\clearpage
\frame{\includegraphics[page=18, width=\textwidth]{figures/A4-clustree-manual-cropped}}\clearpage
\frame{\includegraphics[page=19, width=\textwidth]{figures/A4-clustree-manual-cropped}}\clearpage
\frame{\includegraphics[page=20, width=\textwidth]{figures/A4-clustree-manual-cropped}}\clearpage
\frame{\includegraphics[page=21, width=\textwidth]{figures/A4-clustree-manual-cropped}}\clearpage

\hypertarget{organoid-paper}{%
\chapter{Kidney organoid publication}\label{organoid-paper}}

\frame{\includegraphics[page=1, width=\textwidth]{figures/A5-organoid-paper-cropped}}\clearpage
\frame{\includegraphics[page=2, width=\textwidth]{figures/A5-organoid-paper-cropped}}\clearpage
\frame{\includegraphics[page=3, width=\textwidth]{figures/A5-organoid-paper-cropped}}\clearpage
\frame{\includegraphics[page=4, width=\textwidth]{figures/A5-organoid-paper-cropped}}\clearpage
\frame{\includegraphics[page=5, width=\textwidth]{figures/A5-organoid-paper-cropped}}\clearpage
\frame{\includegraphics[page=6, width=\textwidth]{figures/A5-organoid-paper-cropped}}\clearpage
\frame{\includegraphics[page=7, width=\textwidth]{figures/A5-organoid-paper-cropped}}\clearpage
\frame{\includegraphics[page=8, width=\textwidth]{figures/A5-organoid-paper-cropped}}\clearpage
\frame{\includegraphics[page=9, width=\textwidth]{figures/A5-organoid-paper-cropped}}\clearpage
\frame{\includegraphics[page=10, width=\textwidth]{figures/A5-organoid-paper-cropped}}\clearpage
\frame{\includegraphics[page=11, width=\textwidth]{figures/A5-organoid-paper-cropped}}\clearpage
\frame{\includegraphics[page=12, width=\textwidth]{figures/A5-organoid-paper-cropped}}\clearpage
\frame{\includegraphics[page=13, width=\textwidth]{figures/A5-organoid-paper-cropped}}\clearpage
\frame{\includegraphics[page=14, width=\textwidth]{figures/A5-organoid-paper-cropped}}\clearpage
\frame{\includegraphics[page=15, width=\textwidth]{figures/A5-organoid-paper-cropped}}\clearpage

\hypertarget{analysis-methods}{%
\chapter{Analysis methods}\label{analysis-methods}}

\clearpage

\hypertarget{pre-processing-1}{%
\section{Pre-processing}\label{pre-processing-1}}

The Cell Ranger pipeline (v3.0.1) {[}34{]} was used to perform sample demultiplexing, barcode processing and single-cell gene counting. Briefly,samples were demultiplexed to produce a pair of FASTQ files for each sample. Reads containing sequence information were aligned using the reference provided with Cell Ranger (v3.0.0) based on the GRCh38 reference genome and ENSEMBL gene annotation (v93). PCR duplicates were removed by selecting unique combinations of cell barcodes, unique molecular identifiers and gene ids with the final results being a gene expression matrix that was used for further analysis. The three samples were aggregated using Cell Ranger with no normalisation and treated as a single dataset containing 945038 droplets. The R statistical programming language (v3.5.0) {[}228{]} was used for further analysis. Count data was read into R and used to construct a SingleCellExperiment object (v1.4.1) {[}229{]}.

\hypertarget{droplet-selection-1}{%
\subsection{Droplet selection}\label{droplet-selection-1}}

The droplet selection method in this version of Cell Ranger identified 8291 cell containing droplets. This is more than the traditional Cell Ranger approach based on a simple threshold on total counts which identified 7006 cells. The EmptyDrops method {[}99{]} available in the DropletUtils package (v1.2.2) was also applied. Droplets with less than 100 total counts were used to construct the ambient RNA profile and an FDR threshold of less than 0.01 was used to select 9858 droplets. The droplets selected by either the Cell Ranger v3 method or EmptyDrops were kept giving a dataset with 10002 cells. Gene annotation information was added from BioMart {[}220{]} using the biomaRt package (v2.38.0) {[}221{]} and cells were assigned cell cycle scores using the cyclone {[}222{]} function in the scran package (v1.10.2) {[}109{]}.

\hypertarget{alevin}{%
\subsection{Alevin}\label{alevin}}

An alternative quantification was performed using the alevin method {[}91{]} available as part of the salmon software package (v0.12.0) {[}12{]}. This produced a dataset with 10423 cells and 37247 genes.

\hypertarget{quality-control-1}{%
\section{Quality control}\label{quality-control-1}}

The scater package (v1.10.1) {[}101{]} was used to produce a series of diagnostic quality control plots and select thresholds for filtering. Cells were removed from the dataset if they had \(\log_{10}\) total counts less than 2.309 (4 MADs from the median), \(\log_{10}\) total features expressed less than 2.3659 (4 MADs) or more than 8.2411 percent of counts (4 MADs) assigned to mitochondrial genes. An automated filtering method method available in scater that detects outliers in a PCA of cells based on technical factors was also performed for comparison but not used for filtering. After filtering doublet scores for each cell were calculated using the simulation-based method available in scran and cells with a score greater than 1.5572 were removed. A minimal filter was used to remove genes that did not have at least 1 count in at least 2 cells. After quality control the dataset had 8059 cells and 22739 genes with a median of 7838 counts per cell and a median of 2506 genes expressed.

\hypertarget{clustering-1}{%
\section{Clustering}\label{clustering-1}}

\hypertarget{gene-selection-1}{%
\subsection{Gene selection}\label{gene-selection-1}}

Two methods were used to select genes to use for clustering. The default selection method in the Seurat package (v2.3.4) {[}129{]} uses thresholds on mean expression (between 0.0125 and 3.5) and dispersion (greater than 1), which selected 2457 genes. The M3Drop (v3.10.3) {[}104{]} method was also used to test for genes with significantly more zero counts than expected and 1952 genes with an adjusted p-value less than 0.01 were selected. Following comparisons between these two methods the genes identified by M3Drop were used for clustering.

\hypertarget{graph-based-clustering}{%
\subsection{Graph-based clustering}\label{graph-based-clustering}}

Seurat was used to perform graph-based clustering. PCA was performed using the selected genes and the first 15 principal components were used to construct a shared nearest neighbour graph using the overlap between the 30 nearest neighbours of each cell. Louvain modularity optimisation {[}133{]} was used to partition this graph with a resolution parameter between 0 and 1. Clustering tree visualisations {[}208{]} were produced using the clustree package (v0.3.0) showing the expression of previously identified marker genes. By inspecting these trees a resolution of 0.4 which chosen which produced 13 clusters. To compare these cluster to those that had been previously published the Jaccard index was calculated between pairs of clusters from each analysis and visualised as a heatmap.

\hypertarget{marker-genes-1}{%
\section{Marker genes}\label{marker-genes-1}}

Marker genes for each cluster were identified using edgeR (v3.24.3) {[}15,16{]}. Additional filtering was performed to remove genes with a \(\log_{10}\) maximum average counts in any cluster less than -1.3505. The edgeR negative binomial model was then fitted to the dataset using a design matrix that included the detection rate (scaled proportion of genes expressed in each cell). The quasi-likelihood F-test was used to test cells in one cluster against all other cells. Genes were considered significant markers for a cluster if they had an FDR less than 0.05 and a log fold change greater than 1. Identities were assigned to each cluster by comparing the detected genes to previously published lists of cell type markers.

\hypertarget{partition-based-graph-abstraction-paga}{%
\section{Partition-based graph abstraction (PAGA)}\label{partition-based-graph-abstraction-paga}}

Partition-based graph abstraction (PAGA) {[}226{]} was performed using the scanpy library (v1.4) {[}227{]} for the Python programming language (v3.6.8). A shared nearest neighbour graph was built using the same parameters as were used by Seurat and connectivity between the Seurat clusters was calculated using the PAGA algorithm. A ForceAtlas2 {[}230{]} layout of the cell graph was calculated based on the PAGA cluster graph. The results of the PAGA analysis were read into R for visualisation.

\hypertarget{cell-velocity-1}{%
\section{Cell velocity}\label{cell-velocity-1}}

The velocyto (v0.17.16) {[}151{]} Python program was used to calculate spliced and unspliced count matrices from the aligned reads. The resulting Loom files for each sample were then read into R and merged. The remaining analysis was performed using the velocyto R package (v0.6). Genes were removed from the spliced matrix if they had did not have a mean expression greater than 0.2 in at least one cluster or from the unspliced matrix if they did not have a mean expression greater than 0.05 in at least one cluster. Relative velocity estimates were then calculated for each cell and projected onto reduced dimensional spaces for visualisation.

\hypertarget{other-packages}{%
\section{Other packages}\label{other-packages}}

Visualisations and figures were primarily created using the ggplot2 (v3.1.0) {[}231{]} and cowplot (v0.9.4) {[}232{]} packages using the viridis colour palette (v0.5.1) for continuous data. UpSet plots {[}218{]} were produced using the UpSetR package (v1.3.3) {[}233{]} with help from the gridExtra package (v2.3) {[}234{]} and SinaPlots {[}224{]} using the ggforce package (v0.1.3) {[}235{]}. Data manipulation was performed using other packages in the tidyverse (v1.2.1) {[}236{]} particularly dplyr (v0.7.8) {[}237{]}, tidyr (v0.7.8) {[}238{]} and purrr (v0.3.0) {[}239{]}. Loom files were exported from R using the LoomExperiment package (v1.0.2) {[}240{]}. The analysis project was managed using the workflowr (v1.1.1) {[}241{]} package which was also used to produce the publicly available website displaying the analysis code, results and output. Reproducible reports were produced using knitr (v1.21) {[}242--244{]} and R Markdown (v1.11) {[}245,246{]} and converted to HTML using Pandoc (v1.17.2).

\hypertarget{session-information-1}{%
\chapter{Session information}\label{session-information-1}}

\clearpage

\hypertarget{important-packages}{%
\section{Important packages}\label{important-packages}}

Details of the main packages used to build this thesis.

\begin{itemize}
\tightlist
\item
  bookdown (v0.7) -- writing complex documents using R Markdown {[}247,248{]}
\item
  clustree (v0.3.0) -- producing clustering tree visualisations {[}208,249{]}
\item
  cowplot (v0.9.3) -- constructing complex panel figures {[}232{]}
\item
  dplyr (v0.7.8) -- manipulation of data frames {[}237{]}
\item
  forcats (v0.3.0) -- manipulation of factors {[}250{]}
\item
  ggplot2 (v3.1.0) -- data visualisation {[}231,251{]}
\item
  glue (v1.3.0) -- complex joining of strings {[}252{]}
\item
  googleAnalyticsR (v0.6.0) -- downloading Google Analytics data {[}253{]}
\item
  googleAuthR (v0.7.0) -- Google API authentication {[}254{]}
\item
  here (v0.1) -- locating files {[}255{]}
\item
  jsonlite (v1.6) -- reading JSON files {[}256,257{]}
\item
  knitr (v1.20) -- generating dynamic reports using R {[}242--244{]}
\item
  lubridate (v1.7.4) -- manipulation of dates {[}258,259{]}
\item
  magrittr (v1.5) -- piping between functions {[}260{]}
\item
  packrat (v0.4.9.3) -- management of package dependencies {[}261{]}
\item
  purrr (v0.2.5) -- functional programming for lists {[}239{]}
\item
  RColorBrewer (v1.1.2) -- ColorBrewer colour palettes {[}262{]}
\item
  readr (v1.3.1) -- input of data files {[}263{]}
\item
  remotes (v2.0.0) -- installation of packages from GitHub {[}264{]}
\item
  rmarkdown (v1.10) -- dynamic documents for R {[}246{]}
\item
  stringr (v1.3.1) -- manipulation of strings {[}265{]}
\item
  tidyr (v0.8.2) -- tidying of data {[}238{]}
\item
  viridis (v0.5.1) -- viridis colour palettes {[}266{]}
\end{itemize}

\hypertarget{full-session-information}{%
\section{Full session information}\label{full-session-information}}

\begin{verbatim}
## Session info -------------------------------------------------------------
\end{verbatim}

\begin{verbatim}
##  setting  value                       
##  version  R version 3.6.1 (2017-01-27)
##  system   x86_64, linux-gnu           
##  ui       X11                         
##  language en_US.UTF-8                 
##  collate  en_US.UTF-8                 
##  tz       UTC                         
##  date     2019-08-12
\end{verbatim}

\begin{verbatim}
## Packages -----------------------------------------------------------------
\end{verbatim}

\begin{verbatim}
##  package      * version  date       source        
##  assertthat     0.2.0    2017-04-11 CRAN (R 3.6.1)
##  backports      1.1.2    2017-12-13 CRAN (R 3.6.1)
##  base         * 3.6.1    2019-07-22 local         
##  bindr          0.1.1    2018-03-13 CRAN (R 3.6.1)
##  bindrcpp     * 0.2.2    2018-03-29 CRAN (R 3.6.1)
##  bookdown       0.7      2018-02-18 CRAN (R 3.6.1)
##  checkmate      1.9.1    2019-01-15 CRAN (R 3.6.1)
##  clustree     * 0.3.0    2019-02-24 CRAN (R 3.6.1)
##  colorspace     1.3-2    2016-12-14 CRAN (R 3.6.1)
##  compiler       3.6.1    2019-07-22 local         
##  cowplot      * 0.9.3    2018-07-15 CRAN (R 3.6.1)
##  crayon         1.3.4    2017-09-16 CRAN (R 3.6.1)
##  datasets     * 3.6.1    2019-07-22 local         
##  devtools       1.13.6   2018-06-27 CRAN (R 3.6.1)
##  digest         0.6.18   2018-10-10 CRAN (R 3.6.1)
##  dplyr        * 0.7.8    2018-11-10 CRAN (R 3.6.1)
##  evaluate       0.12     2018-10-09 CRAN (R 3.6.1)
##  farver         1.1.0    2018-11-20 CRAN (R 3.6.1)
##  forcats        0.3.0    2018-02-19 CRAN (R 3.6.1)
##  ggforce        0.2.2    2019-04-23 CRAN (R 3.6.1)
##  ggplot2      * 3.1.0    2018-10-25 CRAN (R 3.6.1)
##  ggraph       * 1.0.2    2018-07-07 CRAN (R 3.6.1)
##  ggrepel        0.8.0    2018-05-09 CRAN (R 3.6.1)
##  glue           1.3.0    2018-07-17 CRAN (R 3.6.1)
##  graphics     * 3.6.1    2019-07-22 local         
##  grDevices    * 3.6.1    2019-07-22 local         
##  grid           3.6.1    2019-07-22 local         
##  gridExtra      2.3      2017-09-09 CRAN (R 3.6.1)
##  gtable         0.2.0    2016-02-26 CRAN (R 3.6.1)
##  here         * 0.1      2017-05-28 CRAN (R 3.6.1)
##  hms            0.4.2    2018-03-10 CRAN (R 3.6.1)
##  htmltools      0.3.6    2017-04-28 CRAN (R 3.6.1)
##  igraph         1.2.2    2018-07-27 CRAN (R 3.6.1)
##  jsonlite     * 1.6      2018-12-07 CRAN (R 3.6.1)
##  knitr        * 1.20     2018-02-20 CRAN (R 3.6.1)
##  labeling       0.3      2014-08-23 CRAN (R 3.6.1)
##  lazyeval       0.2.1    2017-10-29 CRAN (R 3.6.1)
##  magrittr       1.5      2014-11-22 CRAN (R 3.6.1)
##  maps         * 3.3.0    2018-04-03 CRAN (R 3.6.1)
##  MASS           7.3-51.4 2019-03-31 CRAN (R 3.6.1)
##  memoise        1.1.0    2017-04-21 CRAN (R 3.6.1)
##  methods      * 3.6.1    2019-07-22 local         
##  munsell        0.5.0    2018-06-12 CRAN (R 3.6.1)
##  packrat        0.4.9-3  2018-06-01 CRAN (R 3.6.1)
##  pillar         1.3.0    2018-07-14 CRAN (R 3.6.1)
##  pkgconfig      2.0.2    2018-08-16 CRAN (R 3.6.1)
##  plyr           1.8.4    2016-06-08 CRAN (R 3.6.1)
##  polyclip       1.9-1    2018-07-27 CRAN (R 3.6.1)
##  purrr          0.2.5    2018-05-29 CRAN (R 3.6.1)
##  R6             2.3.0    2018-10-04 CRAN (R 3.6.1)
##  RColorBrewer * 1.1-2    2014-12-07 CRAN (R 3.6.1)
##  Rcpp           1.0.0    2018-11-07 CRAN (R 3.6.1)
##  readr        * 1.3.1    2018-12-21 CRAN (R 3.6.1)
##  rlang          0.3.0.1  2018-10-25 CRAN (R 3.6.1)
##  rmarkdown      1.10     2018-06-11 CRAN (R 3.6.1)
##  rprojroot      1.3-2    2018-01-03 CRAN (R 3.6.1)
##  rstudioapi     0.8      2018-10-02 CRAN (R 3.6.1)
##  scales         1.0.0    2018-08-09 CRAN (R 3.6.1)
##  stats        * 3.6.1    2019-07-22 local         
##  stringi        1.2.4    2018-07-20 CRAN (R 3.6.1)
##  stringr      * 1.3.1    2018-05-10 CRAN (R 3.6.1)
##  tibble         1.4.2    2018-01-22 CRAN (R 3.6.1)
##  tidygraph      1.1.1    2018-11-20 CRAN (R 3.6.1)
##  tidyr        * 0.8.2    2018-10-28 CRAN (R 3.6.1)
##  tidyselect     0.2.5    2018-10-11 CRAN (R 3.6.1)
##  tools          3.6.1    2019-07-22 local         
##  tweenr         1.0.1    2018-12-14 CRAN (R 3.6.1)
##  utils        * 3.6.1    2019-07-22 local         
##  viridis        0.5.1    2018-03-29 CRAN (R 3.6.1)
##  viridisLite    0.3.0    2018-02-01 CRAN (R 3.6.1)
##  withr          2.1.2    2018-03-15 CRAN (R 3.6.1)
##  xfun           0.3      2018-07-06 CRAN (R 3.6.1)
##  yaml           2.2.0    2018-07-25 CRAN (R 3.6.1)
\end{verbatim}

\End{tcother}
\end{mainmatter}

\end{document}
