## Introduction

For a computational analysis method to be accepted and used by the community it
needs to demonstrate that is is effective at the task it aims to complete.
Ideally this can be achieved by evaluating the performance of the method on a
real dataset where the results are already known. Unfortunately, in many cases
such gold standard datasets are not available. This is particularly true for
genomic data where it is difficult to know what the truth is or the known truth
is limited to only small sections of the genome. It is possible to create some
genomic datasets where the truth is known, for example through carefully
performed mixing experiments [@Tian2018-ok], but these often do not capture the
true biological complexity. In many cases the most effective way to evaluate an
analysis method during development is by testing it on simulated datasets.
Simulations have the additional advantage of being relatively cheap and easy to
produce, allowing exploration of a wide range of possible parameters. This is
the approach taken by many early methods for scRNA-seq analysis. Unfortunately,
these simulations were often not well explained, code for reproducing them was
not available and perhaps most importantly they didn't show that the synthetic
datasets were similar to real scRNA-seq data.

This chapter presents Splatter, a software package for simulating scRNA-seq
datasets presented in a publication in _Genome Biology_ [@Zappia2017-sg].
Splatter is designed to provide a consistent, easy-to-use interface for
multiple previously developed scRNA-seq simulation models. We do this by
providing a consistent framework for each model defined by two functions: one
which estimates parameters from a real dataset and a second that generates a
synthetic dataset using those parameters. Each simulation model has different
assumptions and reproduces different aspects of scRNA-seq data, and we explain
these differences in our publication. We also present Splat, our own simulation
model based on the Gamma-Poisson distribution. This model includes several
important aspects of scRNA-seq data including highly expressed outliers genes,
differences in library sizes between cells, a relationship between the mean and
the variance of each gene and the ability to add a dropout effect linked to
gene expression. When designing the Splat simulation our goal was to accurately
reproduce scRNA-seq count data rather than test a specific method, with the
result being that the model is highly flexible and able to generate a range of
scenarios including datasets with multiple groups of cells, batch effects and
continuous trajectories.

In the publication we compare how well each simulation reproduces a range of
scRNA-seq datasets, including UMI and full-length protocols, different capture
methods, and homogenous and complex tissues. We found that the Splat simulation
was a good match for some of these datasets across a range of methods, however
it was also clear that some models more faithfully reproduced different aspects
of the data, particularly for datasets from different sources. The Splatter R
package is available for download from Bioconductor
(https://bioconductor.org/packages/splatter). Following the publication
(Section \@ref(splatter-publication)) I discuss updates to the software package
and provide an updated comparison of the current simulation models (Section
\@ref(splatter-update)). A vignette describing the use of Splatter is included
as Appendix \@ref(splatter-vignette) and the full manual in Appendix
\@ref(splatter-manual).
