## Introduction

When I began my PhD in early 2016, single-cell RNA sequencing technologies were
just beginning to become widely available. Since then there has been a rapid
uptake of the technologies and there are now many studies using this approach.
Along with the growth in the adoption of scRNA-seq technologies, there has been
an explosion in the number of software tools for analysing these datasets. This
chapter charts the growth in the scRNA-seq analysis landscape over time.

In 2016 there were relatively few analysis methods available and to answer
questions like how many tools perform a particular task, which areas are
developers focusing on or is there a tool for doing a particular task, I began
to record details about them. Inspired by similar projects such as Sean Davis's
Awesome Single Cell page [@Davis_undated-gy] I decided to make this collection
public. This turned out to be useful to other researchers and over time a
simple spreadsheet became the scRNA-tools database and website
(https://scRNA-tools.org). Our paper published in _PLoS Computational Biology_
describing this resource forms the main part of this chapter [@Zappia2018-ph].

By having access to details about existing analysis tools we were able to
explore how the field has developed. We found that computational researchers
had focused their efforts on analysis tasks specific to scRNA-seq data, such as
clustering and ordering of cells or handling the larger numbers of zero counts.
We also saw that many of the tools performed tasks common to several stages of
analysis, including various approaches to dimensionality reduction and
visualisation of data and results. Details recorded in the database show that
developers of scRNA-seq analysis tools tend to embrace an open-source and
open-science approach, for example most tools are developed on GitHub where
others can ask questions and submit improvements. The majority of tools are
also available under open-source licenses allowing their code to be reused for
other purposes, although there is also a significant proportion that do not
have any associated license. Tools are commonly made public by releasing a
preprint publication, making them quickly available to the community and giving
early adopters a chance to contribute to their development.

Sections at the end of this chapter present a version of some the analysis
presented in the paper based on the most recent version of the database
(Section \@ref(tools-current)), as well as details about the usage of the
scRNA-tools website over time (Section \@ref(tools-usage)).
