## Single-cell RNA sequencing {#intro-scRNAseq}

Traditional bulk RNA-seq experiments average the transcriptome across the
millions of cells in a sample. Recently it has become possible to perform
single-cell RNA sequencing (scRNA-seq) and investigate the transcriptome at the
resolution of individual cells. There are many situations where it is important
to understand how specific cell types react to development or perturbations.
This is often hindered in bulk analyses which may be affected by the unknown
proportions of cell types in a sample. Studies into gene expression in specific
cell types previously required selecting and isolating the cells of interest,
which separates them from the other cell types they are usually associated with
and makes it impossible to investigate interactions between them. With
scRNA-seq technologies it is now possible to look at the transcriptome of all
the cell types in a tissue simultaneously, which has led to a better
understanding of what makes cell types distinct and the discovery of previously
unknown cell types.

### Early single-cell capture technologies

The first scRNA-seq protocol was published in 2009 [@Tang2009-jd], just a year
after the first bulk RNA-seq publication [@Mortazavi2008-vu]. While this
approach allowed measurements of the transcriptome in individual cells it
required manual manipulation and was restricted to inspecting a few precious
cells. Further studies quickly showed that cell types could be identified
without sorting cells and approaches were developed to allow unbiased capture
of the whole transcriptome. Since then, many scRNA-seq protocols have been
developed (including CEL-Seq [@Hashimshony2012-dh], CEL-Seq2
[@Hashimshony2016-yi], Quartz-Seq [@Sasagawa2013-uc], Quartz-Seq2
[@Sasagawa2018-ox] and Smart-seq2 [@Picelli2013-or]) and the number of cells in
scRNA-seq experiments has scaled exponentially (Figure \@ref(fig:technologies))
[@Svensson2018-dx]. The first commercially available cell capture platform was
the Fluidigm C1. This system uses microfluidics to passively separate cells
into individual wells on a plate where they are lysed, reverse transcribed and
the collected cDNA is PCR amplified. After this stage the product is extracted
from the plate and libraries prepared for Illumina sequencing. Most C1 data has
been produced using a 96 well plate but more recently an 800 well plate has
become available, greatly increasing the number of cells that can be captured
at a time. One of the disadvantages of microfluidic plate-based cell capture
technologies is that the chips used have a fixed size window, meaning that only
cells of a particular sizes can be captured in a single run. However, as cells
are captured in individual wells they can be imaged before lysis, potentially
identifying damaged or broken cells, empty wells or wells containing more than
one cell. Capturing multiple cells is a known issue, with Macosko et al.
finding that when preparing a mixture of mouse and human cells, 30 percent of
the resulting libraries contained transcripts from both species but only about
a third of these doublets were visible in microscopy images [@Macosko2015-rl].
The newer Polaris system from Fluidigm also uses microfluidics to capture cells
but can select particular cells based on staining or fluorescent reporter
expression and then hold them for up to 24 hours while introducing various
stimuli. The cells can be imaged during this time before being lysed and
prepared for RNA sequencing. This platform provides opportunities for a range
of experiments that aren't possible using other capture technologies.

```{r technologies, fig.cap = "(ref:cap-technologies)", fig.scap = "(ref:scap-technologies)", out.width = "100%"}
knitr::include_graphics(here::here("figures/01-technologies.jpg"))
```

(ref:scap-technologies) Exponential increase in the scale of scRNA-seq experiments.

(ref:cap-technologies) Exponential increase in the scale of scRNA-seq experiments. (a) Key jumps in scRNA-seq capture technologies including manual handling, plate-based microfluidic capture and droplet-based capture. (b) Cell numbers in representative studies using key technologies by publication date. Image from Svensson, Vento-Tormo and Teichmann "Exponential scaling of single-cell RNA-seq in the past decade" [@Svensson2018-dx].

### Droplet-based cell capture

An alternative to using microfluidics to capture cells in wells is to capture
them in nano-droplets. In this process a dissociated cell mixture is fed into a
microfluidic device, while beads coated in primers enter at another input. The
device is designed to form aqueous droplets within mineral oil and the inputs
are arranged so that cells and beads can be simultaneously captured within a
droplet. When this happens the reagents carried along with the bead lyse the
cell and any poly(A) tagged RNA molecules present can bind to the capture
probes on the bead. Reverse transcription and PCR amplification then begins and
an individual cDNA library is produced for each cell, tagged with the unique
barcode sequence present on the bead. The main advantage of droplet-based
capture technologies is the ability to capture many more cells at one time, up
to tens of thousands. These approaches are also less selective about cell size
and produce fewer doublets. As a result they are much cheaper per cell,
although as sequencing costs are fixed studies using droplet-based captures
typically sequence individual cells at a much lower depth.

Droplet-based capture was popularised by the publication of the Drop-seq
[@Macosko2015-rl] and InDrop [@Klein2015-iw] platforms in 2015 and the updated
InDrops in 2017 [@Zilionis2017-gu]. These are both DIY systems and although
they differ in how the beads are produced, when the droplets are broken and in
some aspects of the chemistry they can both be constructed on a lab bench from
syringes, automatic plungers, a microscope and a small custom-made microfluidic
chip. A similar commercially available platform is the 10x Genomics Chromium
device which automates and streamlines much of the process (Figure
\@ref(fig:10xChromium)) [@Zheng2017-mm]. This device uses droplet-based
technologies for a range of applications including capture of cells for
scRNA-seq. More specialised captures, such as those aimed at profiling immune
cell receptors, are also possible and the company has recently announced kits
for single-cell Assay for Transposase-Accessible Chromatin using sequencing
(scATAC-seq) capture. A recent review has suggested that the 10x Chromium
provides the most appropriate droplet-based capture for most experiments that
aren't cost sensitive or require custom protocols [@Zhang2019-vw].

```{r 10xChromium, fig.cap = "(ref:cap-10xChromium)", fig.scap = "(ref:scap-10xChromium)", out.width = "100%"}
knitr::include_graphics(here::here("figures/01-10xChromium.png"))
```

(ref:scap-10xChromium) Diagram of the 10x Genomics Chromium cell capture process.

(ref:cap-10xChromium) Diagram of the 10x Genomics Chromium cell capture process. (A) Steps in the cell capture process. Barcoded gel beads are passed into a microfluidic device along with dissociated cells where they are captured in aqueous droplets in an oil solution. Cells are lysed within the droplets and mRNA is reverse transcribed to produced barcoded cDNA. Droplets are then broken and the cDNA collected for sequencing. (B) Structure of the gel bead capture probe including adaptor, cell barcode, UMI and poly(T) tail. Adapted using images from 10x Genomics.

### Unique Molecular Identifiers

In contrast to plate-based capture methods, which usually provide reads along
the length of RNA transcripts, droplet-based capture methods typically employ
protocols which include short random nucleotide sequences known as Unique
Molecular Identifiers (UMIs) [@Kivioja2012-bq]. Individual cells contain very
small amounts of RNA (around 10--30 pg, less than 5 percent of which is mRNA)
and to obtain enough cDNA for sequencing a PCR amplification step is necessary.
Depending on their nucleotide sequence different transcripts may be amplified
at different rates which can distort their relative proportions within a
library. UMIs attempt to improve the quantification of gene expression by
allowing the removal of PCR duplicates produced during amplification (Figure
\@ref(fig:UMIs)) [@Islam2014-gq]. The nucleotide probes used in droplet-based
capture protocols include a poly(T) sequence which binds to mature mRNA
molecules, a barcode sequence which is the same for every probe on a bead and
8--10 bases of UMI sequence which is unique to each probe. The UMI sequences are
long enough that the probability of capturing two copies of a transcript on two
probes with the same UMI is extremely low. After reverse transcription,
amplification, sequencing and alignment, de-duplication can be performed by
identifying reads with the same UMI that align to the same position and
therefore should be PCR duplicates rather than truly expressed copies of a
transcript.

```{r UMIs, fig.cap = "(ref:cap-UMIs)", fig.scap = "(ref:scap-UMIs)", out.width = "100%"}
knitr::include_graphics(here::here("figures/01-UMIs.png"))
```

(ref:scap-UMIs) Unique Molecular Identifiers (UMIs) can improve scRNA-seq quantification.

(ref:cap-UMIs) Unique Molecular Identifiers (UMIs) can improve scRNA-seq quantification. (A) UMIs are random 8--10 base pair sequences included as part of the mRNA capture probe along with the cell barcode (BC) and PCR handle. (B) The mRNA sequence can be aligned to a reference genome. (C) Each UMI is only counted once at each location, removing PCR duplicates and improving quantification.

For this method to be effective each read must be associated with a UMI which
means that only a small section at the 3' end of each transcript is sequenced.
This has the side effect of reducing the amount of cDNA that needs to be
sequenced and therefore increasing the number of cells that can be sequenced at
a time. While the improvement in quantification of gene expression levels is
useful for many downstream analyses, it comes at the cost of coverage across
the length of a transcript, which is required for applications such as variant
detection and de-novo assembly. However, reads along the length of genes have
been observed in UMI datasets and are believed to come from unannotated
transcription start sites or regions that contain enough adenine nucleotides to
bind to the capture probes. Datasets with UMIs need extra processing steps
which can be complicated by the possibility of sequencing errors in the UMI
itself. Statistical methods designed for full-length data may also be affected
by the different properties of a UMI dataset.

### Recent advances in scRNA-seq protocols

Although droplet-based techniques are currently the most commonly used cell
capture technologies, other approaches have been proposed that promise to
capture even more cells at a lower cost per cell. These include approaches
based around sub-nanolitre sized wells, for example the Seq-Well protocol
[@Gierahn2017-kr]. A simple flow cell is constructed with an array of
microwells made from polydimethylsiloxane (PDMS), a silicone rubber. A solution
containing dissociated cells is flowed over the array and cells are captured in
the wells by gravity. This is repeated to capture barcoded beads before adding
reagents and sealing the wells. Once reactions are complete the beads can be
retrieved and processed in a similar way to droplet-based capture
[@Bose2015-vn]. Some cell types are difficult to capture due to their size,
shape or other properties, and in some cases, particularly in tissues such as
the brain, it has been suggested that single-nucleus rather than single-cell
RNA-seq may be more effective [@Hu2017-jm; @Gao2017-tq; @Habib2017-uh;
@Wu2018-xf]. In these protocols Nuclei are captured and processed in a similar
way to cells, however the RNA within them is immature and unspliced so
different reference annotations are required during analysis.

Extensions to the standard protocols have also been proposed that allow
multiple measurements from the same cell (Figure \@ref(fig:multimodal)). One
such protocol is CITE-seq which enables measurement of the levels of selected
proteins at the same time as the whole transcriptome (Figure
\@ref(fig:CITE-seq)A) [@Stoeckius2017-wm]. Antibodies for the proteins of
interest are labelled with short nucleotide sequences. These antibodies can
then be applied to the dissociated cells and any that remain unbound are washed
away before cell capture. The antibody labels are then captured along with mRNA
transcripts and a size selection step is applied to separate them before
library preparation. Similar antibodies can be used to allow multiplexing of
samples through a process known as cell hashing (Figure \@ref(fig:CITE-seq)B)
[@Stoeckius2018-ij]. In a typical scRNA-seq experiment each batch corresponds
to a single sample. This complicates analysis as it is impossible to tell what
is noise due to cells being processed in the same way and what is true
biological signal. Cell hashing uses an antibody to a ubiquitously expressed
protein but with a different nucleotide sequence for each sample. The samples
can then be mixed, processed in batches and then the cells computationally
separated based on which sequence they are associated with. An added benefit of
this approach is the simple detection of doublets containing cells from
different samples.

```{r multimodal, fig.cap = "(ref:cap-multimodal)", fig.scap = "(ref:scap-multimodal)", out.width = "100%"}
knitr::include_graphics(here::here("figures/01-multimodal.png"))
```

(ref:scap-multimodal) The range of multimodal scRNA-seq technologies.

(ref:cap-multimodal) The range of multimodal scRNA-seq technologies. Many protocols have been developed to enable multiple measurements from the same individual cells with a selection shown here. These other measurements (shaded colours) include genotype (rose bud), methylation (cornsilk), chromatin state (muted lime), transcription factor binding (periwinkle), protein expression (jade lime), activation or repression (cabbage), lineage tracing (aquamarine blue), spatial location (light sky blue) or electrophysiology (cotton candy). Image from https://github.com/arnavm/multimodal-scRNA-seq available under a Creative Commons Attribution-NonCommercial-ShareAlike 4.0 International (CC BY-NC-SA 4.0) license [@Moudgil2019-bl].

```{r CITE-seq, fig.cap = "(ref:cap-CITE-seq)", fig.scap = "(ref:scap-CITE-seq)", out.width = "100%"}
knitr::include_graphics(here::here("figures/01-CITE-seq.png"))
```

(ref:scap-CITE-seq) Extensions to droplet-based capture protocols using nucleotide tagged antibodies.

(ref:cap-CITE-seq) Extensions to droplet-based capture protocols using nucleotide tagged antibodies. (A) CITE-seq allows measurement of selected cell surface markers along with the transcriptome. Nucleotide tagged antibodies are applied to cells before droplet-based capture and processing. A size selection step is used to separate antibody nucleotides before library preparation. (B) Cell hashing allows multiplexing of samples. Ubiquitous antibodies are used to tag cells in each sample with a hashtag oligonucleotide. The cells are then mixed for cell capture and processing and the hashtags used to computationally identify the origin of each cell. Images from https://cite-seq.com [@CITEseq_web].

CRISPR-Cas9 gene editing has also been developed as an extension to scRNA-seq
protocols. One possibility is to introduce a mutation at a known location that
can then be used to demultiplex samples processed together [@Spanjaard2018-gn].
It is possible to do this with samples from different individuals or cell lines
but the advantage of a gene editing based approach is that the genetic
background remains similar between samples. It is also possible to investigate
the effects of introducing a mutation. Protocols like Perturb-Seq
[@Dixit2016-wp] introduce a range of guide RNA molecules to a cell culture,
subject the cells to some stimulus then perform single-cell RNA sequencing. The
introduced mutation can then be linked to the response of the cells to the
stimulus and the associated broader changes in gene expression. More recently
it has been proposed that CITE-seq can be combined with CRISPR-based approaches
[@Mimitou2018-pn]. Another approach is CellTagging which introduces a barcoded
lentiviral construct into cells [@Biddy2018-uz]. A unique combination of
barcodes is then passed on to daughter cells and can be used to trace the
lineage of a cell population.

Other approaches that allow multiple measurements from the same individual
cells include G&T-seq [@Macaulay2015-dp] and SIDR [@Han2017-sc] which measure
RNA and genomic DNA, similar approaches which measure the exome (or
protein-coding part of the genome) along with the transcriptome [@Wang2018-xp],
scMT-seq which measures RNA and DNA methylation [@Hu2016-il], Patch-seq which
combines patch-clamp recording to measure the electrophysiology of neurons with
scRNA-seq [@Cadwell2017-wu] and scTrio-seq which is able to measure the genome,
transcriptome and methylome simultaneously [@Bian2018-rt].
